<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Working with Sequences: Raw Data &amp; Quality Control | Genomic Data Analysis Course Exercises</title>
  <meta name="description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Working with Sequences: Raw Data &amp; Quality Control | Genomic Data Analysis Course Exercises" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="github-repo" content="clstacy/GenomicDataAnalysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Working with Sequences: Raw Data &amp; Quality Control | Genomic Data Analysis Course Exercises" />
  
  <meta name="twitter:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  

<meta name="author" content="Carson Stacy" />


<meta name="date" content="2023-10-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gene-ontology.html"/>
<link rel="next" href="trimming.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i><b>1.1</b> Usage</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#render-book"><i class="fa fa-check"></i><b>1.2</b> Render book</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#preview-book"><i class="fa fa-check"></i><b>1.3</b> Preview book</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html"><i class="fa fa-check"></i><b>2</b> Getting Started in R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#installing-packages"><i class="fa fa-check"></i><b>2.1</b> Installing Packages</a></li>
<li class="chapter" data-level="2.2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exercise-description"><i class="fa fa-check"></i><b>2.2</b> Exercise Description</a></li>
<li class="chapter" data-level="2.3" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#learning-outcomes"><i class="fa fa-check"></i><b>2.3</b> Learning outcomes</a></li>
<li class="chapter" data-level="2.4" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#using-r-and-rstudio"><i class="fa fa-check"></i><b>2.4</b> Using R and RStudio</a></li>
<li class="chapter" data-level="2.5" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#load-data-directly-from-the-url"><i class="fa fa-check"></i><b>2.5</b> Load data directly from the URL</a></li>
<li class="chapter" data-level="2.6" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#working-with-data-in-r"><i class="fa fa-check"></i><b>2.6</b> Working with data in R</a></li>
<li class="chapter" data-level="2.7" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#looking-at-data-in-rstudio"><i class="fa fa-check"></i><b>2.7</b> Looking at Data in RStudio</a></li>
<li class="chapter" data-level="2.8" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exploring-the-data"><i class="fa fa-check"></i><b>2.8</b> Exploring the data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-ontology.html"><a href="gene-ontology.html"><i class="fa fa-check"></i><b>3</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gene-ontology.html"><a href="gene-ontology.html#installing-packages-1"><i class="fa fa-check"></i><b>3.1</b> Installing Packages</a></li>
<li class="chapter" data-level="3.2" data-path="gene-ontology.html"><a href="gene-ontology.html#description"><i class="fa fa-check"></i><b>3.2</b> Description</a></li>
<li class="chapter" data-level="3.3" data-path="gene-ontology.html"><a href="gene-ontology.html#learning-outcomes-1"><i class="fa fa-check"></i><b>3.3</b> Learning outcomes</a></li>
<li class="chapter" data-level="3.4" data-path="gene-ontology.html"><a href="gene-ontology.html#analysis-workflow"><i class="fa fa-check"></i><b>3.4</b> Analysis Workflow</a></li>
<li class="chapter" data-level="3.5" data-path="gene-ontology.html"><a href="gene-ontology.html#get-de-gene-list"><i class="fa fa-check"></i><b>3.5</b> Get DE gene list</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="gene-ontology.html"><a href="gene-ontology.html#saving-ggplot-output-to-a-file"><i class="fa fa-check"></i><b>3.5.1</b> Saving ggplot output to a file</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="gene-ontology.html"><a href="gene-ontology.html#the-hypergeometric-distribution-in-practice"><i class="fa fa-check"></i><b>3.6</b> The Hypergeometric Distribution in practice</a></li>
<li class="chapter" data-level="3.7" data-path="gene-ontology.html"><a href="gene-ontology.html#now-it-is-your-turn"><i class="fa fa-check"></i><b>3.7</b> Now it is your turn</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="gene-ontology.html"><a href="gene-ontology.html#run-enrichment"><i class="fa fa-check"></i><b>3.7.1</b> Run Enrichment</a></li>
<li class="chapter" data-level="3.7.2" data-path="gene-ontology.html"><a href="gene-ontology.html#see-the-data"><i class="fa fa-check"></i><b>3.7.2</b> see the data</a></li>
<li class="chapter" data-level="3.7.3" data-path="gene-ontology.html"><a href="gene-ontology.html#create-plots"><i class="fa fa-check"></i><b>3.7.3</b> create plots</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="gene-ontology.html"><a href="gene-ontology.html#questions"><i class="fa fa-check"></i><b>3.8</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html"><i class="fa fa-check"></i><b>4</b> Working with Sequences: Raw Data &amp; Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#package-install"><i class="fa fa-check"></i><b>4.1</b> Package Install</a></li>
<li class="chapter" data-level="4.2" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#description-1"><i class="fa fa-check"></i><b>4.2</b> Description</a></li>
<li class="chapter" data-level="4.3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#learning-outcomes-2"><i class="fa fa-check"></i><b>4.3</b> Learning outcomes</a></li>
<li class="chapter" data-level="4.4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#download-fastq"><i class="fa fa-check"></i><b>4.4</b> Download fastq</a></li>
<li class="chapter" data-level="4.5" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#examining-fastq"><i class="fa fa-check"></i><b>4.5</b> Examining fastq</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="trimming.html"><a href="trimming.html"><i class="fa fa-check"></i><b>5</b> Trimming</a>
<ul>
<li class="chapter" data-level="5.1" data-path="trimming.html"><a href="trimming.html#is-there-any-way-we-can-automate-some-of-this"><i class="fa fa-check"></i><b>5.1</b> Is there any way we can automate some of this?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="batch-file-processing.html"><a href="batch-file-processing.html"><i class="fa fa-check"></i><b>6</b> Batch file processing</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="batch-file-processing.html"><a href="batch-file-processing.html#running-rfastp-creates-several-files"><i class="fa fa-check"></i><b>6.0.1</b> Running RfastP creates several files:</a></li>
<li class="chapter" data-level="6.1" data-path="batch-file-processing.html"><a href="batch-file-processing.html#qc-and-adapters"><i class="fa fa-check"></i><b>6.1</b> QC and adapters</a></li>
<li class="chapter" data-level="6.2" data-path="batch-file-processing.html"><a href="batch-file-processing.html#running-fastqc"><i class="fa fa-check"></i><b>6.2</b> Running fastqc</a></li>
<li class="chapter" data-level="6.3" data-path="batch-file-processing.html"><a href="batch-file-processing.html#multiqc-for-qc-on-mutliple-samples"><i class="fa fa-check"></i><b>6.3</b> Multiqc for QC on mutliple samples</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="read-mapping.html"><a href="read-mapping.html"><i class="fa fa-check"></i><b>7</b> Read Mapping</a>
<ul>
<li class="chapter" data-level="7.1" data-path="read-mapping.html"><a href="read-mapping.html#alignment"><i class="fa fa-check"></i><b>7.1</b> Alignment</a></li>
<li class="chapter" data-level="7.2" data-path="read-mapping.html"><a href="read-mapping.html#retrieve-the-genome"><i class="fa fa-check"></i><b>7.2</b> Retrieve the genome</a></li>
<li class="chapter" data-level="7.3" data-path="read-mapping.html"><a href="read-mapping.html#build-rsubread-index"><i class="fa fa-check"></i><b>7.3</b> Build Rsubread Index</a></li>
<li class="chapter" data-level="7.4" data-path="read-mapping.html"><a href="read-mapping.html#pseudomapping-with-salmon"><i class="fa fa-check"></i><b>7.4</b> Pseudomapping with Salmon</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="read-mapping.html"><a href="read-mapping.html#create-conda-env"><i class="fa fa-check"></i><b>7.4.1</b> Create Conda Env</a></li>
<li class="chapter" data-level="7.4.2" data-path="read-mapping.html"><a href="read-mapping.html#download-transcriptome"><i class="fa fa-check"></i><b>7.4.2</b> Download transcriptome</a></li>
<li class="chapter" data-level="7.4.3" data-path="read-mapping.html"><a href="read-mapping.html#building-the-salmon-index"><i class="fa fa-check"></i><b>7.4.3</b> Building the Salmon index</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="read-mapping.html"><a href="read-mapping.html#questions-1"><i class="fa fa-check"></i><b>7.5</b> Questions</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="read-mapping.html"><a href="read-mapping.html#with-rsubread"><i class="fa fa-check"></i><b>7.5.1</b> With Rsubread:</a></li>
<li class="chapter" data-level="7.5.2" data-path="read-mapping.html"><a href="read-mapping.html#with-salmon"><i class="fa fa-check"></i><b>7.5.2</b> With Salmon:</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Genomic Data Analysis Course Exercises</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="working-with-sequences-raw-data-quality-control" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Working with Sequences: Raw Data &amp; Quality Control<a href="working-with-sequences-raw-data-quality-control.html#working-with-sequences-raw-data-quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>last updated: 2023-10-26</p>
<div id="package-install" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Package Install<a href="working-with-sequences-raw-data-quality-control.html#package-install" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As usual, make sure we have the right packages for this exercise</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="working-with-sequences-raw-data-quality-control.html#cb48-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;pacman&quot;</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;pacman&quot;</span>); <span class="fu">library</span>(pacman)</span>
<span id="cb48-2"><a href="working-with-sequences-raw-data-quality-control.html#cb48-2" tabindex="-1"></a></span>
<span id="cb48-3"><a href="working-with-sequences-raw-data-quality-control.html#cb48-3" tabindex="-1"></a><span class="co"># let&#39;s load all of the files we were using and want to have again today</span></span>
<span id="cb48-4"><a href="working-with-sequences-raw-data-quality-control.html#cb48-4" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;knitr&quot;</span>, <span class="st">&quot;readr&quot;</span>,</span>
<span id="cb48-5"><a href="working-with-sequences-raw-data-quality-control.html#cb48-5" tabindex="-1"></a>       <span class="st">&quot;pander&quot;</span>, <span class="st">&quot;BiocManager&quot;</span>, </span>
<span id="cb48-6"><a href="working-with-sequences-raw-data-quality-control.html#cb48-6" tabindex="-1"></a>       <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;stringr&quot;</span>)</span>
<span id="cb48-7"><a href="working-with-sequences-raw-data-quality-control.html#cb48-7" tabindex="-1"></a></span>
<span id="cb48-8"><a href="working-with-sequences-raw-data-quality-control.html#cb48-8" tabindex="-1"></a><span class="co"># We also need the bioconductor packages &quot;ShortRead&quot; and &quot;rfastp&quot; for today&#39;s activity.</span></span>
<span id="cb48-9"><a href="working-with-sequences-raw-data-quality-control.html#cb48-9" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;Rfastp&quot;</span>, <span class="st">&quot;ShortRead&quot;</span>)</span></code></pre></div>
</div>
<div id="description-1" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Description<a href="working-with-sequences-raw-data-quality-control.html#description-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This activity is intended to familiarize you with raw bioinformatic sequence files. Specifically, we’ll be working with short read sequencing data generated from an Illumina platform.</p>
</div>
<div id="learning-outcomes-2" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Learning outcomes<a href="working-with-sequences-raw-data-quality-control.html#learning-outcomes-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>At the end of this exercise, you should be able to:</p>
<ul>
<li>Load and read into R a raw gzipped fastq file.</li>
<li>Inspect sequence quality and evaluate results.</li>
<li>Perform quality control on raw data and save the processed output.</li>
</ul>
<p>Note that instead of <code>{r}</code>, the below chunk uses <code>{bash}</code>, meaning this isn’t r code but bash code (the language used in the terminal). The <code>-nc</code> flag ensures the files are only downloaded if they don’t already exist where you are downloading them.</p>
<p>This may take awhile the first time you run it. The below script is a bash command that downloads these files to your computer</p>
</div>
<div id="download-fastq" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Download fastq<a href="working-with-sequences-raw-data-quality-control.html#download-fastq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb49-1"><a href="working-with-sequences-raw-data-quality-control.html#cb49-1" tabindex="-1"></a><span class="co"># Be sure to change this file path to the path you want your data to go</span></span>
<span id="cb49-2"><a href="working-with-sequences-raw-data-quality-control.html#cb49-2" tabindex="-1"></a><span class="va">RAW_DATA_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Data/Raw&quot;</span></span>
<span id="cb49-3"><a href="working-with-sequences-raw-data-quality-control.html#cb49-3" tabindex="-1"></a></span>
<span id="cb49-4"><a href="working-with-sequences-raw-data-quality-control.html#cb49-4" tabindex="-1"></a><span class="co">#if you&#39;re using Windows 10,</span></span>
<span id="cb49-5"><a href="working-with-sequences-raw-data-quality-control.html#cb49-5" tabindex="-1"></a><span class="co"># in RStudio, go to Tools&gt;Global Options... &gt; Terminal &gt; New Terminals open with...</span></span>
<span id="cb49-6"><a href="working-with-sequences-raw-data-quality-control.html#cb49-6" tabindex="-1"></a><span class="co"># and choose WSL bash or git bash</span></span>
<span id="cb49-7"><a href="working-with-sequences-raw-data-quality-control.html#cb49-7" tabindex="-1"></a><span class="co"># next, use: (be sure to put in the correct username)</span></span>
<span id="cb49-8"><a href="working-with-sequences-raw-data-quality-control.html#cb49-8" tabindex="-1"></a><span class="co">#RAW_DATA_DIR=&quot;/mnt/c/Users/$USER/Desktop/Genomic_Data_Analysis/Data/Raw&quot;</span></span>
<span id="cb49-9"><a href="working-with-sequences-raw-data-quality-control.html#cb49-9" tabindex="-1"></a></span>
<span id="cb49-10"><a href="working-with-sequences-raw-data-quality-control.html#cb49-10" tabindex="-1"></a><span class="co"># create the destination directory if it doesn&#39;t already exist</span></span>
<span id="cb49-11"><a href="working-with-sequences-raw-data-quality-control.html#cb49-11" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$RAW_DATA_DIR</span></span>
<span id="cb49-12"><a href="working-with-sequences-raw-data-quality-control.html#cb49-12" tabindex="-1"></a></span>
<span id="cb49-13"><a href="working-with-sequences-raw-data-quality-control.html#cb49-13" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$RAW_DATA_DIR</span></span>
<span id="cb49-14"><a href="working-with-sequences-raw-data-quality-control.html#cb49-14" tabindex="-1"></a></span>
<span id="cb49-15"><a href="working-with-sequences-raw-data-quality-control.html#cb49-15" tabindex="-1"></a><span class="co"># change to that directory (for this code chunk only)</span></span>
<span id="cb49-16"><a href="working-with-sequences-raw-data-quality-control.html#cb49-16" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$RAW_DATA_DIR</span></span>
<span id="cb49-17"><a href="working-with-sequences-raw-data-quality-control.html#cb49-17" tabindex="-1"></a><span class="bu">pwd</span></span>
<span id="cb49-18"><a href="working-with-sequences-raw-data-quality-control.html#cb49-18" tabindex="-1"></a><span class="co"># Download the files.</span></span>
<span id="cb49-19"><a href="working-with-sequences-raw-data-quality-control.html#cb49-19" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: curl doesn&#39;t work with relative paths</span></span>
<span id="cb49-20"><a href="working-with-sequences-raw-data-quality-control.html#cb49-20" tabindex="-1"></a><span class="co"># WT unstressed (mock)</span></span>
<span id="cb49-21"><a href="working-with-sequences-raw-data-quality-control.html#cb49-21" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_MOCK_REP1.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-22"><a href="working-with-sequences-raw-data-quality-control.html#cb49-22" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_MOCK_REP2.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-23"><a href="working-with-sequences-raw-data-quality-control.html#cb49-23" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_MOCK_REP3.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-24"><a href="working-with-sequences-raw-data-quality-control.html#cb49-24" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_MOCK_REP4.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-25"><a href="working-with-sequences-raw-data-quality-control.html#cb49-25" tabindex="-1"></a><span class="co"># WT EtOH</span></span>
<span id="cb49-26"><a href="working-with-sequences-raw-data-quality-control.html#cb49-26" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_ETOH_REP1.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-27"><a href="working-with-sequences-raw-data-quality-control.html#cb49-27" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_ETOH_REP2.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-28"><a href="working-with-sequences-raw-data-quality-control.html#cb49-28" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_ETOH_REP3.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-29"><a href="working-with-sequences-raw-data-quality-control.html#cb49-29" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_ETOH_REP4.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-30"><a href="working-with-sequences-raw-data-quality-control.html#cb49-30" tabindex="-1"></a><span class="co"># msn2/4dd unstressed (mock)</span></span>
<span id="cb49-31"><a href="working-with-sequences-raw-data-quality-control.html#cb49-31" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_MOCK_REP1.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-32"><a href="working-with-sequences-raw-data-quality-control.html#cb49-32" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_MOCK_REP2.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-33"><a href="working-with-sequences-raw-data-quality-control.html#cb49-33" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_MOCK_REP3.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-34"><a href="working-with-sequences-raw-data-quality-control.html#cb49-34" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_MOCK_REP4.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-35"><a href="working-with-sequences-raw-data-quality-control.html#cb49-35" tabindex="-1"></a><span class="co"># msn2/4dd EtOH</span></span>
<span id="cb49-36"><a href="working-with-sequences-raw-data-quality-control.html#cb49-36" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_ETOH_REP1.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-37"><a href="working-with-sequences-raw-data-quality-control.html#cb49-37" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_ETOH_REP2.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-38"><a href="working-with-sequences-raw-data-quality-control.html#cb49-38" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_ETOH_REP3.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-39"><a href="working-with-sequences-raw-data-quality-control.html#cb49-39" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_ETOH_REP4.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-40"><a href="working-with-sequences-raw-data-quality-control.html#cb49-40" tabindex="-1"></a></span>
<span id="cb49-41"><a href="working-with-sequences-raw-data-quality-control.html#cb49-41" tabindex="-1"></a><span class="co"># These are subsamples of raw fastq files from a current project in our lab.</span></span>
<span id="cb49-42"><a href="working-with-sequences-raw-data-quality-control.html#cb49-42" tabindex="-1"></a></span>
<span id="cb49-43"><a href="working-with-sequences-raw-data-quality-control.html#cb49-43" tabindex="-1"></a><span class="co"># Make sure names are as desired</span></span>
<span id="cb49-44"><a href="working-with-sequences-raw-data-quality-control.html#cb49-44" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$RAW_DATA_DIR</span></span>
<span id="cb49-45"><a href="working-with-sequences-raw-data-quality-control.html#cb49-45" tabindex="-1"></a></span>
<span id="cb49-46"><a href="working-with-sequences-raw-data-quality-control.html#cb49-46" tabindex="-1"></a><span class="co"># This loops through and removes the suffix file for any OS that doesn&#39;t auto do so.</span></span>
<span id="cb49-47"><a href="working-with-sequences-raw-data-quality-control.html#cb49-47" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb49-48"><a href="working-with-sequences-raw-data-quality-control.html#cb49-48" tabindex="-1"></a>    <span class="va">newname</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/\?raw=TRUE//&#39;</span><span class="va">)</span></span>
<span id="cb49-49"><a href="working-with-sequences-raw-data-quality-control.html#cb49-49" tabindex="-1"></a>    <span class="fu">mv</span> <span class="st">&quot;</span><span class="va">$file</span><span class="st">&quot;</span> <span class="st">&quot;</span><span class="va">$newname</span><span class="st">&quot;</span></span>
<span id="cb49-50"><a href="working-with-sequences-raw-data-quality-control.html#cb49-50" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb49-51"><a href="working-with-sequences-raw-data-quality-control.html#cb49-51" tabindex="-1"></a></span>
<span id="cb49-52"><a href="working-with-sequences-raw-data-quality-control.html#cb49-52" tabindex="-1"></a><span class="co"># Let&#39;s see what one of these files contains:</span></span>
<span id="cb49-53"><a href="working-with-sequences-raw-data-quality-control.html#cb49-53" tabindex="-1"></a><span class="co"># if you&#39;re on windows or linux, delete the g from gzcat below</span></span>
<span id="cb49-54"><a href="working-with-sequences-raw-data-quality-control.html#cb49-54" tabindex="-1"></a><span class="ex">gzcat</span> <span class="va">$RAW_DATA_DIR</span>/YPS606_WT_MOCK_REP1.fastq.gz <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n8</span></span></code></pre></div>
<pre><code>## /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw
## /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 7257k  100 7257k    0     0  7961k      0 --:--:-- --:--:-- --:--:-- 7961k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6113k  100 6113k    0     0  8316k      0 --:--:-- --:--:-- --:--:-- 8316k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 7352k  100 7352k    0     0  6882k      0  0:00:01  0:00:01 --:--:-- 12.9M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6746k  100 6746k    0     0  9030k      0 --:--:-- --:--:-- --:--:-- 9030k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0 5774k    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 5774k  100 5774k    0     0  7207k      0 --:--:-- --:--:-- --:--:-- 19.7M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0 6438k    0 31831    0     0  45709      0  0:02:24 --:--:--  0:02:24 45709
 56 6438k   56 3609k    0     0  2184k      0  0:00:02  0:00:01  0:00:01 3742k
100 6438k  100 6438k    0     0  3502k      0  0:00:01  0:00:01 --:--:-- 5615k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 6908k  100 6908k    0     0  4237k      0  0:00:01  0:00:01 --:--:-- 8582k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6020k  100 6020k    0     0  8446k      0 --:--:-- --:--:-- --:--:-- 8446k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 5429k  100 5429k    0     0  6232k      0 --:--:-- --:--:-- --:--:-- 6232k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 5525k  100 5525k    0     0  7484k      0 --:--:-- --:--:-- --:--:-- 13.9M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  7 6866k    7  494k    0     0   873k      0  0:00:07 --:--:--  0:00:07  873k
100 6866k  100 6866k    0     0  8877k      0 --:--:-- --:--:-- --:--:-- 30.0M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
 60 6797k   60 4081k    0     0  5477k      0  0:00:01 --:--:--  0:00:01 5477k
100 6797k  100 6797k    0     0  7781k      0 --:--:-- --:--:-- --:--:-- 20.7M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
 66 7521k   66 4996k    0     0  6212k      0  0:00:01 --:--:--  0:00:01 6212k
100 7521k  100 7521k    0     0  8430k      0 --:--:-- --:--:-- --:--:-- 28.3M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6936k  100 6936k    0     0  8400k      0 --:--:-- --:--:-- --:--:-- 8400k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6426k  100 6426k    0     0  8192k      0 --:--:-- --:--:-- --:--:-- 8192k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6601k  100 6601k    0     0  8491k      0 --:--:-- --:--:-- --:--:-- 8491k
## @K00242:669:HFYYJBBXY:2:1209:22455:7908 1:N:0:GTCCGCAC
## GCAATGGTTTACACCCCACCGTGAGATTAGTATGCAATTTAGATCCATTA
## +
## AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
## @K00242:669:HFYYJBBXY:2:1219:16254:46926 1:N:0:GTCCGCAC
## GTCTGATTTGTCTAGATTCTTCGCAAATTTCCAGCCTTCAGAGGCTTCGC
## +
## AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ</code></pre>
<p>We have the data downloaded onto our system now, so let’s first take a look at some of these files ourselves</p>
<p>The R package ShortRead allows us to look at and process raw fastq files. It has many more features than we will use today.</p>
</div>
<div id="examining-fastq" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Examining fastq<a href="working-with-sequences-raw-data-quality-control.html#examining-fastq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s take a look at a fastq file</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="working-with-sequences-raw-data-quality-control.html#cb51-1" tabindex="-1"></a><span class="co"># If you&#39;re using windows, put your username below and uncomment this code before continuing</span></span>
<span id="cb51-2"><a href="working-with-sequences-raw-data-quality-control.html#cb51-2" tabindex="-1"></a><span class="cf">if</span>(.Platform<span class="sc">$</span>OS.type <span class="sc">==</span> <span class="st">&quot;windows&quot;</span>) {</span>
<span id="cb51-3"><a href="working-with-sequences-raw-data-quality-control.html#cb51-3" tabindex="-1"></a>  <span class="fu">Sys.setenv</span>(<span class="at">R_USER =</span> <span class="st">&quot;C:/Users/$USERNAME&quot;</span>)</span>
<span id="cb51-4"><a href="working-with-sequences-raw-data-quality-control.html#cb51-4" tabindex="-1"></a>}</span>
<span id="cb51-5"><a href="working-with-sequences-raw-data-quality-control.html#cb51-5" tabindex="-1"></a></span>
<span id="cb51-6"><a href="working-with-sequences-raw-data-quality-control.html#cb51-6" tabindex="-1"></a><span class="co"># change this directory here to where you have the file saved</span></span>
<span id="cb51-7"><a href="working-with-sequences-raw-data-quality-control.html#cb51-7" tabindex="-1"></a>path_fastq_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">path.expand</span>(<span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_MOCK_REP1.fastq.gz&quot;</span>)</span>
<span id="cb51-8"><a href="working-with-sequences-raw-data-quality-control.html#cb51-8" tabindex="-1"></a></span>
<span id="cb51-9"><a href="working-with-sequences-raw-data-quality-control.html#cb51-9" tabindex="-1"></a></span>
<span id="cb51-10"><a href="working-with-sequences-raw-data-quality-control.html#cb51-10" tabindex="-1"></a></span>
<span id="cb51-11"><a href="working-with-sequences-raw-data-quality-control.html#cb51-11" tabindex="-1"></a>fastq_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">readFastq</span>(path_fastq_WT_MOCK_REP1)</span>
<span id="cb51-12"><a href="working-with-sequences-raw-data-quality-control.html#cb51-12" tabindex="-1"></a></span>
<span id="cb51-13"><a href="working-with-sequences-raw-data-quality-control.html#cb51-13" tabindex="-1"></a><span class="co"># file too big? swap readFastq() for:</span></span>
<span id="cb51-14"><a href="working-with-sequences-raw-data-quality-control.html#cb51-14" tabindex="-1"></a>subsampled_fastq_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">yield</span>(<span class="fu">FastqSampler</span>(path_fastq_WT_MOCK_REP1, <span class="at">n=</span><span class="dv">10000</span>)) <span class="co"># where n is the number of reads you want to sample</span></span>
<span id="cb51-15"><a href="working-with-sequences-raw-data-quality-control.html#cb51-15" tabindex="-1"></a><span class="co"># the fastq files we downloaded are smaller than a normal fastq file, </span></span>
<span id="cb51-16"><a href="working-with-sequences-raw-data-quality-control.html#cb51-16" tabindex="-1"></a><span class="co"># because they have been subsampled down from their full size for demonstration.</span></span></code></pre></div>
<p>A few quick ways to examine the fastq data object</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="working-with-sequences-raw-data-quality-control.html#cb52-1" tabindex="-1"></a><span class="co"># Typing the name of the object gives us a simple summary</span></span>
<span id="cb52-2"><a href="working-with-sequences-raw-data-quality-control.html#cb52-2" tabindex="-1"></a>fastq_WT_MOCK_REP1</span></code></pre></div>
<pre><code>## class: ShortReadQ
## length: 223565 reads; width: 50 cycles</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="working-with-sequences-raw-data-quality-control.html#cb54-1" tabindex="-1"></a><span class="co"># the length() function gives us the total number of reads</span></span>
<span id="cb54-2"><a href="working-with-sequences-raw-data-quality-control.html#cb54-2" tabindex="-1"></a><span class="fu">length</span>(fastq_WT_MOCK_REP1)</span></code></pre></div>
<pre><code>## [1] 223565</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="working-with-sequences-raw-data-quality-control.html#cb56-1" tabindex="-1"></a><span class="co"># We can use the width() function to find the size of each read/sequence in fastq</span></span>
<span id="cb56-2"><a href="working-with-sequences-raw-data-quality-control.html#cb56-2" tabindex="-1"></a><span class="fu">width</span>(fastq_WT_MOCK_REP1) <span class="sc">|&gt;</span> <span class="fu">head</span>() <span class="co"># add head() pipe to only print first 10</span></span></code></pre></div>
<pre><code>## [1] 50 50 50 50 50 50</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="working-with-sequences-raw-data-quality-control.html#cb58-1" tabindex="-1"></a><span class="co">#sread() - Retrieve sequence of reads.</span></span>
<span id="cb58-2"><a href="working-with-sequences-raw-data-quality-control.html#cb58-2" tabindex="-1"></a><span class="fu">sread</span>(fastq_WT_MOCK_REP1)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 223565:
##          width seq
##      [1]    50 GCAATGGTTTACACCCCACCGTGAGATTAGTATGCAATTTAGATCCATTA
##      [2]    50 GTCTGATTTGTCTAGATTCTTCGCAAATTTCCAGCCTTCAGAGGCTTCGC
##      [3]    50 CTTGAAGTAAGCTTCATCAGCTTCCAACATACCATCGAACCATGGCAACA
##      [4]    50 GCACCGGCAATCTTGTTACCCATAGCATCGAATTTATCTTCTTCGTCTTC
##      [5]    50 CTGTTACCAACTTGTTGTGACATCTTTCTAGTATAATTTTTAAAGTTCTA
##      ...   ... ...
## [223561]    50 GGTGATCAACTGGATTCATGGCAACACCACGGGTCTTTGGCCAAGAGTTT
## [223562]    50 CTTCCAAGTCGATGTATCTCTTGTGGATTCTCATTTCGTAGGTTTCCCAA
## [223563]    50 CGTTAGCATCAACTTCGAAGATAGCTTCCAAGACTGGTTCACCAGCTGGC
## [223564]    50 GCTTCTTTTCTTGGACCTTTTTCAATCTATAAAATTCTTCTCTGTCCAAC
## [223565]    50 ACCGAATGGAAGATTGGTCACCCTCGGAACCCATGATATCTTCGAATGGG</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="working-with-sequences-raw-data-quality-control.html#cb60-1" tabindex="-1"></a><span class="co">#quality() - Retrieve quality of reads as ASCII scores.</span></span>
<span id="cb60-2"><a href="working-with-sequences-raw-data-quality-control.html#cb60-2" tabindex="-1"></a><span class="fu">quality</span>(fastq_WT_MOCK_REP1)</span></code></pre></div>
<pre><code>## class: FastqQuality
## quality:
## BStringSet object of length 223565:
##          width seq
##      [1]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
##      [2]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
##      [3]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
##      [4]    50 AAFFFJJJJJJJJJJJJJJFJJFJJJJFJJJJJJJJJJJFJFJJJJJJJJ
##      [5]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJ
##      ...   ... ...
## [223561]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJ
## [223562]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
## [223563]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
## [223564]    50 A&lt;FFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
## [223565]    50 AAFFFJJJJJJJJJJJJJJJJJJJJ-7FJJJJJJJJJJJJJJJJJJJJJJ</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="working-with-sequences-raw-data-quality-control.html#cb62-1" tabindex="-1"></a><span class="co">#id() - Retrieve IDs of reads</span></span>
<span id="cb62-2"><a href="working-with-sequences-raw-data-quality-control.html#cb62-2" tabindex="-1"></a><span class="fu">id</span>(fastq_WT_MOCK_REP1)</span></code></pre></div>
<pre><code>## BStringSet object of length 223565:
##          width seq
##      [1]    53 K00242:669:HFYYJBBXY:2:1209:22455:7908 1:N:0:GTCCGCAC
##      [2]    54 K00242:669:HFYYJBBXY:2:1219:16254:46926 1:N:0:GTCCGCAC
##      [3]    53 K00242:669:HFYYJBBXY:2:2223:29985:7029 1:N:0:GTCCGCAC
##      [4]    52 K00242:669:HFYYJBBXY:2:2213:7770:7750 1:N:0:GTCCGCAC
##      [5]    54 K00242:669:HFYYJBBXY:2:1223:27813:29606 1:N:0:GTCCGCAC
##      ...   ... ...
## [223561]    54 K00242:669:HFYYJBBXY:2:1120:22820:46205 1:N:0:GTCCGCAC
## [223562]    54 K00242:669:HFYYJBBXY:2:2209:17239:49054 1:N:0:GTCCGCAC
## [223563]    53 K00242:669:HFYYJBBXY:2:2206:3244:20656 1:N:0:GTCCGCAC
## [223564]    54 K00242:669:HFYYJBBXY:2:2102:28229:16506 1:N:0:GTCCGCAC
## [223565]    53 K00242:669:HFYYJBBXY:2:2204:17381:6712 1:N:0:GTCCGCAC</code></pre>
<p>The output of <code>sread()</code> is a DNAStringSet object, so we can use all of the commands from the Biostrings library on the output object.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="working-with-sequences-raw-data-quality-control.html#cb64-1" tabindex="-1"></a><span class="co"># first, let&#39;s save the output of sread as an object</span></span>
<span id="cb64-2"><a href="working-with-sequences-raw-data-quality-control.html#cb64-2" tabindex="-1"></a>sequence_of_reads <span class="ot">&lt;-</span> <span class="fu">sread</span>(fastq_WT_MOCK_REP1)</span>
<span id="cb64-3"><a href="working-with-sequences-raw-data-quality-control.html#cb64-3" tabindex="-1"></a></span>
<span id="cb64-4"><a href="working-with-sequences-raw-data-quality-control.html#cb64-4" tabindex="-1"></a><span class="co"># Now, let&#39;s use the biostrings function alphabetFrequency to see</span></span>
<span id="cb64-5"><a href="working-with-sequences-raw-data-quality-control.html#cb64-5" tabindex="-1"></a><span class="co"># the occurrence of nucleotide bases in reads.</span></span>
<span id="cb64-6"><a href="working-with-sequences-raw-data-quality-control.html#cb64-6" tabindex="-1"></a>alph_freq <span class="ot">&lt;-</span> <span class="fu">alphabetFrequency</span>(sequence_of_reads)</span>
<span id="cb64-7"><a href="working-with-sequences-raw-data-quality-control.html#cb64-7" tabindex="-1"></a></span>
<span id="cb64-8"><a href="working-with-sequences-raw-data-quality-control.html#cb64-8" tabindex="-1"></a><span class="co"># subset just the first two reads</span></span>
<span id="cb64-9"><a href="working-with-sequences-raw-data-quality-control.html#cb64-9" tabindex="-1"></a>alph_freq[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,]</span></code></pre></div>
<pre><code>##       A  C  G  T M R W S Y K V H D B N - + .
## [1,] 15 11  9 15 0 0 0 0 0 0 0 0 0 0 0 0 0 0
## [2,]  9 13 10 18 0 0 0 0 0 0 0 0 0 0 0 0 0 0</code></pre>
<p>We see most of the nucleotides are assigned to A, C, G, or T, with one base in each read an N.</p>
<p>A fundamental difference between fasta and fastq files is the Quality scores containined in fastQ.</p>
<p>Quality scores are stored as ASCII characters representing -log10 probability of base being wrong (Larger scores would be associated to more confident base calls).</p>
<p>A comprehensive description of phred quality can be found on the wiki page for FastQ.</p>
<p>To see the fastq encodings, we can run:</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="working-with-sequences-raw-data-quality-control.html#cb66-1" tabindex="-1"></a><span class="fu">encoding</span>(<span class="fu">quality</span>(fastq_WT_MOCK_REP1))</span></code></pre></div>
<pre><code>##  !  &quot;  #  $  %  &amp;  &#39;  (  )  *  +  ,  -  .  /  0  1  2  3  4  5  6  7  8  9  : 
##  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 
##  ;  &lt;  =  &gt;  ?  @  A  B  C  D  E  F  G  H  I  J  K  L  M  N  O  P  Q  R  S  T 
## 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 
##  U  V  W  X  Y  Z  [ \\  ]  ^  _  `  a  b  c  d  e  f  g  h  i  j  k  l  m  n 
## 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 
##  o  p  q  r  s  t  u  v  w  x  y  z  {  |  }  ~ 
## 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93</code></pre>
<p>The ShortRead package has many functions available to allow us to collect useful metrics from our ShortRead object.</p>
<p>One very useful function is the <code>alphabetByCycle()</code> function which provides a quick method to summarise base occurrence of cycles.</p>
<p>Here we apply <code>alphabetByCycle()</code> function to the sequence information and show the occurrence of main 4 bases over first 15 cycles.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="working-with-sequences-raw-data-quality-control.html#cb68-1" tabindex="-1"></a>alph_by_cycle <span class="ot">&lt;-</span> <span class="fu">alphabetByCycle</span>(sequence_of_reads)</span>
<span id="cb68-2"><a href="working-with-sequences-raw-data-quality-control.html#cb68-2" tabindex="-1"></a>alph_by_cycle[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>]</span></code></pre></div>
<pre><code>##         cycle
## alphabet  [,1]  [,2]  [,3]  [,4]  [,5]  [,6]   [,7]  [,8]  [,9] [,10] [,11]
##        A 21976 31733 40716 54511 67035 76250  52744 50858 52179 82996 65197
##        C 80979 59940 67520 56801 42469 38578  34934 49605 44865 33367 47832
##        G 91512 46565 44726 59238 53934 41211  34113 40354 43781 41308 50801
##        T 28656 85309 70603 52948 60117 67525 101774 82748 82740 65894 59735
##         cycle
## alphabet [,12] [,13] [,14] [,15]
##        A 56240 62322 63066 62605
##        C 50963 43951 43562 46231
##        G 45250 44735 43818 43027
##        T 71112 72557 73119 71702</code></pre>
<p>We can use the table function to identify the number of times a sequence appears in our FastQ file’s sequence reads.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="working-with-sequences-raw-data-quality-control.html#cb70-1" tabindex="-1"></a>readOccurence <span class="ot">&lt;-</span> <span class="fu">table</span>(sequence_of_reads)</span>
<span id="cb70-2"><a href="working-with-sequences-raw-data-quality-control.html#cb70-2" tabindex="-1"></a></span>
<span id="cb70-3"><a href="working-with-sequences-raw-data-quality-control.html#cb70-3" tabindex="-1"></a><span class="co"># see the top 3 sequences that appear the highest number of times</span></span>
<span id="cb70-4"><a href="working-with-sequences-raw-data-quality-control.html#cb70-4" tabindex="-1"></a><span class="fu">sort</span>(readOccurence,<span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## sequence_of_reads
## CTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT 
##                                                600 
## CCCCCCCCCTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT 
##                                                496 
## CCCCCCCCTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT 
##                                                392</code></pre>
<p>We can identify duplicated reads (potentially arising from PCR over amplification) by using the <code>srduplicated()</code> function and the ShortReadQ object.</p>
<p>This returns a logical vector identifying which reads’ sequences are duplicates (occur more than once in file). Note that the first time a sequence appears in file is not a duplicate but the second, third, fourth times etc are.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="working-with-sequences-raw-data-quality-control.html#cb72-1" tabindex="-1"></a>duplicates <span class="ot">&lt;-</span> <span class="fu">srduplicated</span>(fastq_WT_MOCK_REP1)</span>
<span id="cb72-2"><a href="working-with-sequences-raw-data-quality-control.html#cb72-2" tabindex="-1"></a>duplicates[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## [1] FALSE FALSE FALSE</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="working-with-sequences-raw-data-quality-control.html#cb74-1" tabindex="-1"></a><span class="co"># we can use table() to get a quick summary of the seq duplication rate</span></span>
<span id="cb74-2"><a href="working-with-sequences-raw-data-quality-control.html#cb74-2" tabindex="-1"></a><span class="fu">table</span>(duplicates)</span></code></pre></div>
<pre><code>## duplicates
##  FALSE   TRUE 
## 140931  82634</code></pre>
<p>The ShortRead package also contains a function to generate a simple quality control report.</p>
<p>The <code>qa()</code> function accepts a FastQ file and returns a FastqQA object.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="working-with-sequences-raw-data-quality-control.html#cb76-1" tabindex="-1"></a>qa_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">qa</span>(path_fastq_WT_MOCK_REP1)</span>
<span id="cb76-2"><a href="working-with-sequences-raw-data-quality-control.html#cb76-2" tabindex="-1"></a>qa_WT_MOCK_REP1</span></code></pre></div>
<pre><code>## class: FastqQA(10)
## QA elements (access with qa[[&quot;elt&quot;]]):
##   readCounts: data.frame(1 3)
##   baseCalls: data.frame(1 5)
##   readQualityScore: data.frame(512 4)
##   baseQuality: data.frame(95 3)
##   alignQuality: data.frame(1 3)
##   frequentSequences: data.frame(50 4)
##   sequenceDistribution: data.frame(76 4)
##   perCycle: list(2)
##     baseCall: data.frame(231 4)
##     quality: data.frame(322 5)
##   perTile: list(2)
##     readCounts: data.frame(0 4)
##     medianReadQualityScore: data.frame(0 4)
##   adapterContamination: data.frame(1 1)</code></pre>
<p>We can then use the report() function to generate a simple report.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="working-with-sequences-raw-data-quality-control.html#cb78-1" tabindex="-1"></a>myReport_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">report</span>(qa_WT_MOCK_REP1)</span>
<span id="cb78-2"><a href="working-with-sequences-raw-data-quality-control.html#cb78-2" tabindex="-1"></a>myReport_WT_MOCK_REP1</span></code></pre></div>
<pre><code>## [1] &quot;/var/folders/y1/f7wyg4vj50dgrg8drv0bn4nc0000gn/T//RtmpbCeGU7/file16b718f864a/index.html&quot;</code></pre>
<p>Finally we can review the report in a browser or use the browseURL function to open it in a browser from R.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="working-with-sequences-raw-data-quality-control.html#cb80-1" tabindex="-1"></a><span class="fu">browseURL</span>(myReport_WT_MOCK_REP1)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gene-ontology.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="trimming.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-Working_with_Sequences.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
