<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Trimming | Genomic Data Analysis Course Exercises</title>
  <meta name="description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Trimming | Genomic Data Analysis Course Exercises" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="github-repo" content="clstacy/GenomicDataAnalysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Trimming | Genomic Data Analysis Course Exercises" />
  
  <meta name="twitter:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  

<meta name="author" content="Carson Stacy" />


<meta name="date" content="2023-10-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="working-with-sequences-raw-data-quality-control.html"/>
<link rel="next" href="batch-file-processing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i><b>1.1</b> Usage</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#render-book"><i class="fa fa-check"></i><b>1.2</b> Render book</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#preview-book"><i class="fa fa-check"></i><b>1.3</b> Preview book</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html"><i class="fa fa-check"></i><b>2</b> Getting Started in R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#installing-packages"><i class="fa fa-check"></i><b>2.1</b> Installing Packages</a></li>
<li class="chapter" data-level="2.2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exercise-description"><i class="fa fa-check"></i><b>2.2</b> Exercise Description</a></li>
<li class="chapter" data-level="2.3" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#learning-outcomes"><i class="fa fa-check"></i><b>2.3</b> Learning outcomes</a></li>
<li class="chapter" data-level="2.4" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#using-r-and-rstudio"><i class="fa fa-check"></i><b>2.4</b> Using R and RStudio</a></li>
<li class="chapter" data-level="2.5" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#load-data-directly-from-the-url"><i class="fa fa-check"></i><b>2.5</b> Load data directly from the URL</a></li>
<li class="chapter" data-level="2.6" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#working-with-data-in-r"><i class="fa fa-check"></i><b>2.6</b> Working with data in R</a></li>
<li class="chapter" data-level="2.7" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#looking-at-data-in-rstudio"><i class="fa fa-check"></i><b>2.7</b> Looking at Data in RStudio</a></li>
<li class="chapter" data-level="2.8" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exploring-the-data"><i class="fa fa-check"></i><b>2.8</b> Exploring the data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-ontology.html"><a href="gene-ontology.html"><i class="fa fa-check"></i><b>3</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gene-ontology.html"><a href="gene-ontology.html#installing-packages-1"><i class="fa fa-check"></i><b>3.1</b> Installing Packages</a></li>
<li class="chapter" data-level="3.2" data-path="gene-ontology.html"><a href="gene-ontology.html#description"><i class="fa fa-check"></i><b>3.2</b> Description</a></li>
<li class="chapter" data-level="3.3" data-path="gene-ontology.html"><a href="gene-ontology.html#learning-outcomes-1"><i class="fa fa-check"></i><b>3.3</b> Learning outcomes</a></li>
<li class="chapter" data-level="3.4" data-path="gene-ontology.html"><a href="gene-ontology.html#analysis-workflow"><i class="fa fa-check"></i><b>3.4</b> Analysis Workflow</a></li>
<li class="chapter" data-level="3.5" data-path="gene-ontology.html"><a href="gene-ontology.html#get-de-gene-list"><i class="fa fa-check"></i><b>3.5</b> Get DE gene list</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="gene-ontology.html"><a href="gene-ontology.html#saving-ggplot-output-to-a-file"><i class="fa fa-check"></i><b>3.5.1</b> Saving ggplot output to a file</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="gene-ontology.html"><a href="gene-ontology.html#the-hypergeometric-distribution-in-practice"><i class="fa fa-check"></i><b>3.6</b> The Hypergeometric Distribution in practice</a></li>
<li class="chapter" data-level="3.7" data-path="gene-ontology.html"><a href="gene-ontology.html#now-it-is-your-turn"><i class="fa fa-check"></i><b>3.7</b> Now it is your turn</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="gene-ontology.html"><a href="gene-ontology.html#run-enrichment"><i class="fa fa-check"></i><b>3.7.1</b> Run Enrichment</a></li>
<li class="chapter" data-level="3.7.2" data-path="gene-ontology.html"><a href="gene-ontology.html#see-the-data"><i class="fa fa-check"></i><b>3.7.2</b> see the data</a></li>
<li class="chapter" data-level="3.7.3" data-path="gene-ontology.html"><a href="gene-ontology.html#create-plots"><i class="fa fa-check"></i><b>3.7.3</b> create plots</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="gene-ontology.html"><a href="gene-ontology.html#questions"><i class="fa fa-check"></i><b>3.8</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html"><i class="fa fa-check"></i><b>4</b> Working with Sequences: Raw Data &amp; Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#package-install"><i class="fa fa-check"></i><b>4.1</b> Package Install</a></li>
<li class="chapter" data-level="4.2" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#description-1"><i class="fa fa-check"></i><b>4.2</b> Description</a></li>
<li class="chapter" data-level="4.3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#learning-outcomes-2"><i class="fa fa-check"></i><b>4.3</b> Learning outcomes</a></li>
<li class="chapter" data-level="4.4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#download-fastq"><i class="fa fa-check"></i><b>4.4</b> Download fastq</a></li>
<li class="chapter" data-level="4.5" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#examining-fastq"><i class="fa fa-check"></i><b>4.5</b> Examining fastq</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="trimming.html"><a href="trimming.html"><i class="fa fa-check"></i><b>5</b> Trimming</a>
<ul>
<li class="chapter" data-level="5.1" data-path="trimming.html"><a href="trimming.html#is-there-any-way-we-can-automate-some-of-this"><i class="fa fa-check"></i><b>5.1</b> Is there any way we can automate some of this?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="batch-file-processing.html"><a href="batch-file-processing.html"><i class="fa fa-check"></i><b>6</b> Batch file processing</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="batch-file-processing.html"><a href="batch-file-processing.html#running-rfastp-creates-several-files"><i class="fa fa-check"></i><b>6.0.1</b> Running RfastP creates several files:</a></li>
<li class="chapter" data-level="6.1" data-path="batch-file-processing.html"><a href="batch-file-processing.html#qc-and-adapters"><i class="fa fa-check"></i><b>6.1</b> QC and adapters</a></li>
<li class="chapter" data-level="6.2" data-path="batch-file-processing.html"><a href="batch-file-processing.html#running-fastqc"><i class="fa fa-check"></i><b>6.2</b> Running fastqc</a></li>
<li class="chapter" data-level="6.3" data-path="batch-file-processing.html"><a href="batch-file-processing.html#multiqc-for-qc-on-mutliple-samples"><i class="fa fa-check"></i><b>6.3</b> Multiqc for QC on mutliple samples</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="read-mapping.html"><a href="read-mapping.html"><i class="fa fa-check"></i><b>7</b> Read Mapping</a>
<ul>
<li class="chapter" data-level="7.1" data-path="read-mapping.html"><a href="read-mapping.html#alignment"><i class="fa fa-check"></i><b>7.1</b> Alignment</a></li>
<li class="chapter" data-level="7.2" data-path="read-mapping.html"><a href="read-mapping.html#retrieve-the-genome"><i class="fa fa-check"></i><b>7.2</b> Retrieve the genome</a></li>
<li class="chapter" data-level="7.3" data-path="read-mapping.html"><a href="read-mapping.html#build-rsubread-index"><i class="fa fa-check"></i><b>7.3</b> Build Rsubread Index</a></li>
<li class="chapter" data-level="7.4" data-path="read-mapping.html"><a href="read-mapping.html#pseudomapping-with-salmon"><i class="fa fa-check"></i><b>7.4</b> Pseudomapping with Salmon</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="read-mapping.html"><a href="read-mapping.html#create-conda-env"><i class="fa fa-check"></i><b>7.4.1</b> Create Conda Env</a></li>
<li class="chapter" data-level="7.4.2" data-path="read-mapping.html"><a href="read-mapping.html#download-transcriptome"><i class="fa fa-check"></i><b>7.4.2</b> Download transcriptome</a></li>
<li class="chapter" data-level="7.4.3" data-path="read-mapping.html"><a href="read-mapping.html#building-the-salmon-index"><i class="fa fa-check"></i><b>7.4.3</b> Building the Salmon index</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="read-mapping.html"><a href="read-mapping.html#questions-1"><i class="fa fa-check"></i><b>7.5</b> Questions</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="read-mapping.html"><a href="read-mapping.html#with-rsubread"><i class="fa fa-check"></i><b>7.5.1</b> With Rsubread:</a></li>
<li class="chapter" data-level="7.5.2" data-path="read-mapping.html"><a href="read-mapping.html#with-salmon"><i class="fa fa-check"></i><b>7.5.2</b> With Salmon:</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Genomic Data Analysis Course Exercises</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="trimming" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Trimming<a href="trimming.html#trimming" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>When we observe low quality at the end of reads we may wish to remove the low quality bases for later alignment to the genome. The <code>trimTails()</code> function trims reads from the 3’, removing bases which fall below a desired quality. The <code>trimTails()</code> function accepts arguments specifying the ShortReadQ object, the minimum number of successive bases required to be below quality cut-off for trimming and the actual cut-off score.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="trimming.html#cb81-1" tabindex="-1"></a>trimmed_fastq_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">trimTails</span>(fastq_WT_MOCK_REP1, <span class="co"># ShortReadQ object to trim</span></span>
<span id="cb81-2"><a href="trimming.html#cb81-2" tabindex="-1"></a>                          <span class="at">k=</span><span class="dv">10</span>, <span class="co"># integer number of failing letters to trigger trim</span></span>
<span id="cb81-3"><a href="trimming.html#cb81-3" tabindex="-1"></a>                          <span class="at">a=</span><span class="st">&quot;5&quot;</span>) <span class="co"># character giving letter at or below to &quot;fail&quot;</span></span>
<span id="cb81-4"><a href="trimming.html#cb81-4" tabindex="-1"></a>trimmed_fastq_WT_MOCK_REP1</span></code></pre></div>
<pre><code>## class: ShortReadQ
## length: 223565 reads; width: 16..50 cycles</code></pre>
<p>Now we have trimmed our FastQ reads, we can export these reads for further analysis using the writeFastq() function</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="trimming.html#cb83-1" tabindex="-1"></a><span class="fu">writeFastq</span>(trimmed_fastq_WT_MOCK_REP1,</span>
<span id="cb83-2"><a href="trimming.html#cb83-2" tabindex="-1"></a>           <span class="st">&quot;~/Desktop/Genomic_Data_Analysis/WT_MOCK_REP1_shortread_trimmed.fastq.gz&quot;</span>) <span class="co">#path to save file</span></span></code></pre></div>
<div id="is-there-any-way-we-can-automate-some-of-this" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Is there any way we can automate some of this?<a href="trimming.html#is-there-any-way-we-can-automate-some-of-this" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There are several utility programs that will provide you with QC and trim your data for you, with less input from you. We like fastp as it does some basic QC and trims your fastq files, and it does it very quickly. To make this available in R, it has been made available in the Bioconductor package Rfastp.</p>
<p>By default, fastp will make a html report to summarize your result. But the Rfastp wrapper allows you to look at some of them in R.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="trimming.html#cb84-1" tabindex="-1"></a><span class="co"># create a directory for the output to go into if not already present</span></span>
<span id="cb84-2"><a href="trimming.html#cb84-2" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">dirname</span>(<span class="fu">dirname</span>(path_fastq_WT_MOCK_REP1)), <span class="st">&quot;/Trimmed_rfastp&quot;</span>) </span>
<span id="cb84-3"><a href="trimming.html#cb84-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(output_dir)) {<span class="fu">dir.create</span>(output_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>)}</span>
<span id="cb84-4"><a href="trimming.html#cb84-4" tabindex="-1"></a></span>
<span id="cb84-5"><a href="trimming.html#cb84-5" tabindex="-1"></a><span class="co"># if we wanted to just run a single file, we would do so like this:</span></span>
<span id="cb84-6"><a href="trimming.html#cb84-6" tabindex="-1"></a>rfastp_report <span class="ot">&lt;-</span> <span class="fu">rfastp</span>(<span class="at">read1 =</span> path_fastq_WT_MOCK_REP1,</span>
<span id="cb84-7"><a href="trimming.html#cb84-7" tabindex="-1"></a>                        <span class="at">outputFastq =</span> <span class="fu">paste0</span>(output_dir, <span class="st">&quot;/YPS606_WT_MOCK_REP1&quot;</span>))</span>
<span id="cb84-8"><a href="trimming.html#cb84-8" tabindex="-1"></a></span>
<span id="cb84-9"><a href="trimming.html#cb84-9" tabindex="-1"></a><span class="co"># print out the qc summary for this sample</span></span>
<span id="cb84-10"><a href="trimming.html#cb84-10" tabindex="-1"></a>df_summary <span class="ot">&lt;-</span> <span class="fu">qcSummary</span>(rfastp_report)</span>
<span id="cb84-11"><a href="trimming.html#cb84-11" tabindex="-1"></a>df_summary <span class="sc">|&gt;</span> <span class="fu">print.data.frame</span>()</span></code></pre></div>
<pre><code>##                      Before_QC     After_QC
## total_reads       2.235650e+05 2.235390e+05
## total_bases       1.117825e+07 1.117695e+07
## q20_bases         1.110373e+07 1.110280e+07
## q30_bases         1.096271e+07 1.096189e+07
## q20_rate          9.933340e-01 9.933660e-01
## q30_rate          9.807180e-01 9.807590e-01
## read1_mean_length 5.000000e+01 5.000000e+01
## gc_content        4.165340e-01 4.165420e-01</code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="working-with-sequences-raw-data-quality-control.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="batch-file-processing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-Working_with_Sequences.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
