<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Gene Ontology | Genomic Data Analysis Course Exercises</title>
  <meta name="description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Gene Ontology | Genomic Data Analysis Course Exercises" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="github-repo" content="clstacy/GenomicDataAnalysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Gene Ontology | Genomic Data Analysis Course Exercises" />
  
  <meta name="twitter:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  

<meta name="author" content="Carson Stacy &amp; Jeffrey Lewis" />


<meta name="date" content="2023-10-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="getting-started-in-r.html"/>
<link rel="next" href="working-with-sequences-raw-data-quality-control.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/core-js-2.5.3/shim.min.js"></script>
<script src="libs/react-18.2.0/react.min.js"></script>
<script src="libs/react-18.2.0/react-dom.min.js"></script>
<script src="libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="libs/reactable-binding-0.4.4/reactable.js"></script>
<link href="libs/datatables-1/datatables.css" rel="stylesheet" />
<script src="libs/datatables-1/datatables.min.js"></script>
<link href="libs/vega-1/vega_plots.css" rel="stylesheet" />
<script src="libs/vega-1/vega.min.js"></script>
<link href="libs/vega-tooltip-1/vega-tooltip.css" rel="stylesheet" />
<script src="libs/vega-tooltip-1/vega-tooltip.js"></script>
<script src="libs/FileSaver-1/FileSaver.js"></script>
<script src="libs/GlimmaV2-1/XYSpecs.js"></script>
<script src="libs/GlimmaV2-1/MDSSpecs.js"></script>
<script src="libs/GlimmaV2-1/expressionSpec.js"></script>
<script src="libs/GlimmaV2-1/makeTooltip.js"></script>
<script src="libs/GlimmaV2-1/saveTools.js"></script>
<script src="libs/glimmaXY-binding-2.10.0/glimmaXY.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Genomic Data Analysis Exercises</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i><b>0.1</b> Usage</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#key-features"><i class="fa fa-check"></i><b>0.2</b> <strong>Key Features</strong></a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#disclaimer"><i class="fa fa-check"></i><b>0.3</b> Disclaimer</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html"><i class="fa fa-check"></i><b>1</b> Getting Started in R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exercise-description"><i class="fa fa-check"></i><b>1.1</b> Exercise Description</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#learning-outcomes"><i class="fa fa-check"></i><b>1.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#using-r-and-rstudio"><i class="fa fa-check"></i><b>1.3</b> Using R and RStudio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#load-data-directly-from-the-url"><i class="fa fa-check"></i><b>1.4</b> Load data directly from the URL</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#working-with-data-in-r"><i class="fa fa-check"></i><b>1.5</b> Working with data in R</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#looking-at-data-in-rstudio"><i class="fa fa-check"></i><b>1.6</b> Looking at Data in RStudio</a></li>
<li class="chapter" data-level="1.7" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exploring-the-data"><i class="fa fa-check"></i><b>1.7</b> Exploring the data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gene-ontology.html"><a href="gene-ontology.html"><i class="fa fa-check"></i><b>2</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gene-ontology.html"><a href="gene-ontology.html#description"><i class="fa fa-check"></i><b>2.1</b> Description</a></li>
<li class="chapter" data-level="2.2" data-path="gene-ontology.html"><a href="gene-ontology.html#learning-outcomes-1"><i class="fa fa-check"></i><b>2.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="2.3" data-path="gene-ontology.html"><a href="gene-ontology.html#analysis-workflow"><i class="fa fa-check"></i><b>2.3</b> Analysis Workflow</a></li>
<li class="chapter" data-level="2.4" data-path="gene-ontology.html"><a href="gene-ontology.html#get-de-gene-list"><i class="fa fa-check"></i><b>2.4</b> Get DE gene list</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="gene-ontology.html"><a href="gene-ontology.html#saving-ggplot-output-to-a-file"><i class="fa fa-check"></i><b>2.4.1</b> Saving ggplot output to a file</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="gene-ontology.html"><a href="gene-ontology.html#the-hypergeometric-distribution-in-practice"><i class="fa fa-check"></i><b>2.5</b> The Hypergeometric Distribution in practice</a></li>
<li class="chapter" data-level="2.6" data-path="gene-ontology.html"><a href="gene-ontology.html#now-it-is-your-turn"><i class="fa fa-check"></i><b>2.6</b> Now it is your turn</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="gene-ontology.html"><a href="gene-ontology.html#run-enrichment"><i class="fa fa-check"></i><b>2.6.1</b> Run Enrichment</a></li>
<li class="chapter" data-level="2.6.2" data-path="gene-ontology.html"><a href="gene-ontology.html#see-the-data"><i class="fa fa-check"></i><b>2.6.2</b> see the data</a></li>
<li class="chapter" data-level="2.6.3" data-path="gene-ontology.html"><a href="gene-ontology.html#create-plots"><i class="fa fa-check"></i><b>2.6.3</b> create plots</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="gene-ontology.html"><a href="gene-ontology.html#questions"><i class="fa fa-check"></i><b>2.7</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html"><i class="fa fa-check"></i><b>3</b> Working with Sequences: Raw Data &amp; Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#description-1"><i class="fa fa-check"></i><b>3.1</b> Description</a></li>
<li class="chapter" data-level="3.2" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#learning-outcomes-2"><i class="fa fa-check"></i><b>3.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="3.3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#download-fastq"><i class="fa fa-check"></i><b>3.3</b> Download fastq</a></li>
<li class="chapter" data-level="3.4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#examining-fastq"><i class="fa fa-check"></i><b>3.4</b> Examining fastq</a></li>
<li class="chapter" data-level="3.5" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#trimming"><i class="fa fa-check"></i><b>3.5</b> Trimming</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#automate-for-list-of-files"><i class="fa fa-check"></i><b>3.5.1</b> Automate for list of files</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#batch-file-processing"><i class="fa fa-check"></i><b>3.6</b> Batch file processing</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#running-rfastp-creates-several-files"><i class="fa fa-check"></i><b>3.6.1</b> Running RfastP creates several files:</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#qc-and-adapters"><i class="fa fa-check"></i><b>3.7</b> QC and adapters</a></li>
<li class="chapter" data-level="3.8" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#running-fastqc"><i class="fa fa-check"></i><b>3.8</b> Running fastqc</a></li>
<li class="chapter" data-level="3.9" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#multiqc-for-qc-on-mutliple-samples"><i class="fa fa-check"></i><b>3.9</b> Multiqc for QC on mutliple samples</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="read-mapping.html"><a href="read-mapping.html"><i class="fa fa-check"></i><b>4</b> Read Mapping</a>
<ul>
<li class="chapter" data-level="4.1" data-path="read-mapping.html"><a href="read-mapping.html#alignment"><i class="fa fa-check"></i><b>4.1</b> Alignment</a></li>
<li class="chapter" data-level="4.2" data-path="read-mapping.html"><a href="read-mapping.html#retrieve-the-genome"><i class="fa fa-check"></i><b>4.2</b> Retrieve the genome</a></li>
<li class="chapter" data-level="4.3" data-path="read-mapping.html"><a href="read-mapping.html#build-rsubread-index"><i class="fa fa-check"></i><b>4.3</b> Build Rsubread Index</a></li>
<li class="chapter" data-level="4.4" data-path="read-mapping.html"><a href="read-mapping.html#pseudomapping-with-salmon"><i class="fa fa-check"></i><b>4.4</b> Pseudomapping with Salmon</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="read-mapping.html"><a href="read-mapping.html#create-conda-env"><i class="fa fa-check"></i><b>4.4.1</b> Create Conda Env</a></li>
<li class="chapter" data-level="4.4.2" data-path="read-mapping.html"><a href="read-mapping.html#download-transcriptome"><i class="fa fa-check"></i><b>4.4.2</b> Download transcriptome</a></li>
<li class="chapter" data-level="4.4.3" data-path="read-mapping.html"><a href="read-mapping.html#building-the-salmon-index"><i class="fa fa-check"></i><b>4.4.3</b> Building the Salmon index</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="read-mapping.html"><a href="read-mapping.html#questions-1"><i class="fa fa-check"></i><b>4.5</b> Questions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="read-mapping.html"><a href="read-mapping.html#with-rsubread"><i class="fa fa-check"></i><b>4.5.1</b> With Rsubread:</a></li>
<li class="chapter" data-level="4.5.2" data-path="read-mapping.html"><a href="read-mapping.html#with-salmon"><i class="fa fa-check"></i><b>4.5.2</b> With Salmon:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="read-counting.html"><a href="read-counting.html"><i class="fa fa-check"></i><b>5</b> Read Counting</a>
<ul>
<li class="chapter" data-level="5.1" data-path="read-counting.html"><a href="read-counting.html#featurecounts"><i class="fa fa-check"></i><b>5.1</b> featureCounts</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="read-counting.html"><a href="read-counting.html#locate-bam-files"><i class="fa fa-check"></i><b>5.1.1</b> Locate BAM files</a></li>
<li class="chapter" data-level="5.1.2" data-path="read-counting.html"><a href="read-counting.html#retrieve-the-genome-annotation"><i class="fa fa-check"></i><b>5.1.2</b> Retrieve the genome annotation</a></li>
<li class="chapter" data-level="5.1.3" data-path="read-counting.html"><a href="read-counting.html#counting-with-featurecounts"><i class="fa fa-check"></i><b>5.1.3</b> Counting with FeatureCounts</a></li>
<li class="chapter" data-level="5.1.4" data-path="read-counting.html"><a href="read-counting.html#counts-object"><i class="fa fa-check"></i><b>5.1.4</b> Counts object</a></li>
<li class="chapter" data-level="5.1.5" data-path="read-counting.html"><a href="read-counting.html#saving-fc-object-for-future-use"><i class="fa fa-check"></i><b>5.1.5</b> Saving <code>fc</code> object for future use</a></li>
<li class="chapter" data-level="5.1.6" data-path="read-counting.html"><a href="read-counting.html#rsubread-qc"><i class="fa fa-check"></i><b>5.1.6</b> RSubread QC</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="read-counting.html"><a href="read-counting.html#salmon"><i class="fa fa-check"></i><b>5.2</b> Salmon</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="read-counting.html"><a href="read-counting.html#pseudomapping-counting"><i class="fa fa-check"></i><b>5.2.1</b> Pseudomapping &amp; Counting</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="read-counting.html"><a href="read-counting.html#questions-2"><i class="fa fa-check"></i><b>5.3</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html"><i class="fa fa-check"></i><b>6</b> Differential Expression: EdgeR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#description-2"><i class="fa fa-check"></i><b>6.1</b> Description</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#learning-outcomes-3"><i class="fa fa-check"></i><b>6.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="6.3" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#loading-in-the-featurecounts-object"><i class="fa fa-check"></i><b>6.3</b> Loading in the featureCounts object</a></li>
<li class="chapter" data-level="6.4" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#count-loading-and-annotation"><i class="fa fa-check"></i><b>6.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="6.5" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#filtering-to-remove-low-counts"><i class="fa fa-check"></i><b>6.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="6.6" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#normalization-for-composition-bias"><i class="fa fa-check"></i><b>6.6</b> Normalization for composition bias</a></li>
<li class="chapter" data-level="6.7" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#mds-plots"><i class="fa fa-check"></i><b>6.7</b> MDS plots</a></li>
<li class="chapter" data-level="6.8" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#exploring-differences-between-libraries"><i class="fa fa-check"></i><b>6.8</b> Exploring differences between libraries</a></li>
<li class="chapter" data-level="6.9" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#estimate-dispersion"><i class="fa fa-check"></i><b>6.9</b> Estimate Dispersion</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#testing-for-differential-expression"><i class="fa fa-check"></i><b>6.10</b> Testing for differential expression</a></li>
<li class="chapter" data-level="6.11" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#looking-at-all-contrasts-at-once"><i class="fa fa-check"></i><b>6.11</b> Looking at all contrasts at once</a></li>
<li class="chapter" data-level="6.12" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#questions-3"><i class="fa fa-check"></i><b>6.12</b> Questions</a></li>
<li class="chapter" data-level="6.13" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#a-template-set-of-code-chunks-for-doing-this-is-below"><i class="fa fa-check"></i><b>6.13</b> A template set of code chunks for doing this is below:</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html"><i class="fa fa-check"></i><b>7</b> Differential Expression: DESeq2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#description-3"><i class="fa fa-check"></i><b>7.1</b> Description</a></li>
<li class="chapter" data-level="7.2" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#learning-outcomes-4"><i class="fa fa-check"></i><b>7.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="7.3" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#loading-in-the-featurecounts-object-1"><i class="fa fa-check"></i><b>7.3</b> Loading in the featureCounts object</a></li>
<li class="chapter" data-level="7.4" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#count-loading-and-annotation-1"><i class="fa fa-check"></i><b>7.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="7.5" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#filtering-to-remove-low-counts-1"><i class="fa fa-check"></i><b>7.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="7.6" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#testing-for-differential-expression-1"><i class="fa fa-check"></i><b>7.6</b> Testing for differential expression</a></li>
<li class="chapter" data-level="7.7" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#questions-4"><i class="fa fa-check"></i><b>7.7</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html"><i class="fa fa-check"></i><b>8</b> Differential Expression: limma</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#description-4"><i class="fa fa-check"></i><b>8.1</b> Description</a></li>
<li class="chapter" data-level="8.2" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#learning-objectives"><i class="fa fa-check"></i><b>8.2</b> Learning Objectives</a></li>
<li class="chapter" data-level="8.3" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#loading-in-the-count-data-file"><i class="fa fa-check"></i><b>8.3</b> Loading in the count data file</a></li>
<li class="chapter" data-level="8.4" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#count-loading-and-annotation-2"><i class="fa fa-check"></i><b>8.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="8.5" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#filtering-to-remove-low-counts-2"><i class="fa fa-check"></i><b>8.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="8.6" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#normalization-for-composition-bias-1"><i class="fa fa-check"></i><b>8.6</b> Normalization for composition bias</a></li>
<li class="chapter" data-level="8.7" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#exploring-differences-between-libraries-1"><i class="fa fa-check"></i><b>8.7</b> Exploring differences between libraries</a></li>
<li class="chapter" data-level="8.8" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#estimate-dispersion-1"><i class="fa fa-check"></i><b>8.8</b> Estimate Dispersion</a></li>
<li class="chapter" data-level="8.9" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#testing-for-differential-expression-2"><i class="fa fa-check"></i><b>8.9</b> Testing for differential expression</a></li>
<li class="chapter" data-level="8.10" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#examining-a-specific-contrast"><i class="fa fa-check"></i><b>8.10</b> Examining a specific contrast</a></li>
<li class="chapter" data-level="8.11" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#visualization"><i class="fa fa-check"></i><b>8.11</b> Visualization</a>
<ul>
<li class="chapter" data-level="8.11.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#ma-lot"><i class="fa fa-check"></i><b>8.11.1</b> MA lot</a></li>
</ul></li>
<li class="chapter" data-level="8.12" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#treat-testing"><i class="fa fa-check"></i><b>8.12</b> <code>treat()</code> testing</a>
<ul>
<li class="chapter" data-level="8.12.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#visualize-de-genes-from-treat-using-lfc1"><i class="fa fa-check"></i><b>8.12.1</b> Visualize DE genes from <code>Treat</code> using lfc=1</a></li>
</ul></li>
<li class="chapter" data-level="8.13" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#comparing-de-analysis-softwares"><i class="fa fa-check"></i><b>8.13</b> Comparing DE analysis softwares</a></li>
<li class="chapter" data-level="8.14" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#correlation-between-logfc-estimates-across-softwares"><i class="fa fa-check"></i><b>8.14</b> Correlation between logFC estimates across softwares</a></li>
<li class="chapter" data-level="8.15" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#questions-5"><i class="fa fa-check"></i><b>8.15</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html"><i class="fa fa-check"></i><b>9</b> Visualizing Differential Expression Results</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#description-5"><i class="fa fa-check"></i><b>9.1</b> Description</a></li>
<li class="chapter" data-level="9.2" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#learning-outcomes-5"><i class="fa fa-check"></i><b>9.2</b> Learning Outcomes</a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#ma-plot"><i class="fa fa-check"></i><b>9.3</b> MA-plot</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#volcano-plot"><i class="fa fa-check"></i><b>9.4</b> Volcano Plot</a></li>
<li class="chapter" data-level="9.5" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#using-glimma-for-an-interactive-visualization"><i class="fa fa-check"></i><b>9.5</b> Using Glimma for an interactive visualization</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#ma-plots"><i class="fa fa-check"></i><b>9.5.1</b> MA plots</a></li>
<li class="chapter" data-level="9.5.2" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#volcano-plots"><i class="fa fa-check"></i><b>9.5.2</b> Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#generating-bar-graph-summaries"><i class="fa fa-check"></i><b>9.6</b> Generating bar graph summaries</a></li>
<li class="chapter" data-level="9.7" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#exercise"><i class="fa fa-check"></i><b>9.7</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Clustering</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#description-6"><i class="fa fa-check"></i><b>10.1</b> Description</a></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#learning-outcomes-6"><i class="fa fa-check"></i><b>10.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#cluster-3.0"><i class="fa fa-check"></i><b>10.3</b> Cluster 3.0</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="clustering.html"><a href="clustering.html#generating-a-pcl-file"><i class="fa fa-check"></i><b>10.3.1</b> Generating a PCL file</a></li>
<li class="chapter" data-level="10.3.2" data-path="clustering.html"><a href="clustering.html#filtering-data"><i class="fa fa-check"></i><b>10.3.2</b> Filtering data</a></li>
<li class="chapter" data-level="10.3.3" data-path="clustering.html"><a href="clustering.html#hierarchical-clustering"><i class="fa fa-check"></i><b>10.3.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="10.3.4" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.3.4</b> <em>K</em>-Means Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#visualizing-clusters-with-java-treeview"><i class="fa fa-check"></i><b>10.4</b> Visualizing Clusters with Java TreeView</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="clustering.html"><a href="clustering.html#changing-pixel-settings."><i class="fa fa-check"></i><b>10.4.1</b> Changing pixel settings.</a></li>
<li class="chapter" data-level="10.4.2" data-path="clustering.html"><a href="clustering.html#selecting-groups-of-genes."><i class="fa fa-check"></i><b>10.4.2</b> Selecting groups of genes.</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#performing-clustering-on-yeast-stress-data"><i class="fa fa-check"></i><b>10.5</b> Performing clustering on yeast stress data</a></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#questions-6"><i class="fa fa-check"></i><b>10.6</b> Questions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Genomic Data Analysis Course Exercises</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gene-ontology" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Gene Ontology<a href="gene-ontology.html#gene-ontology" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>last updated: 2023-10-26</p>
<p><strong>Installing Packages</strong></p>
<p>The following code installs all of the packages used in this document – if not already installed – and then loads the packages into R. We need to install packages specific to our gene ontology bioinformatic analysis. Many of these packages aren’t available on the R CRAN package repository, instead they are hosted on BioConductor repository that is focused on packages used in biological research. Today, we need to install the package clusterProfiler with the code below. The <code>p_load()</code> function will check the bioconductor repository if the package isn’t on CRAN</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="gene-ontology.html#cb16-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;pacman&quot;</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;pacman&quot;</span>); <span class="fu">library</span>(pacman)</span>
<span id="cb16-2"><a href="gene-ontology.html#cb16-2" tabindex="-1"></a></span>
<span id="cb16-3"><a href="gene-ontology.html#cb16-3" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;here&quot;</span>, <span class="st">&quot;knitr&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="co"># already downloaded last activity</span></span>
<span id="cb16-4"><a href="gene-ontology.html#cb16-4" tabindex="-1"></a>       <span class="st">&quot;readr&quot;</span>,<span class="st">&quot;pander&quot;</span>, <span class="st">&quot;BiocManager&quot;</span>, <span class="co"># also from last activity</span></span>
<span id="cb16-5"><a href="gene-ontology.html#cb16-5" tabindex="-1"></a>       <span class="st">&quot;janitor&quot;</span>, <span class="co"># for cleaning column names</span></span>
<span id="cb16-6"><a href="gene-ontology.html#cb16-6" tabindex="-1"></a>       <span class="st">&quot;igraph&quot;</span>, <span class="st">&quot;tidytree&quot;</span>, <span class="co"># dependencies that require explicit download on latest Mac OS</span></span>
<span id="cb16-7"><a href="gene-ontology.html#cb16-7" tabindex="-1"></a>       <span class="st">&quot;ggVennDiagram&quot;</span>, <span class="co"># visualization venn diagram</span></span>
<span id="cb16-8"><a href="gene-ontology.html#cb16-8" tabindex="-1"></a>       <span class="st">&quot;clusterProfiler&quot;</span>, <span class="co"># for GO enrichment</span></span>
<span id="cb16-9"><a href="gene-ontology.html#cb16-9" tabindex="-1"></a>       <span class="st">&quot;AnnotationDbi&quot;</span>, <span class="co"># database of common genome annotations</span></span>
<span id="cb16-10"><a href="gene-ontology.html#cb16-10" tabindex="-1"></a>       <span class="st">&quot;org.Sc.sgd.db&quot;</span> <span class="co"># annotation database for S. cerevesiae</span></span>
<span id="cb16-11"><a href="gene-ontology.html#cb16-11" tabindex="-1"></a>       )</span>
<span id="cb16-12"><a href="gene-ontology.html#cb16-12" tabindex="-1"></a></span>
<span id="cb16-13"><a href="gene-ontology.html#cb16-13" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<div id="description" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Description<a href="gene-ontology.html#description" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This activity is intended to familiarize you with Gene Ontology analysis and some of the unique challenges that come from working with bioinformatic data.</p>
</div>
<div id="learning-outcomes-1" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Learning outcomes<a href="gene-ontology.html#learning-outcomes-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>At the end of this exercise, you should be able to:</p>
<ul>
<li>Understand gene ontology and its significance in functional annotation</li>
<li>learn to perform a GO enrichment &amp; appropriate statistical methods (hypergeometric &amp; Fisher’s exact test) for the enrichment analysis</li>
<li>interpret &amp; critically evaluate the results of GO enrichment &amp; limitations/challenges</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="gene-ontology.html#cb17-1" tabindex="-1"></a><span class="co"># we don&#39;t have to run this, but if you install without pacman, we have to do load libraries</span></span>
<span id="cb17-2"><a href="gene-ontology.html#cb17-2" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb17-3"><a href="gene-ontology.html#cb17-3" tabindex="-1"></a><span class="fu">library</span>(org.Sc.sgd.db)</span></code></pre></div>
</div>
<div id="analysis-workflow" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Analysis Workflow<a href="gene-ontology.html#analysis-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s use the same file from last class, this time performing GO term enrichment</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="gene-ontology.html#cb18-1" tabindex="-1"></a><span class="co"># assign url to a variable</span></span>
<span id="cb18-2"><a href="gene-ontology.html#cb18-2" tabindex="-1"></a>DE_data_url <span class="ot">&lt;-</span> <span class="st">&quot;https://raw.githubusercontent.com/clstacy/GenomicDataAnalysis_Fa23/main/data/ethanol_stress/msn2-4_mutants_EtOH.txt&quot;</span></span>
<span id="cb18-3"><a href="gene-ontology.html#cb18-3" tabindex="-1"></a></span>
<span id="cb18-4"><a href="gene-ontology.html#cb18-4" tabindex="-1"></a><span class="co"># download the data from the web</span></span>
<span id="cb18-5"><a href="gene-ontology.html#cb18-5" tabindex="-1"></a>DE_results_msn24_EtOH <span class="ot">&lt;-</span></span>
<span id="cb18-6"><a href="gene-ontology.html#cb18-6" tabindex="-1"></a>  <span class="fu">read_tsv</span>(<span class="at">file=</span>DE_data_url)</span></code></pre></div>
<pre><code>## Warning: One or more parsing issues, call `problems()` on your data frame for details,
## e.g.:
##   dat &lt;- vroom(...)
##   problems(dat)</code></pre>
<pre><code>## Rows: 5756 Columns: 18
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: &quot;\t&quot;
## chr  (3): Gene ID, Common Name, Annotation
## dbl (15): logFC: YPS606 (WT) EtOH response, Pvalue: YPS606 (WT) EtOH respons...
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="gene-ontology.html#cb21-1" tabindex="-1"></a>msn24_EtOH <span class="ot">&lt;-</span> <span class="co"># assign a new object name</span></span>
<span id="cb21-2"><a href="gene-ontology.html#cb21-2" tabindex="-1"></a>  DE_results_msn24_EtOH <span class="sc">|&gt;</span> <span class="co"># our object with messy names</span></span>
<span id="cb21-3"><a href="gene-ontology.html#cb21-3" tabindex="-1"></a>  <span class="fu">clean_names</span>() <span class="co"># function from janitor package to make names consistent</span></span></code></pre></div>
</div>
<div id="get-de-gene-list" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Get DE gene list<a href="gene-ontology.html#get-de-gene-list" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We need a list of deferentially expressed genes to test for over or under enrichment of terms. Here we choose genes with significantly (FDR&lt;0.05) higher expression (log<sub>2</sub>-fold change (logFC) greater than 1) in the <em>msn2/4ΔΔ</em> mutant’s EtOH response compared to the wild-type strains EtOH response (positive values in the logFC column of WT vs <em>msn2/4ΔΔ</em>: EtOH response).</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="gene-ontology.html#cb22-1" tabindex="-1"></a><span class="co"># subset to just genes with significant fdr &amp; log2FC&gt;1</span></span>
<span id="cb22-2"><a href="gene-ontology.html#cb22-2" tabindex="-1"></a>msn24_EtOH <span class="sc">|&gt;</span></span>
<span id="cb22-3"><a href="gene-ontology.html#cb22-3" tabindex="-1"></a>  <span class="fu">filter</span>(log_fc_wt_v_msn2_4dd_et_oh_response <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> fdr_wt_v_msn2_4dd_et_oh_response <span class="sc">&lt;</span> <span class="fl">0.05</span>) </span></code></pre></div>
<pre><code>## # A tibble: 94 × 18
##    gene_id  common_name annotation log_fc_yps606_wt_et_…¹ pvalue_yps606_wt_et_…²
##    &lt;chr&gt;    &lt;chr&gt;       &lt;chr&gt;                       &lt;dbl&gt;                  &lt;dbl&gt;
##  1 YOR315W  SFG1        Putative …                  -5.52               4.33e-31
##  2 YFL051C  YFL051C     &lt;NA&gt;                        -4.54               6.37e-27
##  3 YMR016C  SOK2        Nuclear p…                  -3.09               1.32e-32
##  4 YPL061W  ALD6        Cytosolic…                  -7.04               4.96e-26
##  5 YER073W  ALD5        Mitochond…                  -1.88               6.14e-17
##  6 YBL005W… YBL005W-B   Retrotran…                   1.91               5.46e-13
##  7 YBL039C  URA7        Major CTP…                  -6.95               3.62e-41
##  8 YJL050W  MTR4        RNA duple…                  -4.59               4.73e-36
##  9 YMR241W  YHM2        Citrate a…                  -1.72               1.72e-20
## 10 YIL131C  FKH1        Forkhead …                  -2.20               1.23e-27
## # ℹ 84 more rows
## # ℹ abbreviated names: ¹​log_fc_yps606_wt_et_oh_response,
## #   ²​pvalue_yps606_wt_et_oh_response
## # ℹ 13 more variables: fdr_yps606_wt_et_oh_response &lt;dbl&gt;,
## #   log_fc_yps606_msn2_4dd_et_oh_response &lt;dbl&gt;,
## #   pvalue_yps606_msn2_4dd_et_oh_response &lt;dbl&gt;,
## #   fdr_yps606_msn2_4dd_et_oh_response &lt;dbl&gt;, …</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="gene-ontology.html#cb24-1" tabindex="-1"></a><span class="co"># the above command gave us what we want, here it is again but saved to a new variable:</span></span>
<span id="cb24-2"><a href="gene-ontology.html#cb24-2" tabindex="-1"></a>DE_genes_upregulated_msn24_EtOH <span class="ot">&lt;-</span> </span>
<span id="cb24-3"><a href="gene-ontology.html#cb24-3" tabindex="-1"></a>  msn24_EtOH <span class="sc">|&gt;</span></span>
<span id="cb24-4"><a href="gene-ontology.html#cb24-4" tabindex="-1"></a>  <span class="fu">filter</span>(log_fc_wt_v_msn2_4dd_et_oh_response <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> fdr_wt_v_msn2_4dd_et_oh_response <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb24-5"><a href="gene-ontology.html#cb24-5" tabindex="-1"></a>  <span class="fu">pull</span>(gene_id) <span class="co"># get just the gene names</span></span></code></pre></div>
<p>Now we have a list of genes (saved as DE_genes_upregulated_msn24_EtOH) that we want to perform GO term enrichment on. Let’s do that now, using the clusterProfiler package’s <code>enrichGO</code> function</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="gene-ontology.html#cb25-1" tabindex="-1"></a>GO_msn24_EtOH_up_results <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(</span>
<span id="cb25-2"><a href="gene-ontology.html#cb25-2" tabindex="-1"></a>  <span class="at">gene =</span> DE_genes_upregulated_msn24_EtOH,</span>
<span id="cb25-3"><a href="gene-ontology.html#cb25-3" tabindex="-1"></a>  <span class="at">OrgDb =</span> <span class="st">&quot;org.Sc.sgd.db&quot;</span>,</span>
<span id="cb25-4"><a href="gene-ontology.html#cb25-4" tabindex="-1"></a>  <span class="at">universe =</span> msn24_EtOH<span class="sc">$</span>gene_id,</span>
<span id="cb25-5"><a href="gene-ontology.html#cb25-5" tabindex="-1"></a>  <span class="at">keyType =</span> <span class="st">&quot;ORF&quot;</span>,</span>
<span id="cb25-6"><a href="gene-ontology.html#cb25-6" tabindex="-1"></a>  <span class="at">ont=</span> <span class="st">&quot;BP&quot;</span></span>
<span id="cb25-7"><a href="gene-ontology.html#cb25-7" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb25-8"><a href="gene-ontology.html#cb25-8" tabindex="-1"></a>  <span class="co"># let&#39;s add a &#39;richFactor&#39; column that gives us the proportion of genes DE in the term</span></span>
<span id="cb25-9"><a href="gene-ontology.html#cb25-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">richFactor =</span> Count <span class="sc">/</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">&quot;/</span><span class="sc">\\</span><span class="st">d+&quot;</span>, <span class="st">&quot;&quot;</span>, BgRatio)))</span></code></pre></div>
<p>Now, we can look at the results in table form.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="gene-ontology.html#cb26-1" tabindex="-1"></a><span class="co"># open up the results in a data frame to examine</span></span>
<span id="cb26-2"><a href="gene-ontology.html#cb26-2" tabindex="-1"></a>GO_msn24_EtOH_up_results <span class="sc">|&gt;</span></span>
<span id="cb26-3"><a href="gene-ontology.html#cb26-3" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb26-4"><a href="gene-ontology.html#cb26-4" tabindex="-1"></a>  <span class="fu">View</span>()</span>
<span id="cb26-5"><a href="gene-ontology.html#cb26-5" tabindex="-1"></a></span>
<span id="cb26-6"><a href="gene-ontology.html#cb26-6" tabindex="-1"></a><span class="co"># Here is how we could write this result into a text file:</span></span>
<span id="cb26-7"><a href="gene-ontology.html#cb26-7" tabindex="-1"></a>GO_msn24_EtOH_up_results <span class="sc">|&gt;</span></span>
<span id="cb26-8"><a href="gene-ontology.html#cb26-8" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb26-9"><a href="gene-ontology.html#cb26-9" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="st">&quot;~/Desktop/GO_msn24_EtOH_up_results.tsv&quot;</span>)</span></code></pre></div>
<p>Now we can visualize the enrichment results, which shows us gene ontology categories that are enriched in genes with higher expression (upregulated) in the WT vs <em>msn2/4ΔΔ</em>: EtOH response.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="gene-ontology.html#cb27-1" tabindex="-1"></a><span class="co"># a simple visualization</span></span>
<span id="cb27-2"><a href="gene-ontology.html#cb27-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">barplot</span>(GO_msn24_EtOH_up_results, <span class="at">showCategory =</span> <span class="dv">10</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/visualize-GOterms_upregulated_phenol-GO-1.png" width="672" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="gene-ontology.html#cb28-1" tabindex="-1"></a><span class="co"># a more complicated visualization, with more information density</span></span>
<span id="cb28-2"><a href="gene-ontology.html#cb28-2" tabindex="-1"></a><span class="fu">ggplot</span>(GO_msn24_EtOH_up_results,</span>
<span id="cb28-3"><a href="gene-ontology.html#cb28-3" tabindex="-1"></a>       <span class="at">showCategory =</span> <span class="dv">15</span>,</span>
<span id="cb28-4"><a href="gene-ontology.html#cb28-4" tabindex="-1"></a>       <span class="fu">aes</span>(richFactor, <span class="fu">fct_reorder</span>(Description, richFactor))) <span class="sc">+</span></span>
<span id="cb28-5"><a href="gene-ontology.html#cb28-5" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">xend =</span> <span class="dv">0</span>, <span class="at">yend =</span> Description)) <span class="sc">+</span></span>
<span id="cb28-6"><a href="gene-ontology.html#cb28-6" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> p.adjust, <span class="at">size =</span> Count)) <span class="sc">+</span></span>
<span id="cb28-7"><a href="gene-ontology.html#cb28-7" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(</span>
<span id="cb28-8"><a href="gene-ontology.html#cb28-8" tabindex="-1"></a>    <span class="at">colours =</span> <span class="fu">c</span>(<span class="st">&quot;#f7ca64&quot;</span>, <span class="st">&quot;#46bac2&quot;</span>, <span class="st">&quot;#7e62a3&quot;</span>),</span>
<span id="cb28-9"><a href="gene-ontology.html#cb28-9" tabindex="-1"></a>    <span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>,</span>
<span id="cb28-10"><a href="gene-ontology.html#cb28-10" tabindex="-1"></a>    <span class="at">guide =</span> <span class="fu">guide_colorbar</span>(<span class="at">reverse =</span> <span class="cn">TRUE</span>, <span class="at">order =</span> <span class="dv">1</span>)</span>
<span id="cb28-11"><a href="gene-ontology.html#cb28-11" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb28-12"><a href="gene-ontology.html#cb28-12" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb28-13"><a href="gene-ontology.html#cb28-13" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Rich Factor&quot;</span>) <span class="sc">+</span></span>
<span id="cb28-14"><a href="gene-ontology.html#cb28-14" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb28-15"><a href="gene-ontology.html#cb28-15" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Biological Processes&quot;</span>) <span class="sc">+</span></span>
<span id="cb28-16"><a href="gene-ontology.html#cb28-16" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/visualize-GOterms_upregulated_phenol-GO-2.png" width="672" /></p>
<p>You can try adjusting the size of the output figures by clicking the gear icon in the top right of the code chunk and click “use custom figure size”. Note this updates the chunk header so the change is saved.</p>
<div id="saving-ggplot-output-to-a-file" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Saving ggplot output to a file<a href="gene-ontology.html#saving-ggplot-output-to-a-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We usually want to save our visualizations for later. When plotting with the ggplot package, there is an easy way to do this. See below:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="gene-ontology.html#cb29-1" tabindex="-1"></a><span class="co"># First, let&#39;s create a folder to save our visualizations</span></span>
<span id="cb29-2"><a href="gene-ontology.html#cb29-2" tabindex="-1"></a>dir_visualization <span class="ot">&lt;-</span> <span class="fu">path.expand</span>(<span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Visualization/&quot;</span>)</span>
<span id="cb29-3"><a href="gene-ontology.html#cb29-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(dir_visualization)) {<span class="fu">dir.create</span>(dir_visualization, <span class="at">recursive =</span> <span class="cn">TRUE</span>)}</span>
<span id="cb29-4"><a href="gene-ontology.html#cb29-4" tabindex="-1"></a></span>
<span id="cb29-5"><a href="gene-ontology.html#cb29-5" tabindex="-1"></a><span class="co"># type ?ggsave in the console for more information via the help page.</span></span>
<span id="cb29-6"><a href="gene-ontology.html#cb29-6" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb29-7"><a href="gene-ontology.html#cb29-7" tabindex="-1"></a>  <span class="st">&quot;GO_BP_msn24_EtOH_up_results_lollipopPlot.pdf&quot;</span>, </span>
<span id="cb29-8"><a href="gene-ontology.html#cb29-8" tabindex="-1"></a>  <span class="co"># if we don&#39;t need the image to go to a certain spot, we only need the file name above.</span></span>
<span id="cb29-9"><a href="gene-ontology.html#cb29-9" tabindex="-1"></a>  <span class="at">plot =</span> <span class="fu">last_plot</span>(), <span class="co"># either the last plot, or name of a ggplot object you&#39;ve saved.</span></span>
<span id="cb29-10"><a href="gene-ontology.html#cb29-10" tabindex="-1"></a>  <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>, <span class="co">#Can be &quot;png&quot;, &quot;eps&quot;, &quot;ps&quot;, &quot;tex&quot; (pictex), &quot;pdf&quot;, &quot;jpeg&quot;, &quot;tiff&quot;, &quot;png&quot;, &quot;bmp&quot;, &quot;svg&quot; or &quot;wmf&quot; (windows only).</span></span>
<span id="cb29-11"><a href="gene-ontology.html#cb29-11" tabindex="-1"></a>  <span class="co"># note that pdf, eps, svg are vector/line art, so zooming doesn&#39;t pixelate.</span></span>
<span id="cb29-12"><a href="gene-ontology.html#cb29-12" tabindex="-1"></a>  <span class="at">path =</span> dir_visualization, <span class="co"># Path of the directory to save plot to. defaults to work dir.</span></span>
<span id="cb29-13"><a href="gene-ontology.html#cb29-13" tabindex="-1"></a>  <span class="at">scale =</span> <span class="dv">2</span>, <span class="co"># multiplicative scaling factor </span></span>
<span id="cb29-14"><a href="gene-ontology.html#cb29-14" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">12</span>,</span>
<span id="cb29-15"><a href="gene-ontology.html#cb29-15" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">8</span>,</span>
<span id="cb29-16"><a href="gene-ontology.html#cb29-16" tabindex="-1"></a>  <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>, <span class="co"># must be one of: &quot;in&quot;, &quot;cm&quot;, &quot;mm&quot;, &quot;px&quot;</span></span>
<span id="cb29-17"><a href="gene-ontology.html#cb29-17" tabindex="-1"></a>  <span class="at">dpi =</span> <span class="dv">300</span>,  <span class="co"># adjusting this larger gives higher quality plot, making a larger file.</span></span>
<span id="cb29-18"><a href="gene-ontology.html#cb29-18" tabindex="-1"></a>  <span class="at">limitsize =</span> <span class="cn">TRUE</span>, <span class="co"># prevents accidentally making it massive, defaults to TRUE</span></span>
<span id="cb29-19"><a href="gene-ontology.html#cb29-19" tabindex="-1"></a>  <span class="at">bg =</span> <span class="cn">NULL</span> <span class="co"># Background colour. If NULL, uses the plot.background fill value from the plot theme.</span></span>
<span id="cb29-20"><a href="gene-ontology.html#cb29-20" tabindex="-1"></a>)</span></code></pre></div>
<p>Recall that when we knit this Rmarkdown notebook, we keep a copy of the plots/images there as well, in the same place as the code and analysis used to generate it. However, we may want a higher resolution file of just the image, or the image in a different format. In this case, saving the plot is a useful option for us. The journal Science has the following <a href="https://www.science.org/do/10.5555/page.2385607/full/author_figure_prep_guide_2022-1689707679870.pdf">recommendations</a>: “We prefer prefer ai, eps, pdf, layered psd, tif, and jpeg files. …minimum file resolution of 300 dpi.”</p>
</div>
</div>
<div id="the-hypergeometric-distribution-in-practice" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> The Hypergeometric Distribution in practice<a href="gene-ontology.html#the-hypergeometric-distribution-in-practice" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Notice that the DNA integration process does not have very many genes in the category, but they appear to be highly present in the the upregulated gene list. Specifically, DE genes have this GO term, where in the entire genome, there are only genes. What are the odds that we see this by random chance? let’s do the math:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="gene-ontology.html#cb30-1" tabindex="-1"></a><span class="co"># number of genes that have GO:0015074 (DNA integration)</span></span>
<span id="cb30-2"><a href="gene-ontology.html#cb30-2" tabindex="-1"></a>integration_genes <span class="ot">=</span> <span class="dv">23</span></span>
<span id="cb30-3"><a href="gene-ontology.html#cb30-3" tabindex="-1"></a><span class="co"># number of genes that are DE (msn2/4 EtOH response, logFC&gt;1)</span></span>
<span id="cb30-4"><a href="gene-ontology.html#cb30-4" tabindex="-1"></a>DE_genes <span class="ot">=</span> <span class="dv">91</span></span>
<span id="cb30-5"><a href="gene-ontology.html#cb30-5" tabindex="-1"></a><span class="co"># number of genes that are both DE and DNA integration genes</span></span>
<span id="cb30-6"><a href="gene-ontology.html#cb30-6" tabindex="-1"></a>Overlap <span class="ot">=</span> <span class="dv">6</span></span>
<span id="cb30-7"><a href="gene-ontology.html#cb30-7" tabindex="-1"></a><span class="co"># total number of genes in experiment</span></span>
<span id="cb30-8"><a href="gene-ontology.html#cb30-8" tabindex="-1"></a>total <span class="ot">=</span> <span class="dv">5538</span> <span class="co"># number of genes in genome</span></span></code></pre></div>
<p><strong>Without doing the math, do you expect these to be underrepresented, overrepresented, or neither?</strong></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="gene-ontology.html#cb31-1" tabindex="-1"></a><span class="co"># test for underrepresentation (depletion)</span></span>
<span id="cb31-2"><a href="gene-ontology.html#cb31-2" tabindex="-1"></a><span class="fu">phyper</span>(<span class="at">q =</span> Overlap, <span class="co"># number of integration genes that were DE</span></span>
<span id="cb31-3"><a href="gene-ontology.html#cb31-3" tabindex="-1"></a>       <span class="at">m =</span> DE_genes, <span class="co"># number of DE genes</span></span>
<span id="cb31-4"><a href="gene-ontology.html#cb31-4" tabindex="-1"></a>       <span class="at">n =</span> total<span class="sc">-</span>DE_genes, <span class="co"># number of non DE genes</span></span>
<span id="cb31-5"><a href="gene-ontology.html#cb31-5" tabindex="-1"></a>       <span class="at">k =</span> integration_genes, <span class="co"># number of observed DE DNA integration genes</span></span>
<span id="cb31-6"><a href="gene-ontology.html#cb31-6" tabindex="-1"></a>       <span class="at">lower.tail =</span> <span class="cn">TRUE</span>) <span class="co"># the probability that X &lt;= x</span></span></code></pre></div>
<pre><code>## [1] 0.9999999</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="gene-ontology.html#cb33-1" tabindex="-1"></a><span class="co"># test for overrepresentation (enrichmen t)</span></span>
<span id="cb33-2"><a href="gene-ontology.html#cb33-2" tabindex="-1"></a><span class="fu">phyper</span>(<span class="at">q =</span> Overlap<span class="dv">-1</span>, <span class="co"># number of integration genes that were DE</span></span>
<span id="cb33-3"><a href="gene-ontology.html#cb33-3" tabindex="-1"></a>                      <span class="co"># we subtract 1 b/c of lower.tail=FALSE means greater than</span></span>
<span id="cb33-4"><a href="gene-ontology.html#cb33-4" tabindex="-1"></a>                      <span class="co"># without equality, so have to do one less</span></span>
<span id="cb33-5"><a href="gene-ontology.html#cb33-5" tabindex="-1"></a>       <span class="at">m =</span> DE_genes, <span class="co"># number of DE genes</span></span>
<span id="cb33-6"><a href="gene-ontology.html#cb33-6" tabindex="-1"></a>       <span class="at">n =</span> total<span class="sc">-</span>DE_genes, <span class="co"># number of non DE genes</span></span>
<span id="cb33-7"><a href="gene-ontology.html#cb33-7" tabindex="-1"></a>       <span class="at">k =</span> integration_genes, <span class="co"># number of observed DE integration genes</span></span>
<span id="cb33-8"><a href="gene-ontology.html#cb33-8" tabindex="-1"></a>       <span class="at">lower.tail =</span> <span class="cn">FALSE</span>) <span class="co"># the probability that X &gt; x</span></span></code></pre></div>
<pre><code>## [1] 1.344447e-06</code></pre>
<p>As we see, there is strong evidence that the number of genes with this GO term is unlikely to be seen due to chance. In layman’s terms, this GO term is enriched in upregulated genes in this contrast. The test for underrepresenation shows there is no support for a hypothesis that this gene is underrepresented in the DE gene list.</p>
<p>Interestingly, the hypergeometric distribution is the same thing as the Fisher’s Exact test, so we can rerun the same tests above with a different command:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="gene-ontology.html#cb35-1" tabindex="-1"></a><span class="co">#fisher test for underrepresentation</span></span>
<span id="cb35-2"><a href="gene-ontology.html#cb35-2" tabindex="-1"></a><span class="fu">fisher.test</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(Overlap, DE_genes<span class="sc">-</span>Overlap, integration_genes<span class="sc">-</span>Overlap, total<span class="sc">-</span>DE_genes<span class="sc">-</span>integration_genes <span class="sc">+</span> Overlap), <span class="dv">2</span>, <span class="dv">2</span>), <span class="at">alternative=</span><span class="st">&#39;less&#39;</span>)<span class="sc">$</span>p.value</span></code></pre></div>
<pre><code>## [1] 0.9999999</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="gene-ontology.html#cb37-1" tabindex="-1"></a><span class="co">#fisher test for overrepresentation</span></span>
<span id="cb37-2"><a href="gene-ontology.html#cb37-2" tabindex="-1"></a><span class="fu">fisher.test</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(Overlap, DE_genes<span class="sc">-</span>Overlap, integration_genes<span class="sc">-</span>Overlap, total<span class="sc">-</span>DE_genes<span class="sc">-</span>integration_genes <span class="sc">+</span> Overlap), <span class="dv">2</span>, <span class="dv">2</span>), <span class="at">alternative=</span><span class="st">&#39;greater&#39;</span>)<span class="sc">$</span>p.value</span></code></pre></div>
<pre><code>## [1] 1.344447e-06</code></pre>
<p>How does the p-value that we get from this test compare to the results table? They should match.</p>
<p><br><br><br><br></p>
</div>
<div id="now-it-is-your-turn" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Now it is your turn<a href="gene-ontology.html#now-it-is-your-turn" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Try running your own GO enrichment with a different gene list. Some options could be:</p>
<ul>
<li>Start with the WT vs <em>msn2/4ΔΔ</em>: EtOH response again, and this time change to “downregulated” (i.e., genes with higher expression in the wild-type strain compared to the <em>msn2/4ΔΔ</em> mutant). These would potentially include genes with defective induction.</li>
<li>See what happens when you change the FDR threshold from a liberal one (0.05) to a more conservative one (0.01).</li>
<li>Try different logFC cutoffs.</li>
<li>Look at different comparisons in the data file (there are 5 total)</li>
<li>Look at a different GO category (we only looked at BP, not MF or CC)</li>
<li>Advanced: include multiple filters (e.g., genes upregulated by EtOH stress in the WT strain that ALSO have defective induction during ethanol stress in the <em>msn2/4ΔΔ</em> mutant).</li>
</ul>
<p>The code below is a template for you to modify to complete this activity. The example code below looks at the downregulated genes in response to stress in the WT (choose something else for your gene list)</p>
<hr />
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="gene-ontology.html#cb39-1" tabindex="-1"></a><span class="co"># subset to just genes meeting your requirements</span></span>
<span id="cb39-2"><a href="gene-ontology.html#cb39-2" tabindex="-1"></a>DE_genes_GIVE_NAME <span class="ot">&lt;-</span> msn24_EtOH <span class="sc">|&gt;</span></span>
<span id="cb39-3"><a href="gene-ontology.html#cb39-3" tabindex="-1"></a>  <span class="co"># change the below line for the filters that you want</span></span>
<span id="cb39-4"><a href="gene-ontology.html#cb39-4" tabindex="-1"></a>  <span class="fu">filter</span>(log_fc_yps606_wt_et_oh_response <span class="sc">&lt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> pvalue_yps606_wt_et_oh_response<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">|&gt;</span> </span>
<span id="cb39-5"><a href="gene-ontology.html#cb39-5" tabindex="-1"></a>  <span class="fu">pull</span>(gene_id) <span class="co"># grabbing just the gene names</span></span></code></pre></div>
<div id="run-enrichment" class="section level3 hasAnchor" number="2.6.1">
<h3><span class="header-section-number">2.6.1</span> Run Enrichment<a href="gene-ontology.html#run-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="gene-ontology.html#cb40-1" tabindex="-1"></a>GO_GIVE_NAME_results <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(</span>
<span id="cb40-2"><a href="gene-ontology.html#cb40-2" tabindex="-1"></a>  <span class="at">gene =</span> DE_genes_GIVE_NAME,</span>
<span id="cb40-3"><a href="gene-ontology.html#cb40-3" tabindex="-1"></a>  <span class="at">OrgDb =</span> <span class="st">&quot;org.Sc.sgd.db&quot;</span>,</span>
<span id="cb40-4"><a href="gene-ontology.html#cb40-4" tabindex="-1"></a>  <span class="at">universe =</span> msn24_EtOH<span class="sc">$</span>gene_id,</span>
<span id="cb40-5"><a href="gene-ontology.html#cb40-5" tabindex="-1"></a>  <span class="at">keyType =</span> <span class="st">&quot;ORF&quot;</span>,</span>
<span id="cb40-6"><a href="gene-ontology.html#cb40-6" tabindex="-1"></a>  <span class="at">ont=</span> <span class="st">&quot;BP&quot;</span></span>
<span id="cb40-7"><a href="gene-ontology.html#cb40-7" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb40-8"><a href="gene-ontology.html#cb40-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">richFactor =</span> Count <span class="sc">/</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">&quot;/</span><span class="sc">\\</span><span class="st">d+&quot;</span>, <span class="st">&quot;&quot;</span>, BgRatio)))</span></code></pre></div>
</div>
<div id="see-the-data" class="section level3 hasAnchor" number="2.6.2">
<h3><span class="header-section-number">2.6.2</span> see the data<a href="gene-ontology.html#see-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="gene-ontology.html#cb41-1" tabindex="-1"></a><span class="co"># open up the results in a data frame to examine</span></span>
<span id="cb41-2"><a href="gene-ontology.html#cb41-2" tabindex="-1"></a>GO_GIVE_NAME_results <span class="sc">|&gt;</span></span>
<span id="cb41-3"><a href="gene-ontology.html#cb41-3" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-4"><a href="gene-ontology.html#cb41-4" tabindex="-1"></a>  <span class="fu">View</span>()</span>
<span id="cb41-5"><a href="gene-ontology.html#cb41-5" tabindex="-1"></a></span>
<span id="cb41-6"><a href="gene-ontology.html#cb41-6" tabindex="-1"></a><span class="co"># write out your results to a text file</span></span>
<span id="cb41-7"><a href="gene-ontology.html#cb41-7" tabindex="-1"></a>GO_GIVE_NAME_results <span class="sc">|&gt;</span></span>
<span id="cb41-8"><a href="gene-ontology.html#cb41-8" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-9"><a href="gene-ontology.html#cb41-9" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="st">&quot;~/Desktop/GO_GIVE_NAME_DIRECTION_results.tsv&quot;</span>)</span></code></pre></div>
</div>
<div id="create-plots" class="section level3 hasAnchor" number="2.6.3">
<h3><span class="header-section-number">2.6.3</span> create plots<a href="gene-ontology.html#create-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="gene-ontology.html#cb42-1" tabindex="-1"></a><span class="co"># a simple visualization</span></span>
<span id="cb42-2"><a href="gene-ontology.html#cb42-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">barplot</span>(GO_GIVE_NAME_results, <span class="at">showCategory =</span> <span class="dv">10</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/visualize-GOterms_(direction)(treatment)-GO-1.png" width="672" /></p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="gene-ontology.html#cb43-1" tabindex="-1"></a><span class="co"># built in visualization with dots instead</span></span>
<span id="cb43-2"><a href="gene-ontology.html#cb43-2" tabindex="-1"></a><span class="fu">dotplot</span>(GO_GIVE_NAME_results, <span class="at">showCategory=</span><span class="dv">10</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/visualize-GOterms_(direction)(treatment)-GO-2.png" width="672" /></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="gene-ontology.html#cb44-1" tabindex="-1"></a><span class="co"># a more complicated visualization, with more information density</span></span>
<span id="cb44-2"><a href="gene-ontology.html#cb44-2" tabindex="-1"></a><span class="fu">ggplot</span>(GO_GIVE_NAME_results,</span>
<span id="cb44-3"><a href="gene-ontology.html#cb44-3" tabindex="-1"></a>       <span class="at">showCategory =</span> <span class="dv">15</span>,</span>
<span id="cb44-4"><a href="gene-ontology.html#cb44-4" tabindex="-1"></a>       <span class="fu">aes</span>(richFactor, <span class="fu">fct_reorder</span>(Description, richFactor))) <span class="sc">+</span></span>
<span id="cb44-5"><a href="gene-ontology.html#cb44-5" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">xend =</span> <span class="dv">0</span>, <span class="at">yend =</span> Description)) <span class="sc">+</span></span>
<span id="cb44-6"><a href="gene-ontology.html#cb44-6" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> p.adjust, <span class="at">size =</span> Count)) <span class="sc">+</span></span>
<span id="cb44-7"><a href="gene-ontology.html#cb44-7" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(</span>
<span id="cb44-8"><a href="gene-ontology.html#cb44-8" tabindex="-1"></a>    <span class="at">colours =</span> <span class="fu">c</span>(<span class="st">&quot;#f7ca64&quot;</span>, <span class="st">&quot;#46bac2&quot;</span>, <span class="st">&quot;#7e62a3&quot;</span>),</span>
<span id="cb44-9"><a href="gene-ontology.html#cb44-9" tabindex="-1"></a>    <span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>,</span>
<span id="cb44-10"><a href="gene-ontology.html#cb44-10" tabindex="-1"></a>    <span class="at">guide =</span> <span class="fu">guide_colorbar</span>(<span class="at">reverse =</span> <span class="cn">TRUE</span>, <span class="at">order =</span> <span class="dv">1</span>)</span>
<span id="cb44-11"><a href="gene-ontology.html#cb44-11" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb44-12"><a href="gene-ontology.html#cb44-12" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb44-13"><a href="gene-ontology.html#cb44-13" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">label =</span> <span class="cf">function</span>(x) stringr<span class="sc">::</span><span class="fu">str_trunc</span>(x, <span class="dv">50</span>)) <span class="sc">+</span> <span class="co"># cut off long names</span></span>
<span id="cb44-14"><a href="gene-ontology.html#cb44-14" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Rich Factor&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-15"><a href="gene-ontology.html#cb44-15" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb44-16"><a href="gene-ontology.html#cb44-16" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Biological Processes&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-17"><a href="gene-ontology.html#cb44-17" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/visualize-GOterms_(direction)(treatment)-GO-3.png" width="672" /></p>
</div>
</div>
<div id="questions" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Questions<a href="gene-ontology.html#questions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Answer the following questions:</p>
<ol style="list-style-type: decimal">
<li><p>Which GO term had the smallest adjusted p-value in the upregulated comparison example that we did together?</p></li>
<li><p>What percent of the genes would we expect to have that GO term in the DE list under the null hypothesis? What percent of the DE genes actually had that GO term?</p></li>
<li><p>For the upregulated comparision, what GO terms are enriched for genes with pval &lt; 0.01 but fdr &gt; 0.01 and what is their average/median log fold change?</p></li>
<li><p>For one of your own novel comparisons, explain what comparison you were interested in, and your rationale for the cutoffs you chose for your gene list.</p></li>
<li><p>For that novel gene list you chose for yourself, which GO term had the smallest adjusted p-value?</p></li>
<li><p>In simple terms, how would you describe what the “Rich Factor” tells about a given GO term in the gene list.</p></li>
<li><p>Challenge: create a venn diagram of the GO terms in the GO analysis you ran comparing to the upregulated comparison example.</p></li>
</ol>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="gene-ontology.html#cb45-1" tabindex="-1"></a><span class="co"># create a list of the data we want to compare</span></span>
<span id="cb45-2"><a href="gene-ontology.html#cb45-2" tabindex="-1"></a>GO_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">data.frame</span>(GO_msn24_EtOH_up_results)<span class="sc">$</span>ID,</span>
<span id="cb45-3"><a href="gene-ontology.html#cb45-3" tabindex="-1"></a>                        <span class="fu">data.frame</span>(GO_GIVE_NAME_results)<span class="sc">$</span>ID)</span>
<span id="cb45-4"><a href="gene-ontology.html#cb45-4" tabindex="-1"></a></span>
<span id="cb45-5"><a href="gene-ontology.html#cb45-5" tabindex="-1"></a><span class="co"># visualize the GO results list as a venn diagram</span></span>
<span id="cb45-6"><a href="gene-ontology.html#cb45-6" tabindex="-1"></a><span class="fu">ggVennDiagram</span>(GO_results_list,</span>
<span id="cb45-7"><a href="gene-ontology.html#cb45-7" tabindex="-1"></a>              <span class="at">category.names =</span> <span class="fu">c</span>(<span class="st">&quot;msn24_EtOH_upregulated&quot;</span>, <span class="st">&quot;[GIVE_NAME]&quot;</span>)) <span class="sc">+</span></span>
<span id="cb45-8"><a href="gene-ontology.html#cb45-8" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> .<span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb45-9"><a href="gene-ontology.html#cb45-9" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">palette =</span> <span class="st">&quot;RdBu&quot;</span></span>
<span id="cb45-10"><a href="gene-ontology.html#cb45-10" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="_main_files/figure-html/create-vennDiagram-1.png" width="672" /></p>
<p>Be sure to knit this file into a pdf or html file once you’re finished.</p>
<p><br></p>
<p>System information for reproducibility:</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="gene-ontology.html#cb46-1" tabindex="-1"></a>pander<span class="sc">::</span><span class="fu">pander</span>(<span class="fu">sessionInfo</span>())</span></code></pre></div>
<p><strong>R version 4.3.1 (2023-06-16)</strong></p>
<p><strong>Platform:</strong> aarch64-apple-darwin20 (64-bit)</p>
<p><strong>locale:</strong>
en_US.UTF-8||en_US.UTF-8||en_US.UTF-8||C||en_US.UTF-8||en_US.UTF-8</p>
<p><strong>attached base packages:</strong>
<em>stats4</em>, <em>stats</em>, <em>graphics</em>, <em>grDevices</em>, <em>utils</em>, <em>datasets</em>, <em>methods</em> and <em>base</em></p>
<p><strong>other attached packages:</strong>
<em>org.Sc.sgd.db(v.3.17.0)</em>, <em>AnnotationDbi(v.1.62.2)</em>, <em>IRanges(v.2.34.1)</em>, <em>S4Vectors(v.0.38.2)</em>, <em>Biobase(v.2.60.0)</em>, <em>BiocGenerics(v.0.46.0)</em>, <em>clusterProfiler(v.4.8.2)</em>, <em>ggVennDiagram(v.1.2.3)</em>, <em>tidytree(v.0.4.5)</em>, <em>igraph(v.1.5.1)</em>, <em>janitor(v.2.2.0)</em>, <em>BiocManager(v.1.30.22)</em>, <em>pander(v.0.6.5)</em>, <em>knitr(v.1.44)</em>, <em>here(v.1.0.1)</em>, <em>lubridate(v.1.9.3)</em>, <em>forcats(v.1.0.0)</em>, <em>stringr(v.1.5.0)</em>, <em>dplyr(v.1.1.3)</em>, <em>purrr(v.1.0.2)</em>, <em>readr(v.2.1.4)</em>, <em>tidyr(v.1.3.0)</em>, <em>tibble(v.3.2.1)</em>, <em>ggplot2(v.3.4.4)</em>, <em>tidyverse(v.2.0.0)</em> and <em>pacman(v.0.5.1)</em></p>
<p><strong>loaded via a namespace (and not attached):</strong>
<em>RColorBrewer(v.1.1-3)</em>, <em>rstudioapi(v.0.15.0)</em>, <em>jsonlite(v.1.8.7)</em>, <em>magrittr(v.2.0.3)</em>, <em>farver(v.2.1.1)</em>, <em>rmarkdown(v.2.25)</em>, <em>ragg(v.1.2.6)</em>, <em>fs(v.1.6.3)</em>, <em>zlibbioc(v.1.46.0)</em>, <em>vctrs(v.0.6.4)</em>, <em>memoise(v.2.0.1)</em>, <em>RCurl(v.1.98-1.12)</em>, <em>ggtree(v.3.8.2)</em>, <em>htmltools(v.0.5.6.1)</em>, <em>curl(v.5.1.0)</em>, <em>gridGraphics(v.0.5-1)</em>, <em>sass(v.0.4.7)</em>, <em>KernSmooth(v.2.23-22)</em>, <em>bslib(v.0.5.1)</em>, <em>plyr(v.1.8.9)</em>, <em>cachem(v.1.0.8)</em>, <em>lifecycle(v.1.0.3)</em>, <em>pkgconfig(v.2.0.3)</em>, <em>Matrix(v.1.6-1.1)</em>, <em>R6(v.2.5.1)</em>, <em>fastmap(v.1.1.1)</em>, <em>gson(v.0.1.0)</em>, <em>GenomeInfoDbData(v.1.2.10)</em>, <em>snakecase(v.0.11.1)</em>, <em>digest(v.0.6.33)</em>, <em>aplot(v.0.2.2)</em>, <em>enrichplot(v.1.20.0)</em>, <em>colorspace(v.2.1-0)</em>, <em>patchwork(v.1.1.3)</em>, <em>rprojroot(v.2.0.3)</em>, <em>textshaping(v.0.3.7)</em>, <em>RSQLite(v.2.3.1)</em>, <em>labeling(v.0.4.3)</em>, <em>fansi(v.1.0.5)</em>, <em>timechange(v.0.2.0)</em>, <em>httr(v.1.4.7)</em>, <em>polyclip(v.1.10-6)</em>, <em>compiler(v.4.3.1)</em>, <em>proxy(v.0.4-27)</em>, <em>bit64(v.4.0.5)</em>, <em>withr(v.2.5.1)</em>, <em>downloader(v.0.4)</em>, <em>BiocParallel(v.1.34.2)</em>, <em>viridis(v.0.6.4)</em>, <em>DBI(v.1.1.3)</em>, <em>ggforce(v.0.4.1)</em>, <em>MASS(v.7.3-60)</em>, <em>classInt(v.0.4-10)</em>, <em>HDO.db(v.0.99.1)</em>, <em>units(v.0.8-4)</em>, <em>tools(v.4.3.1)</em>, <em>ape(v.5.7-1)</em>, <em>scatterpie(v.0.2.1)</em>, <em>glue(v.1.6.2)</em>, <em>nlme(v.3.1-163)</em>, <em>GOSemSim(v.2.26.1)</em>, <em>sf(v.1.0-14)</em>, <em>grid(v.4.3.1)</em>, <em>shadowtext(v.0.1.2)</em>, <em>reshape2(v.1.4.4)</em>, <em>fgsea(v.1.26.0)</em>, <em>generics(v.0.1.3)</em>, <em>gtable(v.0.3.4)</em>, <em>tzdb(v.0.4.0)</em>, <em>class(v.7.3-22)</em>, <em>data.table(v.1.14.8)</em>, <em>hms(v.1.1.3)</em>, <em>tidygraph(v.1.2.3)</em>, <em>utf8(v.1.2.3)</em>, <em>XVector(v.0.40.0)</em>, <em>ggrepel(v.0.9.4)</em>, <em>pillar(v.1.9.0)</em>, <em>yulab.utils(v.0.1.0)</em>, <em>vroom(v.1.6.4)</em>, <em>splines(v.4.3.1)</em>, <em>tweenr(v.2.0.2)</em>, <em>treeio(v.1.24.3)</em>, <em>lattice(v.0.21-9)</em>, <em>bit(v.4.0.5)</em>, <em>tidyselect(v.1.2.0)</em>, <em>GO.db(v.3.17.0)</em>, <em>Biostrings(v.2.68.1)</em>, <em>gridExtra(v.2.3)</em>, <em>bookdown(v.0.36)</em>, <em>xfun(v.0.40)</em>, <em>graphlayouts(v.1.0.1)</em>, <em>stringi(v.1.7.12)</em>, <em>lazyeval(v.0.2.2)</em>, <em>ggfun(v.0.1.3)</em>, <em>yaml(v.2.3.7)</em>, <em>evaluate(v.0.22)</em>, <em>codetools(v.0.2-19)</em>, <em>ggraph(v.2.1.0)</em>, <em>qvalue(v.2.32.0)</em>, <em>RVenn(v.1.1.0)</em>, <em>ggplotify(v.0.1.2)</em>, <em>cli(v.3.6.1)</em>, <em>systemfonts(v.1.0.5)</em>, <em>munsell(v.0.5.0)</em>, <em>jquerylib(v.0.1.4)</em>, <em>Rcpp(v.1.0.11)</em>, <em>GenomeInfoDb(v.1.36.4)</em>, <em>png(v.0.1-8)</em>, <em>parallel(v.4.3.1)</em>, <em>blob(v.1.2.4)</em>, <em>DOSE(v.3.26.1)</em>, <em>bitops(v.1.0-7)</em>, <em>viridisLite(v.0.4.2)</em>, <em>e1071(v.1.7-13)</em>, <em>scales(v.1.2.1)</em>, <em>crayon(v.1.5.2)</em>, <em>rlang(v.1.1.1)</em>, <em>cowplot(v.1.1.1)</em>, <em>fastmatch(v.1.1-4)</em> and <em>KEGGREST(v.1.40.1)</em></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="getting-started-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="working-with-sequences-raw-data-quality-control.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-Gene_Ontology.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
