<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Read Mapping | Genomic Data Analysis Course Exercises</title>
  <meta name="description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Read Mapping | Genomic Data Analysis Course Exercises" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="github-repo" content="clstacy/GenomicDataAnalysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Read Mapping | Genomic Data Analysis Course Exercises" />
  
  <meta name="twitter:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  

<meta name="author" content="Carson Stacy &amp; Jeffrey Lewis" />


<meta name="date" content="2023-10-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="working-with-sequences-raw-data-quality-control.html"/>
<link rel="next" href="read-counting.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/core-js-2.5.3/shim.min.js"></script>
<script src="libs/react-18.2.0/react.min.js"></script>
<script src="libs/react-18.2.0/react-dom.min.js"></script>
<script src="libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="libs/reactable-binding-0.4.4/reactable.js"></script>
<link href="libs/datatables-1/datatables.css" rel="stylesheet" />
<script src="libs/datatables-1/datatables.min.js"></script>
<link href="libs/vega-1/vega_plots.css" rel="stylesheet" />
<script src="libs/vega-1/vega.min.js"></script>
<link href="libs/vega-tooltip-1/vega-tooltip.css" rel="stylesheet" />
<script src="libs/vega-tooltip-1/vega-tooltip.js"></script>
<script src="libs/FileSaver-1/FileSaver.js"></script>
<script src="libs/GlimmaV2-1/XYSpecs.js"></script>
<script src="libs/GlimmaV2-1/MDSSpecs.js"></script>
<script src="libs/GlimmaV2-1/expressionSpec.js"></script>
<script src="libs/GlimmaV2-1/makeTooltip.js"></script>
<script src="libs/GlimmaV2-1/saveTools.js"></script>
<script src="libs/glimmaXY-binding-2.10.0/glimmaXY.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Genomic Data Analysis Exercises</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i><b>0.1</b> Usage</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#key-features"><i class="fa fa-check"></i><b>0.2</b> <strong>Key Features</strong></a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#disclaimer"><i class="fa fa-check"></i><b>0.3</b> Disclaimer</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html"><i class="fa fa-check"></i><b>1</b> Getting Started in R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exercise-description"><i class="fa fa-check"></i><b>1.1</b> Exercise Description</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#learning-outcomes"><i class="fa fa-check"></i><b>1.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#using-r-and-rstudio"><i class="fa fa-check"></i><b>1.3</b> Using R and RStudio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#load-data-directly-from-the-url"><i class="fa fa-check"></i><b>1.4</b> Load data directly from the URL</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#working-with-data-in-r"><i class="fa fa-check"></i><b>1.5</b> Working with data in R</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#looking-at-data-in-rstudio"><i class="fa fa-check"></i><b>1.6</b> Looking at Data in RStudio</a></li>
<li class="chapter" data-level="1.7" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exploring-the-data"><i class="fa fa-check"></i><b>1.7</b> Exploring the data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gene-ontology.html"><a href="gene-ontology.html"><i class="fa fa-check"></i><b>2</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gene-ontology.html"><a href="gene-ontology.html#description"><i class="fa fa-check"></i><b>2.1</b> Description</a></li>
<li class="chapter" data-level="2.2" data-path="gene-ontology.html"><a href="gene-ontology.html#learning-outcomes-1"><i class="fa fa-check"></i><b>2.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="2.3" data-path="gene-ontology.html"><a href="gene-ontology.html#analysis-workflow"><i class="fa fa-check"></i><b>2.3</b> Analysis Workflow</a></li>
<li class="chapter" data-level="2.4" data-path="gene-ontology.html"><a href="gene-ontology.html#get-de-gene-list"><i class="fa fa-check"></i><b>2.4</b> Get DE gene list</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="gene-ontology.html"><a href="gene-ontology.html#saving-ggplot-output-to-a-file"><i class="fa fa-check"></i><b>2.4.1</b> Saving ggplot output to a file</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="gene-ontology.html"><a href="gene-ontology.html#the-hypergeometric-distribution-in-practice"><i class="fa fa-check"></i><b>2.5</b> The Hypergeometric Distribution in practice</a></li>
<li class="chapter" data-level="2.6" data-path="gene-ontology.html"><a href="gene-ontology.html#now-it-is-your-turn"><i class="fa fa-check"></i><b>2.6</b> Now it is your turn</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="gene-ontology.html"><a href="gene-ontology.html#run-enrichment"><i class="fa fa-check"></i><b>2.6.1</b> Run Enrichment</a></li>
<li class="chapter" data-level="2.6.2" data-path="gene-ontology.html"><a href="gene-ontology.html#see-the-data"><i class="fa fa-check"></i><b>2.6.2</b> see the data</a></li>
<li class="chapter" data-level="2.6.3" data-path="gene-ontology.html"><a href="gene-ontology.html#create-plots"><i class="fa fa-check"></i><b>2.6.3</b> create plots</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="gene-ontology.html"><a href="gene-ontology.html#questions"><i class="fa fa-check"></i><b>2.7</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html"><i class="fa fa-check"></i><b>3</b> Working with Sequences: Raw Data &amp; Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#description-1"><i class="fa fa-check"></i><b>3.1</b> Description</a></li>
<li class="chapter" data-level="3.2" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#learning-outcomes-2"><i class="fa fa-check"></i><b>3.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="3.3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#download-fastq"><i class="fa fa-check"></i><b>3.3</b> Download fastq</a></li>
<li class="chapter" data-level="3.4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#examining-fastq"><i class="fa fa-check"></i><b>3.4</b> Examining fastq</a></li>
<li class="chapter" data-level="3.5" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#trimming"><i class="fa fa-check"></i><b>3.5</b> Trimming</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#automate-for-list-of-files"><i class="fa fa-check"></i><b>3.5.1</b> Automate for list of files</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#batch-file-processing"><i class="fa fa-check"></i><b>3.6</b> Batch file processing</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#running-rfastp-creates-several-files"><i class="fa fa-check"></i><b>3.6.1</b> Running RfastP creates several files:</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#qc-and-adapters"><i class="fa fa-check"></i><b>3.7</b> QC and adapters</a></li>
<li class="chapter" data-level="3.8" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#running-fastqc"><i class="fa fa-check"></i><b>3.8</b> Running fastqc</a></li>
<li class="chapter" data-level="3.9" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#multiqc-for-qc-on-mutliple-samples"><i class="fa fa-check"></i><b>3.9</b> Multiqc for QC on mutliple samples</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="read-mapping.html"><a href="read-mapping.html"><i class="fa fa-check"></i><b>4</b> Read Mapping</a>
<ul>
<li class="chapter" data-level="4.1" data-path="read-mapping.html"><a href="read-mapping.html#alignment"><i class="fa fa-check"></i><b>4.1</b> Alignment</a></li>
<li class="chapter" data-level="4.2" data-path="read-mapping.html"><a href="read-mapping.html#retrieve-the-genome"><i class="fa fa-check"></i><b>4.2</b> Retrieve the genome</a></li>
<li class="chapter" data-level="4.3" data-path="read-mapping.html"><a href="read-mapping.html#build-rsubread-index"><i class="fa fa-check"></i><b>4.3</b> Build Rsubread Index</a></li>
<li class="chapter" data-level="4.4" data-path="read-mapping.html"><a href="read-mapping.html#pseudomapping-with-salmon"><i class="fa fa-check"></i><b>4.4</b> Pseudomapping with Salmon</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="read-mapping.html"><a href="read-mapping.html#create-conda-env"><i class="fa fa-check"></i><b>4.4.1</b> Create Conda Env</a></li>
<li class="chapter" data-level="4.4.2" data-path="read-mapping.html"><a href="read-mapping.html#download-transcriptome"><i class="fa fa-check"></i><b>4.4.2</b> Download transcriptome</a></li>
<li class="chapter" data-level="4.4.3" data-path="read-mapping.html"><a href="read-mapping.html#building-the-salmon-index"><i class="fa fa-check"></i><b>4.4.3</b> Building the Salmon index</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="read-mapping.html"><a href="read-mapping.html#questions-1"><i class="fa fa-check"></i><b>4.5</b> Questions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="read-mapping.html"><a href="read-mapping.html#with-rsubread"><i class="fa fa-check"></i><b>4.5.1</b> With Rsubread:</a></li>
<li class="chapter" data-level="4.5.2" data-path="read-mapping.html"><a href="read-mapping.html#with-salmon"><i class="fa fa-check"></i><b>4.5.2</b> With Salmon:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="read-counting.html"><a href="read-counting.html"><i class="fa fa-check"></i><b>5</b> Read Counting</a>
<ul>
<li class="chapter" data-level="5.1" data-path="read-counting.html"><a href="read-counting.html#featurecounts"><i class="fa fa-check"></i><b>5.1</b> featureCounts</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="read-counting.html"><a href="read-counting.html#locate-bam-files"><i class="fa fa-check"></i><b>5.1.1</b> Locate BAM files</a></li>
<li class="chapter" data-level="5.1.2" data-path="read-counting.html"><a href="read-counting.html#retrieve-the-genome-annotation"><i class="fa fa-check"></i><b>5.1.2</b> Retrieve the genome annotation</a></li>
<li class="chapter" data-level="5.1.3" data-path="read-counting.html"><a href="read-counting.html#counting-with-featurecounts"><i class="fa fa-check"></i><b>5.1.3</b> Counting with FeatureCounts</a></li>
<li class="chapter" data-level="5.1.4" data-path="read-counting.html"><a href="read-counting.html#counts-object"><i class="fa fa-check"></i><b>5.1.4</b> Counts object</a></li>
<li class="chapter" data-level="5.1.5" data-path="read-counting.html"><a href="read-counting.html#saving-fc-object-for-future-use"><i class="fa fa-check"></i><b>5.1.5</b> Saving <code>fc</code> object for future use</a></li>
<li class="chapter" data-level="5.1.6" data-path="read-counting.html"><a href="read-counting.html#rsubread-qc"><i class="fa fa-check"></i><b>5.1.6</b> RSubread QC</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="read-counting.html"><a href="read-counting.html#salmon"><i class="fa fa-check"></i><b>5.2</b> Salmon</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="read-counting.html"><a href="read-counting.html#pseudomapping-counting"><i class="fa fa-check"></i><b>5.2.1</b> Pseudomapping &amp; Counting</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="read-counting.html"><a href="read-counting.html#questions-2"><i class="fa fa-check"></i><b>5.3</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html"><i class="fa fa-check"></i><b>6</b> Differential Expression: EdgeR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#description-2"><i class="fa fa-check"></i><b>6.1</b> Description</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#learning-outcomes-3"><i class="fa fa-check"></i><b>6.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="6.3" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#loading-in-the-featurecounts-object"><i class="fa fa-check"></i><b>6.3</b> Loading in the featureCounts object</a></li>
<li class="chapter" data-level="6.4" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#count-loading-and-annotation"><i class="fa fa-check"></i><b>6.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="6.5" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#filtering-to-remove-low-counts"><i class="fa fa-check"></i><b>6.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="6.6" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#normalization-for-composition-bias"><i class="fa fa-check"></i><b>6.6</b> Normalization for composition bias</a></li>
<li class="chapter" data-level="6.7" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#mds-plots"><i class="fa fa-check"></i><b>6.7</b> MDS plots</a></li>
<li class="chapter" data-level="6.8" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#exploring-differences-between-libraries"><i class="fa fa-check"></i><b>6.8</b> Exploring differences between libraries</a></li>
<li class="chapter" data-level="6.9" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#estimate-dispersion"><i class="fa fa-check"></i><b>6.9</b> Estimate Dispersion</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#testing-for-differential-expression"><i class="fa fa-check"></i><b>6.10</b> Testing for differential expression</a></li>
<li class="chapter" data-level="6.11" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#looking-at-all-contrasts-at-once"><i class="fa fa-check"></i><b>6.11</b> Looking at all contrasts at once</a></li>
<li class="chapter" data-level="6.12" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#questions-3"><i class="fa fa-check"></i><b>6.12</b> Questions</a></li>
<li class="chapter" data-level="6.13" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#a-template-set-of-code-chunks-for-doing-this-is-below"><i class="fa fa-check"></i><b>6.13</b> A template set of code chunks for doing this is below:</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html"><i class="fa fa-check"></i><b>7</b> Differential Expression: DESeq2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#description-3"><i class="fa fa-check"></i><b>7.1</b> Description</a></li>
<li class="chapter" data-level="7.2" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#learning-outcomes-4"><i class="fa fa-check"></i><b>7.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="7.3" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#loading-in-the-featurecounts-object-1"><i class="fa fa-check"></i><b>7.3</b> Loading in the featureCounts object</a></li>
<li class="chapter" data-level="7.4" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#count-loading-and-annotation-1"><i class="fa fa-check"></i><b>7.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="7.5" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#filtering-to-remove-low-counts-1"><i class="fa fa-check"></i><b>7.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="7.6" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#testing-for-differential-expression-1"><i class="fa fa-check"></i><b>7.6</b> Testing for differential expression</a></li>
<li class="chapter" data-level="7.7" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#questions-4"><i class="fa fa-check"></i><b>7.7</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html"><i class="fa fa-check"></i><b>8</b> Differential Expression: limma</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#description-4"><i class="fa fa-check"></i><b>8.1</b> Description</a></li>
<li class="chapter" data-level="8.2" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#learning-objectives"><i class="fa fa-check"></i><b>8.2</b> Learning Objectives</a></li>
<li class="chapter" data-level="8.3" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#loading-in-the-count-data-file"><i class="fa fa-check"></i><b>8.3</b> Loading in the count data file</a></li>
<li class="chapter" data-level="8.4" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#count-loading-and-annotation-2"><i class="fa fa-check"></i><b>8.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="8.5" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#filtering-to-remove-low-counts-2"><i class="fa fa-check"></i><b>8.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="8.6" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#normalization-for-composition-bias-1"><i class="fa fa-check"></i><b>8.6</b> Normalization for composition bias</a></li>
<li class="chapter" data-level="8.7" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#exploring-differences-between-libraries-1"><i class="fa fa-check"></i><b>8.7</b> Exploring differences between libraries</a></li>
<li class="chapter" data-level="8.8" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#estimate-dispersion-1"><i class="fa fa-check"></i><b>8.8</b> Estimate Dispersion</a></li>
<li class="chapter" data-level="8.9" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#testing-for-differential-expression-2"><i class="fa fa-check"></i><b>8.9</b> Testing for differential expression</a></li>
<li class="chapter" data-level="8.10" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#examining-a-specific-contrast"><i class="fa fa-check"></i><b>8.10</b> Examining a specific contrast</a></li>
<li class="chapter" data-level="8.11" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#visualization"><i class="fa fa-check"></i><b>8.11</b> Visualization</a>
<ul>
<li class="chapter" data-level="8.11.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#ma-lot"><i class="fa fa-check"></i><b>8.11.1</b> MA lot</a></li>
</ul></li>
<li class="chapter" data-level="8.12" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#treat-testing"><i class="fa fa-check"></i><b>8.12</b> <code>treat()</code> testing</a>
<ul>
<li class="chapter" data-level="8.12.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#visualize-de-genes-from-treat-using-lfc1"><i class="fa fa-check"></i><b>8.12.1</b> Visualize DE genes from <code>Treat</code> using lfc=1</a></li>
</ul></li>
<li class="chapter" data-level="8.13" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#comparing-de-analysis-softwares"><i class="fa fa-check"></i><b>8.13</b> Comparing DE analysis softwares</a></li>
<li class="chapter" data-level="8.14" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#correlation-between-logfc-estimates-across-softwares"><i class="fa fa-check"></i><b>8.14</b> Correlation between logFC estimates across softwares</a></li>
<li class="chapter" data-level="8.15" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#questions-5"><i class="fa fa-check"></i><b>8.15</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html"><i class="fa fa-check"></i><b>9</b> Visualizing Differential Expression Results</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#description-5"><i class="fa fa-check"></i><b>9.1</b> Description</a></li>
<li class="chapter" data-level="9.2" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#learning-outcomes-5"><i class="fa fa-check"></i><b>9.2</b> Learning Outcomes</a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#ma-plot"><i class="fa fa-check"></i><b>9.3</b> MA-plot</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#volcano-plot"><i class="fa fa-check"></i><b>9.4</b> Volcano Plot</a></li>
<li class="chapter" data-level="9.5" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#using-glimma-for-an-interactive-visualization"><i class="fa fa-check"></i><b>9.5</b> Using Glimma for an interactive visualization</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#ma-plots"><i class="fa fa-check"></i><b>9.5.1</b> MA plots</a></li>
<li class="chapter" data-level="9.5.2" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#volcano-plots"><i class="fa fa-check"></i><b>9.5.2</b> Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#generating-bar-graph-summaries"><i class="fa fa-check"></i><b>9.6</b> Generating bar graph summaries</a></li>
<li class="chapter" data-level="9.7" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#exercise"><i class="fa fa-check"></i><b>9.7</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Clustering</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#description-6"><i class="fa fa-check"></i><b>10.1</b> Description</a></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#learning-outcomes-6"><i class="fa fa-check"></i><b>10.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#cluster-3.0"><i class="fa fa-check"></i><b>10.3</b> Cluster 3.0</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="clustering.html"><a href="clustering.html#generating-a-pcl-file"><i class="fa fa-check"></i><b>10.3.1</b> Generating a PCL file</a></li>
<li class="chapter" data-level="10.3.2" data-path="clustering.html"><a href="clustering.html#filtering-data"><i class="fa fa-check"></i><b>10.3.2</b> Filtering data</a></li>
<li class="chapter" data-level="10.3.3" data-path="clustering.html"><a href="clustering.html#hierarchical-clustering"><i class="fa fa-check"></i><b>10.3.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="10.3.4" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.3.4</b> <em>K</em>-Means Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#visualizing-clusters-with-java-treeview"><i class="fa fa-check"></i><b>10.4</b> Visualizing Clusters with Java TreeView</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="clustering.html"><a href="clustering.html#changing-pixel-settings."><i class="fa fa-check"></i><b>10.4.1</b> Changing pixel settings.</a></li>
<li class="chapter" data-level="10.4.2" data-path="clustering.html"><a href="clustering.html#selecting-groups-of-genes."><i class="fa fa-check"></i><b>10.4.2</b> Selecting groups of genes.</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#performing-clustering-on-yeast-stress-data"><i class="fa fa-check"></i><b>10.5</b> Performing clustering on yeast stress data</a></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#questions-6"><i class="fa fa-check"></i><b>10.6</b> Questions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Genomic Data Analysis Course Exercises</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="read-mapping" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Read Mapping<a href="read-mapping.html#read-mapping" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>last updated: 2023-10-26</p>
<p>As usual, make sure we have the right packages for this exercise</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="read-mapping.html#cb99-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;pacman&quot;</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;pacman&quot;</span>); <span class="fu">library</span>(pacman)</span>
<span id="cb99-2"><a href="read-mapping.html#cb99-2" tabindex="-1"></a></span>
<span id="cb99-3"><a href="read-mapping.html#cb99-3" tabindex="-1"></a><span class="co"># let&#39;s load all of the files we were using and want to have again today</span></span>
<span id="cb99-4"><a href="read-mapping.html#cb99-4" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;knitr&quot;</span>, <span class="st">&quot;readr&quot;</span>,</span>
<span id="cb99-5"><a href="read-mapping.html#cb99-5" tabindex="-1"></a>       <span class="st">&quot;pander&quot;</span>, <span class="st">&quot;BiocManager&quot;</span>, </span>
<span id="cb99-6"><a href="read-mapping.html#cb99-6" tabindex="-1"></a>       <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;stringr&quot;</span>)</span>
<span id="cb99-7"><a href="read-mapping.html#cb99-7" tabindex="-1"></a></span>
<span id="cb99-8"><a href="read-mapping.html#cb99-8" tabindex="-1"></a><span class="co"># We also need the Bioconductor packages &quot;Rsubread&quot; for today&#39;s activity.</span></span>
<span id="cb99-9"><a href="read-mapping.html#cb99-9" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;Rsubread&quot;</span>)</span></code></pre></div>
<p>Previously, we filtered and trimmed our raw fastq files. They should be in the folder below, unless you chose a different place to store them.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="read-mapping.html#cb100-1" tabindex="-1"></a>dir_trimmed.fq_files <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp&quot;</span></span>
<span id="cb100-2"><a href="read-mapping.html#cb100-2" tabindex="-1"></a></span>
<span id="cb100-3"><a href="read-mapping.html#cb100-3" tabindex="-1"></a>trimmed_fastq_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> dir_trimmed.fq_files, </span>
<span id="cb100-4"><a href="read-mapping.html#cb100-4" tabindex="-1"></a>                                  <span class="at">pattern =</span> <span class="st">&quot;.fastq.gz$&quot;</span>, </span>
<span id="cb100-5"><a href="read-mapping.html#cb100-5" tabindex="-1"></a>                                  <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb100-6"><a href="read-mapping.html#cb100-6" tabindex="-1"></a>trimmed_fastq_files</span></code></pre></div>
<pre><code>##  [1] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP1_R1.fastq.gz&quot;
##  [2] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP2_R1.fastq.gz&quot;
##  [3] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP3_R1.fastq.gz&quot;
##  [4] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP4_R1.fastq.gz&quot;
##  [5] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP1_R1.fastq.gz&quot;
##  [6] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP2_R1.fastq.gz&quot;
##  [7] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP3_R1.fastq.gz&quot;
##  [8] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP4_R1.fastq.gz&quot;
##  [9] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP1_R1.fastq.gz&quot;   
## [10] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP2_R1.fastq.gz&quot;   
## [11] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP3_R1.fastq.gz&quot;   
## [12] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP4_R1.fastq.gz&quot;   
## [13] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP1_R1.fastq.gz&quot;   
## [14] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP2_R1.fastq.gz&quot;   
## [15] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP3_R1.fastq.gz&quot;   
## [16] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP4_R1.fastq.gz&quot;</code></pre>
<p>You should see the full paths to all 16 trimmed fastq files that we will be mapping to the reference genome today.</p>
<div id="alignment" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Alignment<a href="read-mapping.html#alignment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Read sequences are stored in compressed (gzipped) FASTQ files. Before the differential expression analysis can proceed, these reads must be aligned to the yeast genome and counted into annotated genes. This can be achieved with functions in the Rsubread package.</p>
</div>
<div id="retrieve-the-genome" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Retrieve the genome<a href="read-mapping.html#retrieve-the-genome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will use a bash code chunk to download the latest genome</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb102-1"><a href="read-mapping.html#cb102-1" tabindex="-1"></a><span class="co"># Define the destination file path</span></span>
<span id="cb102-2"><a href="read-mapping.html#cb102-2" tabindex="-1"></a><span class="co"># You can change this file path to the path you want your data to go, or leave it.</span></span>
<span id="cb102-3"><a href="read-mapping.html#cb102-3" tabindex="-1"></a><span class="va">REF_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Reference&quot;</span></span>
<span id="cb102-4"><a href="read-mapping.html#cb102-4" tabindex="-1"></a></span>
<span id="cb102-5"><a href="read-mapping.html#cb102-5" tabindex="-1"></a><span class="co"># make that directory if it doesn&#39;t already</span></span>
<span id="cb102-6"><a href="read-mapping.html#cb102-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$REF_DIR</span></span>
<span id="cb102-7"><a href="read-mapping.html#cb102-7" tabindex="-1"></a></span>
<span id="cb102-8"><a href="read-mapping.html#cb102-8" tabindex="-1"></a><span class="co"># Define the URL of reference genome</span></span>
<span id="cb102-9"><a href="read-mapping.html#cb102-9" tabindex="-1"></a><span class="co"># (latest from ensembl)</span></span>
<span id="cb102-10"><a href="read-mapping.html#cb102-10" tabindex="-1"></a><span class="va">url</span><span class="op">=</span><span class="st">&quot;ftp://ftp.ensembl.org/pub/release-110/fasta/saccharomyces_cerevisiae/dna/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz&quot;</span></span>
<span id="cb102-11"><a href="read-mapping.html#cb102-11" tabindex="-1"></a></span>
<span id="cb102-12"><a href="read-mapping.html#cb102-12" tabindex="-1"></a></span>
<span id="cb102-13"><a href="read-mapping.html#cb102-13" tabindex="-1"></a><span class="co"># Check if the file already exists at the destination location</span></span>
<span id="cb102-14"><a href="read-mapping.html#cb102-14" tabindex="-1"></a><span class="cf">if</span> <span class="bu">[</span> <span class="ot">!</span> <span class="ot">-f</span> <span class="st">&quot;</span><span class="va">$REF_DIR</span><span class="st">/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz&quot;</span> <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb102-15"><a href="read-mapping.html#cb102-15" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;Reference genome not found, downloading...&quot;</span></span>
<span id="cb102-16"><a href="read-mapping.html#cb102-16" tabindex="-1"></a>    <span class="co"># If the file does not exist, download it using curl</span></span>
<span id="cb102-17"><a href="read-mapping.html#cb102-17" tabindex="-1"></a>    <span class="ex">curl</span> <span class="at">-o</span> <span class="st">&quot;</span><span class="va">$REF_DIR</span><span class="st">/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz&quot;</span> <span class="st">&quot;</span><span class="va">$url</span><span class="st">&quot;</span></span>
<span id="cb102-18"><a href="read-mapping.html#cb102-18" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;Downloading finished&quot;</span></span>
<span id="cb102-19"><a href="read-mapping.html#cb102-19" tabindex="-1"></a><span class="cf">else</span></span>
<span id="cb102-20"><a href="read-mapping.html#cb102-20" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;File already exists at </span><span class="va">$REF_DIR</span><span class="st"> Skipping download.&quot;</span></span>
<span id="cb102-21"><a href="read-mapping.html#cb102-21" tabindex="-1"></a><span class="cf">fi</span></span></code></pre></div>
<pre><code>## File already exists at /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference Skipping download.</code></pre>
</div>
<div id="build-rsubread-index" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Build Rsubread Index<a href="read-mapping.html#build-rsubread-index" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The first step in performing the alignment is to build an index. In order to build an index you need to have the fasta file (.fa), which can be downloaded from the UCSC genome browser. This may take several minutes to run. Building the full index using the whole genome usually takes about 30 minutes to an hr on a server for larger Eukaryotic genomes. Because yeast has a relatively small genome size, we are able to build the full index in class.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="read-mapping.html#cb104-1" tabindex="-1"></a><span class="fu">library</span>(Rsubread)</span>
<span id="cb104-2"><a href="read-mapping.html#cb104-2" tabindex="-1"></a></span>
<span id="cb104-3"><a href="read-mapping.html#cb104-3" tabindex="-1"></a><span class="co"># Set path of the reference fasta file</span></span>
<span id="cb104-4"><a href="read-mapping.html#cb104-4" tabindex="-1"></a>reference_genome <span class="ot">=</span> <span class="fu">path.expand</span>(<span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Reference/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz&quot;</span>)</span>
<span id="cb104-5"><a href="read-mapping.html#cb104-5" tabindex="-1"></a></span>
<span id="cb104-6"><a href="read-mapping.html#cb104-6" tabindex="-1"></a>index_reference_genome <span class="ot">=</span> <span class="fu">path.expand</span>(<span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Reference/index_rsubread_Saccharomyces_cerevisiae.R64-1-1&quot;</span>)</span>
<span id="cb104-7"><a href="read-mapping.html#cb104-7" tabindex="-1"></a></span>
<span id="cb104-8"><a href="read-mapping.html#cb104-8" tabindex="-1"></a><span class="co"># build the index</span></span>
<span id="cb104-9"><a href="read-mapping.html#cb104-9" tabindex="-1"></a><span class="fu">buildindex</span>(<span class="at">basename=</span>index_reference_genome, <span class="at">reference=</span>reference_genome)</span></code></pre></div>
<pre><code>## 
##         ==========     _____ _    _ ____  _____  ______          _____  
##         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ 
##           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |
##             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |
##               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |
##         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/
##        Rsubread 2.14.2
## 
## //================================= setting ==================================\\
## ||                                                                            ||
## ||                Index name : index_rsubread_Saccharomyces_cerevisiae.R6 ... ||
## ||               Index space : base space                                     ||
## ||               Index split : no-split                                       ||
## ||          Repeat threshold : 100 repeats                                    ||
## ||              Gapped index : no                                             ||
## ||                                                                            ||
## ||       Free / total memory : 1.4GB / 8.0GB                                  ||
## ||                                                                            ||
## ||               Input files : 1 file in total                                ||
## ||                             o Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz ||
## ||                                                                            ||
## ||                                                                            ||
## ||   WARNING: the free memory is lower than 3.0GB.                            ||
## ||            the program may run very slow or crash.                         ||
## ||                                                                            ||
## \\============================================================================//
## 
## //================================= Running ==================================\\
## ||                                                                            ||
## || Check the integrity of provided reference sequences ...                    ||
## || No format issues were found                                                ||
## || Scan uninformative subreads in reference sequences ...                     ||
## || 11 uninformative subreads were found.                                      ||
## || These subreads were excluded from index building.                          ||
## || Estimate the index size...                                                 ||
## ||    8%,   0 mins elapsed, rate=9960.4k bps/s                                ||
## ||   16%,   0 mins elapsed, rate=11494.5k bps/s                               ||
## ||   24%,   0 mins elapsed, rate=12057.2k bps/s                               ||
## ||   33%,   0 mins elapsed, rate=12376.9k bps/s                               ||
## ||   41%,   0 mins elapsed, rate=12547.2k bps/s                               ||
## ||   49%,   0 mins elapsed, rate=12702.2k bps/s                               ||
## ||   58%,   0 mins elapsed, rate=12812.2k bps/s                               ||
## ||   66%,   0 mins elapsed, rate=12918.7k bps/s                               ||
## ||   74%,   0 mins elapsed, rate=12985.2k bps/s                               ||
## ||   83%,   0 mins elapsed, rate=13044.8k bps/s                               ||
## ||   91%,   0 mins elapsed, rate=13068.8k bps/s                               ||
## ||                                                                            ||
## ||              WARNING: available memory is lower than 3.0 GB.               ||
## ||                           The program may run very slow.                   ||
## || Build a gapped index and/or split index into blocks to reduce memory use.  ||
## ||                                                                            ||
## || Build the index...                                                         ||
## ||    8%,   0 mins elapsed, rate=217.5k bps/s                                 ||
## ||   16%,   0 mins elapsed, rate=249.5k bps/s                                 ||
## ||   24%,   0 mins elapsed, rate=240.3k bps/s                                 ||
## ||   33%,   0 mins elapsed, rate=247.0k bps/s                                 ||
## ||   41%,   0 mins elapsed, rate=251.0k bps/s                                 ||
## ||   49%,   0 mins elapsed, rate=256.0k bps/s                                 ||
## ||   58%,   0 mins elapsed, rate=243.3k bps/s                                 ||
## ||   66%,   0 mins elapsed, rate=232.4k bps/s                                 ||
## ||   74%,   0 mins elapsed, rate=231.8k bps/s                                 ||
## ||   83%,   0 mins elapsed, rate=230.8k bps/s                                 ||
## ||   91%,   0 mins elapsed, rate=231.0k bps/s                                 ||
## || Save current index block...                                                ||
## ||  [ 0.0% finished ]                                                         ||
## ||  [ 10.0% finished ]                                                        ||
## ||  [ 20.0% finished ]                                                        ||
## ||  [ 30.0% finished ]                                                        ||
## ||  [ 40.0% finished ]                                                        ||
## ||  [ 50.0% finished ]                                                        ||
## ||  [ 60.0% finished ]                                                        ||
## ||  [ 70.0% finished ]                                                        ||
## ||  [ 80.0% finished ]                                                        ||
## ||  [ 90.0% finished ]                                                        ||
## ||  [ 100.0% finished ]                                                       ||
## ||                                                                            ||
## ||                      Total running time: 1.4 minutes.                      ||
## ||Index /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_rsu ... ||
## ||                                                                            ||
## \\============================================================================//</code></pre>
<p>We can see the arguments available with the align function from the Rsubread package</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="read-mapping.html#cb106-1" tabindex="-1"></a><span class="fu">args</span>(align)</span></code></pre></div>
<pre><code>## function (index, readfile1, readfile2 = NULL, type = &quot;rna&quot;, input_format = &quot;gzFASTQ&quot;, 
##     output_format = &quot;BAM&quot;, output_file = paste(readfile1, &quot;subread&quot;, 
##         output_format, sep = &quot;.&quot;), phredOffset = 33, nsubreads = 10, 
##     TH1 = 3, TH2 = 1, maxMismatches = 3, unique = FALSE, nBestLocations = 1, 
##     indels = 5, complexIndels = FALSE, nTrim5 = 0, nTrim3 = 0, 
##     minFragLength = 50, maxFragLength = 600, PE_orientation = &quot;fr&quot;, 
##     nthreads = 1, readGroupID = NULL, readGroup = NULL, keepReadOrder = FALSE, 
##     sortReadsByCoordinates = FALSE, color2base = FALSE, DP_GapOpenPenalty = -1, 
##     DP_GapExtPenalty = 0, DP_MismatchPenalty = 0, DP_MatchScore = 2, 
##     detectSV = FALSE, useAnnotation = FALSE, annot.inbuilt = &quot;mm39&quot;, 
##     annot.ext = NULL, isGTF = FALSE, GTF.featureType = &quot;exon&quot;, 
##     GTF.attrType = &quot;gene_id&quot;, chrAliases = NULL) 
## NULL</code></pre>
<p>This process takes some time to finish.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="read-mapping.html#cb108-1" tabindex="-1"></a><span class="co"># run the alignment on all of the trimmed_fastq_files</span></span>
<span id="cb108-2"><a href="read-mapping.html#cb108-2" tabindex="-1"></a><span class="fu">align</span>(<span class="at">index=</span>index_reference_genome, </span>
<span id="cb108-3"><a href="read-mapping.html#cb108-3" tabindex="-1"></a>      <span class="at">readfile1=</span>trimmed_fastq_files,</span>
<span id="cb108-4"><a href="read-mapping.html#cb108-4" tabindex="-1"></a>      <span class="at">type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb108-5"><a href="read-mapping.html#cb108-5" tabindex="-1"></a>      <span class="at">input_format =</span> <span class="st">&quot;gzFASTQ&quot;</span>,</span>
<span id="cb108-6"><a href="read-mapping.html#cb108-6" tabindex="-1"></a>      <span class="at">output_format =</span> <span class="st">&quot;BAM&quot;</span>,</span>
<span id="cb108-7"><a href="read-mapping.html#cb108-7" tabindex="-1"></a>      <span class="at">unique =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-8"><a href="read-mapping.html#cb108-8" tabindex="-1"></a>      <span class="at">nBestLocations =</span> <span class="dv">1</span>,</span>
<span id="cb108-9"><a href="read-mapping.html#cb108-9" tabindex="-1"></a>      <span class="at">sortReadsByCoordinates =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-10"><a href="read-mapping.html#cb108-10" tabindex="-1"></a>      <span class="at">nthreads=</span><span class="dv">6</span></span>
<span id="cb108-11"><a href="read-mapping.html#cb108-11" tabindex="-1"></a>      )</span></code></pre></div>
<p>The output of the alignment are bam corresponding to each fastq file.</p>
<p>We can get a summary of the proportion of reads that mapped to the reference genome using the propmapped function.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="read-mapping.html#cb109-1" tabindex="-1"></a><span class="co"># create an object in R listing</span></span>
<span id="cb109-2"><a href="read-mapping.html#cb109-2" tabindex="-1"></a>bam_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> dir_trimmed.fq_files, <span class="at">pattern =</span> <span class="st">&quot;.BAM$&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb109-3"><a href="read-mapping.html#cb109-3" tabindex="-1"></a>bam_files</span></code></pre></div>
<pre><code>##  [1] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP1_R1.fastq.gz.subread.BAM&quot;
##  [2] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP2_R1.fastq.gz.subread.BAM&quot;
##  [3] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP3_R1.fastq.gz.subread.BAM&quot;
##  [4] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP4_R1.fastq.gz.subread.BAM&quot;
##  [5] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP1_R1.fastq.gz.subread.BAM&quot;
##  [6] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP2_R1.fastq.gz.subread.BAM&quot;
##  [7] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP3_R1.fastq.gz.subread.BAM&quot;
##  [8] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP4_R1.fastq.gz.subread.BAM&quot;
##  [9] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP1_R1.fastq.gz.subread.BAM&quot;   
## [10] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP2_R1.fastq.gz.subread.BAM&quot;   
## [11] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP3_R1.fastq.gz.subread.BAM&quot;   
## [12] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP4_R1.fastq.gz.subread.BAM&quot;   
## [13] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP1_R1.fastq.gz.subread.BAM&quot;   
## [14] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP2_R1.fastq.gz.subread.BAM&quot;   
## [15] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP3_R1.fastq.gz.subread.BAM&quot;</code></pre>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="read-mapping.html#cb111-1" tabindex="-1"></a><span class="co"># find the proportion of reads that mapped for each sample</span></span>
<span id="cb111-2"><a href="read-mapping.html#cb111-2" tabindex="-1"></a>props <span class="ot">&lt;-</span> <span class="fu">propmapped</span>(<span class="at">files=</span>bam_files)</span>
<span id="cb111-3"><a href="read-mapping.html#cb111-3" tabindex="-1"></a></span>
<span id="cb111-4"><a href="read-mapping.html#cb111-4" tabindex="-1"></a>props <span class="sc">|&gt;</span> <span class="fu">print</span>()</span></code></pre></div>
<pre><code>##                                                NumTotal NumMapped PropMapped
## YPS606_MSN24_ETOH_REP1_R1.fastq.gz.subread.BAM        0         0        NaN
## YPS606_MSN24_ETOH_REP2_R1.fastq.gz.subread.BAM        0         0        NaN
## YPS606_MSN24_ETOH_REP3_R1.fastq.gz.subread.BAM   199076    173693   0.872496
## YPS606_MSN24_ETOH_REP4_R1.fastq.gz.subread.BAM   205792    178785   0.868766
## YPS606_MSN24_MOCK_REP1_R1.fastq.gz.subread.BAM   167075    143114   0.856585
## YPS606_MSN24_MOCK_REP2_R1.fastq.gz.subread.BAM   169754    146302   0.861847
## YPS606_MSN24_MOCK_REP3_R1.fastq.gz.subread.BAM   210001    178664   0.850777
## YPS606_MSN24_MOCK_REP4_R1.fastq.gz.subread.BAM   208329    177749   0.853213
## YPS606_WT_ETOH_REP1_R1.fastq.gz.subread.BAM      181587    159200   0.876715
## YPS606_WT_ETOH_REP2_R1.fastq.gz.subread.BAM      201551    176904   0.877713
## YPS606_WT_ETOH_REP3_R1.fastq.gz.subread.BAM      214745    188499   0.877781
## YPS606_WT_ETOH_REP4_R1.fastq.gz.subread.BAM      187319    164152   0.876323
## YPS606_WT_MOCK_REP1_R1.fastq.gz.subread.BAM      223539    193407   0.865205
## YPS606_WT_MOCK_REP2_R1.fastq.gz.subread.BAM      187469    161251   0.860148
## YPS606_WT_MOCK_REP3_R1.fastq.gz.subread.BAM           0         0        NaN</code></pre>
</div>
<div id="pseudomapping-with-salmon" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Pseudomapping with Salmon<a href="read-mapping.html#pseudomapping-with-salmon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Salmon is a widely used pseudomapper. It is not available to use in R, but we can use bash code chunks to run it in the same markdown document.</p>
<div id="create-conda-env" class="section level3 hasAnchor" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Create Conda Env<a href="read-mapping.html#create-conda-env" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we need to create a new conda environment for salmon.</p>
<p><strong>Depending on your computer, we might need to run this code in terminal.</strong></p>
<div class="sourceCode" id="cb113"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb113-1"><a href="read-mapping.html#cb113-1" tabindex="-1"></a><span class="co">## Warning, if you did not complete Working_with_Sequences.Rmd activity, </span></span>
<span id="cb113-2"><a href="read-mapping.html#cb113-2" tabindex="-1"></a><span class="co">#    your conda might not be set up correctly for this code.</span></span>
<span id="cb113-3"><a href="read-mapping.html#cb113-3" tabindex="-1"></a></span>
<span id="cb113-4"><a href="read-mapping.html#cb113-4" tabindex="-1"></a><span class="co"># create an environment for our pseudomapping with Salmon</span></span>
<span id="cb113-5"><a href="read-mapping.html#cb113-5" tabindex="-1"></a><span class="co"># this code is &quot;extra&quot; because it only creates env if not already existing.</span></span>
<span id="cb113-6"><a href="read-mapping.html#cb113-6" tabindex="-1"></a><span class="cf">if</span> <span class="ex">conda</span> info <span class="at">--envs</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-q</span> salmon<span class="kw">;</span> <span class="cf">then</span> <span class="bu">echo</span> <span class="st">&quot;environment &#39;salmon&#39; already exists&quot;</span><span class="kw">;</span> <span class="cf">else</span> <span class="va">CONDA_SUBDIR</span><span class="op">=</span>osx-64 <span class="ex">conda</span> create <span class="at">-y</span> <span class="at">-n</span> salmon <span class="at">-c</span> conda-forge <span class="at">-c</span> bioconda salmon=1.10.0<span class="kw">;</span> <span class="cf">fi</span></span>
<span id="cb113-7"><a href="read-mapping.html#cb113-7" tabindex="-1"></a><span class="co"># the channel priority order above is needed to get a recent version via conda.</span></span>
<span id="cb113-8"><a href="read-mapping.html#cb113-8" tabindex="-1"></a></span>
<span id="cb113-9"><a href="read-mapping.html#cb113-9" tabindex="-1"></a><span class="co"># see available conda environments</span></span>
<span id="cb113-10"><a href="read-mapping.html#cb113-10" tabindex="-1"></a><span class="ex">conda</span> env list</span>
<span id="cb113-11"><a href="read-mapping.html#cb113-11" tabindex="-1"></a></span>
<span id="cb113-12"><a href="read-mapping.html#cb113-12" tabindex="-1"></a><span class="co"># activate our QC environment</span></span>
<span id="cb113-13"><a href="read-mapping.html#cb113-13" tabindex="-1"></a><span class="ex">conda</span> activate salmon</span>
<span id="cb113-14"><a href="read-mapping.html#cb113-14" tabindex="-1"></a></span>
<span id="cb113-15"><a href="read-mapping.html#cb113-15" tabindex="-1"></a><span class="co"># make sure desired packages are working</span></span>
<span id="cb113-16"><a href="read-mapping.html#cb113-16" tabindex="-1"></a><span class="fu">which</span> salmon</span>
<span id="cb113-17"><a href="read-mapping.html#cb113-17" tabindex="-1"></a></span>
<span id="cb113-18"><a href="read-mapping.html#cb113-18" tabindex="-1"></a><span class="co"># help page for using salmon</span></span>
<span id="cb113-19"><a href="read-mapping.html#cb113-19" tabindex="-1"></a><span class="ex">salmon</span> <span class="at">-h</span></span>
<span id="cb113-20"><a href="read-mapping.html#cb113-20" tabindex="-1"></a></span>
<span id="cb113-21"><a href="read-mapping.html#cb113-21" tabindex="-1"></a><span class="co"># it&#39;s always good coding practice to deactivate </span></span>
<span id="cb113-22"><a href="read-mapping.html#cb113-22" tabindex="-1"></a><span class="co"># a conda environment at the end of a chunk</span></span>
<span id="cb113-23"><a href="read-mapping.html#cb113-23" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
</div>
<div id="download-transcriptome" class="section level3 hasAnchor" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> Download transcriptome<a href="read-mapping.html#download-transcriptome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To make an index for Salmon, we need transcript sequences in the FASTA format.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb114-1"><a href="read-mapping.html#cb114-1" tabindex="-1"></a><span class="co"># Define the destination file path</span></span>
<span id="cb114-2"><a href="read-mapping.html#cb114-2" tabindex="-1"></a><span class="co"># Be sure to change this file path to the path you want your data to go</span></span>
<span id="cb114-3"><a href="read-mapping.html#cb114-3" tabindex="-1"></a><span class="va">REF_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Reference&quot;</span></span>
<span id="cb114-4"><a href="read-mapping.html#cb114-4" tabindex="-1"></a></span>
<span id="cb114-5"><a href="read-mapping.html#cb114-5" tabindex="-1"></a><span class="co"># make that directory if it doesn&#39;t already</span></span>
<span id="cb114-6"><a href="read-mapping.html#cb114-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$REF_DIR</span></span>
<span id="cb114-7"><a href="read-mapping.html#cb114-7" tabindex="-1"></a></span>
<span id="cb114-8"><a href="read-mapping.html#cb114-8" tabindex="-1"></a><span class="co"># Define the URL of reference transcriptome</span></span>
<span id="cb114-9"><a href="read-mapping.html#cb114-9" tabindex="-1"></a><span class="co"># (latest from ensembl)</span></span>
<span id="cb114-10"><a href="read-mapping.html#cb114-10" tabindex="-1"></a><span class="va">url</span><span class="op">=</span><span class="st">&quot;ftp://ftp.ensembl.org/pub/release-110/fasta/saccharomyces_cerevisiae/cdna/Saccharomyces_cerevisiae.R64-1-1.cdna.all.fa.gz&quot;</span></span>
<span id="cb114-11"><a href="read-mapping.html#cb114-11" tabindex="-1"></a></span>
<span id="cb114-12"><a href="read-mapping.html#cb114-12" tabindex="-1"></a></span>
<span id="cb114-13"><a href="read-mapping.html#cb114-13" tabindex="-1"></a><span class="co"># Check if the file already exists at the destination location</span></span>
<span id="cb114-14"><a href="read-mapping.html#cb114-14" tabindex="-1"></a><span class="cf">if</span> <span class="bu">[</span> <span class="ot">!</span> <span class="ot">-f</span> <span class="st">&quot;</span><span class="va">$REF_DIR</span><span class="st">/Saccharomyces_cerevisiae.R64-1-1.cdna.all.fa.gz&quot;</span> <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb114-15"><a href="read-mapping.html#cb114-15" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;Reference transcriptome not found, downloading...&quot;</span></span>
<span id="cb114-16"><a href="read-mapping.html#cb114-16" tabindex="-1"></a>    <span class="co"># If the file does not exist, download it using curl</span></span>
<span id="cb114-17"><a href="read-mapping.html#cb114-17" tabindex="-1"></a>    <span class="ex">curl</span> <span class="at">-o</span> <span class="st">&quot;</span><span class="va">$REF_DIR</span><span class="st">/Saccharomyces_cerevisiae.R64-1-1.cdna.all.fa.gz&quot;</span> <span class="st">&quot;</span><span class="va">$url</span><span class="st">&quot;</span></span>
<span id="cb114-18"><a href="read-mapping.html#cb114-18" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;Downloading finished&quot;</span></span>
<span id="cb114-19"><a href="read-mapping.html#cb114-19" tabindex="-1"></a><span class="cf">else</span></span>
<span id="cb114-20"><a href="read-mapping.html#cb114-20" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;File already exists at </span><span class="va">$REF_DIR</span><span class="st"> Skipping download.&quot;</span></span>
<span id="cb114-21"><a href="read-mapping.html#cb114-21" tabindex="-1"></a><span class="cf">fi</span></span></code></pre></div>
<pre><code>## File already exists at /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference Skipping download.</code></pre>
</div>
<div id="building-the-salmon-index" class="section level3 hasAnchor" number="4.4.3">
<h3><span class="header-section-number">4.4.3</span> Building the Salmon index<a href="read-mapping.html#building-the-salmon-index" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Salmon can index by using the command <code>salmon index</code>. A recent feature update to Salmon includes an option to map to decoys, we will use the entire genome as the decoy for our index, because the <em>S. cerevesiae</em> genome is small. You can read more at: <a href="https://salmon.readthedocs.io/en/latest/salmon.html#preparing-transcriptome-indices-mapping-based-mode" class="uri">https://salmon.readthedocs.io/en/latest/salmon.html#preparing-transcriptome-indices-mapping-based-mode</a>.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb116-1"><a href="read-mapping.html#cb116-1" tabindex="-1"></a><span class="co"># We need to set a variable for where the transcriptome file is</span></span>
<span id="cb116-2"><a href="read-mapping.html#cb116-2" tabindex="-1"></a><span class="va">REF_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Reference&quot;</span></span>
<span id="cb116-3"><a href="read-mapping.html#cb116-3" tabindex="-1"></a><span class="va">TRANSCRIPTOME</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Reference/Saccharomyces_cerevisiae.R64-1-1.cdna.all.fa.gz&quot;</span></span>
<span id="cb116-4"><a href="read-mapping.html#cb116-4" tabindex="-1"></a><span class="va">GENOME</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Reference/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz&quot;</span></span>
<span id="cb116-5"><a href="read-mapping.html#cb116-5" tabindex="-1"></a></span>
<span id="cb116-6"><a href="read-mapping.html#cb116-6" tabindex="-1"></a><span class="co"># Activate conda salmon environment</span></span>
<span id="cb116-7"><a href="read-mapping.html#cb116-7" tabindex="-1"></a><span class="ex">conda</span> activate salmon</span>
<span id="cb116-8"><a href="read-mapping.html#cb116-8" tabindex="-1"></a></span>
<span id="cb116-9"><a href="read-mapping.html#cb116-9" tabindex="-1"></a><span class="co"># Run a script that generates a decoy.txt file from the genome we downloaded</span></span>
<span id="cb116-10"><a href="read-mapping.html#cb116-10" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&quot;^&gt;&quot;</span> <span class="op">&lt;(</span><span class="fu">gunzip</span> <span class="at">-c</span> <span class="va">$GENOME</span><span class="op">)</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot; &quot;</span> <span class="at">-f</span> 1 <span class="op">&gt;</span> <span class="va">$REF_DIR</span>/decoys.txt</span>
<span id="cb116-11"><a href="read-mapping.html#cb116-11" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i.bak</span> <span class="at">-e</span> <span class="st">&#39;s/&gt;//g&#39;</span> <span class="va">$REF_DIR</span>/decoys.txt</span>
<span id="cb116-12"><a href="read-mapping.html#cb116-12" tabindex="-1"></a></span>
<span id="cb116-13"><a href="read-mapping.html#cb116-13" tabindex="-1"></a><span class="co"># Combine the transcriptome and genome into a single file for indexing</span></span>
<span id="cb116-14"><a href="read-mapping.html#cb116-14" tabindex="-1"></a><span class="fu">cat</span> <span class="va">$TRANSCRIPTOME</span> <span class="va">$GENOME</span> <span class="op">&gt;</span> <span class="va">$REF_DIR</span>/gentrome.fasta.gz</span>
<span id="cb116-15"><a href="read-mapping.html#cb116-15" tabindex="-1"></a></span>
<span id="cb116-16"><a href="read-mapping.html#cb116-16" tabindex="-1"></a></span>
<span id="cb116-17"><a href="read-mapping.html#cb116-17" tabindex="-1"></a><span class="co"># We will use the yeast, but it needs to be indexed by salmon</span></span>
<span id="cb116-18"><a href="read-mapping.html#cb116-18" tabindex="-1"></a><span class="ex">salmon</span> index <span class="at">-t</span> <span class="va">$REF_DIR</span>/gentrome.fasta.gz <span class="at">-d</span> <span class="va">$REF_DIR</span>/decoys.txt <span class="at">-p</span> 4 <span class="at">-i</span> <span class="va">$REF_DIR</span>/index_salmon_Saccharomyces_cerevisiae.R64-1-1</span>
<span id="cb116-19"><a href="read-mapping.html#cb116-19" tabindex="-1"></a></span>
<span id="cb116-20"><a href="read-mapping.html#cb116-20" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
<pre><code>## Version Info: This is the most recent version of salmon.
## [2023-10-26 12:12:52.539] [jLog] [info] building index
## out : /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1
## [2023-10-26 12:12:52.540] [puff::index::jointLog] [info] Running fixFasta
## 
## [Step 1 of 4] : counting k-mers
## 
## [2023-10-26 12:12:53.052] [puff::index::jointLog] [warning] Removed 41 transcripts that were sequence duplicates of indexed transcripts.
## [2023-10-26 12:12:53.052] [puff::index::jointLog] [warning] If you wish to retain duplicate transcripts, please use the `--keepDuplicates` flag
## [2023-10-26 12:12:53.053] [puff::index::jointLog] [info] Replaced 0 non-ATCG nucleotides
## [2023-10-26 12:12:53.053] [puff::index::jointLog] [info] Clipped poly-A tails from 0 transcripts
## wrote 6588 cleaned references
## [2023-10-26 12:12:53.150] [puff::index::jointLog] [info] Filter size not provided; estimating from number of distinct k-mers
## [2023-10-26 12:12:53.603] [puff::index::jointLog] [info] ntHll estimated 11513300 distinct k-mers, setting filter size to 2^28
## Threads = 4
## Vertex length = 31
## Hash functions = 5
## Filter size = 268435456
## Capacity = 2
## Files: 
## /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1/ref_k31_fixed.fa
## --------------------------------------------------------------------------------
## Round 0, 0:268435456
## Pass Filling Filtering
## 1    2   3   
## 2    1   0
## True junctions count = 20631
## False junctions count = 49701
## Hash table size = 70332
## Candidate marks count = 194448
## --------------------------------------------------------------------------------
## Reallocating bifurcations time: 0
## True marks count: 93214
## Edges construction time: 1
## --------------------------------------------------------------------------------
## Distinct junctions = 20631
## 
## TwoPaCo::buildGraphMain:: allocated with scalable_malloc; freeing.
## TwoPaCo::buildGraphMain:: Calling scalable_allocation_command(TBBMALLOC_CLEAN_ALL_BUFFERS, 0);
## allowedIn: 14
## Max Junction ID: 20809
## seen.size():166481 kmerInfo.size():20810
## approximateContigTotalLength: 11070364
## counters for complex kmers:
## (prec&gt;1 &amp; succ&gt;1)=327 | (succ&gt;1 &amp; isStart)=7 | (prec&gt;1 &amp; isEnd)=7 | (isStart &amp; isEnd)=2
## contig count: 25029 element count: 12321058 complex nodes: 343
## # of ones in rank vector: 25028
## [2023-10-26 12:13:01.661] [puff::index::jointLog] [info] Starting the Pufferfish indexing by reading the GFA binary file.
## [2023-10-26 12:13:01.664] [puff::index::jointLog] [info] Setting the index/BinaryGfa directory /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1
## size = 12321058
## -----------------------------------------
## | Loading contigs | Time = 5.9218 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading contig boundaries | Time = 3.1299 ms
## -----------------------------------------
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## 25028
## [2023-10-26 12:13:01.699] [puff::index::jointLog] [info] Done wrapping the rank vector with a rank9sel structure.
## [2023-10-26 12:13:01.700] [puff::index::jointLog] [info] contig count for validation: 25,028
## [2023-10-26 12:13:01.708] [puff::index::jointLog] [info] Total # of Contigs : 25,028
## [2023-10-26 12:13:01.708] [puff::index::jointLog] [info] Total # of numerical Contigs : 25,028
## [2023-10-26 12:13:01.708] [puff::index::jointLog] [info] Total # of contig vec entries: 88,496
## [2023-10-26 12:13:01.708] [puff::index::jointLog] [info] bits per offset entry 17
## [2023-10-26 12:13:01.709] [puff::index::jointLog] [info] Done constructing the contig vector. 25029
## [2023-10-26 12:13:01.721] [puff::index::jointLog] [info] # segments = 25,028
## [2023-10-26 12:13:01.721] [puff::index::jointLog] [info] total length = 12,321,058
## [2023-10-26 12:13:01.723] [puff::index::jointLog] [info] Reading the reference files ...
## [2023-10-26 12:13:01.849] [puff::index::jointLog] [info] positional integer width = 24
## [2023-10-26 12:13:01.849] [puff::index::jointLog] [info] seqSize = 12,321,058
## [2023-10-26 12:13:01.849] [puff::index::jointLog] [info] rankSize = 12,321,058
## [2023-10-26 12:13:01.849] [puff::index::jointLog] [info] edgeVecSize = 0
## [2023-10-26 12:13:01.849] [puff::index::jointLog] [info] num keys = 11,570,218
## 
[Building BooPHF]  0.319%   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  0.425%   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  0.433%   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  0.456%   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  0.668%   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  0.71 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  0.79 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.03 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.09 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.09 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.26 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.26 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.51 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.52 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.61 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.69 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.95 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  1.99 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2    %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.04 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.2  %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.39 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.43 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.57 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.67 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.75 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.95 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  2.99 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.01 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.06 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.37 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.39 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.41 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.52 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.62 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.7  %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.9  %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  3.94 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.05 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.17 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.28 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.35 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.4  %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.61 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.71 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.78 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  4.83 %   elapsed:   0 min 0  sec   remaining:   0 min 1  sec
[Building BooPHF]  5.03 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.05 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.1  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.26 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.36 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.47 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.54 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.59 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.76 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.93 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  5.97 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.02 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.2  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.25 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.3  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.45 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.51 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.69 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.73 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.77 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  6.95 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.12 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.16 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.21 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.39 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.44 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.48 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.64 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.7  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.88 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.92 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  7.97 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.15 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.28 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.35 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.38 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.62 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.66 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.67 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.92 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  8.92 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.02 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.18 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.26 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.33 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.36 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.59 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.66 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.77 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  9.9  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  10.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  11.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  12.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  13.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  14.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  15.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  16.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  17.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  18.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  19.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  20.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  21.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  22.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  23.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  24.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  25.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  26.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  27.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  28.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  29.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  30.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  31.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  32.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  33.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  34.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  35.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  36.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  37.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  38.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  39.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  40.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  41.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  42.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  43.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  44.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  45.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  46.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  47.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  48.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  49.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  50.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  51.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  52.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  53.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  54.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  55.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  56.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  57.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  58.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  59.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  60.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  61.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  62.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  63.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  64.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  65.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  66.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  67.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  68.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  69.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  70.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  71.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  72.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  73.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  74.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  75.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  76.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  77.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  78.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  79.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  80.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  81.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  82.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  83.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  84.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  85.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  86.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  87.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  88.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  89.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  90.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  91.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.5 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.7 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  92.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.2 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.6 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.8 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  93.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  94   %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.1 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.3 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.4 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.6 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.6 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.9 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  94.9 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.1 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.1 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.2 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.4 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.5 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.6 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.7 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  95.9 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.1 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.2 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.2 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.4 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.5 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.5 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.5 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.7 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.9 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  96.9 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97   %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.2 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.2 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.4 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.5 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.5 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.7 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  97.9 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.2 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.2 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.2 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.3 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.5 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.6 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.6 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  98.9 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99   %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.1 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.3 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.3 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.5 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.6 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.6 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.8 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  99.9 %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
[Building BooPHF]  100  %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec
## [2023-10-26 12:13:02.402] [puff::index::jointLog] [info] mphf size = 7.22767 MB
## [2023-10-26 12:13:02.411] [puff::index::jointLog] [info] chunk size = 3,080,265
## [2023-10-26 12:13:02.411] [puff::index::jointLog] [info] chunk 0 = [0, 3,080,282)
## [2023-10-26 12:13:02.411] [puff::index::jointLog] [info] chunk 1 = [3,080,282, 6,160,547)
## [2023-10-26 12:13:02.411] [puff::index::jointLog] [info] chunk 2 = [6,160,547, 9,240,812)
## [2023-10-26 12:13:02.411] [puff::index::jointLog] [info] chunk 3 = [9,240,812, 12,321,028)
## [2023-10-26 12:13:02.916] [puff::index::jointLog] [info] finished populating pos vector
## [2023-10-26 12:13:02.916] [puff::index::jointLog] [info] writing index components
## [2023-10-26 12:13:03.170] [puff::index::jointLog] [info] finished writing dense pufferfish index
## [2023-10-26 12:13:03.174] [jLog] [info] done building index
## for info, total work write each  : 2.331    total work inram from level 3 : 4.322  total work raw : 25.000 
## Bitarray        60630080  bits (100.00 %)   (array + ranks )
## final hash             0  bits (0.00 %) (nb in final hash 0)</code></pre>
<p>Notice that we combined the fasta file of the transcriptome with the fasta file of the entire genome (in that order) into the gentrome.fasta.gz file which was then indexed.</p>
<p>Salmon is a pseudomapper, so it doesn’t create sam/bam files and is instead able to count directly from the fastq files. We will do the pseudomapping and counting all in one step in the next activity.</p>
</div>
</div>
<div id="questions-1" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Questions<a href="read-mapping.html#questions-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="with-rsubread" class="section level3 hasAnchor" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> With Rsubread:<a href="read-mapping.html#with-rsubread" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Question 1: Try aligning the fastq files allowing multi-mapping reads (set unique = FALSE), allowing for up to 6 “best” locations to be reported (nBestLocations = 6), and allow reads to be fractionally counted (fraction = TRUE). Specify the output file names (bam_files_multi) by substituting “.fastq.gz” with “.multi.bam” so we don’t overwrite our unique alignment bam files.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="read-mapping.html#cb118-1" tabindex="-1"></a><span class="co"># Define the pattern and replacement</span></span>
<span id="cb118-2"><a href="read-mapping.html#cb118-2" tabindex="-1"></a>pattern <span class="ot">&lt;-</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.fastq</span><span class="sc">\\</span><span class="st">.gz$&quot;</span></span>
<span id="cb118-3"><a href="read-mapping.html#cb118-3" tabindex="-1"></a>replacement <span class="ot">&lt;-</span> <span class="st">&quot;subread.multi.bam&quot;</span></span>
<span id="cb118-4"><a href="read-mapping.html#cb118-4" tabindex="-1"></a></span>
<span id="cb118-5"><a href="read-mapping.html#cb118-5" tabindex="-1"></a><span class="co"># Create the new file names</span></span>
<span id="cb118-6"><a href="read-mapping.html#cb118-6" tabindex="-1"></a>bam_files_multi <span class="ot">&lt;-</span> <span class="fu">gsub</span>(pattern, replacement, trimmed_fastq_files)</span>
<span id="cb118-7"><a href="read-mapping.html#cb118-7" tabindex="-1"></a></span>
<span id="cb118-8"><a href="read-mapping.html#cb118-8" tabindex="-1"></a><span class="co"># update this code to run with Rsubread multimapping, as described above.</span></span>
<span id="cb118-9"><a href="read-mapping.html#cb118-9" tabindex="-1"></a><span class="fu">align</span>(<span class="at">index=</span>index_reference_genome, </span>
<span id="cb118-10"><a href="read-mapping.html#cb118-10" tabindex="-1"></a>      <span class="at">readfile1=</span>trimmed_fastq_files,</span>
<span id="cb118-11"><a href="read-mapping.html#cb118-11" tabindex="-1"></a>      <span class="at">output_file =</span> ___________,</span>
<span id="cb118-12"><a href="read-mapping.html#cb118-12" tabindex="-1"></a>      <span class="at">type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb118-13"><a href="read-mapping.html#cb118-13" tabindex="-1"></a>      <span class="at">input_format =</span> <span class="st">&quot;gzFASTQ&quot;</span>,</span>
<span id="cb118-14"><a href="read-mapping.html#cb118-14" tabindex="-1"></a>      <span class="at">output_format =</span> <span class="st">&quot;BAM&quot;</span>,</span>
<span id="cb118-15"><a href="read-mapping.html#cb118-15" tabindex="-1"></a>      <span class="at">unique =</span> ____,</span>
<span id="cb118-16"><a href="read-mapping.html#cb118-16" tabindex="-1"></a>      <span class="at">nBestLocations =</span> ____,</span>
<span id="cb118-17"><a href="read-mapping.html#cb118-17" tabindex="-1"></a>      <span class="at">nthreads=</span><span class="dv">6</span></span>
<span id="cb118-18"><a href="read-mapping.html#cb118-18" tabindex="-1"></a>      )</span></code></pre></div>
<p>Question 2: Look at the proportion of reads mapped and see if we get any more reads mapping by specifying a less stringent criteria.</p>
</div>
<div id="with-salmon" class="section level3 hasAnchor" number="4.5.2">
<h3><span class="header-section-number">4.5.2</span> With Salmon:<a href="read-mapping.html#with-salmon" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Question 3: What are the pros and cons of using Salmon vs subread for mapping reads?</p>
<p>Be sure to knit this file into a pdf or html file once you’re finished.</p>
<p>System information for reproducibility:</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="read-mapping.html#cb119-1" tabindex="-1"></a>pander<span class="sc">::</span><span class="fu">pander</span>(<span class="fu">sessionInfo</span>())</span></code></pre></div>
<p><strong>R version 4.3.1 (2023-06-16)</strong></p>
<p><strong>Platform:</strong> aarch64-apple-darwin20 (64-bit)</p>
<p><strong>locale:</strong>
en_US.UTF-8||en_US.UTF-8||en_US.UTF-8||C||en_US.UTF-8||en_US.UTF-8</p>
<p><strong>attached base packages:</strong>
<em>stats4</em>, <em>stats</em>, <em>graphics</em>, <em>grDevices</em>, <em>utils</em>, <em>datasets</em>, <em>methods</em> and <em>base</em></p>
<p><strong>other attached packages:</strong>
<em>Rsubread(v.2.14.2)</em>, <em>ShortRead(v.1.58.0)</em>, <em>GenomicAlignments(v.1.36.0)</em>, <em>SummarizedExperiment(v.1.30.2)</em>, <em>MatrixGenerics(v.1.12.3)</em>, <em>matrixStats(v.1.0.0)</em>, <em>Rsamtools(v.2.16.0)</em>, <em>GenomicRanges(v.1.52.1)</em>, <em>Biostrings(v.2.68.1)</em>, <em>GenomeInfoDb(v.1.36.4)</em>, <em>XVector(v.0.40.0)</em>, <em>BiocParallel(v.1.34.2)</em>, <em>Rfastp(v.1.10.0)</em>, <em>org.Sc.sgd.db(v.3.17.0)</em>, <em>AnnotationDbi(v.1.62.2)</em>, <em>IRanges(v.2.34.1)</em>, <em>S4Vectors(v.0.38.2)</em>, <em>Biobase(v.2.60.0)</em>, <em>BiocGenerics(v.0.46.0)</em>, <em>clusterProfiler(v.4.8.2)</em>, <em>ggVennDiagram(v.1.2.3)</em>, <em>tidytree(v.0.4.5)</em>, <em>igraph(v.1.5.1)</em>, <em>janitor(v.2.2.0)</em>, <em>BiocManager(v.1.30.22)</em>, <em>pander(v.0.6.5)</em>, <em>knitr(v.1.44)</em>, <em>here(v.1.0.1)</em>, <em>lubridate(v.1.9.3)</em>, <em>forcats(v.1.0.0)</em>, <em>stringr(v.1.5.0)</em>, <em>dplyr(v.1.1.3)</em>, <em>purrr(v.1.0.2)</em>, <em>readr(v.2.1.4)</em>, <em>tidyr(v.1.3.0)</em>, <em>tibble(v.3.2.1)</em>, <em>ggplot2(v.3.4.4)</em>, <em>tidyverse(v.2.0.0)</em> and <em>pacman(v.0.5.1)</em></p>
<p><strong>loaded via a namespace (and not attached):</strong>
<em>RColorBrewer(v.1.1-3)</em>, <em>rstudioapi(v.0.15.0)</em>, <em>jsonlite(v.1.8.7)</em>, <em>magrittr(v.2.0.3)</em>, <em>farver(v.2.1.1)</em>, <em>rmarkdown(v.2.25)</em>, <em>fs(v.1.6.3)</em>, <em>zlibbioc(v.1.46.0)</em>, <em>vctrs(v.0.6.4)</em>, <em>memoise(v.2.0.1)</em>, <em>RCurl(v.1.98-1.12)</em>, <em>ggtree(v.3.8.2)</em>, <em>S4Arrays(v.1.0.6)</em>, <em>htmltools(v.0.5.6.1)</em>, <em>gridGraphics(v.0.5-1)</em>, <em>sass(v.0.4.7)</em>, <em>bslib(v.0.5.1)</em>, <em>plyr(v.1.8.9)</em>, <em>cachem(v.1.0.8)</em>, <em>lifecycle(v.1.0.3)</em>, <em>pkgconfig(v.2.0.3)</em>, <em>Matrix(v.1.6-1.1)</em>, <em>R6(v.2.5.1)</em>, <em>fastmap(v.1.1.1)</em>, <em>gson(v.0.1.0)</em>, <em>GenomeInfoDbData(v.1.2.10)</em>, <em>snakecase(v.0.11.1)</em>, <em>digest(v.0.6.33)</em>, <em>aplot(v.0.2.2)</em>, <em>enrichplot(v.1.20.0)</em>, <em>colorspace(v.2.1-0)</em>, <em>patchwork(v.1.1.3)</em>, <em>rprojroot(v.2.0.3)</em>, <em>RSQLite(v.2.3.1)</em>, <em>hwriter(v.1.3.2.1)</em>, <em>fansi(v.1.0.5)</em>, <em>timechange(v.0.2.0)</em>, <em>abind(v.1.4-5)</em>, <em>httr(v.1.4.7)</em>, <em>polyclip(v.1.10-6)</em>, <em>compiler(v.4.3.1)</em>, <em>bit64(v.4.0.5)</em>, <em>withr(v.2.5.1)</em>, <em>downloader(v.0.4)</em>, <em>viridis(v.0.6.4)</em>, <em>DBI(v.1.1.3)</em>, <em>ggforce(v.0.4.1)</em>, <em>MASS(v.7.3-60)</em>, <em>DelayedArray(v.0.26.7)</em>, <em>rjson(v.0.2.21)</em>, <em>HDO.db(v.0.99.1)</em>, <em>tools(v.4.3.1)</em>, <em>ape(v.5.7-1)</em>, <em>scatterpie(v.0.2.1)</em>, <em>glue(v.1.6.2)</em>, <em>nlme(v.3.1-163)</em>, <em>GOSemSim(v.2.26.1)</em>, <em>grid(v.4.3.1)</em>, <em>shadowtext(v.0.1.2)</em>, <em>reshape2(v.1.4.4)</em>, <em>fgsea(v.1.26.0)</em>, <em>generics(v.0.1.3)</em>, <em>gtable(v.0.3.4)</em>, <em>tzdb(v.0.4.0)</em>, <em>data.table(v.1.14.8)</em>, <em>hms(v.1.1.3)</em>, <em>tidygraph(v.1.2.3)</em>, <em>utf8(v.1.2.3)</em>, <em>ggrepel(v.0.9.4)</em>, <em>pillar(v.1.9.0)</em>, <em>yulab.utils(v.0.1.0)</em>, <em>splines(v.4.3.1)</em>, <em>tweenr(v.2.0.2)</em>, <em>treeio(v.1.24.3)</em>, <em>lattice(v.0.21-9)</em>, <em>deldir(v.1.0-9)</em>, <em>bit(v.4.0.5)</em>, <em>tidyselect(v.1.2.0)</em>, <em>GO.db(v.3.17.0)</em>, <em>gridExtra(v.2.3)</em>, <em>bookdown(v.0.36)</em>, <em>xfun(v.0.40)</em>, <em>graphlayouts(v.1.0.1)</em>, <em>stringi(v.1.7.12)</em>, <em>lazyeval(v.0.2.2)</em>, <em>ggfun(v.0.1.3)</em>, <em>yaml(v.2.3.7)</em>, <em>evaluate(v.0.22)</em>, <em>codetools(v.0.2-19)</em>, <em>interp(v.1.1-4)</em>, <em>ggraph(v.2.1.0)</em>, <em>qvalue(v.2.32.0)</em>, <em>RVenn(v.1.1.0)</em>, <em>ggplotify(v.0.1.2)</em>, <em>cli(v.3.6.1)</em>, <em>munsell(v.0.5.0)</em>, <em>jquerylib(v.0.1.4)</em>, <em>Rcpp(v.1.0.11)</em>, <em>png(v.0.1-8)</em>, <em>parallel(v.4.3.1)</em>, <em>blob(v.1.2.4)</em>, <em>jpeg(v.0.1-10)</em>, <em>latticeExtra(v.0.6-30)</em>, <em>DOSE(v.3.26.1)</em>, <em>bitops(v.1.0-7)</em>, <em>viridisLite(v.0.4.2)</em>, <em>scales(v.1.2.1)</em>, <em>crayon(v.1.5.2)</em>, <em>rlang(v.1.1.1)</em>, <em>cowplot(v.1.1.1)</em>, <em>fastmatch(v.1.1-4)</em> and <em>KEGGREST(v.1.40.1)</em></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="working-with-sequences-raw-data-quality-control.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="read-counting.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/04-Read_Mapping.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
