<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Read Counting | Genomic Data Analysis Course Exercises</title>
  <meta name="description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Read Counting | Genomic Data Analysis Course Exercises" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="github-repo" content="clstacy/GenomicDataAnalysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Read Counting | Genomic Data Analysis Course Exercises" />
  
  <meta name="twitter:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  

<meta name="author" content="Carson Stacy &amp; Jeffrey Lewis" />


<meta name="date" content="2023-10-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="read-mapping.html"/>
<link rel="next" href="differential-expression-edger.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/core-js-2.5.3/shim.min.js"></script>
<script src="libs/react-18.2.0/react.min.js"></script>
<script src="libs/react-18.2.0/react-dom.min.js"></script>
<script src="libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="libs/reactable-binding-0.4.4/reactable.js"></script>
<link href="libs/datatables-1/datatables.css" rel="stylesheet" />
<script src="libs/datatables-1/datatables.min.js"></script>
<link href="libs/vega-1/vega_plots.css" rel="stylesheet" />
<script src="libs/vega-1/vega.min.js"></script>
<link href="libs/vega-tooltip-1/vega-tooltip.css" rel="stylesheet" />
<script src="libs/vega-tooltip-1/vega-tooltip.js"></script>
<script src="libs/FileSaver-1/FileSaver.js"></script>
<script src="libs/GlimmaV2-1/XYSpecs.js"></script>
<script src="libs/GlimmaV2-1/MDSSpecs.js"></script>
<script src="libs/GlimmaV2-1/expressionSpec.js"></script>
<script src="libs/GlimmaV2-1/makeTooltip.js"></script>
<script src="libs/GlimmaV2-1/saveTools.js"></script>
<script src="libs/glimmaXY-binding-2.10.0/glimmaXY.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Genomic Data Analysis Exercises</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i><b>0.1</b> Usage</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#key-features"><i class="fa fa-check"></i><b>0.2</b> <strong>Key Features</strong></a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#disclaimer"><i class="fa fa-check"></i><b>0.3</b> Disclaimer</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html"><i class="fa fa-check"></i><b>1</b> Getting Started in R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exercise-description"><i class="fa fa-check"></i><b>1.1</b> Exercise Description</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#learning-outcomes"><i class="fa fa-check"></i><b>1.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#using-r-and-rstudio"><i class="fa fa-check"></i><b>1.3</b> Using R and RStudio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#load-data-directly-from-the-url"><i class="fa fa-check"></i><b>1.4</b> Load data directly from the URL</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#working-with-data-in-r"><i class="fa fa-check"></i><b>1.5</b> Working with data in R</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#looking-at-data-in-rstudio"><i class="fa fa-check"></i><b>1.6</b> Looking at Data in RStudio</a></li>
<li class="chapter" data-level="1.7" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exploring-the-data"><i class="fa fa-check"></i><b>1.7</b> Exploring the data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gene-ontology.html"><a href="gene-ontology.html"><i class="fa fa-check"></i><b>2</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gene-ontology.html"><a href="gene-ontology.html#description"><i class="fa fa-check"></i><b>2.1</b> Description</a></li>
<li class="chapter" data-level="2.2" data-path="gene-ontology.html"><a href="gene-ontology.html#learning-outcomes-1"><i class="fa fa-check"></i><b>2.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="2.3" data-path="gene-ontology.html"><a href="gene-ontology.html#analysis-workflow"><i class="fa fa-check"></i><b>2.3</b> Analysis Workflow</a></li>
<li class="chapter" data-level="2.4" data-path="gene-ontology.html"><a href="gene-ontology.html#get-de-gene-list"><i class="fa fa-check"></i><b>2.4</b> Get DE gene list</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="gene-ontology.html"><a href="gene-ontology.html#saving-ggplot-output-to-a-file"><i class="fa fa-check"></i><b>2.4.1</b> Saving ggplot output to a file</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="gene-ontology.html"><a href="gene-ontology.html#the-hypergeometric-distribution-in-practice"><i class="fa fa-check"></i><b>2.5</b> The Hypergeometric Distribution in practice</a></li>
<li class="chapter" data-level="2.6" data-path="gene-ontology.html"><a href="gene-ontology.html#now-it-is-your-turn"><i class="fa fa-check"></i><b>2.6</b> Now it is your turn</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="gene-ontology.html"><a href="gene-ontology.html#run-enrichment"><i class="fa fa-check"></i><b>2.6.1</b> Run Enrichment</a></li>
<li class="chapter" data-level="2.6.2" data-path="gene-ontology.html"><a href="gene-ontology.html#see-the-data"><i class="fa fa-check"></i><b>2.6.2</b> see the data</a></li>
<li class="chapter" data-level="2.6.3" data-path="gene-ontology.html"><a href="gene-ontology.html#create-plots"><i class="fa fa-check"></i><b>2.6.3</b> create plots</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="gene-ontology.html"><a href="gene-ontology.html#questions"><i class="fa fa-check"></i><b>2.7</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html"><i class="fa fa-check"></i><b>3</b> Working with Sequences: Raw Data &amp; Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#description-1"><i class="fa fa-check"></i><b>3.1</b> Description</a></li>
<li class="chapter" data-level="3.2" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#learning-outcomes-2"><i class="fa fa-check"></i><b>3.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="3.3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#download-fastq"><i class="fa fa-check"></i><b>3.3</b> Download fastq</a></li>
<li class="chapter" data-level="3.4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#examining-fastq"><i class="fa fa-check"></i><b>3.4</b> Examining fastq</a></li>
<li class="chapter" data-level="3.5" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#trimming"><i class="fa fa-check"></i><b>3.5</b> Trimming</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#automate-for-list-of-files"><i class="fa fa-check"></i><b>3.5.1</b> Automate for list of files</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#batch-file-processing"><i class="fa fa-check"></i><b>3.6</b> Batch file processing</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#running-rfastp-creates-several-files"><i class="fa fa-check"></i><b>3.6.1</b> Running RfastP creates several files:</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#qc-and-adapters"><i class="fa fa-check"></i><b>3.7</b> QC and adapters</a></li>
<li class="chapter" data-level="3.8" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#running-fastqc"><i class="fa fa-check"></i><b>3.8</b> Running fastqc</a></li>
<li class="chapter" data-level="3.9" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#multiqc-for-qc-on-mutliple-samples"><i class="fa fa-check"></i><b>3.9</b> Multiqc for QC on mutliple samples</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="read-mapping.html"><a href="read-mapping.html"><i class="fa fa-check"></i><b>4</b> Read Mapping</a>
<ul>
<li class="chapter" data-level="4.1" data-path="read-mapping.html"><a href="read-mapping.html#alignment"><i class="fa fa-check"></i><b>4.1</b> Alignment</a></li>
<li class="chapter" data-level="4.2" data-path="read-mapping.html"><a href="read-mapping.html#retrieve-the-genome"><i class="fa fa-check"></i><b>4.2</b> Retrieve the genome</a></li>
<li class="chapter" data-level="4.3" data-path="read-mapping.html"><a href="read-mapping.html#build-rsubread-index"><i class="fa fa-check"></i><b>4.3</b> Build Rsubread Index</a></li>
<li class="chapter" data-level="4.4" data-path="read-mapping.html"><a href="read-mapping.html#pseudomapping-with-salmon"><i class="fa fa-check"></i><b>4.4</b> Pseudomapping with Salmon</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="read-mapping.html"><a href="read-mapping.html#create-conda-env"><i class="fa fa-check"></i><b>4.4.1</b> Create Conda Env</a></li>
<li class="chapter" data-level="4.4.2" data-path="read-mapping.html"><a href="read-mapping.html#download-transcriptome"><i class="fa fa-check"></i><b>4.4.2</b> Download transcriptome</a></li>
<li class="chapter" data-level="4.4.3" data-path="read-mapping.html"><a href="read-mapping.html#building-the-salmon-index"><i class="fa fa-check"></i><b>4.4.3</b> Building the Salmon index</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="read-mapping.html"><a href="read-mapping.html#questions-1"><i class="fa fa-check"></i><b>4.5</b> Questions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="read-mapping.html"><a href="read-mapping.html#with-rsubread"><i class="fa fa-check"></i><b>4.5.1</b> With Rsubread:</a></li>
<li class="chapter" data-level="4.5.2" data-path="read-mapping.html"><a href="read-mapping.html#with-salmon"><i class="fa fa-check"></i><b>4.5.2</b> With Salmon:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="read-counting.html"><a href="read-counting.html"><i class="fa fa-check"></i><b>5</b> Read Counting</a>
<ul>
<li class="chapter" data-level="5.1" data-path="read-counting.html"><a href="read-counting.html#featurecounts"><i class="fa fa-check"></i><b>5.1</b> featureCounts</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="read-counting.html"><a href="read-counting.html#locate-bam-files"><i class="fa fa-check"></i><b>5.1.1</b> Locate BAM files</a></li>
<li class="chapter" data-level="5.1.2" data-path="read-counting.html"><a href="read-counting.html#retrieve-the-genome-annotation"><i class="fa fa-check"></i><b>5.1.2</b> Retrieve the genome annotation</a></li>
<li class="chapter" data-level="5.1.3" data-path="read-counting.html"><a href="read-counting.html#counting-with-featurecounts"><i class="fa fa-check"></i><b>5.1.3</b> Counting with FeatureCounts</a></li>
<li class="chapter" data-level="5.1.4" data-path="read-counting.html"><a href="read-counting.html#counts-object"><i class="fa fa-check"></i><b>5.1.4</b> Counts object</a></li>
<li class="chapter" data-level="5.1.5" data-path="read-counting.html"><a href="read-counting.html#saving-fc-object-for-future-use"><i class="fa fa-check"></i><b>5.1.5</b> Saving <code>fc</code> object for future use</a></li>
<li class="chapter" data-level="5.1.6" data-path="read-counting.html"><a href="read-counting.html#rsubread-qc"><i class="fa fa-check"></i><b>5.1.6</b> RSubread QC</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="read-counting.html"><a href="read-counting.html#salmon"><i class="fa fa-check"></i><b>5.2</b> Salmon</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="read-counting.html"><a href="read-counting.html#pseudomapping-counting"><i class="fa fa-check"></i><b>5.2.1</b> Pseudomapping &amp; Counting</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="read-counting.html"><a href="read-counting.html#questions-2"><i class="fa fa-check"></i><b>5.3</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html"><i class="fa fa-check"></i><b>6</b> Differential Expression: EdgeR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#description-2"><i class="fa fa-check"></i><b>6.1</b> Description</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#learning-outcomes-3"><i class="fa fa-check"></i><b>6.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="6.3" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#loading-in-the-featurecounts-object"><i class="fa fa-check"></i><b>6.3</b> Loading in the featureCounts object</a></li>
<li class="chapter" data-level="6.4" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#count-loading-and-annotation"><i class="fa fa-check"></i><b>6.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="6.5" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#filtering-to-remove-low-counts"><i class="fa fa-check"></i><b>6.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="6.6" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#normalization-for-composition-bias"><i class="fa fa-check"></i><b>6.6</b> Normalization for composition bias</a></li>
<li class="chapter" data-level="6.7" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#mds-plots"><i class="fa fa-check"></i><b>6.7</b> MDS plots</a></li>
<li class="chapter" data-level="6.8" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#exploring-differences-between-libraries"><i class="fa fa-check"></i><b>6.8</b> Exploring differences between libraries</a></li>
<li class="chapter" data-level="6.9" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#estimate-dispersion"><i class="fa fa-check"></i><b>6.9</b> Estimate Dispersion</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#testing-for-differential-expression"><i class="fa fa-check"></i><b>6.10</b> Testing for differential expression</a></li>
<li class="chapter" data-level="6.11" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#looking-at-all-contrasts-at-once"><i class="fa fa-check"></i><b>6.11</b> Looking at all contrasts at once</a></li>
<li class="chapter" data-level="6.12" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#questions-3"><i class="fa fa-check"></i><b>6.12</b> Questions</a></li>
<li class="chapter" data-level="6.13" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#a-template-set-of-code-chunks-for-doing-this-is-below"><i class="fa fa-check"></i><b>6.13</b> A template set of code chunks for doing this is below:</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html"><i class="fa fa-check"></i><b>7</b> Differential Expression: DESeq2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#description-3"><i class="fa fa-check"></i><b>7.1</b> Description</a></li>
<li class="chapter" data-level="7.2" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#learning-outcomes-4"><i class="fa fa-check"></i><b>7.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="7.3" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#loading-in-the-featurecounts-object-1"><i class="fa fa-check"></i><b>7.3</b> Loading in the featureCounts object</a></li>
<li class="chapter" data-level="7.4" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#count-loading-and-annotation-1"><i class="fa fa-check"></i><b>7.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="7.5" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#filtering-to-remove-low-counts-1"><i class="fa fa-check"></i><b>7.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="7.6" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#testing-for-differential-expression-1"><i class="fa fa-check"></i><b>7.6</b> Testing for differential expression</a></li>
<li class="chapter" data-level="7.7" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#questions-4"><i class="fa fa-check"></i><b>7.7</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html"><i class="fa fa-check"></i><b>8</b> Differential Expression: limma</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#description-4"><i class="fa fa-check"></i><b>8.1</b> Description</a></li>
<li class="chapter" data-level="8.2" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#learning-objectives"><i class="fa fa-check"></i><b>8.2</b> Learning Objectives</a></li>
<li class="chapter" data-level="8.3" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#loading-in-the-count-data-file"><i class="fa fa-check"></i><b>8.3</b> Loading in the count data file</a></li>
<li class="chapter" data-level="8.4" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#count-loading-and-annotation-2"><i class="fa fa-check"></i><b>8.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="8.5" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#filtering-to-remove-low-counts-2"><i class="fa fa-check"></i><b>8.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="8.6" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#normalization-for-composition-bias-1"><i class="fa fa-check"></i><b>8.6</b> Normalization for composition bias</a></li>
<li class="chapter" data-level="8.7" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#exploring-differences-between-libraries-1"><i class="fa fa-check"></i><b>8.7</b> Exploring differences between libraries</a></li>
<li class="chapter" data-level="8.8" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#estimate-dispersion-1"><i class="fa fa-check"></i><b>8.8</b> Estimate Dispersion</a></li>
<li class="chapter" data-level="8.9" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#testing-for-differential-expression-2"><i class="fa fa-check"></i><b>8.9</b> Testing for differential expression</a></li>
<li class="chapter" data-level="8.10" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#examining-a-specific-contrast"><i class="fa fa-check"></i><b>8.10</b> Examining a specific contrast</a></li>
<li class="chapter" data-level="8.11" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#visualization"><i class="fa fa-check"></i><b>8.11</b> Visualization</a>
<ul>
<li class="chapter" data-level="8.11.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#ma-lot"><i class="fa fa-check"></i><b>8.11.1</b> MA lot</a></li>
</ul></li>
<li class="chapter" data-level="8.12" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#treat-testing"><i class="fa fa-check"></i><b>8.12</b> <code>treat()</code> testing</a>
<ul>
<li class="chapter" data-level="8.12.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#visualize-de-genes-from-treat-using-lfc1"><i class="fa fa-check"></i><b>8.12.1</b> Visualize DE genes from <code>Treat</code> using lfc=1</a></li>
</ul></li>
<li class="chapter" data-level="8.13" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#comparing-de-analysis-softwares"><i class="fa fa-check"></i><b>8.13</b> Comparing DE analysis softwares</a></li>
<li class="chapter" data-level="8.14" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#correlation-between-logfc-estimates-across-softwares"><i class="fa fa-check"></i><b>8.14</b> Correlation between logFC estimates across softwares</a></li>
<li class="chapter" data-level="8.15" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#questions-5"><i class="fa fa-check"></i><b>8.15</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html"><i class="fa fa-check"></i><b>9</b> Visualizing Differential Expression Results</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#description-5"><i class="fa fa-check"></i><b>9.1</b> Description</a></li>
<li class="chapter" data-level="9.2" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#learning-outcomes-5"><i class="fa fa-check"></i><b>9.2</b> Learning Outcomes</a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#ma-plot"><i class="fa fa-check"></i><b>9.3</b> MA-plot</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#volcano-plot"><i class="fa fa-check"></i><b>9.4</b> Volcano Plot</a></li>
<li class="chapter" data-level="9.5" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#using-glimma-for-an-interactive-visualization"><i class="fa fa-check"></i><b>9.5</b> Using Glimma for an interactive visualization</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#ma-plots"><i class="fa fa-check"></i><b>9.5.1</b> MA plots</a></li>
<li class="chapter" data-level="9.5.2" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#volcano-plots"><i class="fa fa-check"></i><b>9.5.2</b> Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#generating-bar-graph-summaries"><i class="fa fa-check"></i><b>9.6</b> Generating bar graph summaries</a></li>
<li class="chapter" data-level="9.7" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#exercise"><i class="fa fa-check"></i><b>9.7</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Clustering</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#description-6"><i class="fa fa-check"></i><b>10.1</b> Description</a></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#learning-outcomes-6"><i class="fa fa-check"></i><b>10.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#cluster-3.0"><i class="fa fa-check"></i><b>10.3</b> Cluster 3.0</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="clustering.html"><a href="clustering.html#generating-a-pcl-file"><i class="fa fa-check"></i><b>10.3.1</b> Generating a PCL file</a></li>
<li class="chapter" data-level="10.3.2" data-path="clustering.html"><a href="clustering.html#filtering-data"><i class="fa fa-check"></i><b>10.3.2</b> Filtering data</a></li>
<li class="chapter" data-level="10.3.3" data-path="clustering.html"><a href="clustering.html#hierarchical-clustering"><i class="fa fa-check"></i><b>10.3.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="10.3.4" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.3.4</b> <em>K</em>-Means Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#visualizing-clusters-with-java-treeview"><i class="fa fa-check"></i><b>10.4</b> Visualizing Clusters with Java TreeView</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="clustering.html"><a href="clustering.html#changing-pixel-settings."><i class="fa fa-check"></i><b>10.4.1</b> Changing pixel settings.</a></li>
<li class="chapter" data-level="10.4.2" data-path="clustering.html"><a href="clustering.html#selecting-groups-of-genes."><i class="fa fa-check"></i><b>10.4.2</b> Selecting groups of genes.</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#performing-clustering-on-yeast-stress-data"><i class="fa fa-check"></i><b>10.5</b> Performing clustering on yeast stress data</a></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#questions-6"><i class="fa fa-check"></i><b>10.6</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="kegg-analysis.html"><a href="kegg-analysis.html"><i class="fa fa-check"></i><b>11</b> KEGG Analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="kegg-analysis.html"><a href="kegg-analysis.html#description-7"><i class="fa fa-check"></i><b>11.1</b> Description</a></li>
<li class="chapter" data-level="11.2" data-path="kegg-analysis.html"><a href="kegg-analysis.html#learning-outcomes-7"><i class="fa fa-check"></i><b>11.2</b> Learning Outcomes</a></li>
<li class="chapter" data-level="11.3" data-path="kegg-analysis.html"><a href="kegg-analysis.html#loading-in-the-edger-de-gene-file-output."><i class="fa fa-check"></i><b>11.3</b> Loading in the edgeR DE gene file output.</a></li>
<li class="chapter" data-level="11.4" data-path="kegg-analysis.html"><a href="kegg-analysis.html#kegg-analysis-1"><i class="fa fa-check"></i><b>11.4</b> KEGG Analysis</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="kegg-analysis.html"><a href="kegg-analysis.html#using-kegga-from-limma"><i class="fa fa-check"></i><b>11.4.1</b> Using <code>kegga()</code> from limma</a></li>
<li class="chapter" data-level="11.4.2" data-path="kegg-analysis.html"><a href="kegg-analysis.html#pathway-visualization"><i class="fa fa-check"></i><b>11.4.2</b> Pathway Visualization</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="kegg-analysis.html"><a href="kegg-analysis.html#visualize-on-the-kegg-website"><i class="fa fa-check"></i><b>11.5</b> Visualize on the KEGG website</a></li>
<li class="chapter" data-level="11.6" data-path="kegg-analysis.html"><a href="kegg-analysis.html#comparing-paralogs-in-common-pathways"><i class="fa fa-check"></i><b>11.6</b> Comparing Paralogs in common pathways</a></li>
<li class="chapter" data-level="11.7" data-path="kegg-analysis.html"><a href="kegg-analysis.html#additional-kegg-related-analyses"><i class="fa fa-check"></i><b>11.7</b> Additional KEGG-related analyses</a></li>
<li class="chapter" data-level="11.8" data-path="kegg-analysis.html"><a href="kegg-analysis.html#dotplot"><i class="fa fa-check"></i><b>11.8</b> Dotplot</a></li>
<li class="chapter" data-level="11.9" data-path="kegg-analysis.html"><a href="kegg-analysis.html#cnetplot"><i class="fa fa-check"></i><b>11.9</b> cnetplot</a></li>
<li class="chapter" data-level="11.10" data-path="kegg-analysis.html"><a href="kegg-analysis.html#heatplot"><i class="fa fa-check"></i><b>11.10</b> heatplot</a></li>
<li class="chapter" data-level="11.11" data-path="kegg-analysis.html"><a href="kegg-analysis.html#upsetplot"><i class="fa fa-check"></i><b>11.11</b> upsetplot</a></li>
<li class="chapter" data-level="11.12" data-path="kegg-analysis.html"><a href="kegg-analysis.html#emapplot"><i class="fa fa-check"></i><b>11.12</b> emapplot</a></li>
<li class="chapter" data-level="11.13" data-path="kegg-analysis.html"><a href="kegg-analysis.html#gsea"><i class="fa fa-check"></i><b>11.13</b> GSEA</a>
<ul>
<li class="chapter" data-level="11.13.1" data-path="kegg-analysis.html"><a href="kegg-analysis.html#gsekegg-vs-enrichkegg"><i class="fa fa-check"></i><b>11.13.1</b> <code>gseKEGG</code> vs <code>enrichKEGG</code></a></li>
<li class="chapter" data-level="11.13.2" data-path="kegg-analysis.html"><a href="kegg-analysis.html#gsea-visualization"><i class="fa fa-check"></i><b>11.13.2</b> GSEA visualization</a></li>
</ul></li>
<li class="chapter" data-level="11.14" data-path="kegg-analysis.html"><a href="kegg-analysis.html#questions-7"><i class="fa fa-check"></i><b>11.14</b> Questions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Genomic Data Analysis Course Exercises</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="read-counting" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Read Counting<a href="read-counting.html#read-counting" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>last updated: 2023-10-27</p>
<p>As usual, make sure we have the right packages for this exercise</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="read-counting.html#cb120-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;pacman&quot;</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;pacman&quot;</span>); <span class="fu">library</span>(pacman)</span>
<span id="cb120-2"><a href="read-counting.html#cb120-2" tabindex="-1"></a></span>
<span id="cb120-3"><a href="read-counting.html#cb120-3" tabindex="-1"></a><span class="co"># let&#39;s load all of the files we were using and want to have again today</span></span>
<span id="cb120-4"><a href="read-counting.html#cb120-4" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;knitr&quot;</span>, <span class="st">&quot;readr&quot;</span>,</span>
<span id="cb120-5"><a href="read-counting.html#cb120-5" tabindex="-1"></a>       <span class="st">&quot;pander&quot;</span>, <span class="st">&quot;BiocManager&quot;</span>, </span>
<span id="cb120-6"><a href="read-counting.html#cb120-6" tabindex="-1"></a>       <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;stringr&quot;</span>)</span>
<span id="cb120-7"><a href="read-counting.html#cb120-7" tabindex="-1"></a></span>
<span id="cb120-8"><a href="read-counting.html#cb120-8" tabindex="-1"></a><span class="co"># We also need the Bioconductor packages &quot;Rsubread&quot; for today&#39;s activity.</span></span>
<span id="cb120-9"><a href="read-counting.html#cb120-9" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;Rsubread&quot;</span>)</span></code></pre></div>
<div id="featurecounts" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> featureCounts<a href="read-counting.html#featurecounts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will first show how to use the `<code>featureCounts()</code> function in the <code>Rsubread</code> package to generate counts from the mapped .bam files.</p>
<div id="locate-bam-files" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Locate BAM files<a href="read-counting.html#locate-bam-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Previously, we aligned our fastq files to the reference genome, generating BAM files. They should be in your “~/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp” folder, unless you chose a different place to store them.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="read-counting.html#cb121-1" tabindex="-1"></a><span class="co"># Where the bam files are located (default same as trimmed fastq file location)</span></span>
<span id="cb121-2"><a href="read-counting.html#cb121-2" tabindex="-1"></a>bam_file_dir <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/&quot;</span></span>
<span id="cb121-3"><a href="read-counting.html#cb121-3" tabindex="-1"></a></span>
<span id="cb121-4"><a href="read-counting.html#cb121-4" tabindex="-1"></a><span class="co"># save list of all of those files with their full path</span></span>
<span id="cb121-5"><a href="read-counting.html#cb121-5" tabindex="-1"></a>bam.files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> bam_file_dir, </span>
<span id="cb121-6"><a href="read-counting.html#cb121-6" tabindex="-1"></a>                                  <span class="at">pattern =</span> <span class="st">&quot;.subread.BAM$&quot;</span>, </span>
<span id="cb121-7"><a href="read-counting.html#cb121-7" tabindex="-1"></a>                                  <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb121-8"><a href="read-counting.html#cb121-8" tabindex="-1"></a><span class="co"># make sure we see what we expect.</span></span>
<span id="cb121-9"><a href="read-counting.html#cb121-9" tabindex="-1"></a>bam.files</span></code></pre></div>
<pre><code>##  [1] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_MSN24_ETOH_REP1_R1.fastq.gz.subread.BAM&quot;
##  [2] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_MSN24_ETOH_REP2_R1.fastq.gz.subread.BAM&quot;
##  [3] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_MSN24_ETOH_REP3_R1.fastq.gz.subread.BAM&quot;
##  [4] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_MSN24_ETOH_REP4_R1.fastq.gz.subread.BAM&quot;
##  [5] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_MSN24_MOCK_REP1_R1.fastq.gz.subread.BAM&quot;
##  [6] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_MSN24_MOCK_REP2_R1.fastq.gz.subread.BAM&quot;
##  [7] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_MSN24_MOCK_REP3_R1.fastq.gz.subread.BAM&quot;
##  [8] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_MSN24_MOCK_REP4_R1.fastq.gz.subread.BAM&quot;
##  [9] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_WT_ETOH_REP1_R1.fastq.gz.subread.BAM&quot;   
## [10] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_WT_ETOH_REP2_R1.fastq.gz.subread.BAM&quot;   
## [11] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_WT_ETOH_REP3_R1.fastq.gz.subread.BAM&quot;   
## [12] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_WT_ETOH_REP4_R1.fastq.gz.subread.BAM&quot;   
## [13] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_WT_MOCK_REP1_R1.fastq.gz.subread.BAM&quot;   
## [14] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_WT_MOCK_REP2_R1.fastq.gz.subread.BAM&quot;   
## [15] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_WT_MOCK_REP3_R1.fastq.gz.subread.BAM&quot;   
## [16] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp//YPS606_WT_MOCK_REP4_R1.fastq.gz.subread.BAM&quot;</code></pre>
<p>You should see the full paths to all 16 trimmed fastq bam files that we will be mapping to the reference genome today.</p>
</div>
<div id="retrieve-the-genome-annotation" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Retrieve the genome annotation<a href="read-counting.html#retrieve-the-genome-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We currently have our raw reads mapped to the genome in the form of bam files. Before the differential expression analysis can proceed, these reads must be assigned and counted towards annotated genes. This can be achieved with functions in the Rsubread package, we all also see how to do this with Salmon.</p>
<p>We will use a bash code chunk to download the latest genome annotation</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb123-1"><a href="read-counting.html#cb123-1" tabindex="-1"></a><span class="co"># Define the destination file path</span></span>
<span id="cb123-2"><a href="read-counting.html#cb123-2" tabindex="-1"></a><span class="va">REF_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Reference&quot;</span></span>
<span id="cb123-3"><a href="read-counting.html#cb123-3" tabindex="-1"></a><span class="co"># If this directory doesn&#39;t exist, you need to first complete the Read_Mapping.Rmd exercise. </span></span>
<span id="cb123-4"><a href="read-counting.html#cb123-4" tabindex="-1"></a></span>
<span id="cb123-5"><a href="read-counting.html#cb123-5" tabindex="-1"></a><span class="co"># Define the URL of reference genome annotation (gtf)</span></span>
<span id="cb123-6"><a href="read-counting.html#cb123-6" tabindex="-1"></a><span class="co"># (latest from ensembl)</span></span>
<span id="cb123-7"><a href="read-counting.html#cb123-7" tabindex="-1"></a><span class="va">url</span><span class="op">=</span><span class="st">&quot;ftp://ftp.ensembl.org/pub/release-110/gtf/saccharomyces_cerevisiae/Saccharomyces_cerevisiae.R64-1-1.110.gtf.gz&quot;</span></span>
<span id="cb123-8"><a href="read-counting.html#cb123-8" tabindex="-1"></a></span>
<span id="cb123-9"><a href="read-counting.html#cb123-9" tabindex="-1"></a><span class="co"># Check if the file already exists at the destination location</span></span>
<span id="cb123-10"><a href="read-counting.html#cb123-10" tabindex="-1"></a><span class="cf">if</span> <span class="bu">[</span> <span class="ot">!</span> <span class="ot">-f</span> <span class="st">&quot;</span><span class="va">$REF_DIR</span><span class="st">/Saccharomyces_cerevisiae.R64-1-1.110.gtf.gz&quot;</span> <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb123-11"><a href="read-counting.html#cb123-11" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;Reference genome annotation not found, downloading...&quot;</span></span>
<span id="cb123-12"><a href="read-counting.html#cb123-12" tabindex="-1"></a>    <span class="co"># If the file does not exist, download it using curl</span></span>
<span id="cb123-13"><a href="read-counting.html#cb123-13" tabindex="-1"></a>    <span class="ex">curl</span> <span class="at">-o</span> <span class="st">&quot;</span><span class="va">$REF_DIR</span><span class="st">/Saccharomyces_cerevisiae.R64-1-1.110.gtf.gz&quot;</span> <span class="st">&quot;</span><span class="va">$url</span><span class="st">&quot;</span></span>
<span id="cb123-14"><a href="read-counting.html#cb123-14" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;Downloading finished&quot;</span></span>
<span id="cb123-15"><a href="read-counting.html#cb123-15" tabindex="-1"></a><span class="cf">else</span></span>
<span id="cb123-16"><a href="read-counting.html#cb123-16" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;File already exists at </span><span class="va">$REF_DIR</span><span class="st"> Skipping download.&quot;</span></span>
<span id="cb123-17"><a href="read-counting.html#cb123-17" tabindex="-1"></a><span class="cf">fi</span></span></code></pre></div>
<pre><code>## File already exists at /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference Skipping download.</code></pre>
<p>Let’s take a look at the first few lines of the gtf file</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="read-counting.html#cb125-1" tabindex="-1"></a><span class="co"># see the header columns with metadata starting with #! and delimited with \t</span></span>
<span id="cb125-2"><a href="read-counting.html#cb125-2" tabindex="-1"></a><span class="fu">read.delim</span>(</span>
<span id="cb125-3"><a href="read-counting.html#cb125-3" tabindex="-1"></a>  <span class="fu">path.expand</span>(</span>
<span id="cb125-4"><a href="read-counting.html#cb125-4" tabindex="-1"></a>    <span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Reference/Saccharomyces_cerevisiae.R64-1-1.110.gtf.gz&quot;</span></span>
<span id="cb125-5"><a href="read-counting.html#cb125-5" tabindex="-1"></a>  ),</span>
<span id="cb125-6"><a href="read-counting.html#cb125-6" tabindex="-1"></a>  <span class="at">header =</span> F,</span>
<span id="cb125-7"><a href="read-counting.html#cb125-7" tabindex="-1"></a>  <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb125-8"><a href="read-counting.html#cb125-8" tabindex="-1"></a>  <span class="at">nrows =</span> <span class="dv">10</span></span>
<span id="cb125-9"><a href="read-counting.html#cb125-9" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>##                                                                                V1
## 1                                                          #!genome-build R64-1-1
## 2                                                        #!genome-version R64-1-1
## 3                                                           #!genome-date 2011-09
## 4                                        #!genome-build-accession GCA_000146045.2
## 5                                                #!genebuild-last-updated 2018-10
## 6                                                                              IV
## 7                                                                             sgd
## 8                                                                            gene
## 9                                                                            8683
## 10                                                                           9756
## 11                                                                              .
## 12                                                                              -
## 13                                                                              .
## 14 gene_id YDL246C; gene_name SOR2; gene_source sgd; gene_biotype protein_coding;</code></pre>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="read-counting.html#cb127-1" tabindex="-1"></a><span class="co"># We can also take a look at the first few entries to see the columns</span></span>
<span id="cb127-2"><a href="read-counting.html#cb127-2" tabindex="-1"></a><span class="fu">read.delim</span>(</span>
<span id="cb127-3"><a href="read-counting.html#cb127-3" tabindex="-1"></a>  <span class="fu">path.expand</span>(</span>
<span id="cb127-4"><a href="read-counting.html#cb127-4" tabindex="-1"></a>    <span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Reference/Saccharomyces_cerevisiae.R64-1-1.110.gtf.gz&quot;</span></span>
<span id="cb127-5"><a href="read-counting.html#cb127-5" tabindex="-1"></a>  ),</span>
<span id="cb127-6"><a href="read-counting.html#cb127-6" tabindex="-1"></a>  <span class="at">header =</span> F,</span>
<span id="cb127-7"><a href="read-counting.html#cb127-7" tabindex="-1"></a>  <span class="at">comment.char =</span> <span class="st">&quot;#&quot;</span>,</span>
<span id="cb127-8"><a href="read-counting.html#cb127-8" tabindex="-1"></a>  <span class="at">strip.white =</span> T,</span>
<span id="cb127-9"><a href="read-counting.html#cb127-9" tabindex="-1"></a>  <span class="at">nrows =</span> <span class="dv">20</span> <span class="co">#just the first 20 lines</span></span>
<span id="cb127-10"><a href="read-counting.html#cb127-10" tabindex="-1"></a>) </span></code></pre></div>
<pre><code>##    V1  V2          V3      V4      V5 V6 V7 V8
## 1  IV sgd        gene    8683    9756  .  -  .
## 2  IV sgd  transcript    8683    9756  .  -  .
## 3  IV sgd        exon    8683    9756  .  -  .
## 4  IV sgd         CDS    8686    9756  .  -  0
## 5  IV sgd start_codon    9754    9756  .  -  0
## 6  IV sgd  stop_codon    8683    8685  .  -  0
## 7  IV sgd        gene   17577   18566  .  -  .
## 8  IV sgd  transcript   17577   18566  .  -  .
## 9  IV sgd        exon   17577   18566  .  -  .
## 10 IV sgd         CDS   17580   18566  .  -  0
## 11 IV sgd start_codon   18564   18566  .  -  0
## 12 IV sgd  stop_codon   17577   17579  .  -  0
## 13 IV sgd        gene 1248154 1249821  .  -  .
## 14 IV sgd  transcript 1248154 1249821  .  -  .
## 15 IV sgd        exon 1248154 1249821  .  -  .
## 16 IV sgd         CDS 1248157 1249821  .  -  0
## 17 IV sgd start_codon 1249819 1249821  .  -  0
## 18 IV sgd  stop_codon 1248154 1248156  .  -  0
## 19 IV sgd        gene  289572  290081  .  -  .
## 20 IV sgd  transcript  289572  290081  .  -  .
##                                                                                                                                                                                                                                                             V9
## 1                                                                                                                                                                               gene_id YDL246C; gene_name SOR2; gene_source sgd; gene_biotype protein_coding;
## 2                                            gene_id YDL246C; transcript_id YDL246C_mRNA; gene_name SOR2; gene_source sgd; gene_biotype protein_coding; transcript_name SOR2; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 3    gene_id YDL246C; transcript_id YDL246C_mRNA; exon_number 1; gene_name SOR2; gene_source sgd; gene_biotype protein_coding; transcript_name SOR2; transcript_source sgd; transcript_biotype protein_coding; exon_id YDL246C_mRNA-E1; tag Ensembl_canonical;
## 4         gene_id YDL246C; transcript_id YDL246C_mRNA; exon_number 1; gene_name SOR2; gene_source sgd; gene_biotype protein_coding; transcript_name SOR2; transcript_source sgd; transcript_biotype protein_coding; protein_id YDL246C; tag Ensembl_canonical;
## 5                             gene_id YDL246C; transcript_id YDL246C_mRNA; exon_number 1; gene_name SOR2; gene_source sgd; gene_biotype protein_coding; transcript_name SOR2; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 6                             gene_id YDL246C; transcript_id YDL246C_mRNA; exon_number 1; gene_name SOR2; gene_source sgd; gene_biotype protein_coding; transcript_name SOR2; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 7                                                                                                                                                                               gene_id YDL243C; gene_name AAD4; gene_source sgd; gene_biotype protein_coding;
## 8                                            gene_id YDL243C; transcript_id YDL243C_mRNA; gene_name AAD4; gene_source sgd; gene_biotype protein_coding; transcript_name AAD4; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 9    gene_id YDL243C; transcript_id YDL243C_mRNA; exon_number 1; gene_name AAD4; gene_source sgd; gene_biotype protein_coding; transcript_name AAD4; transcript_source sgd; transcript_biotype protein_coding; exon_id YDL243C_mRNA-E1; tag Ensembl_canonical;
## 10        gene_id YDL243C; transcript_id YDL243C_mRNA; exon_number 1; gene_name AAD4; gene_source sgd; gene_biotype protein_coding; transcript_name AAD4; transcript_source sgd; transcript_biotype protein_coding; protein_id YDL243C; tag Ensembl_canonical;
## 11                            gene_id YDL243C; transcript_id YDL243C_mRNA; exon_number 1; gene_name AAD4; gene_source sgd; gene_biotype protein_coding; transcript_name AAD4; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 12                            gene_id YDL243C; transcript_id YDL243C_mRNA; exon_number 1; gene_name AAD4; gene_source sgd; gene_biotype protein_coding; transcript_name AAD4; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 13                                                                                                                                                                             gene_id YDR387C; gene_name CIN10; gene_source sgd; gene_biotype protein_coding;
## 14                                         gene_id YDR387C; transcript_id YDR387C_mRNA; gene_name CIN10; gene_source sgd; gene_biotype protein_coding; transcript_name CIN10; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 15 gene_id YDR387C; transcript_id YDR387C_mRNA; exon_number 1; gene_name CIN10; gene_source sgd; gene_biotype protein_coding; transcript_name CIN10; transcript_source sgd; transcript_biotype protein_coding; exon_id YDR387C_mRNA-E1; tag Ensembl_canonical;
## 16      gene_id YDR387C; transcript_id YDR387C_mRNA; exon_number 1; gene_name CIN10; gene_source sgd; gene_biotype protein_coding; transcript_name CIN10; transcript_source sgd; transcript_biotype protein_coding; protein_id YDR387C; tag Ensembl_canonical;
## 17                          gene_id YDR387C; transcript_id YDR387C_mRNA; exon_number 1; gene_name CIN10; gene_source sgd; gene_biotype protein_coding; transcript_name CIN10; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 18                          gene_id YDR387C; transcript_id YDR387C_mRNA; exon_number 1; gene_name CIN10; gene_source sgd; gene_biotype protein_coding; transcript_name CIN10; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;
## 19                                                                                                                                                                                              gene_id YDL094C; gene_source sgd; gene_biotype protein_coding;
## 20                                                                                 gene_id YDL094C; transcript_id YDL094C_mRNA; gene_source sgd; gene_biotype protein_coding; transcript_source sgd; transcript_biotype protein_coding; tag Ensembl_canonical;</code></pre>
<p>There are 9 columns in a standard gtf file, information about each is available here: <a href="https://useast.ensembl.org/info/website/upload/gff.html" class="uri">https://useast.ensembl.org/info/website/upload/gff.html</a></p>
<p>Note that version 2 of gff is identical to the gtf format.</p>
</div>
<div id="counting-with-featurecounts" class="section level3 hasAnchor" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> Counting with FeatureCounts<a href="read-counting.html#counting-with-featurecounts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="read-counting.html#cb129-1" tabindex="-1"></a><span class="fu">library</span>(Rsubread)</span>
<span id="cb129-2"><a href="read-counting.html#cb129-2" tabindex="-1"></a></span>
<span id="cb129-3"><a href="read-counting.html#cb129-3" tabindex="-1"></a><span class="co"># Set path of the reference annotation gzipped gtf file</span></span>
<span id="cb129-4"><a href="read-counting.html#cb129-4" tabindex="-1"></a>reference_annotation <span class="ot">=</span> <span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Reference/Saccharomyces_cerevisiae.R64-1-1.110.gtf.gz&quot;</span></span></code></pre></div>
<p>We can see the arguments available with the align function from the Rsubread package</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="read-counting.html#cb130-1" tabindex="-1"></a><span class="fu">args</span>(featureCounts)</span></code></pre></div>
<pre><code>## function (files, annot.inbuilt = &quot;mm39&quot;, annot.ext = NULL, isGTFAnnotationFile = FALSE, 
##     GTF.featureType = &quot;exon&quot;, GTF.attrType = &quot;gene_id&quot;, GTF.attrType.extra = NULL, 
##     chrAliases = NULL, useMetaFeatures = TRUE, allowMultiOverlap = FALSE, 
##     minOverlap = 1, fracOverlap = 0, fracOverlapFeature = 0, 
##     largestOverlap = FALSE, nonOverlap = NULL, nonOverlapFeature = NULL, 
##     readShiftType = &quot;upstream&quot;, readShiftSize = 0, readExtension5 = 0, 
##     readExtension3 = 0, read2pos = NULL, countMultiMappingReads = TRUE, 
##     fraction = FALSE, isLongRead = FALSE, minMQS = 0, splitOnly = FALSE, 
##     nonSplitOnly = FALSE, primaryOnly = FALSE, ignoreDup = FALSE, 
##     strandSpecific = 0, juncCounts = FALSE, genome = NULL, isPairedEnd = FALSE, 
##     countReadPairs = TRUE, requireBothEndsMapped = FALSE, checkFragLength = FALSE, 
##     minFragLength = 50, maxFragLength = 600, countChimericFragments = TRUE, 
##     autosort = TRUE, nthreads = 1, byReadGroup = FALSE, reportReads = NULL, 
##     reportReadsPath = NULL, maxMOp = 10, tmpDir = &quot;.&quot;, verbose = FALSE) 
## NULL</code></pre>
<p>The Phred offset determines the encoding for the base-calling quality string in the FASTQ file. For the Illumina 1.8 format onwards, this encoding is set at +33. However, older formats may use a +64 encoding. Users should ensure that the correct encoding is specified during alignment. If unsure, one can examine the first several quality strings in the FASTQ file. A good rule of thumb is to check whether lower-case letters are present (+64 encoding) or absent (+33).</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="read-counting.html#cb132-1" tabindex="-1"></a><span class="co"># This command counts the number of each feature per fastq file, </span></span>
<span id="cb132-2"><a href="read-counting.html#cb132-2" tabindex="-1"></a><span class="co">#.  generating an output we can use later.</span></span>
<span id="cb132-3"><a href="read-counting.html#cb132-3" tabindex="-1"></a>fc <span class="ot">&lt;-</span> <span class="fu">featureCounts</span>(bam.files,</span>
<span id="cb132-4"><a href="read-counting.html#cb132-4" tabindex="-1"></a>                    <span class="at">annot.ext =</span> reference_annotation,</span>
<span id="cb132-5"><a href="read-counting.html#cb132-5" tabindex="-1"></a>                    <span class="at">isGTFAnnotationFile =</span> <span class="cn">TRUE</span>,</span>
<span id="cb132-6"><a href="read-counting.html#cb132-6" tabindex="-1"></a>                    <span class="at">GTF.featureType =</span> <span class="st">&quot;exon&quot;</span></span>
<span id="cb132-7"><a href="read-counting.html#cb132-7" tabindex="-1"></a>                    )</span></code></pre></div>
<pre><code>## 
##         ==========     _____ _    _ ____  _____  ______          _____  
##         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ 
##           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |
##             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |
##               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |
##         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/
##        Rsubread 2.14.2
## 
## //========================== featureCounts setting ===========================\\
## ||                                                                            ||
## ||             Input files : 16 BAM files                                     ||
## ||                                                                            ||
## ||                           YPS606_MSN24_ETOH_REP1_R1.fastq.gz.subread.BAM   ||
## ||                           YPS606_MSN24_ETOH_REP2_R1.fastq.gz.subread.BAM   ||
## ||                           YPS606_MSN24_ETOH_REP3_R1.fastq.gz.subread.BAM   ||
## ||                           YPS606_MSN24_ETOH_REP4_R1.fastq.gz.subread.BAM   ||
## ||                           YPS606_MSN24_MOCK_REP1_R1.fastq.gz.subread.BAM   ||
## ||                           YPS606_MSN24_MOCK_REP2_R1.fastq.gz.subread.BAM   ||
## ||                           YPS606_MSN24_MOCK_REP3_R1.fastq.gz.subread.BAM   ||
## ||                           YPS606_MSN24_MOCK_REP4_R1.fastq.gz.subread.BAM   ||
## ||                           YPS606_WT_ETOH_REP1_R1.fastq.gz.subread.BAM      ||
## ||                           YPS606_WT_ETOH_REP2_R1.fastq.gz.subread.BAM      ||
## ||                           YPS606_WT_ETOH_REP3_R1.fastq.gz.subread.BAM      ||
## ||                           YPS606_WT_ETOH_REP4_R1.fastq.gz.subread.BAM      ||
## ||                           YPS606_WT_MOCK_REP1_R1.fastq.gz.subread.BAM      ||
## ||                           YPS606_WT_MOCK_REP2_R1.fastq.gz.subread.BAM      ||
## ||                           YPS606_WT_MOCK_REP3_R1.fastq.gz.subread.BAM      ||
## ||                           YPS606_WT_MOCK_REP4_R1.fastq.gz.subread.BAM      ||
## ||                                                                            ||
## ||              Paired-end : no                                               ||
## ||        Count read pairs : no                                               ||
## ||              Annotation : Saccharomyces_cerevisiae.R64-1-1.110.gtf.gz  ... ||
## ||      Dir for temp files : .                                                ||
## ||                 Threads : 1                                                ||
## ||                   Level : meta-feature level                               ||
## ||      Multimapping reads : counted                                          ||
## || Multi-overlapping reads : not counted                                      ||
## ||   Min overlapping bases : 1                                                ||
## ||                                                                            ||
## \\============================================================================//
## 
## //================================= Running ==================================\\
## ||                                                                            ||
## || Load annotation file Saccharomyces_cerevisiae.R64-1-1.110.gtf.gz ...       ||
## ||    Features : 7507                                                         ||
## ||    Meta-features : 7127                                                    ||
## ||    Chromosomes/contigs : 17                                                ||
## ||                                                                            ||
## || Process BAM file YPS606_MSN24_ETOH_REP1_R1.fastq.gz.subread.BAM...         ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 233278                                               ||
## ||    Successfully assigned alignments : 175843 (75.4%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_MSN24_ETOH_REP2_R1.fastq.gz.subread.BAM...         ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 215810                                               ||
## ||    Successfully assigned alignments : 161818 (75.0%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_MSN24_ETOH_REP3_R1.fastq.gz.subread.BAM...         ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 199076                                               ||
## ||    Successfully assigned alignments : 148581 (74.6%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_MSN24_ETOH_REP4_R1.fastq.gz.subread.BAM...         ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 205792                                               ||
## ||    Successfully assigned alignments : 153525 (74.6%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_MSN24_MOCK_REP1_R1.fastq.gz.subread.BAM...         ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 167075                                               ||
## ||    Successfully assigned alignments : 122364 (73.2%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_MSN24_MOCK_REP2_R1.fastq.gz.subread.BAM...         ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 169754                                               ||
## ||    Successfully assigned alignments : 126310 (74.4%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_MSN24_MOCK_REP3_R1.fastq.gz.subread.BAM...         ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 210001                                               ||
## ||    Successfully assigned alignments : 151958 (72.4%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_MSN24_MOCK_REP4_R1.fastq.gz.subread.BAM...         ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 208329                                               ||
## ||    Successfully assigned alignments : 153346 (73.6%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_WT_ETOH_REP1_R1.fastq.gz.subread.BAM...            ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 181587                                               ||
## ||    Successfully assigned alignments : 137526 (75.7%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_WT_ETOH_REP2_R1.fastq.gz.subread.BAM...            ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 201551                                               ||
## ||    Successfully assigned alignments : 151322 (75.1%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_WT_ETOH_REP3_R1.fastq.gz.subread.BAM...            ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 214745                                               ||
## ||    Successfully assigned alignments : 161909 (75.4%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_WT_ETOH_REP4_R1.fastq.gz.subread.BAM...            ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 187319                                               ||
## ||    Successfully assigned alignments : 141422 (75.5%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_WT_MOCK_REP1_R1.fastq.gz.subread.BAM...            ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 223539                                               ||
## ||    Successfully assigned alignments : 165863 (74.2%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_WT_MOCK_REP2_R1.fastq.gz.subread.BAM...            ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 187469                                               ||
## ||    Successfully assigned alignments : 138324 (73.8%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_WT_MOCK_REP3_R1.fastq.gz.subread.BAM...            ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 224767                                               ||
## ||    Successfully assigned alignments : 163337 (72.7%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Process BAM file YPS606_WT_MOCK_REP4_R1.fastq.gz.subread.BAM...            ||
## ||    Single-end reads are included.                                          ||
## ||    Total alignments : 206865                                               ||
## ||    Successfully assigned alignments : 152394 (73.7%)                       ||
## ||    Running time : 0.00 minutes                                             ||
## ||                                                                            ||
## || Write the final count table.                                               ||
## || Write the read assignment summary.                                         ||
## ||                                                                            ||
## \\============================================================================//</code></pre>
<p>We can see what all is stored in the featureCounts output object</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="read-counting.html#cb134-1" tabindex="-1"></a><span class="fu">names</span>(fc)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;     &quot;annotation&quot; &quot;targets&quot;    &quot;stat&quot;</code></pre>
<p>The statistics of the read mapping can be seen with <code>fc$stats</code>. This reports the numbers of unassigned reads and the reasons why they are not assigned (eg. ambiguity, multi-mapping, secondary alignment, mapping quality, fragment length, chimera, read duplicate, non-junction and so on), in addition to the number of successfully assigned reads for each library.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="read-counting.html#cb136-1" tabindex="-1"></a>fc<span class="sc">$</span>stat</span></code></pre></div>
<pre><code>##                           Status YPS606_MSN24_ETOH_REP1_R1.fastq.gz.subread.BAM
## 1                       Assigned                                         175843
## 2            Unassigned_Unmapped                                          28887
## 3           Unassigned_Read_Type                                              0
## 4           Unassigned_Singleton                                              0
## 5      Unassigned_MappingQuality                                              0
## 6             Unassigned_Chimera                                              0
## 7      Unassigned_FragmentLength                                              0
## 8           Unassigned_Duplicate                                              0
## 9        Unassigned_MultiMapping                                              0
## 10          Unassigned_Secondary                                              0
## 11           Unassigned_NonSplit                                              0
## 12         Unassigned_NoFeatures                                          16741
## 13 Unassigned_Overlapping_Length                                              0
## 14          Unassigned_Ambiguity                                          11807
##    YPS606_MSN24_ETOH_REP2_R1.fastq.gz.subread.BAM
## 1                                          161818
## 2                                           26525
## 3                                               0
## 4                                               0
## 5                                               0
## 6                                               0
## 7                                               0
## 8                                               0
## 9                                               0
## 10                                              0
## 11                                              0
## 12                                          15747
## 13                                              0
## 14                                          11720
##    YPS606_MSN24_ETOH_REP3_R1.fastq.gz.subread.BAM
## 1                                          148581
## 2                                           25383
## 3                                               0
## 4                                               0
## 5                                               0
## 6                                               0
## 7                                               0
## 8                                               0
## 9                                               0
## 10                                              0
## 11                                              0
## 12                                          14166
## 13                                              0
## 14                                          10946
##    YPS606_MSN24_ETOH_REP4_R1.fastq.gz.subread.BAM
## 1                                          153525
## 2                                           27007
## 3                                               0
## 4                                               0
## 5                                               0
## 6                                               0
## 7                                               0
## 8                                               0
## 9                                               0
## 10                                              0
## 11                                              0
## 12                                          13608
## 13                                              0
## 14                                          11652
##    YPS606_MSN24_MOCK_REP1_R1.fastq.gz.subread.BAM
## 1                                          122364
## 2                                           23961
## 3                                               0
## 4                                               0
## 5                                               0
## 6                                               0
## 7                                               0
## 8                                               0
## 9                                               0
## 10                                              0
## 11                                              0
## 12                                          12806
## 13                                              0
## 14                                           7944
##    YPS606_MSN24_MOCK_REP2_R1.fastq.gz.subread.BAM
## 1                                          126310
## 2                                           23452
## 3                                               0
## 4                                               0
## 5                                               0
## 6                                               0
## 7                                               0
## 8                                               0
## 9                                               0
## 10                                              0
## 11                                              0
## 12                                          11908
## 13                                              0
## 14                                           8084
##    YPS606_MSN24_MOCK_REP3_R1.fastq.gz.subread.BAM
## 1                                          151958
## 2                                           31337
## 3                                               0
## 4                                               0
## 5                                               0
## 6                                               0
## 7                                               0
## 8                                               0
## 9                                               0
## 10                                              0
## 11                                              0
## 12                                          16741
## 13                                              0
## 14                                           9965
##    YPS606_MSN24_MOCK_REP4_R1.fastq.gz.subread.BAM
## 1                                          153346
## 2                                           30580
## 3                                               0
## 4                                               0
## 5                                               0
## 6                                               0
## 7                                               0
## 8                                               0
## 9                                               0
## 10                                              0
## 11                                              0
## 12                                          14454
## 13                                              0
## 14                                           9949
##    YPS606_WT_ETOH_REP1_R1.fastq.gz.subread.BAM
## 1                                       137526
## 2                                        22387
## 3                                            0
## 4                                            0
## 5                                            0
## 6                                            0
## 7                                            0
## 8                                            0
## 9                                            0
## 10                                           0
## 11                                           0
## 12                                       11511
## 13                                           0
## 14                                       10163
##    YPS606_WT_ETOH_REP2_R1.fastq.gz.subread.BAM
## 1                                       151322
## 2                                        24647
## 3                                            0
## 4                                            0
## 5                                            0
## 6                                            0
## 7                                            0
## 8                                            0
## 9                                            0
## 10                                           0
## 11                                           0
## 12                                       14578
## 13                                           0
## 14                                       11004
##    YPS606_WT_ETOH_REP3_R1.fastq.gz.subread.BAM
## 1                                       161909
## 2                                        26246
## 3                                            0
## 4                                            0
## 5                                            0
## 6                                            0
## 7                                            0
## 8                                            0
## 9                                            0
## 10                                           0
## 11                                           0
## 12                                       15064
## 13                                           0
## 14                                       11526
##    YPS606_WT_ETOH_REP4_R1.fastq.gz.subread.BAM
## 1                                       141422
## 2                                        23167
## 3                                            0
## 4                                            0
## 5                                            0
## 6                                            0
## 7                                            0
## 8                                            0
## 9                                            0
## 10                                           0
## 11                                           0
## 12                                       12670
## 13                                           0
## 14                                       10060
##    YPS606_WT_MOCK_REP1_R1.fastq.gz.subread.BAM
## 1                                       165863
## 2                                        30132
## 3                                            0
## 4                                            0
## 5                                            0
## 6                                            0
## 7                                            0
## 8                                            0
## 9                                            0
## 10                                           0
## 11                                           0
## 12                                       16586
## 13                                           0
## 14                                       10958
##    YPS606_WT_MOCK_REP2_R1.fastq.gz.subread.BAM
## 1                                       138324
## 2                                        26218
## 3                                            0
## 4                                            0
## 5                                            0
## 6                                            0
## 7                                            0
## 8                                            0
## 9                                            0
## 10                                           0
## 11                                           0
## 12                                       13984
## 13                                           0
## 14                                        8943
##    YPS606_WT_MOCK_REP3_R1.fastq.gz.subread.BAM
## 1                                       163337
## 2                                        32663
## 3                                            0
## 4                                            0
## 5                                            0
## 6                                            0
## 7                                            0
## 8                                            0
## 9                                            0
## 10                                           0
## 11                                           0
## 12                                       17777
## 13                                           0
## 14                                       10990
##    YPS606_WT_MOCK_REP4_R1.fastq.gz.subread.BAM
## 1                                       152394
## 2                                        29335
## 3                                            0
## 4                                            0
## 5                                            0
## 6                                            0
## 7                                            0
## 8                                            0
## 9                                            0
## 10                                           0
## 11                                           0
## 12                                       14960
## 13                                           0
## 14                                       10176</code></pre>
</div>
<div id="counts-object" class="section level3 hasAnchor" number="5.1.4">
<h3><span class="header-section-number">5.1.4</span> Counts object<a href="read-counting.html#counts-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The counts for the samples are stored in fc$counts.</p>
<p>We can look at the dimensions of the counts to see how many genes and samples are present. The first number is the number of genes and the second number is the number of samples.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="read-counting.html#cb138-1" tabindex="-1"></a><span class="fu">dim</span>(fc<span class="sc">$</span>counts)</span></code></pre></div>
<pre><code>## [1] 7127   16</code></pre>
<p>let’s take a look at the first few lines of fc$counts</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="read-counting.html#cb140-1" tabindex="-1"></a><span class="fu">head</span>(fc<span class="sc">$</span>counts)</span></code></pre></div>
<pre><code>##         YPS606_MSN24_ETOH_REP1_R1.fastq.gz.subread.BAM
## YDL246C                                              0
## YDL243C                                              2
## YDR387C                                              6
## YDL094C                                              4
## YDR438W                                              5
## YDR523C                                              1
##         YPS606_MSN24_ETOH_REP2_R1.fastq.gz.subread.BAM
## YDL246C                                              0
## YDL243C                                              1
## YDR387C                                             10
## YDL094C                                              5
## YDR438W                                              6
## YDR523C                                              1
##         YPS606_MSN24_ETOH_REP3_R1.fastq.gz.subread.BAM
## YDL246C                                              0
## YDL243C                                              1
## YDR387C                                              7
## YDL094C                                              7
## YDR438W                                              5
## YDR523C                                              0
##         YPS606_MSN24_ETOH_REP4_R1.fastq.gz.subread.BAM
## YDL246C                                              0
## YDL243C                                              4
## YDR387C                                              6
## YDL094C                                              4
## YDR438W                                              3
## YDR523C                                              0
##         YPS606_MSN24_MOCK_REP1_R1.fastq.gz.subread.BAM
## YDL246C                                              0
## YDL243C                                              1
## YDR387C                                              1
## YDL094C                                              3
## YDR438W                                              4
## YDR523C                                              0
##         YPS606_MSN24_MOCK_REP2_R1.fastq.gz.subread.BAM
## YDL246C                                              0
## YDL243C                                              1
## YDR387C                                              3
## YDL094C                                              1
## YDR438W                                              1
## YDR523C                                              0
##         YPS606_MSN24_MOCK_REP3_R1.fastq.gz.subread.BAM
## YDL246C                                              0
## YDL243C                                              0
## YDR387C                                              3
## YDL094C                                              3
## YDR438W                                              3
## YDR523C                                              0
##         YPS606_MSN24_MOCK_REP4_R1.fastq.gz.subread.BAM
## YDL246C                                              0
## YDL243C                                              0
## YDR387C                                              6
## YDL094C                                              1
## YDR438W                                              1
## YDR523C                                              1
##         YPS606_WT_ETOH_REP1_R1.fastq.gz.subread.BAM
## YDL246C                                           0
## YDL243C                                           4
## YDR387C                                           9
## YDL094C                                           2
## YDR438W                                           1
## YDR523C                                           0
##         YPS606_WT_ETOH_REP2_R1.fastq.gz.subread.BAM
## YDL246C                                           0
## YDL243C                                           0
## YDR387C                                           7
## YDL094C                                           4
## YDR438W                                           3
## YDR523C                                           1
##         YPS606_WT_ETOH_REP3_R1.fastq.gz.subread.BAM
## YDL246C                                           0
## YDL243C                                           3
## YDR387C                                          12
## YDL094C                                           2
## YDR438W                                           4
## YDR523C                                           1
##         YPS606_WT_ETOH_REP4_R1.fastq.gz.subread.BAM
## YDL246C                                           0
## YDL243C                                           2
## YDR387C                                           7
## YDL094C                                           5
## YDR438W                                           7
## YDR523C                                           1
##         YPS606_WT_MOCK_REP1_R1.fastq.gz.subread.BAM
## YDL246C                                           0
## YDL243C                                           3
## YDR387C                                           4
## YDL094C                                           2
## YDR438W                                           1
## YDR523C                                           0
##         YPS606_WT_MOCK_REP2_R1.fastq.gz.subread.BAM
## YDL246C                                           0
## YDL243C                                           1
## YDR387C                                           2
## YDL094C                                           3
## YDR438W                                           3
## YDR523C                                           0
##         YPS606_WT_MOCK_REP3_R1.fastq.gz.subread.BAM
## YDL246C                                           0
## YDL243C                                           0
## YDR387C                                           9
## YDL094C                                           6
## YDR438W                                           2
## YDR523C                                           0
##         YPS606_WT_MOCK_REP4_R1.fastq.gz.subread.BAM
## YDL246C                                           0
## YDL243C                                           0
## YDR387C                                           6
## YDL094C                                           5
## YDR438W                                           4
## YDR523C                                           0</code></pre>
<p>The row names of the fc$counts matrix represent the Systematic Name for each gene (can be Entrez gene identifiers for other organisms) and the column names are the output filenames from calling the align function.</p>
<p>The annotation slot shows the annotation information that featureCounts used to summarise reads over genes.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="read-counting.html#cb142-1" tabindex="-1"></a><span class="fu">head</span>(fc<span class="sc">$</span>annotation)</span></code></pre></div>
<pre><code>##    GeneID Chr   Start     End Strand Length
## 1 YDL246C  IV    8683    9756      -   1074
## 2 YDL243C  IV   17577   18566      -    990
## 3 YDR387C  IV 1248154 1249821      -   1668
## 4 YDL094C  IV  289572  290081      -    510
## 5 YDR438W  IV 1338274 1339386      +   1113
## 6 YDR523C  IV 1485566 1487038      -   1473</code></pre>
</div>
<div id="saving-fc-object-for-future-use" class="section level3 hasAnchor" number="5.1.5">
<h3><span class="header-section-number">5.1.5</span> Saving <code>fc</code> object for future use<a href="read-counting.html#saving-fc-object-for-future-use" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will need to use this object in our next class. We can use the R function <code>saveRDS()</code> to save the R object to your computer, so it can be accessed at a later date.</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="read-counting.html#cb144-1" tabindex="-1"></a><span class="co"># create a directory for the count output to go into if not already present</span></span>
<span id="cb144-2"><a href="read-counting.html#cb144-2" tabindex="-1"></a>dir_output_counts <span class="ot">&lt;-</span> <span class="fu">path.expand</span>(<span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Data/Counts/Rsubread/&quot;</span>)</span>
<span id="cb144-3"><a href="read-counting.html#cb144-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(dir_output_counts)) {<span class="fu">dir.create</span>(dir_output_counts, <span class="at">recursive =</span> <span class="cn">TRUE</span>)}</span>
<span id="cb144-4"><a href="read-counting.html#cb144-4" tabindex="-1"></a></span>
<span id="cb144-5"><a href="read-counting.html#cb144-5" tabindex="-1"></a><span class="co"># save the R data object</span></span>
<span id="cb144-6"><a href="read-counting.html#cb144-6" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="at">object =</span> fc, <span class="at">file =</span> <span class="fu">paste0</span>(dir_output_counts,<span class="st">&quot;rsubread.yeast_fc_output.Rds&quot;</span>))</span>
<span id="cb144-7"><a href="read-counting.html#cb144-7" tabindex="-1"></a></span>
<span id="cb144-8"><a href="read-counting.html#cb144-8" tabindex="-1"></a><span class="co"># often, we want to share this file as a tsv file. Here is how we can do that:</span></span>
<span id="cb144-9"><a href="read-counting.html#cb144-9" tabindex="-1"></a><span class="fu">write_tsv</span>(<span class="fu">data.frame</span>(</span>
<span id="cb144-10"><a href="read-counting.html#cb144-10" tabindex="-1"></a>            fc<span class="sc">$</span>annotation[,<span class="st">&quot;GeneID&quot;</span>],</span>
<span id="cb144-11"><a href="read-counting.html#cb144-11" tabindex="-1"></a>            fc<span class="sc">$</span>counts,</span>
<span id="cb144-12"><a href="read-counting.html#cb144-12" tabindex="-1"></a>            <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>),</span>
<span id="cb144-13"><a href="read-counting.html#cb144-13" tabindex="-1"></a>    <span class="at">file=</span><span class="fu">paste0</span>(dir_output_counts,<span class="st">&quot;rsubread.gene_counts.merged.yeast.tsv&quot;</span>))</span></code></pre></div>
</div>
<div id="rsubread-qc" class="section level3 hasAnchor" number="5.1.6">
<h3><span class="header-section-number">5.1.6</span> RSubread QC<a href="read-counting.html#rsubread-qc" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can have a look at the quality scores associated with each base that has been called by the sequencing machine using the qualityScores function in Rsubread.</p>
<p>Let’s extract quality scores for 50 reads for the fastq file .</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="read-counting.html#cb145-1" tabindex="-1"></a><span class="co"># Extract quality scores</span></span>
<span id="cb145-2"><a href="read-counting.html#cb145-2" tabindex="-1"></a>qs <span class="ot">&lt;-</span> <span class="fu">qualityScores</span>(</span>
<span id="cb145-3"><a href="read-counting.html#cb145-3" tabindex="-1"></a>  <span class="at">filename=</span><span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP1_R1.fastq.gz&quot;</span>,</span>
<span id="cb145-4"><a href="read-counting.html#cb145-4" tabindex="-1"></a>                    <span class="at">nreads=</span><span class="dv">50</span>)</span></code></pre></div>
<pre><code>## 
## qualityScores Rsubread 2.14.2
## 
## Scan the input file...
## Totally 233278 reads were scanned; the sampling interval is 4665.
## Now extract read quality information...
## 
## Completed successfully. Quality scores for 50 reads (equally spaced in the file) are returned.</code></pre>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="read-counting.html#cb147-1" tabindex="-1"></a><span class="fu">head</span>(qs)</span></code></pre></div>
<pre><code>##       1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25
## [1,] 32 32 37 37 37 41 41 41 37 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41
## [2,] 32 32 37 37 37 41 41 41 41 41 41 41 37 41 41 41 41 41 41 41 41 41 41 41 41
## [3,] 32 32 37 37 37 41 41 41 41 41 37 41 41 37 41 41 41 37 41 41 41 41 41 41 41
## [4,] 32 32 37 37 37 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41
## [5,] 32 32 37 37 37 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41
## [6,] 32 32 37 37 37 41 41 41 41 41 41 27 37 41 41 41 41 41 41 41 41 41 41 41 41
##      26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50
## [1,] 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41
## [2,] 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41
## [3,] 41 41 41 41 41 41 41 41 41 41 41 41 41 41 37 41 41 41 41 41 41 41 41 41 41
## [4,] 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41
## [5,] 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41
## [6,] 41 41 41 41 37 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41 41</code></pre>
<p>We are randomly sampling 50 reads from the file and seeing the quality scores. A quality score of 30 corresponds to a 1 in 1000 chance of an incorrect base call. (A quality score of 10 is a 1 in 10 chance of an incorrect base call.) To look at the overall distribution of quality scores across the sampled reads, we can look at a boxplot</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="read-counting.html#cb149-1" tabindex="-1"></a><span class="fu">boxplot</span>(qs)</span></code></pre></div>
<p><img src="_main_files/figure-html/visualizeq-qs-count-1.png" width="672" /></p>
</div>
</div>
<div id="salmon" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Salmon<a href="read-counting.html#salmon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s go through using salmon to count reads directly from the trimmed fastq.gz files</p>
<div id="pseudomapping-counting" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Pseudomapping &amp; Counting<a href="read-counting.html#pseudomapping-counting" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb150"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb150-1"><a href="read-counting.html#cb150-1" tabindex="-1"></a><span class="va">DATA_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp&quot;</span></span>
<span id="cb150-2"><a href="read-counting.html#cb150-2" tabindex="-1"></a><span class="va">SALMON_OUT_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon&quot;</span></span>
<span id="cb150-3"><a href="read-counting.html#cb150-3" tabindex="-1"></a><span class="va">SALMON_INDEX_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1&quot;</span></span>
<span id="cb150-4"><a href="read-counting.html#cb150-4" tabindex="-1"></a></span>
<span id="cb150-5"><a href="read-counting.html#cb150-5" tabindex="-1"></a><span class="co"># make the analysis directory if it doesn&#39;t already exist</span></span>
<span id="cb150-6"><a href="read-counting.html#cb150-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$SALMON_OUT_DIR</span></span>
<span id="cb150-7"><a href="read-counting.html#cb150-7" tabindex="-1"></a></span>
<span id="cb150-8"><a href="read-counting.html#cb150-8" tabindex="-1"></a><span class="co"># activate the salmon environment</span></span>
<span id="cb150-9"><a href="read-counting.html#cb150-9" tabindex="-1"></a><span class="ex">conda</span> activate salmon</span>
<span id="cb150-10"><a href="read-counting.html#cb150-10" tabindex="-1"></a></span>
<span id="cb150-11"><a href="read-counting.html#cb150-11" tabindex="-1"></a><span class="co"># loop through all of the fastq files</span></span>
<span id="cb150-12"><a href="read-counting.html#cb150-12" tabindex="-1"></a><span class="cf">for</span> fn <span class="kw">in</span> <span class="va">$DATA_DIR</span>/<span class="pp">*</span>.fastq.gz<span class="kw">;</span></span>
<span id="cb150-13"><a href="read-counting.html#cb150-13" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb150-14"><a href="read-counting.html#cb150-14" tabindex="-1"></a><span class="va">samp</span><span class="op">=</span><span class="kw">`</span><span class="fu">basename</span> <span class="va">${fn}</span><span class="kw">`</span></span>
<span id="cb150-15"><a href="read-counting.html#cb150-15" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;Processing sample </span><span class="va">${samp}</span><span class="st">&quot;</span></span>
<span id="cb150-16"><a href="read-counting.html#cb150-16" tabindex="-1"></a></span>
<span id="cb150-17"><a href="read-counting.html#cb150-17" tabindex="-1"></a><span class="co"># run salmon</span></span>
<span id="cb150-18"><a href="read-counting.html#cb150-18" tabindex="-1"></a><span class="ex">salmon</span> quant <span class="at">-i</span> <span class="va">$SALMON_INDEX_DIR</span> <span class="at">-l</span> A <span class="dt">\</span></span>
<span id="cb150-19"><a href="read-counting.html#cb150-19" tabindex="-1"></a>         <span class="at">-r</span> <span class="va">${fn}</span> <span class="dt">\</span></span>
<span id="cb150-20"><a href="read-counting.html#cb150-20" tabindex="-1"></a>         <span class="at">--useVBOpt</span> <span class="dt">\</span></span>
<span id="cb150-21"><a href="read-counting.html#cb150-21" tabindex="-1"></a>         <span class="at">-p</span> 4 <span class="at">--validateMappings</span> <span class="at">-o</span> <span class="va">$SALMON_OUT_DIR</span>/<span class="va">${samp}</span>_quant</span>
<span id="cb150-22"><a href="read-counting.html#cb150-22" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb150-23"><a href="read-counting.html#cb150-23" tabindex="-1"></a></span>
<span id="cb150-24"><a href="read-counting.html#cb150-24" tabindex="-1"></a><span class="co"># combine all of the output files into a merged count matrix</span></span>
<span id="cb150-25"><a href="read-counting.html#cb150-25" tabindex="-1"></a><span class="ex">salmon</span> quantmerge <span class="at">--quants</span> <span class="va">$SALMON_OUT_DIR</span>/<span class="pp">*</span>_quant <span class="at">--column</span> numreads <span class="at">-o</span> <span class="va">$SALMON_OUT_DIR</span>/salmon.gene_counts.merged.yeast.tsv </span>
<span id="cb150-26"><a href="read-counting.html#cb150-26" tabindex="-1"></a></span>
<span id="cb150-27"><a href="read-counting.html#cb150-27" tabindex="-1"></a><span class="co"># remove the _mRNA from gene name</span></span>
<span id="cb150-28"><a href="read-counting.html#cb150-28" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;&#39;</span> <span class="at">-E</span> <span class="st">&#39;s/^([^\t]+)_mRNA(\t|$)/\1\2/&#39;</span> <span class="va">$SALMON_OUT_DIR</span>/salmon.gene_counts.merged.yeast.tsv</span>
<span id="cb150-29"><a href="read-counting.html#cb150-29" tabindex="-1"></a></span>
<span id="cb150-30"><a href="read-counting.html#cb150-30" tabindex="-1"></a><span class="co"># we can also create a table of tpm values per gene by changing the --column flag</span></span>
<span id="cb150-31"><a href="read-counting.html#cb150-31" tabindex="-1"></a><span class="ex">salmon</span> quantmerge <span class="at">--quants</span> <span class="va">$SALMON_OUT_DIR</span>/<span class="pp">*</span>_quant <span class="at">--column</span> tpm <span class="dt">\</span></span>
<span id="cb150-32"><a href="read-counting.html#cb150-32" tabindex="-1"></a>          <span class="at">-o</span> <span class="va">$SALMON_OUT_DIR</span>/salmon.gene_tpm.merged.yeast.tsv</span>
<span id="cb150-33"><a href="read-counting.html#cb150-33" tabindex="-1"></a></span>
<span id="cb150-34"><a href="read-counting.html#cb150-34" tabindex="-1"></a><span class="co"># remove the _mRNA from gene name</span></span>
<span id="cb150-35"><a href="read-counting.html#cb150-35" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;&#39;</span> <span class="at">-E</span> <span class="st">&#39;s/^([^\t]+)_mRNA(\t|$)/\1\2/&#39;</span> <span class="va">$SALMON_OUT_DIR</span>/salmon.gene_tpm.merged.yeast.tsv</span>
<span id="cb150-36"><a href="read-counting.html#cb150-36" tabindex="-1"></a></span>
<span id="cb150-37"><a href="read-counting.html#cb150-37" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
<pre><code>## Processing sample YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP1_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP1_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP1_R1.fastq.gz_quant/logs
## [2023-10-26 16:15:59.990] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:15:59.991] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:15:59.991] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:15:59.991] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:15:59.991] [jointLog] [info] parsing read library format
## -----------------------------------------
## | Loading contig table | Time = 6.3758 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 1.5446 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 191.33 us
## -----------------------------------------
## [2023-10-26 16:15:59.992] [jointLog] [info] There is 1 library.
## [2023-10-26 16:15:59.994] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:15:59.995] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading mphf table | Time = 11.646 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## -----------------------------------------
## | Loading contig boundaries | Time = 35.129 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 5.7992 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 40.245 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 6.5275 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 579.67 us
## -----------------------------------------
## [2023-10-26 16:16:00.109] [jointLog] [info] done
## [2023-10-26 16:16:00.199] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:00.201] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:00.201] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:00.378] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:00.789] [jointLog] [info] Thread saw mini-batch with a maximum of 1.32% zero probability fragments
## [2023-10-26 16:16:00.808] [jointLog] [info] Thread saw mini-batch with a maximum of 1.40% zero probability fragments
## [2023-10-26 16:16:00.819] [jointLog] [info] Thread saw mini-batch with a maximum of 1.22% zero probability fragments
## [2023-10-26 16:16:00.824] [jointLog] [info] Thread saw mini-batch with a maximum of 1.32% zero probability fragments
## [2023-10-26 16:16:00.847] [jointLog] [info] Computed 5,478 rich equivalence classes for further processing
## [2023-10-26 16:16:00.847] [jointLog] [info] Counted 186,104 total reads in the equivalence classes 
## [2023-10-26 16:16:00.854] [jointLog] [info] Number of mappings discarded because of alignment score : 31,126
## [2023-10-26 16:16:00.854] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 7,997
## [2023-10-26 16:16:00.854] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 6,248
## [2023-10-26 16:16:00.854] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:00.855] [jointLog] [warning] Only 186104 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:00.855] [jointLog] [info] Mapping rate = 79.7778%
## 
## [2023-10-26 16:16:00.855] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:00.857] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:00.863] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:00.875] [jointLog] [info] iteration = 0 | max rel diff. = 2057.34
## [2023-10-26 16:16:02.144] [jointLog] [info] iteration = 100 | max rel diff. = 0.000215242
## [2023-10-26 16:16:02.145] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:02.145] [jointLog] [info] writing output 
## 
## Processing sample YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP2_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP2_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP2_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:02.564] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:02.564] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:02.564] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:02.564] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:02.564] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:02.564] [jointLog] [info] There is 1 library.
## [2023-10-26 16:16:02.564] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:02.564] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig table | Time = 3.4287 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 129.58 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 35.583 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 5.9878 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## -----------------------------------------
## | Loading contig boundaries | Time = 27.749 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.535 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 27.005 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.0361 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 76.584 us
## -----------------------------------------
## [2023-10-26 16:16:02.636] [jointLog] [info] done
## [2023-10-26 16:16:02.711] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:02.712] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:02.712] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:02.889] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:03.250] [jointLog] [info] Thread saw mini-batch with a maximum of 1.32% zero probability fragments
## [2023-10-26 16:16:03.274] [jointLog] [info] Thread saw mini-batch with a maximum of 1.22% zero probability fragments
## [2023-10-26 16:16:03.280] [jointLog] [info] Thread saw mini-batch with a maximum of 1.18% zero probability fragments
## [2023-10-26 16:16:03.281] [jointLog] [info] Thread saw mini-batch with a maximum of 1.40% zero probability fragments
## [2023-10-26 16:16:03.307] [jointLog] [info] Computed 5,293 rich equivalence classes for further processing
## [2023-10-26 16:16:03.307] [jointLog] [info] Counted 173,313 total reads in the equivalence classes 
## [2023-10-26 16:16:03.314] [jointLog] [info] Number of mappings discarded because of alignment score : 25,173
## [2023-10-26 16:16:03.314] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 7,099
## [2023-10-26 16:16:03.314] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 5,772
## [2023-10-26 16:16:03.314] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:03.315] [jointLog] [warning] Only 173313 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:03.315] [jointLog] [info] Mapping rate = 80.3081%
## 
## [2023-10-26 16:16:03.315] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:03.315] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:03.318] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:03.330] [jointLog] [info] iteration = 0 | max rel diff. = 1476.38
## [2023-10-26 16:16:04.568] [jointLog] [info] iteration = 100 | max rel diff. = 0.00017473
## [2023-10-26 16:16:04.568] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:04.568] [jointLog] [info] writing output 
## 
## Processing sample YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP3_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP3_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP3_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:05.148] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:05.148] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:05.148] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:05.148] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:05.148] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:05.148] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 4.4033 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 100.25 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 33.875 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.0534 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## [2023-10-26 16:16:05.149] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:05.149] [jointLog] [info] Loading dense pufferfish index.
## Inventory entries filled: 49
## -----------------------------------------
## | Loading contig boundaries | Time = 27.141 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.5127 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 27.572 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.2762 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 63.959 us
## -----------------------------------------
## [2023-10-26 16:16:05.221] [jointLog] [info] done
## [2023-10-26 16:16:05.289] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:05.290] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:05.290] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:05.463] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:05.726] [jointLog] [info] Thread saw mini-batch with a maximum of 1.58% zero probability fragments
## [2023-10-26 16:16:05.738] [jointLog] [info] Thread saw mini-batch with a maximum of 1.56% zero probability fragments
## [2023-10-26 16:16:05.742] [jointLog] [info] Thread saw mini-batch with a maximum of 1.52% zero probability fragments
## [2023-10-26 16:16:05.743] [jointLog] [info] Thread saw mini-batch with a maximum of 1.55% zero probability fragments
## [2023-10-26 16:16:05.766] [jointLog] [info] Computed 5,306 rich equivalence classes for further processing
## [2023-10-26 16:16:05.766] [jointLog] [info] Counted 158,068 total reads in the equivalence classes 
## [2023-10-26 16:16:05.773] [jointLog] [info] Number of mappings discarded because of alignment score : 19,662
## [2023-10-26 16:16:05.773] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 6,467
## [2023-10-26 16:16:05.773] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 4,929
## [2023-10-26 16:16:05.773] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:05.773] [jointLog] [warning] Only 158068 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:05.773] [jointLog] [info] Mapping rate = 79.4008%
## 
## [2023-10-26 16:16:05.773] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:05.774] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:05.777] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:05.788] [jointLog] [info] iteration = 0 | max rel diff. = 483.863
## [2023-10-26 16:16:07.056] [jointLog] [info] iteration = 100 | max rel diff. = 0.000307156
## [2023-10-26 16:16:07.056] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:07.056] [jointLog] [info] writing output 
## 
## Processing sample YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP4_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP4_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP4_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:07.715] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:07.715] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:07.715] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:07.715] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:07.715] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:07.715] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 3.4127 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 97.084 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 30.875 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.0134 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## [2023-10-26 16:16:07.716] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:07.716] [jointLog] [info] Loading dense pufferfish index.
## Inventory entries filled: 49
## -----------------------------------------
## | Loading contig boundaries | Time = 26.329 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.4595 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 27.063 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.1234 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 66.834 us
## -----------------------------------------
## [2023-10-26 16:16:07.786] [jointLog] [info] done
## [2023-10-26 16:16:07.853] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:07.854] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:07.854] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:08.029] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:08.297] [jointLog] [info] Thread saw mini-batch with a maximum of 1.26% zero probability fragments
## [2023-10-26 16:16:08.298] [jointLog] [info] Thread saw mini-batch with a maximum of 1.32% zero probability fragments
## [2023-10-26 16:16:08.300] [jointLog] [info] Thread saw mini-batch with a maximum of 1.26% zero probability fragments
## [2023-10-26 16:16:08.322] [jointLog] [info] Thread saw mini-batch with a maximum of 1.20% zero probability fragments
## [2023-10-26 16:16:08.342] [jointLog] [info] Computed 5,260 rich equivalence classes for further processing
## [2023-10-26 16:16:08.342] [jointLog] [info] Counted 165,617 total reads in the equivalence classes 
## [2023-10-26 16:16:08.348] [jointLog] [info] Number of mappings discarded because of alignment score : 20,744
## [2023-10-26 16:16:08.348] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 6,497
## [2023-10-26 16:16:08.348] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 5,015
## [2023-10-26 16:16:08.348] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:08.348] [jointLog] [warning] Only 165617 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:08.348] [jointLog] [info] Mapping rate = 80.4779%
## 
## [2023-10-26 16:16:08.348] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:08.348] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:08.351] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:08.361] [jointLog] [info] iteration = 0 | max rel diff. = 1859.21
## [2023-10-26 16:16:09.656] [jointLog] [info] iteration = 100 | max rel diff. = 0.000249168
## [2023-10-26 16:16:09.656] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:09.656] [jointLog] [info] writing output 
## 
## Processing sample YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP1_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP1_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP1_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:10.282] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:10.282] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:10.282] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:10.282] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:10.282] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:10.282] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 3.5464 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 114.71 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 33.167 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 5.8997 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## [2023-10-26 16:16:10.282] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:10.282] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig boundaries | Time = 27.302 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.4788 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 27.364 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.24 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 70.167 us
## -----------------------------------------
## [2023-10-26 16:16:10.354] [jointLog] [info] done
## [2023-10-26 16:16:10.421] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:10.422] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:10.422] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:10.591] [jointLog] [info] Automatically detected most likely library type as SR
## 
## [2023-10-26 16:16:10.756] [jointLog] [info] Thread saw mini-batch with a maximum of 1.16% zero probability fragments
## [2023-10-26 16:16:10.765] [jointLog] [info] Thread saw mini-batch with a maximum of 1.20% zero probability fragments
## [2023-10-26 16:16:10.776] [jointLog] [info] Thread saw mini-batch with a maximum of 1.20% zero probability fragments
## [2023-10-26 16:16:10.780] [jointLog] [info] Thread saw mini-batch with a maximum of 1.18% zero probability fragments
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:10.798] [jointLog] [info] Computed 5,121 rich equivalence classes for further processing
## [2023-10-26 16:16:10.798] [jointLog] [info] Counted 130,772 total reads in the equivalence classes 
## [2023-10-26 16:16:10.804] [jointLog] [info] Number of mappings discarded because of alignment score : 28,846
## [2023-10-26 16:16:10.804] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 6,943
## [2023-10-26 16:16:10.804] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 5,380
## [2023-10-26 16:16:10.804] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:10.804] [jointLog] [warning] Only 130772 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:10.804] [jointLog] [info] Mapping rate = 78.2714%
## 
## [2023-10-26 16:16:10.804] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:10.805] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:10.807] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:10.816] [jointLog] [info] iteration = 0 | max rel diff. = 1687.17
## [2023-10-26 16:16:12.056] [jointLog] [info] iteration = 100 | max rel diff. = 0.000122859
## [2023-10-26 16:16:12.057] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:12.057] [jointLog] [info] writing output 
## 
## Processing sample YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP2_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP2_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP2_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:12.865] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:12.865] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:12.865] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:12.865] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:12.865] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:12.865] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 3.6053 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 97.416 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 30.834 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 5.8202 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## [2023-10-26 16:16:12.865] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:12.865] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig boundaries | Time = 26.288 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.4539 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 27.096 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.1037 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 71.625 us
## -----------------------------------------
## [2023-10-26 16:16:12.935] [jointLog] [info] done
## [2023-10-26 16:16:13.004] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:13.005] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:13.005] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:13.173] [jointLog] [info] Automatically detected most likely library type as SR
## 
## [2023-10-26 16:16:13.346] [jointLog] [info] Thread saw mini-batch with a maximum of 1.10% zero probability fragments
## [2023-10-26 16:16:13.351] [jointLog] [info] Thread saw mini-batch with a maximum of 1.12% zero probability fragments
## [2023-10-26 16:16:13.364] [jointLog] [info] Thread saw mini-batch with a maximum of 1.12% zero probability fragments
## [2023-10-26 16:16:13.371] [jointLog] [info] Thread saw mini-batch with a maximum of 1.14% zero probability fragments
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:13.388] [jointLog] [info] Computed 5,107 rich equivalence classes for further processing
## [2023-10-26 16:16:13.388] [jointLog] [info] Counted 135,238 total reads in the equivalence classes 
## [2023-10-26 16:16:13.394] [jointLog] [info] Number of mappings discarded because of alignment score : 19,440
## [2023-10-26 16:16:13.394] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 5,837
## [2023-10-26 16:16:13.394] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 4,792
## [2023-10-26 16:16:13.394] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:13.394] [jointLog] [warning] Only 135238 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:13.394] [jointLog] [info] Mapping rate = 79.667%
## 
## [2023-10-26 16:16:13.394] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:13.394] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:13.397] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:13.405] [jointLog] [info] iteration = 0 | max rel diff. = 1365.23
## [2023-10-26 16:16:14.702] [jointLog] [info] iteration = 100 | max rel diff. = 0.00138672
## [2023-10-26 16:16:14.703] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:14.703] [jointLog] [info] writing output 
## 
## Processing sample YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP3_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP3_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP3_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:15.601] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:15.601] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:15.601] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:15.601] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:15.601] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:15.601] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 4.5574 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 123.63 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 32.375 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.253 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## [2023-10-26 16:16:15.602] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:15.602] [jointLog] [info] Loading dense pufferfish index.
## Inventory entries filled: 49
## -----------------------------------------
## | Loading contig boundaries | Time = 27.792 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.5702 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 28.717 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.0438 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 64.25 us
## -----------------------------------------
## [2023-10-26 16:16:15.676] [jointLog] [info] done
## [2023-10-26 16:16:15.750] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:15.751] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:15.751] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:15.968] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:16.453] [jointLog] [info] Thread saw mini-batch with a maximum of 1.52% zero probability fragments
## [2023-10-26 16:16:16.463] [jointLog] [info] Thread saw mini-batch with a maximum of 1.30% zero probability fragments
## [2023-10-26 16:16:16.487] [jointLog] [info] Thread saw mini-batch with a maximum of 1.58% zero probability fragments
## [2023-10-26 16:16:16.490] [jointLog] [info] Thread saw mini-batch with a maximum of 1.62% zero probability fragments
## [2023-10-26 16:16:16.519] [jointLog] [info] Computed 5,212 rich equivalence classes for further processing
## [2023-10-26 16:16:16.519] [jointLog] [info] Counted 161,111 total reads in the equivalence classes 
## [2023-10-26 16:16:16.525] [jointLog] [info] Number of mappings discarded because of alignment score : 23,201
## [2023-10-26 16:16:16.525] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 7,784
## [2023-10-26 16:16:16.525] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 6,012
## [2023-10-26 16:16:16.525] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:16.525] [jointLog] [warning] Only 161111 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:16.525] [jointLog] [info] Mapping rate = 76.7192%
## 
## [2023-10-26 16:16:16.525] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:16.526] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:16.529] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:16.540] [jointLog] [info] iteration = 0 | max rel diff. = 1651.79
## [2023-10-26 16:16:17.891] [jointLog] [info] iteration = 100 | max rel diff. = 0.000125455
## [2023-10-26 16:16:17.891] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:17.891] [jointLog] [info] writing output 
## 
## Processing sample YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP4_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP4_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP4_R1.fastq.gz_quant/logs
## -----------------------------------------
## | Loading contig table | Time = 4.4065 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 98.292 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 30.541 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.1712 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## [2023-10-26 16:16:18.767] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:18.767] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:18.767] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:18.767] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:18.767] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:18.767] [jointLog] [info] There is 1 library.
## [2023-10-26 16:16:18.767] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:18.767] [jointLog] [info] Loading dense pufferfish index.
## Inventory entries filled: 49
## -----------------------------------------
## | Loading contig boundaries | Time = 26.763 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.5253 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 27.182 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.0114 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 85.584 us
## -----------------------------------------
## [2023-10-26 16:16:18.839] [jointLog] [info] done
## [2023-10-26 16:16:18.906] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:18.907] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:18.907] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:19.094] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:19.365] [jointLog] [info] Thread saw mini-batch with a maximum of 1.46% zero probability fragments
## [2023-10-26 16:16:19.373] [jointLog] [info] Thread saw mini-batch with a maximum of 1.46% zero probability fragments
## [2023-10-26 16:16:19.386] [jointLog] [info] Thread saw mini-batch with a maximum of 1.30% zero probability fragments
## [2023-10-26 16:16:19.388] [jointLog] [info] Thread saw mini-batch with a maximum of 1.28% zero probability fragments
## [2023-10-26 16:16:19.409] [jointLog] [info] Computed 5,243 rich equivalence classes for further processing
## [2023-10-26 16:16:19.409] [jointLog] [info] Counted 163,485 total reads in the equivalence classes 
## [2023-10-26 16:16:19.415] [jointLog] [info] Number of mappings discarded because of alignment score : 20,601
## [2023-10-26 16:16:19.415] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 7,334
## [2023-10-26 16:16:19.415] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 5,778
## [2023-10-26 16:16:19.415] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:19.416] [jointLog] [warning] Only 163485 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:19.416] [jointLog] [info] Mapping rate = 78.4744%
## 
## [2023-10-26 16:16:19.416] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:19.416] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:19.420] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:19.431] [jointLog] [info] iteration = 0 | max rel diff. = 1711.21
## [2023-10-26 16:16:20.762] [jointLog] [info] iteration = 100 | max rel diff. = 0.000307278
## [2023-10-26 16:16:20.762] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:20.762] [jointLog] [info] writing output 
## 
## Processing sample YPS606_WT_ETOH_REP1_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP1_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP1_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP1_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:21.344] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:21.344] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:21.344] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:21.344] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:21.344] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:21.344] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 3.4204 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 127.21 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 30.709 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 5.9205 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## [2023-10-26 16:16:21.344] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:21.344] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig boundaries | Time = 26.265 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.4185 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 26.725 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.113 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 60.75 us
## -----------------------------------------
## [2023-10-26 16:16:21.414] [jointLog] [info] done
## [2023-10-26 16:16:21.480] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:21.481] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:21.481] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:21.654] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:21.872] [jointLog] [info] Thread saw mini-batch with a maximum of 1.40% zero probability fragments
## [2023-10-26 16:16:21.879] [jointLog] [info] Thread saw mini-batch with a maximum of 1.16% zero probability fragments
## [2023-10-26 16:16:21.882] [jointLog] [info] Thread saw mini-batch with a maximum of 1.36% zero probability fragments
## [2023-10-26 16:16:21.886] [jointLog] [info] Thread saw mini-batch with a maximum of 1.30% zero probability fragments
## [2023-10-26 16:16:21.905] [jointLog] [info] Computed 5,053 rich equivalence classes for further processing
## [2023-10-26 16:16:21.905] [jointLog] [info] Counted 149,230 total reads in the equivalence classes 
## [2023-10-26 16:16:21.910] [jointLog] [info] Number of mappings discarded because of alignment score : 15,755
## [2023-10-26 16:16:21.910] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 5,253
## [2023-10-26 16:16:21.910] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 4,402
## [2023-10-26 16:16:21.910] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:21.911] [jointLog] [warning] Only 149230 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:21.911] [jointLog] [info] Mapping rate = 82.181%
## 
## [2023-10-26 16:16:21.911] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:21.911] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:21.914] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:21.921] [jointLog] [info] iteration = 0 | max rel diff. = 1406.19
## [2023-10-26 16:16:23.115] [jointLog] [info] iteration = 100 | max rel diff. = 2.70713e-05
## [2023-10-26 16:16:23.115] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:23.115] [jointLog] [info] writing output 
## 
## Processing sample YPS606_WT_ETOH_REP2_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP2_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP2_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP2_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:23.931] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:23.931] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:23.931] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:23.931] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:23.931] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:23.931] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 5.807 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 134.04 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 34.542 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.0668 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## [2023-10-26 16:16:23.931] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:23.931] [jointLog] [info] Loading dense pufferfish index.
## Inventory entries filled: 49
## -----------------------------------------
## | Loading contig boundaries | Time = 27.445 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.6328 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 28.627 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.1181 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 75.375 us
## -----------------------------------------
## [2023-10-26 16:16:24.007] [jointLog] [info] done
## [2023-10-26 16:16:24.076] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:24.077] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:24.077] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:24.267] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:24.607] [jointLog] [info] Thread saw mini-batch with a maximum of 1.36% zero probability fragments
## [2023-10-26 16:16:24.614] [jointLog] [info] Thread saw mini-batch with a maximum of 1.46% zero probability fragments
## [2023-10-26 16:16:24.617] [jointLog] [info] Thread saw mini-batch with a maximum of 1.12% zero probability fragments
## [2023-10-26 16:16:24.623] [jointLog] [info] Thread saw mini-batch with a maximum of 1.34% zero probability fragments
## [2023-10-26 16:16:24.649] [jointLog] [info] Computed 5,185 rich equivalence classes for further processing
## [2023-10-26 16:16:24.649] [jointLog] [info] Counted 163,064 total reads in the equivalence classes 
## [2023-10-26 16:16:24.658] [jointLog] [info] Number of mappings discarded because of alignment score : 22,598
## [2023-10-26 16:16:24.658] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 6,365
## [2023-10-26 16:16:24.658] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 5,134
## [2023-10-26 16:16:24.658] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:24.658] [jointLog] [warning] Only 163064 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:24.658] [jointLog] [info] Mapping rate = 80.9046%
## 
## [2023-10-26 16:16:24.658] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:24.660] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:24.664] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:24.678] [jointLog] [info] iteration = 0 | max rel diff. = 2044.58
## [2023-10-26 16:16:26.012] [jointLog] [info] iteration = 100 | max rel diff. = 0.000322112
## [2023-10-26 16:16:26.012] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:26.012] [jointLog] [info] writing output 
## 
## Processing sample YPS606_WT_ETOH_REP3_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP3_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP3_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP3_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:26.521] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:26.521] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:26.521] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:26.521] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:26.521] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:26.521] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 3.8293 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 130.42 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 32.208 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.6345 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## [2023-10-26 16:16:26.522] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:26.522] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig boundaries | Time = 27.488 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.714 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 28.752 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 5.2932 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 84.25 us
## -----------------------------------------
## [2023-10-26 16:16:26.597] [jointLog] [info] done
## [2023-10-26 16:16:26.686] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:26.687] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:26.687] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:26.924] [jointLog] [info] Automatically detected most likely library type as SR
## 
## [2023-10-26 16:16:27.376] [jointLog] [info] Thread saw mini-batch with a maximum of 1.78% zero probability fragments
## [2023-10-26 16:16:27.377] [jointLog] [info] Thread saw mini-batch with a maximum of 1.62% zero probability fragments
## [2023-10-26 16:16:27.389] [jointLog] [info] Thread saw mini-batch with a maximum of 1.72% zero probability fragments
## [2023-10-26 16:16:27.390] [jointLog] [info] Thread saw mini-batch with a maximum of 1.62% zero probability fragments
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:27.429] [jointLog] [info] Computed 5,301 rich equivalence classes for further processing
## [2023-10-26 16:16:27.429] [jointLog] [info] Counted 171,045 total reads in the equivalence classes 
## [2023-10-26 16:16:27.435] [jointLog] [info] Number of mappings discarded because of alignment score : 18,641
## [2023-10-26 16:16:27.435] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 6,542
## [2023-10-26 16:16:27.435] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 5,138
## [2023-10-26 16:16:27.435] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:27.436] [jointLog] [warning] Only 171045 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:27.436] [jointLog] [info] Mapping rate = 79.6503%
## 
## [2023-10-26 16:16:27.436] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:27.437] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:27.440] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:27.450] [jointLog] [info] iteration = 0 | max rel diff. = 2209.07
## [2023-10-26 16:16:28.651] [jointLog] [info] iteration = 100 | max rel diff. = 6.51043e-05
## [2023-10-26 16:16:28.651] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:28.651] [jointLog] [info] writing output 
## 
## Processing sample YPS606_WT_ETOH_REP4_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP4_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP4_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP4_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:29.683] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:29.683] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:29.683] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:29.683] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:29.683] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:29.683] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 4.7789 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 94.041 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 31.209 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 5.8783 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## [2023-10-26 16:16:29.684] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:29.684] [jointLog] [info] Loading dense pufferfish index.
## Inventory entries filled: 49
## -----------------------------------------
## | Loading contig boundaries | Time = 26.255 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.4236 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 26.623 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.1305 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 72.75 us
## -----------------------------------------
## [2023-10-26 16:16:29.754] [jointLog] [info] done
## [2023-10-26 16:16:29.827] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:29.828] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:29.828] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:29.996] [jointLog] [info] Automatically detected most likely library type as SR
## 
## [2023-10-26 16:16:30.194] [jointLog] [info] Thread saw mini-batch with a maximum of 1.36% zero probability fragments
## [2023-10-26 16:16:30.195] [jointLog] [info] Thread saw mini-batch with a maximum of 1.26% zero probability fragments
## [2023-10-26 16:16:30.204] [jointLog] [info] Thread saw mini-batch with a maximum of 1.38% zero probability fragments
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:30.214] [jointLog] [info] Thread saw mini-batch with a maximum of 1.46% zero probability fragments
## [2023-10-26 16:16:30.230] [jointLog] [info] Computed 5,217 rich equivalence classes for further processing
## [2023-10-26 16:16:30.230] [jointLog] [info] Counted 151,376 total reads in the equivalence classes 
## [2023-10-26 16:16:30.235] [jointLog] [info] Number of mappings discarded because of alignment score : 18,141
## [2023-10-26 16:16:30.235] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 5,851
## [2023-10-26 16:16:30.235] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 4,515
## [2023-10-26 16:16:30.235] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:30.236] [jointLog] [warning] Only 151376 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:30.236] [jointLog] [info] Mapping rate = 80.8119%
## 
## [2023-10-26 16:16:30.236] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:30.236] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:30.238] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:30.246] [jointLog] [info] iteration = 0 | max rel diff. = 2134.58
## [2023-10-26 16:16:31.426] [jointLog] [info] iteration = 100 | max rel diff. = 0.000703782
## [2023-10-26 16:16:31.426] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:31.426] [jointLog] [info] writing output 
## 
## Processing sample YPS606_WT_MOCK_REP1_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP1_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP1_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP1_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:32.332] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:32.332] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:32.332] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:32.332] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:32.332] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:32.332] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 3.4363 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 108.08 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 32.042 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.0111 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## [2023-10-26 16:16:32.333] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:32.333] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig boundaries | Time = 26.453 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.5232 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 26.941 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.0996 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 74.041 us
## -----------------------------------------
## [2023-10-26 16:16:32.403] [jointLog] [info] done
## [2023-10-26 16:16:32.470] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:32.471] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:32.471] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:32.635] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:32.901] [jointLog] [info] Thread saw mini-batch with a maximum of 1.38% zero probability fragments
## [2023-10-26 16:16:32.916] [jointLog] [info] Thread saw mini-batch with a maximum of 1.24% zero probability fragments
## [2023-10-26 16:16:32.917] [jointLog] [info] Thread saw mini-batch with a maximum of 1.30% zero probability fragments
## [2023-10-26 16:16:32.922] [jointLog] [info] Thread saw mini-batch with a maximum of 1.20% zero probability fragments
## [2023-10-26 16:16:32.939] [jointLog] [info] Computed 5,305 rich equivalence classes for further processing
## [2023-10-26 16:16:32.939] [jointLog] [info] Counted 177,055 total reads in the equivalence classes 
## [2023-10-26 16:16:32.944] [jointLog] [info] Number of mappings discarded because of alignment score : 31,142
## [2023-10-26 16:16:32.944] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 8,597
## [2023-10-26 16:16:32.944] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 6,941
## [2023-10-26 16:16:32.944] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:32.944] [jointLog] [warning] Only 177055 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:32.944] [jointLog] [info] Mapping rate = 79.2054%
## 
## [2023-10-26 16:16:32.944] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:32.944] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:32.947] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:32.954] [jointLog] [info] iteration = 0 | max rel diff. = 2091.83
## [2023-10-26 16:16:34.249] [jointLog] [info] iteration = 100 | max rel diff. = 0.000208332
## [2023-10-26 16:16:34.249] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:34.249] [jointLog] [info] writing output 
## 
## Processing sample YPS606_WT_MOCK_REP2_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP2_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP2_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP2_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:34.893] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:34.893] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:34.893] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:34.893] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:34.893] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:34.894] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 4.187 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 104.21 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 29.75 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 5.9423 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## [2023-10-26 16:16:34.894] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:34.894] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig boundaries | Time = 26.304 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.5596 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 26.808 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.0102 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 65.5 us
## -----------------------------------------
## [2023-10-26 16:16:34.964] [jointLog] [info] done
## [2023-10-26 16:16:35.030] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:35.031] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:35.031] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:35.202] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:35.429] [jointLog] [info] Thread saw mini-batch with a maximum of 1.26% zero probability fragments
## [2023-10-26 16:16:35.429] [jointLog] [info] Thread saw mini-batch with a maximum of 1.32% zero probability fragments
## [2023-10-26 16:16:35.441] [jointLog] [info] Thread saw mini-batch with a maximum of 1.54% zero probability fragments
## [2023-10-26 16:16:35.451] [jointLog] [info] Thread saw mini-batch with a maximum of 1.46% zero probability fragments
## [2023-10-26 16:16:35.469] [jointLog] [info] Computed 5,175 rich equivalence classes for further processing
## [2023-10-26 16:16:35.469] [jointLog] [info] Counted 147,312 total reads in the equivalence classes 
## [2023-10-26 16:16:35.475] [jointLog] [warning] 0.00160026% of fragments were shorter than the k used to build the index.
## If this fraction is too large, consider re-building the index with a smaller k.
## The minimum read size found was 27.
## 
## 
## [2023-10-26 16:16:35.475] [jointLog] [info] Number of mappings discarded because of alignment score : 24,562
## [2023-10-26 16:16:35.475] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 7,192
## [2023-10-26 16:16:35.475] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 5,915
## [2023-10-26 16:16:35.475] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:35.475] [jointLog] [warning] Only 147312 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:35.475] [jointLog] [info] Mapping rate = 78.5794%
## 
## [2023-10-26 16:16:35.475] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:35.477] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:35.479] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:35.487] [jointLog] [info] iteration = 0 | max rel diff. = 1719.02
## [2023-10-26 16:16:36.895] [jointLog] [info] iteration = 100 | max rel diff. = 0.000465167
## [2023-10-26 16:16:36.895] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:36.895] [jointLog] [info] writing output 
## 
## Processing sample YPS606_WT_MOCK_REP3_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP3_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP3_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP3_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:37.481] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:37.481] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:37.481] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:37.481] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:37.481] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:37.481] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 3.5011 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 93.583 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 31.709 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.4904 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## [2023-10-26 16:16:37.481] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:37.481] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig boundaries | Time = 27.659 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.6508 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 30.155 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 4.43 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 72.25 us
## -----------------------------------------
## [2023-10-26 16:16:37.557] [jointLog] [info] done
## [2023-10-26 16:16:37.631] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:37.632] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:37.632] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:37.835] [jointLog] [info] Automatically detected most likely library type as SR
## 
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:38.275] [jointLog] [info] Thread saw mini-batch with a maximum of 1.58% zero probability fragments
## [2023-10-26 16:16:38.282] [jointLog] [info] Thread saw mini-batch with a maximum of 1.64% zero probability fragments
## [2023-10-26 16:16:38.289] [jointLog] [info] Thread saw mini-batch with a maximum of 1.58% zero probability fragments
## [2023-10-26 16:16:38.310] [jointLog] [info] Thread saw mini-batch with a maximum of 1.54% zero probability fragments
## [2023-10-26 16:16:38.336] [jointLog] [info] Computed 5,175 rich equivalence classes for further processing
## [2023-10-26 16:16:38.336] [jointLog] [info] Counted 173,912 total reads in the equivalence classes 
## [2023-10-26 16:16:38.341] [jointLog] [info] Number of mappings discarded because of alignment score : 25,288
## [2023-10-26 16:16:38.341] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 8,408
## [2023-10-26 16:16:38.341] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 6,477
## [2023-10-26 16:16:38.341] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:38.342] [jointLog] [warning] Only 173912 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:38.342] [jointLog] [info] Mapping rate = 77.3743%
## 
## [2023-10-26 16:16:38.342] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:38.342] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:38.345] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:38.355] [jointLog] [info] iteration = 0 | max rel diff. = 1677.12
## [2023-10-26 16:16:39.603] [jointLog] [info] iteration = 100 | max rel diff. = 7.7896e-05
## [2023-10-26 16:16:39.603] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:39.603] [jointLog] [info] writing output 
## 
## Processing sample YPS606_WT_MOCK_REP4_R1.fastq.gz
## Version Info: This is the most recent version of salmon.
## ### salmon (selective-alignment-based) v1.10.0
## ### [ program ] =&gt; salmon 
## ### [ command ] =&gt; quant 
## ### [ index ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Reference/index_salmon_Saccharomyces_cerevisiae.R64-1-1 }
## ### [ libType ] =&gt; { A }
## ### [ unmatedReads ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP4_R1.fastq.gz }
## ### [ useVBOpt ] =&gt; { }
## ### [ threads ] =&gt; { 4 }
## ### [ validateMappings ] =&gt; { }
## ### [ output ] =&gt; { /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP4_R1.fastq.gz_quant }
## Logs will be written to /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP4_R1.fastq.gz_quant/logs
## [2023-10-26 16:16:40.066] [jointLog] [info] setting maxHashResizeThreads to 4
## [2023-10-26 16:16:40.066] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
## [2023-10-26 16:16:40.066] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
## [2023-10-26 16:16:40.066] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.
## [2023-10-26 16:16:40.066] [jointLog] [info] parsing read library format
## [2023-10-26 16:16:40.066] [jointLog] [info] There is 1 library.
## -----------------------------------------
## | Loading contig table | Time = 3.6218 ms
## -----------------------------------------
## size = 25029
## -----------------------------------------
## | Loading contig offsets | Time = 155.04 us
## -----------------------------------------
## -----------------------------------------
## | Loading reference lengths | Time = 33.083 us
## -----------------------------------------
## -----------------------------------------
## | Loading mphf table | Time = 6.4004 ms
## -----------------------------------------
## size = 12321058
## Number of ones: 25028
## Number of ones per inventory item: 512
## Inventory entries filled: 49
## [2023-10-26 16:16:40.066] [jointLog] [info] Loading pufferfish index
## [2023-10-26 16:16:40.066] [jointLog] [info] Loading dense pufferfish index.
## -----------------------------------------
## | Loading contig boundaries | Time = 26.799 ms
## -----------------------------------------
## size = 12321058
## -----------------------------------------
## | Loading sequence | Time = 2.5237 ms
## -----------------------------------------
## size = 11570218
## -----------------------------------------
## | Loading positions | Time = 38.085 ms
## -----------------------------------------
## size = 20892357
## -----------------------------------------
## | Loading reference sequence | Time = 18.538 ms
## -----------------------------------------
## -----------------------------------------
## | Loading reference accumulative lengths | Time = 122.42 us
## -----------------------------------------
## [2023-10-26 16:16:40.163] [jointLog] [info] done
## [2023-10-26 16:16:40.274] [jointLog] [info] Index contained 6,588 targets
## [2023-10-26 16:16:40.275] [jointLog] [info] Number of decoys : 17
## [2023-10-26 16:16:40.275] [jointLog] [info] First decoy index : 6,571 
## 
## 
## 
## 
## [2023-10-26 16:16:40.539] [jointLog] [info] Automatically detected most likely library type as SR
## 
## [2023-10-26 16:16:40.929] [jointLog] [info] Thread saw mini-batch with a maximum of 1.08% zero probability fragments
## [2023-10-26 16:16:40.931] [jointLog] [info] Thread saw mini-batch with a maximum of 1.16% zero probability fragments
## [2023-10-26 16:16:40.944] [jointLog] [info] Thread saw mini-batch with a maximum of 1.38% zero probability fragments
## [2023-10-26 16:16:40.963] [jointLog] [info] Thread saw mini-batch with a maximum of 1.14% zero probability fragments
## 
## 
## 
## 
## 
## 
## 
## 
## [2023-10-26 16:16:41.005] [jointLog] [info] Computed 5,167 rich equivalence classes for further processing
## [2023-10-26 16:16:41.005] [jointLog] [info] Counted 163,571 total reads in the equivalence classes 
## [2023-10-26 16:16:41.011] [jointLog] [info] Number of mappings discarded because of alignment score : 24,431
## [2023-10-26 16:16:41.011] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 7,543
## [2023-10-26 16:16:41.011] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 5,763
## [2023-10-26 16:16:41.011] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0
## [2023-10-26 16:16:41.011] [jointLog] [warning] Only 163571 fragments were mapped, but the number of burn-in fragments was set to 5000000.
## The effective lengths have been computed using the observed mappings.
## 
## [2023-10-26 16:16:41.011] [jointLog] [info] Mapping rate = 79.0714%
## 
## [2023-10-26 16:16:41.011] [jointLog] [info] finished quantifyLibrary()
## [2023-10-26 16:16:41.012] [jointLog] [info] Starting optimizer
## [2023-10-26 16:16:41.015] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2023-10-26 16:16:41.026] [jointLog] [info] iteration = 0 | max rel diff. = 1702.2
## [2023-10-26 16:16:42.401] [jointLog] [info] iteration = 100 | max rel diff. = 0.000337813
## [2023-10-26 16:16:42.401] [jointLog] [info] Finished optimizer
## [2023-10-26 16:16:42.401] [jointLog] [info] writing output 
## 
## Version Info: This is the most recent version of salmon.
## [2023-10-26 16:16:43.777] [mergeLog] [info] samples: [ /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP1_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP2_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP3_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP4_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP1_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP2_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP3_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP4_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP1_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP2_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP3_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP4_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP1_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP2_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP3_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP4_R1.fastq.gz_quant ]
## [2023-10-26 16:16:43.777] [mergeLog] [info] sample names : [ YPS606_MSN24_ETOH_REP1_R1.fastq.gz_quant, YPS606_MSN24_ETOH_REP2_R1.fastq.gz_quant, YPS606_MSN24_ETOH_REP3_R1.fastq.gz_quant, YPS606_MSN24_ETOH_REP4_R1.fastq.gz_quant, YPS606_MSN24_MOCK_REP1_R1.fastq.gz_quant, YPS606_MSN24_MOCK_REP2_R1.fastq.gz_quant, YPS606_MSN24_MOCK_REP3_R1.fastq.gz_quant, YPS606_MSN24_MOCK_REP4_R1.fastq.gz_quant, YPS606_WT_ETOH_REP1_R1.fastq.gz_quant, YPS606_WT_ETOH_REP2_R1.fastq.gz_quant, YPS606_WT_ETOH_REP3_R1.fastq.gz_quant, YPS606_WT_ETOH_REP4_R1.fastq.gz_quant, YPS606_WT_MOCK_REP1_R1.fastq.gz_quant, YPS606_WT_MOCK_REP2_R1.fastq.gz_quant, YPS606_WT_MOCK_REP3_R1.fastq.gz_quant, YPS606_WT_MOCK_REP4_R1.fastq.gz_quant ]
## [2023-10-26 16:16:43.777] [mergeLog] [info] output column : NUMREADS
## [2023-10-26 16:16:43.777] [mergeLog] [info] output file : /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/salmon.gene_counts.merged.yeast.tsv
## [2023-10-26 16:16:43.777] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP1_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.790] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP2_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.803] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP3_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.815] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP4_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.826] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP1_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.838] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP2_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.849] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP3_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.860] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP4_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.871] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP1_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.885] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP2_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.896] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP3_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.907] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP4_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.918] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP1_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.928] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP2_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.939] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP3_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:43.950] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP4_R1.fastq.gz_quant/quant.sf
## Version Info: This is the most recent version of salmon.
## [2023-10-26 16:16:44.401] [mergeLog] [info] samples: [ /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP1_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP2_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP3_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP4_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP1_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP2_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP3_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP4_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP1_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP2_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP3_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP4_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP1_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP2_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP3_R1.fastq.gz_quant, /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP4_R1.fastq.gz_quant ]
## [2023-10-26 16:16:44.401] [mergeLog] [info] sample names : [ YPS606_MSN24_ETOH_REP1_R1.fastq.gz_quant, YPS606_MSN24_ETOH_REP2_R1.fastq.gz_quant, YPS606_MSN24_ETOH_REP3_R1.fastq.gz_quant, YPS606_MSN24_ETOH_REP4_R1.fastq.gz_quant, YPS606_MSN24_MOCK_REP1_R1.fastq.gz_quant, YPS606_MSN24_MOCK_REP2_R1.fastq.gz_quant, YPS606_MSN24_MOCK_REP3_R1.fastq.gz_quant, YPS606_MSN24_MOCK_REP4_R1.fastq.gz_quant, YPS606_WT_ETOH_REP1_R1.fastq.gz_quant, YPS606_WT_ETOH_REP2_R1.fastq.gz_quant, YPS606_WT_ETOH_REP3_R1.fastq.gz_quant, YPS606_WT_ETOH_REP4_R1.fastq.gz_quant, YPS606_WT_MOCK_REP1_R1.fastq.gz_quant, YPS606_WT_MOCK_REP2_R1.fastq.gz_quant, YPS606_WT_MOCK_REP3_R1.fastq.gz_quant, YPS606_WT_MOCK_REP4_R1.fastq.gz_quant ]
## [2023-10-26 16:16:44.401] [mergeLog] [info] output column : TPM
## [2023-10-26 16:16:44.401] [mergeLog] [info] output file : /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/salmon.gene_tpm.merged.yeast.tsv
## [2023-10-26 16:16:44.401] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP1_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.413] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP2_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.424] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP3_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.436] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_ETOH_REP4_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.446] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP1_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.458] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP2_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.469] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP3_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.480] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_MSN24_MOCK_REP4_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.491] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP1_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.504] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP2_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.515] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP3_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.525] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_ETOH_REP4_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.538] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP1_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.549] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP2_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.560] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP3_R1.fastq.gz_quant/quant.sf
## [2023-10-26 16:16:44.570] [mergeLog] [info] Parsing /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Counts/Salmon/YPS606_WT_MOCK_REP4_R1.fastq.gz_quant/quant.sf</code></pre>
<p>This script loops through each sample and invokes salmon using default mostly options. The <code>-i</code> argument tells salmon where to find the index <code>-l A</code> tells salmon that it should automatically determine the library type of the sequencing reads (e.g. stranded vs. unstranded etc.). The -r arguments tell salmon where to find the SE reads for this sample (notice, salmon will accept gzipped FASTQ files directly). Finally, the <code>-p 4</code> argument tells salmon to make use of 4 threads and the <code>-o</code> argument specifies the directory where salmon’s quantification results should be written. The <code>–useVBOpt</code> flag sets to use variational Bayesian EM algorithm rather than the ‘standard EM’ to optimize abundance estimates (more accurate). Salmon exposes many different options to the user that enable extra features or modify default behavior. However, the purpose and behavior of all of those options is beyond the scope of this introductory tutorial. You can read about salmon’s many options in the documentation.</p>
</div>
</div>
<div id="questions-2" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Questions<a href="read-counting.html#questions-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><p>Identify which gene has the highest counts across all samples for both salmon and Rsubread outputs.</p></li>
<li><p>Redo the counting over the exons, rather than the genes (specify useMetaFeatures = FALSE) with RSubread. Use the bam files generated doing alignment reporting only unique reads, and call the featureCounts object fc.exon. Check the dimension of the counts slot to see how much larger it is.</p></li>
<li><p>What differences do you notice in the count values from Salmon vs Rsubread?</p></li>
<li><p>CHALLENGE: Download the full size fastq files from OneDrive &amp; use Salmon to get the read counts on the non-subsampled files.</p></li>
</ol>
<p>Be sure to knit this file into a pdf or html file once you’re finished.</p>
<p>System information for reproducibility:</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="read-counting.html#cb152-1" tabindex="-1"></a>pander<span class="sc">::</span><span class="fu">pander</span>(<span class="fu">sessionInfo</span>())</span></code></pre></div>
<p><strong>R version 4.3.1 (2023-06-16)</strong></p>
<p><strong>Platform:</strong> aarch64-apple-darwin20 (64-bit)</p>
<p><strong>locale:</strong>
en_US.UTF-8||en_US.UTF-8||en_US.UTF-8||C||en_US.UTF-8||en_US.UTF-8</p>
<p><strong>attached base packages:</strong>
<em>stats4</em>, <em>stats</em>, <em>graphics</em>, <em>grDevices</em>, <em>utils</em>, <em>datasets</em>, <em>methods</em> and <em>base</em></p>
<p><strong>other attached packages:</strong>
<em>Rsubread(v.2.14.2)</em>, <em>ShortRead(v.1.58.0)</em>, <em>GenomicAlignments(v.1.36.0)</em>, <em>SummarizedExperiment(v.1.30.2)</em>, <em>MatrixGenerics(v.1.12.3)</em>, <em>matrixStats(v.1.0.0)</em>, <em>Rsamtools(v.2.16.0)</em>, <em>GenomicRanges(v.1.52.1)</em>, <em>Biostrings(v.2.68.1)</em>, <em>GenomeInfoDb(v.1.36.4)</em>, <em>XVector(v.0.40.0)</em>, <em>BiocParallel(v.1.34.2)</em>, <em>Rfastp(v.1.10.0)</em>, <em>org.Sc.sgd.db(v.3.17.0)</em>, <em>AnnotationDbi(v.1.62.2)</em>, <em>IRanges(v.2.34.1)</em>, <em>S4Vectors(v.0.38.2)</em>, <em>Biobase(v.2.60.0)</em>, <em>BiocGenerics(v.0.46.0)</em>, <em>clusterProfiler(v.4.8.2)</em>, <em>ggVennDiagram(v.1.2.3)</em>, <em>tidytree(v.0.4.5)</em>, <em>igraph(v.1.5.1)</em>, <em>janitor(v.2.2.0)</em>, <em>BiocManager(v.1.30.22)</em>, <em>pander(v.0.6.5)</em>, <em>knitr(v.1.44)</em>, <em>here(v.1.0.1)</em>, <em>lubridate(v.1.9.3)</em>, <em>forcats(v.1.0.0)</em>, <em>stringr(v.1.5.0)</em>, <em>dplyr(v.1.1.3)</em>, <em>purrr(v.1.0.2)</em>, <em>readr(v.2.1.4)</em>, <em>tidyr(v.1.3.0)</em>, <em>tibble(v.3.2.1)</em>, <em>ggplot2(v.3.4.4)</em>, <em>tidyverse(v.2.0.0)</em> and <em>pacman(v.0.5.1)</em></p>
<p><strong>loaded via a namespace (and not attached):</strong>
<em>RColorBrewer(v.1.1-3)</em>, <em>rstudioapi(v.0.15.0)</em>, <em>jsonlite(v.1.8.7)</em>, <em>magrittr(v.2.0.3)</em>, <em>farver(v.2.1.1)</em>, <em>rmarkdown(v.2.25)</em>, <em>fs(v.1.6.3)</em>, <em>zlibbioc(v.1.46.0)</em>, <em>vctrs(v.0.6.4)</em>, <em>memoise(v.2.0.1)</em>, <em>RCurl(v.1.98-1.12)</em>, <em>ggtree(v.3.8.2)</em>, <em>S4Arrays(v.1.0.6)</em>, <em>htmltools(v.0.5.6.1)</em>, <em>gridGraphics(v.0.5-1)</em>, <em>sass(v.0.4.7)</em>, <em>bslib(v.0.5.1)</em>, <em>plyr(v.1.8.9)</em>, <em>cachem(v.1.0.8)</em>, <em>lifecycle(v.1.0.3)</em>, <em>pkgconfig(v.2.0.3)</em>, <em>Matrix(v.1.6-1.1)</em>, <em>R6(v.2.5.1)</em>, <em>fastmap(v.1.1.1)</em>, <em>gson(v.0.1.0)</em>, <em>GenomeInfoDbData(v.1.2.10)</em>, <em>snakecase(v.0.11.1)</em>, <em>digest(v.0.6.33)</em>, <em>aplot(v.0.2.2)</em>, <em>enrichplot(v.1.20.0)</em>, <em>colorspace(v.2.1-0)</em>, <em>patchwork(v.1.1.3)</em>, <em>rprojroot(v.2.0.3)</em>, <em>RSQLite(v.2.3.1)</em>, <em>hwriter(v.1.3.2.1)</em>, <em>fansi(v.1.0.5)</em>, <em>timechange(v.0.2.0)</em>, <em>abind(v.1.4-5)</em>, <em>httr(v.1.4.7)</em>, <em>polyclip(v.1.10-6)</em>, <em>compiler(v.4.3.1)</em>, <em>bit64(v.4.0.5)</em>, <em>withr(v.2.5.1)</em>, <em>downloader(v.0.4)</em>, <em>viridis(v.0.6.4)</em>, <em>DBI(v.1.1.3)</em>, <em>ggforce(v.0.4.1)</em>, <em>MASS(v.7.3-60)</em>, <em>DelayedArray(v.0.26.7)</em>, <em>rjson(v.0.2.21)</em>, <em>HDO.db(v.0.99.1)</em>, <em>tools(v.4.3.1)</em>, <em>ape(v.5.7-1)</em>, <em>scatterpie(v.0.2.1)</em>, <em>glue(v.1.6.2)</em>, <em>nlme(v.3.1-163)</em>, <em>GOSemSim(v.2.26.1)</em>, <em>grid(v.4.3.1)</em>, <em>shadowtext(v.0.1.2)</em>, <em>reshape2(v.1.4.4)</em>, <em>fgsea(v.1.26.0)</em>, <em>generics(v.0.1.3)</em>, <em>gtable(v.0.3.4)</em>, <em>tzdb(v.0.4.0)</em>, <em>data.table(v.1.14.8)</em>, <em>hms(v.1.1.3)</em>, <em>tidygraph(v.1.2.3)</em>, <em>utf8(v.1.2.3)</em>, <em>ggrepel(v.0.9.4)</em>, <em>pillar(v.1.9.0)</em>, <em>vroom(v.1.6.4)</em>, <em>yulab.utils(v.0.1.0)</em>, <em>splines(v.4.3.1)</em>, <em>tweenr(v.2.0.2)</em>, <em>treeio(v.1.24.3)</em>, <em>lattice(v.0.21-9)</em>, <em>deldir(v.1.0-9)</em>, <em>bit(v.4.0.5)</em>, <em>tidyselect(v.1.2.0)</em>, <em>GO.db(v.3.17.0)</em>, <em>gridExtra(v.2.3)</em>, <em>bookdown(v.0.36)</em>, <em>xfun(v.0.40)</em>, <em>graphlayouts(v.1.0.1)</em>, <em>stringi(v.1.7.12)</em>, <em>lazyeval(v.0.2.2)</em>, <em>ggfun(v.0.1.3)</em>, <em>yaml(v.2.3.7)</em>, <em>evaluate(v.0.22)</em>, <em>codetools(v.0.2-19)</em>, <em>interp(v.1.1-4)</em>, <em>ggraph(v.2.1.0)</em>, <em>archive(v.1.1.5)</em>, <em>qvalue(v.2.32.0)</em>, <em>RVenn(v.1.1.0)</em>, <em>ggplotify(v.0.1.2)</em>, <em>cli(v.3.6.1)</em>, <em>munsell(v.0.5.0)</em>, <em>jquerylib(v.0.1.4)</em>, <em>Rcpp(v.1.0.11)</em>, <em>png(v.0.1-8)</em>, <em>parallel(v.4.3.1)</em>, <em>blob(v.1.2.4)</em>, <em>jpeg(v.0.1-10)</em>, <em>latticeExtra(v.0.6-30)</em>, <em>DOSE(v.3.26.1)</em>, <em>bitops(v.1.0-7)</em>, <em>viridisLite(v.0.4.2)</em>, <em>scales(v.1.2.1)</em>, <em>crayon(v.1.5.2)</em>, <em>rlang(v.1.1.1)</em>, <em>cowplot(v.1.1.1)</em>, <em>fastmatch(v.1.1-4)</em> and <em>KEGGREST(v.1.40.1)</em></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="read-mapping.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-expression-edger.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-Read_Counting.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
