<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Working with Sequences: Raw Data &amp; Quality Control | Genomic Data Analysis Course Exercises</title>
  <meta name="description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Working with Sequences: Raw Data &amp; Quality Control | Genomic Data Analysis Course Exercises" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  <meta name="github-repo" content="clstacy/GenomicDataAnalysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Working with Sequences: Raw Data &amp; Quality Control | Genomic Data Analysis Course Exercises" />
  
  <meta name="twitter:description" content="These are exercises created for the 2023 course: Genomic Data Analysis at the University of Arkansas." />
  

<meta name="author" content="Carson Stacy" />


<meta name="date" content="2023-10-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gene-ontology.html"/>
<link rel="next" href="read-mapping.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/core-js-2.5.3/shim.min.js"></script>
<script src="libs/react-18.2.0/react.min.js"></script>
<script src="libs/react-18.2.0/react-dom.min.js"></script>
<script src="libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="libs/reactable-binding-0.4.4/reactable.js"></script>
<link href="libs/datatables-1/datatables.css" rel="stylesheet" />
<script src="libs/datatables-1/datatables.min.js"></script>
<link href="libs/vega-1/vega_plots.css" rel="stylesheet" />
<script src="libs/vega-1/vega.min.js"></script>
<link href="libs/vega-tooltip-1/vega-tooltip.css" rel="stylesheet" />
<script src="libs/vega-tooltip-1/vega-tooltip.js"></script>
<script src="libs/FileSaver-1/FileSaver.js"></script>
<script src="libs/GlimmaV2-1/XYSpecs.js"></script>
<script src="libs/GlimmaV2-1/MDSSpecs.js"></script>
<script src="libs/GlimmaV2-1/expressionSpec.js"></script>
<script src="libs/GlimmaV2-1/makeTooltip.js"></script>
<script src="libs/GlimmaV2-1/saveTools.js"></script>
<script src="libs/glimmaXY-binding-2.10.0/glimmaXY.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i><b>1.1</b> Usage</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#render-book"><i class="fa fa-check"></i><b>1.2</b> Render book</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#preview-book"><i class="fa fa-check"></i><b>1.3</b> Preview book</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html"><i class="fa fa-check"></i><b>2</b> Getting Started in R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exercise-description"><i class="fa fa-check"></i><b>2.1</b> Exercise Description</a></li>
<li class="chapter" data-level="2.2" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#learning-outcomes"><i class="fa fa-check"></i><b>2.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="2.3" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#using-r-and-rstudio"><i class="fa fa-check"></i><b>2.3</b> Using R and RStudio</a></li>
<li class="chapter" data-level="2.4" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#load-data-directly-from-the-url"><i class="fa fa-check"></i><b>2.4</b> Load data directly from the URL</a></li>
<li class="chapter" data-level="2.5" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#working-with-data-in-r"><i class="fa fa-check"></i><b>2.5</b> Working with data in R</a></li>
<li class="chapter" data-level="2.6" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#looking-at-data-in-rstudio"><i class="fa fa-check"></i><b>2.6</b> Looking at Data in RStudio</a></li>
<li class="chapter" data-level="2.7" data-path="getting-started-in-r.html"><a href="getting-started-in-r.html#exploring-the-data"><i class="fa fa-check"></i><b>2.7</b> Exploring the data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-ontology.html"><a href="gene-ontology.html"><i class="fa fa-check"></i><b>3</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gene-ontology.html"><a href="gene-ontology.html#description"><i class="fa fa-check"></i><b>3.1</b> Description</a></li>
<li class="chapter" data-level="3.2" data-path="gene-ontology.html"><a href="gene-ontology.html#learning-outcomes-1"><i class="fa fa-check"></i><b>3.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="3.3" data-path="gene-ontology.html"><a href="gene-ontology.html#analysis-workflow"><i class="fa fa-check"></i><b>3.3</b> Analysis Workflow</a></li>
<li class="chapter" data-level="3.4" data-path="gene-ontology.html"><a href="gene-ontology.html#get-de-gene-list"><i class="fa fa-check"></i><b>3.4</b> Get DE gene list</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="gene-ontology.html"><a href="gene-ontology.html#saving-ggplot-output-to-a-file"><i class="fa fa-check"></i><b>3.4.1</b> Saving ggplot output to a file</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="gene-ontology.html"><a href="gene-ontology.html#the-hypergeometric-distribution-in-practice"><i class="fa fa-check"></i><b>3.5</b> The Hypergeometric Distribution in practice</a></li>
<li class="chapter" data-level="3.6" data-path="gene-ontology.html"><a href="gene-ontology.html#now-it-is-your-turn"><i class="fa fa-check"></i><b>3.6</b> Now it is your turn</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="gene-ontology.html"><a href="gene-ontology.html#run-enrichment"><i class="fa fa-check"></i><b>3.6.1</b> Run Enrichment</a></li>
<li class="chapter" data-level="3.6.2" data-path="gene-ontology.html"><a href="gene-ontology.html#see-the-data"><i class="fa fa-check"></i><b>3.6.2</b> see the data</a></li>
<li class="chapter" data-level="3.6.3" data-path="gene-ontology.html"><a href="gene-ontology.html#create-plots"><i class="fa fa-check"></i><b>3.6.3</b> create plots</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="gene-ontology.html"><a href="gene-ontology.html#questions"><i class="fa fa-check"></i><b>3.7</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html"><i class="fa fa-check"></i><b>4</b> Working with Sequences: Raw Data &amp; Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#description-1"><i class="fa fa-check"></i><b>4.1</b> Description</a></li>
<li class="chapter" data-level="4.2" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#learning-outcomes-2"><i class="fa fa-check"></i><b>4.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="4.3" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#download-fastq"><i class="fa fa-check"></i><b>4.3</b> Download fastq</a></li>
<li class="chapter" data-level="4.4" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#examining-fastq"><i class="fa fa-check"></i><b>4.4</b> Examining fastq</a></li>
<li class="chapter" data-level="4.5" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#trimming"><i class="fa fa-check"></i><b>4.5</b> Trimming</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#automate-for-list-of-files"><i class="fa fa-check"></i><b>4.5.1</b> Automate for list of files</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#batch-file-processing"><i class="fa fa-check"></i><b>4.6</b> Batch file processing</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#running-rfastp-creates-several-files"><i class="fa fa-check"></i><b>4.6.1</b> Running RfastP creates several files:</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#qc-and-adapters"><i class="fa fa-check"></i><b>4.7</b> QC and adapters</a></li>
<li class="chapter" data-level="4.8" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#running-fastqc"><i class="fa fa-check"></i><b>4.8</b> Running fastqc</a></li>
<li class="chapter" data-level="4.9" data-path="working-with-sequences-raw-data-quality-control.html"><a href="working-with-sequences-raw-data-quality-control.html#multiqc-for-qc-on-mutliple-samples"><i class="fa fa-check"></i><b>4.9</b> Multiqc for QC on mutliple samples</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="read-mapping.html"><a href="read-mapping.html"><i class="fa fa-check"></i><b>5</b> Read Mapping</a>
<ul>
<li class="chapter" data-level="5.1" data-path="read-mapping.html"><a href="read-mapping.html#alignment"><i class="fa fa-check"></i><b>5.1</b> Alignment</a></li>
<li class="chapter" data-level="5.2" data-path="read-mapping.html"><a href="read-mapping.html#retrieve-the-genome"><i class="fa fa-check"></i><b>5.2</b> Retrieve the genome</a></li>
<li class="chapter" data-level="5.3" data-path="read-mapping.html"><a href="read-mapping.html#build-rsubread-index"><i class="fa fa-check"></i><b>5.3</b> Build Rsubread Index</a></li>
<li class="chapter" data-level="5.4" data-path="read-mapping.html"><a href="read-mapping.html#pseudomapping-with-salmon"><i class="fa fa-check"></i><b>5.4</b> Pseudomapping with Salmon</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="read-mapping.html"><a href="read-mapping.html#create-conda-env"><i class="fa fa-check"></i><b>5.4.1</b> Create Conda Env</a></li>
<li class="chapter" data-level="5.4.2" data-path="read-mapping.html"><a href="read-mapping.html#download-transcriptome"><i class="fa fa-check"></i><b>5.4.2</b> Download transcriptome</a></li>
<li class="chapter" data-level="5.4.3" data-path="read-mapping.html"><a href="read-mapping.html#building-the-salmon-index"><i class="fa fa-check"></i><b>5.4.3</b> Building the Salmon index</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="read-mapping.html"><a href="read-mapping.html#questions-1"><i class="fa fa-check"></i><b>5.5</b> Questions</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="read-mapping.html"><a href="read-mapping.html#with-rsubread"><i class="fa fa-check"></i><b>5.5.1</b> With Rsubread:</a></li>
<li class="chapter" data-level="5.5.2" data-path="read-mapping.html"><a href="read-mapping.html#with-salmon"><i class="fa fa-check"></i><b>5.5.2</b> With Salmon:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="read-counting.html"><a href="read-counting.html"><i class="fa fa-check"></i><b>6</b> Read Counting</a>
<ul>
<li class="chapter" data-level="6.1" data-path="read-counting.html"><a href="read-counting.html#featurecounts"><i class="fa fa-check"></i><b>6.1</b> featureCounts</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="read-counting.html"><a href="read-counting.html#locate-bam-files"><i class="fa fa-check"></i><b>6.1.1</b> Locate BAM files</a></li>
<li class="chapter" data-level="6.1.2" data-path="read-counting.html"><a href="read-counting.html#retrieve-the-genome-annotation"><i class="fa fa-check"></i><b>6.1.2</b> Retrieve the genome annotation</a></li>
<li class="chapter" data-level="6.1.3" data-path="read-counting.html"><a href="read-counting.html#counting-with-featurecounts"><i class="fa fa-check"></i><b>6.1.3</b> Counting with FeatureCounts</a></li>
<li class="chapter" data-level="6.1.4" data-path="read-counting.html"><a href="read-counting.html#counts-object"><i class="fa fa-check"></i><b>6.1.4</b> Counts object</a></li>
<li class="chapter" data-level="6.1.5" data-path="read-counting.html"><a href="read-counting.html#saving-fc-object-for-future-use"><i class="fa fa-check"></i><b>6.1.5</b> Saving <code>fc</code> object for future use</a></li>
<li class="chapter" data-level="6.1.6" data-path="read-counting.html"><a href="read-counting.html#rsubread-qc"><i class="fa fa-check"></i><b>6.1.6</b> RSubread QC</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="read-counting.html"><a href="read-counting.html#salmon"><i class="fa fa-check"></i><b>6.2</b> Salmon</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="read-counting.html"><a href="read-counting.html#pseudomapping-counting"><i class="fa fa-check"></i><b>6.2.1</b> Pseudomapping &amp; Counting</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="read-counting.html"><a href="read-counting.html#questions-2"><i class="fa fa-check"></i><b>6.3</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html"><i class="fa fa-check"></i><b>7</b> Differential Expression: EdgeR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#description-2"><i class="fa fa-check"></i><b>7.1</b> Description</a></li>
<li class="chapter" data-level="7.2" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#learning-outcomes-3"><i class="fa fa-check"></i><b>7.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="7.3" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#loading-in-the-featurecounts-object"><i class="fa fa-check"></i><b>7.3</b> Loading in the featureCounts object</a></li>
<li class="chapter" data-level="7.4" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#count-loading-and-annotation"><i class="fa fa-check"></i><b>7.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="7.5" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#filtering-to-remove-low-counts"><i class="fa fa-check"></i><b>7.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="7.6" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#normalization-for-composition-bias"><i class="fa fa-check"></i><b>7.6</b> Normalization for composition bias</a></li>
<li class="chapter" data-level="7.7" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#mds-plots"><i class="fa fa-check"></i><b>7.7</b> MDS plots</a></li>
<li class="chapter" data-level="7.8" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#exploring-differences-between-libraries"><i class="fa fa-check"></i><b>7.8</b> Exploring differences between libraries</a></li>
<li class="chapter" data-level="7.9" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#estimate-dispersion"><i class="fa fa-check"></i><b>7.9</b> Estimate Dispersion</a></li>
<li class="chapter" data-level="7.10" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#testing-for-differential-expression"><i class="fa fa-check"></i><b>7.10</b> Testing for differential expression</a></li>
<li class="chapter" data-level="7.11" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#looking-at-all-contrasts-at-once"><i class="fa fa-check"></i><b>7.11</b> Looking at all contrasts at once</a></li>
<li class="chapter" data-level="7.12" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#questions-3"><i class="fa fa-check"></i><b>7.12</b> Questions</a></li>
<li class="chapter" data-level="7.13" data-path="differential-expression-edger.html"><a href="differential-expression-edger.html#a-template-set-of-code-chunks-for-doing-this-is-below"><i class="fa fa-check"></i><b>7.13</b> A template set of code chunks for doing this is below:</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html"><i class="fa fa-check"></i><b>8</b> Differential Expression: DESeq2</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#description-3"><i class="fa fa-check"></i><b>8.1</b> Description</a></li>
<li class="chapter" data-level="8.2" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#learning-outcomes-4"><i class="fa fa-check"></i><b>8.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="8.3" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#loading-in-the-featurecounts-object-1"><i class="fa fa-check"></i><b>8.3</b> Loading in the featureCounts object</a></li>
<li class="chapter" data-level="8.4" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#count-loading-and-annotation-1"><i class="fa fa-check"></i><b>8.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="8.5" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#filtering-to-remove-low-counts-1"><i class="fa fa-check"></i><b>8.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="8.6" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#testing-for-differential-expression-1"><i class="fa fa-check"></i><b>8.6</b> Testing for differential expression</a></li>
<li class="chapter" data-level="8.7" data-path="differential-expression-deseq2.html"><a href="differential-expression-deseq2.html#questions-4"><i class="fa fa-check"></i><b>8.7</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html"><i class="fa fa-check"></i><b>9</b> Differential Expression: limma</a>
<ul>
<li class="chapter" data-level="9.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#description-4"><i class="fa fa-check"></i><b>9.1</b> Description</a></li>
<li class="chapter" data-level="9.2" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#learning-objectives"><i class="fa fa-check"></i><b>9.2</b> Learning Objectives</a></li>
<li class="chapter" data-level="9.3" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#loading-in-the-count-data-file"><i class="fa fa-check"></i><b>9.3</b> Loading in the count data file</a></li>
<li class="chapter" data-level="9.4" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#count-loading-and-annotation-2"><i class="fa fa-check"></i><b>9.4</b> Count loading and Annotation</a></li>
<li class="chapter" data-level="9.5" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#filtering-to-remove-low-counts-2"><i class="fa fa-check"></i><b>9.5</b> Filtering to remove low counts</a></li>
<li class="chapter" data-level="9.6" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#normalization-for-composition-bias-1"><i class="fa fa-check"></i><b>9.6</b> Normalization for composition bias</a></li>
<li class="chapter" data-level="9.7" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#exploring-differences-between-libraries-1"><i class="fa fa-check"></i><b>9.7</b> Exploring differences between libraries</a></li>
<li class="chapter" data-level="9.8" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#estimate-dispersion-1"><i class="fa fa-check"></i><b>9.8</b> Estimate Dispersion</a></li>
<li class="chapter" data-level="9.9" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#testing-for-differential-expression-2"><i class="fa fa-check"></i><b>9.9</b> Testing for differential expression</a></li>
<li class="chapter" data-level="9.10" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#examining-a-specific-contrast"><i class="fa fa-check"></i><b>9.10</b> Examining a specific contrast</a></li>
<li class="chapter" data-level="9.11" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#visualization"><i class="fa fa-check"></i><b>9.11</b> Visualization</a>
<ul>
<li class="chapter" data-level="9.11.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#ma-lot"><i class="fa fa-check"></i><b>9.11.1</b> MA lot</a></li>
</ul></li>
<li class="chapter" data-level="9.12" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#treat-testing"><i class="fa fa-check"></i><b>9.12</b> <code>treat()</code> testing</a>
<ul>
<li class="chapter" data-level="9.12.1" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#visualize-de-genes-from-treat-using-lfc1"><i class="fa fa-check"></i><b>9.12.1</b> Visualize DE genes from <code>Treat</code> using lfc=1</a></li>
</ul></li>
<li class="chapter" data-level="9.13" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#comparing-de-analysis-softwares"><i class="fa fa-check"></i><b>9.13</b> Comparing DE analysis softwares</a></li>
<li class="chapter" data-level="9.14" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#correlation-between-logfc-estimates-across-softwares"><i class="fa fa-check"></i><b>9.14</b> Correlation between logFC estimates across softwares</a></li>
<li class="chapter" data-level="9.15" data-path="differential-expression-limma.html"><a href="differential-expression-limma.html#questions-5"><i class="fa fa-check"></i><b>9.15</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html"><i class="fa fa-check"></i><b>10</b> Visualizing Differential Expression Results</a>
<ul>
<li class="chapter" data-level="10.1" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#description-5"><i class="fa fa-check"></i><b>10.1</b> Description</a></li>
<li class="chapter" data-level="10.2" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#learning-outcomes-5"><i class="fa fa-check"></i><b>10.2</b> Learning Outcomes</a></li>
<li class="chapter" data-level="10.3" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#ma-plot"><i class="fa fa-check"></i><b>10.3</b> MA-plot</a></li>
<li class="chapter" data-level="10.4" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#volcano-plot"><i class="fa fa-check"></i><b>10.4</b> Volcano Plot</a></li>
<li class="chapter" data-level="10.5" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#using-glimma-for-an-interactive-visualization"><i class="fa fa-check"></i><b>10.5</b> Using Glimma for an interactive visualization</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#ma-plots"><i class="fa fa-check"></i><b>10.5.1</b> MA plots</a></li>
<li class="chapter" data-level="10.5.2" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#volcano-plots"><i class="fa fa-check"></i><b>10.5.2</b> Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#generating-bar-graph-summaries"><i class="fa fa-check"></i><b>10.6</b> Generating bar graph summaries</a></li>
<li class="chapter" data-level="10.7" data-path="visualizing-differential-expression-results.html"><a href="visualizing-differential-expression-results.html#exercise"><i class="fa fa-check"></i><b>10.7</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>11</b> Clustering</a>
<ul>
<li class="chapter" data-level="11.1" data-path="clustering.html"><a href="clustering.html#description-6"><i class="fa fa-check"></i><b>11.1</b> Description</a></li>
<li class="chapter" data-level="11.2" data-path="clustering.html"><a href="clustering.html#learning-outcomes-6"><i class="fa fa-check"></i><b>11.2</b> Learning outcomes</a></li>
<li class="chapter" data-level="11.3" data-path="clustering.html"><a href="clustering.html#cluster-3.0"><i class="fa fa-check"></i><b>11.3</b> Cluster 3.0</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="clustering.html"><a href="clustering.html#generating-a-pcl-file"><i class="fa fa-check"></i><b>11.3.1</b> Generating a PCL file</a></li>
<li class="chapter" data-level="11.3.2" data-path="clustering.html"><a href="clustering.html#filtering-data"><i class="fa fa-check"></i><b>11.3.2</b> Filtering data</a></li>
<li class="chapter" data-level="11.3.3" data-path="clustering.html"><a href="clustering.html#hierarchical-clustering"><i class="fa fa-check"></i><b>11.3.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="11.3.4" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>11.3.4</b> <em>K</em>-Means Clustering</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="clustering.html"><a href="clustering.html#visualizing-clusters-with-java-treeview"><i class="fa fa-check"></i><b>11.4</b> Visualizing Clusters with Java TreeView</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="clustering.html"><a href="clustering.html#changing-pixel-settings."><i class="fa fa-check"></i><b>11.4.1</b> Changing pixel settings.</a></li>
<li class="chapter" data-level="11.4.2" data-path="clustering.html"><a href="clustering.html#selecting-groups-of-genes."><i class="fa fa-check"></i><b>11.4.2</b> Selecting groups of genes.</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="clustering.html"><a href="clustering.html#performing-clustering-on-yeast-stress-data"><i class="fa fa-check"></i><b>11.5</b> Performing clustering on yeast stress data</a></li>
<li class="chapter" data-level="11.6" data-path="clustering.html"><a href="clustering.html#questions-6"><i class="fa fa-check"></i><b>11.6</b> Questions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Genomic Data Analysis Course Exercises</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="working-with-sequences-raw-data-quality-control" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Working with Sequences: Raw Data &amp; Quality Control<a href="working-with-sequences-raw-data-quality-control.html#working-with-sequences-raw-data-quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>last updated: 2023-10-26</p>
<p><strong>Package Install</strong></p>
<p>As usual, make sure we have the right packages for this exercise</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="working-with-sequences-raw-data-quality-control.html#cb48-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;pacman&quot;</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;pacman&quot;</span>); <span class="fu">library</span>(pacman)</span>
<span id="cb48-2"><a href="working-with-sequences-raw-data-quality-control.html#cb48-2" tabindex="-1"></a></span>
<span id="cb48-3"><a href="working-with-sequences-raw-data-quality-control.html#cb48-3" tabindex="-1"></a><span class="co"># let&#39;s load all of the files we were using and want to have again today</span></span>
<span id="cb48-4"><a href="working-with-sequences-raw-data-quality-control.html#cb48-4" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;knitr&quot;</span>, <span class="st">&quot;readr&quot;</span>,</span>
<span id="cb48-5"><a href="working-with-sequences-raw-data-quality-control.html#cb48-5" tabindex="-1"></a>       <span class="st">&quot;pander&quot;</span>, <span class="st">&quot;BiocManager&quot;</span>, </span>
<span id="cb48-6"><a href="working-with-sequences-raw-data-quality-control.html#cb48-6" tabindex="-1"></a>       <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;stringr&quot;</span>)</span>
<span id="cb48-7"><a href="working-with-sequences-raw-data-quality-control.html#cb48-7" tabindex="-1"></a></span>
<span id="cb48-8"><a href="working-with-sequences-raw-data-quality-control.html#cb48-8" tabindex="-1"></a><span class="co"># We also need the bioconductor packages &quot;ShortRead&quot; and &quot;rfastp&quot; for today&#39;s activity.</span></span>
<span id="cb48-9"><a href="working-with-sequences-raw-data-quality-control.html#cb48-9" tabindex="-1"></a><span class="fu">p_load</span>(<span class="st">&quot;Rfastp&quot;</span>, <span class="st">&quot;ShortRead&quot;</span>)</span></code></pre></div>
<div id="description-1" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Description<a href="working-with-sequences-raw-data-quality-control.html#description-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This activity is intended to familiarize you with raw bioinformatic sequence files. Specifically, we’ll be working with short read sequencing data generated from an Illumina platform.</p>
</div>
<div id="learning-outcomes-2" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Learning outcomes<a href="working-with-sequences-raw-data-quality-control.html#learning-outcomes-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>At the end of this exercise, you should be able to:</p>
<ul>
<li>Load and read into R a raw gzipped fastq file.</li>
<li>Inspect sequence quality and evaluate results.</li>
<li>Perform quality control on raw data and save the processed output.</li>
</ul>
<p>Note that instead of <code>{r}</code>, the below chunk uses <code>{bash}</code>, meaning this isn’t r code but bash code (the language used in the terminal). The <code>-nc</code> flag ensures the files are only downloaded if they don’t already exist where you are downloading them.</p>
<p>This may take awhile the first time you run it. The below script is a bash command that downloads these files to your computer</p>
</div>
<div id="download-fastq" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Download fastq<a href="working-with-sequences-raw-data-quality-control.html#download-fastq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb49-1"><a href="working-with-sequences-raw-data-quality-control.html#cb49-1" tabindex="-1"></a><span class="co"># Be sure to change this file path to the path you want your data to go</span></span>
<span id="cb49-2"><a href="working-with-sequences-raw-data-quality-control.html#cb49-2" tabindex="-1"></a><span class="va">RAW_DATA_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Data/Raw&quot;</span></span>
<span id="cb49-3"><a href="working-with-sequences-raw-data-quality-control.html#cb49-3" tabindex="-1"></a></span>
<span id="cb49-4"><a href="working-with-sequences-raw-data-quality-control.html#cb49-4" tabindex="-1"></a><span class="co">#if you&#39;re using Windows 10,</span></span>
<span id="cb49-5"><a href="working-with-sequences-raw-data-quality-control.html#cb49-5" tabindex="-1"></a><span class="co"># in RStudio, go to Tools&gt;Global Options... &gt; Terminal &gt; New Terminals open with...</span></span>
<span id="cb49-6"><a href="working-with-sequences-raw-data-quality-control.html#cb49-6" tabindex="-1"></a><span class="co"># and choose WSL bash or git bash</span></span>
<span id="cb49-7"><a href="working-with-sequences-raw-data-quality-control.html#cb49-7" tabindex="-1"></a><span class="co"># next, use: (be sure to put in the correct username)</span></span>
<span id="cb49-8"><a href="working-with-sequences-raw-data-quality-control.html#cb49-8" tabindex="-1"></a><span class="co">#RAW_DATA_DIR=&quot;/mnt/c/Users/$USER/Desktop/Genomic_Data_Analysis/Data/Raw&quot;</span></span>
<span id="cb49-9"><a href="working-with-sequences-raw-data-quality-control.html#cb49-9" tabindex="-1"></a></span>
<span id="cb49-10"><a href="working-with-sequences-raw-data-quality-control.html#cb49-10" tabindex="-1"></a><span class="co"># create the destination directory if it doesn&#39;t already exist</span></span>
<span id="cb49-11"><a href="working-with-sequences-raw-data-quality-control.html#cb49-11" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$RAW_DATA_DIR</span></span>
<span id="cb49-12"><a href="working-with-sequences-raw-data-quality-control.html#cb49-12" tabindex="-1"></a></span>
<span id="cb49-13"><a href="working-with-sequences-raw-data-quality-control.html#cb49-13" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$RAW_DATA_DIR</span></span>
<span id="cb49-14"><a href="working-with-sequences-raw-data-quality-control.html#cb49-14" tabindex="-1"></a></span>
<span id="cb49-15"><a href="working-with-sequences-raw-data-quality-control.html#cb49-15" tabindex="-1"></a><span class="co"># change to that directory (for this code chunk only)</span></span>
<span id="cb49-16"><a href="working-with-sequences-raw-data-quality-control.html#cb49-16" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$RAW_DATA_DIR</span></span>
<span id="cb49-17"><a href="working-with-sequences-raw-data-quality-control.html#cb49-17" tabindex="-1"></a><span class="bu">pwd</span></span>
<span id="cb49-18"><a href="working-with-sequences-raw-data-quality-control.html#cb49-18" tabindex="-1"></a><span class="co"># Download the files.</span></span>
<span id="cb49-19"><a href="working-with-sequences-raw-data-quality-control.html#cb49-19" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: curl doesn&#39;t work with relative paths</span></span>
<span id="cb49-20"><a href="working-with-sequences-raw-data-quality-control.html#cb49-20" tabindex="-1"></a><span class="co"># WT unstressed (mock)</span></span>
<span id="cb49-21"><a href="working-with-sequences-raw-data-quality-control.html#cb49-21" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_MOCK_REP1.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-22"><a href="working-with-sequences-raw-data-quality-control.html#cb49-22" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_MOCK_REP2.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-23"><a href="working-with-sequences-raw-data-quality-control.html#cb49-23" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_MOCK_REP3.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-24"><a href="working-with-sequences-raw-data-quality-control.html#cb49-24" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_MOCK_REP4.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-25"><a href="working-with-sequences-raw-data-quality-control.html#cb49-25" tabindex="-1"></a><span class="co"># WT EtOH</span></span>
<span id="cb49-26"><a href="working-with-sequences-raw-data-quality-control.html#cb49-26" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_ETOH_REP1.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-27"><a href="working-with-sequences-raw-data-quality-control.html#cb49-27" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_ETOH_REP2.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-28"><a href="working-with-sequences-raw-data-quality-control.html#cb49-28" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_ETOH_REP3.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-29"><a href="working-with-sequences-raw-data-quality-control.html#cb49-29" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_WT_ETOH_REP4.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-30"><a href="working-with-sequences-raw-data-quality-control.html#cb49-30" tabindex="-1"></a><span class="co"># msn2/4dd unstressed (mock)</span></span>
<span id="cb49-31"><a href="working-with-sequences-raw-data-quality-control.html#cb49-31" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_MOCK_REP1.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-32"><a href="working-with-sequences-raw-data-quality-control.html#cb49-32" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_MOCK_REP2.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-33"><a href="working-with-sequences-raw-data-quality-control.html#cb49-33" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_MOCK_REP3.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-34"><a href="working-with-sequences-raw-data-quality-control.html#cb49-34" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_MOCK_REP4.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-35"><a href="working-with-sequences-raw-data-quality-control.html#cb49-35" tabindex="-1"></a><span class="co"># msn2/4dd EtOH</span></span>
<span id="cb49-36"><a href="working-with-sequences-raw-data-quality-control.html#cb49-36" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_ETOH_REP1.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-37"><a href="working-with-sequences-raw-data-quality-control.html#cb49-37" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_ETOH_REP2.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-38"><a href="working-with-sequences-raw-data-quality-control.html#cb49-38" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_ETOH_REP3.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-39"><a href="working-with-sequences-raw-data-quality-control.html#cb49-39" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-L</span> <span class="at">-C</span> <span class="at">-</span> <span class="at">-O</span> https://github.com/clstacy/GenomicDataAnalysis_Fa23/raw/main/data/ethanol_stress/fastq/YPS606_MSN24_ETOH_REP4.fastq.gz<span class="pp">?</span>raw=TRUE</span>
<span id="cb49-40"><a href="working-with-sequences-raw-data-quality-control.html#cb49-40" tabindex="-1"></a></span>
<span id="cb49-41"><a href="working-with-sequences-raw-data-quality-control.html#cb49-41" tabindex="-1"></a><span class="co"># These are subsamples of raw fastq files from a current project in our lab.</span></span>
<span id="cb49-42"><a href="working-with-sequences-raw-data-quality-control.html#cb49-42" tabindex="-1"></a></span>
<span id="cb49-43"><a href="working-with-sequences-raw-data-quality-control.html#cb49-43" tabindex="-1"></a><span class="co"># Make sure names are as desired</span></span>
<span id="cb49-44"><a href="working-with-sequences-raw-data-quality-control.html#cb49-44" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$RAW_DATA_DIR</span></span>
<span id="cb49-45"><a href="working-with-sequences-raw-data-quality-control.html#cb49-45" tabindex="-1"></a></span>
<span id="cb49-46"><a href="working-with-sequences-raw-data-quality-control.html#cb49-46" tabindex="-1"></a><span class="co"># This loops through and removes the suffix file for any OS that doesn&#39;t auto do so.</span></span>
<span id="cb49-47"><a href="working-with-sequences-raw-data-quality-control.html#cb49-47" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb49-48"><a href="working-with-sequences-raw-data-quality-control.html#cb49-48" tabindex="-1"></a>    <span class="va">newname</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/\?raw=TRUE//&#39;</span><span class="va">)</span></span>
<span id="cb49-49"><a href="working-with-sequences-raw-data-quality-control.html#cb49-49" tabindex="-1"></a>    <span class="fu">mv</span> <span class="st">&quot;</span><span class="va">$file</span><span class="st">&quot;</span> <span class="st">&quot;</span><span class="va">$newname</span><span class="st">&quot;</span></span>
<span id="cb49-50"><a href="working-with-sequences-raw-data-quality-control.html#cb49-50" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb49-51"><a href="working-with-sequences-raw-data-quality-control.html#cb49-51" tabindex="-1"></a></span>
<span id="cb49-52"><a href="working-with-sequences-raw-data-quality-control.html#cb49-52" tabindex="-1"></a><span class="co"># Let&#39;s see what one of these files contains:</span></span>
<span id="cb49-53"><a href="working-with-sequences-raw-data-quality-control.html#cb49-53" tabindex="-1"></a><span class="co"># if you&#39;re on windows or linux, delete the g from gzcat below</span></span>
<span id="cb49-54"><a href="working-with-sequences-raw-data-quality-control.html#cb49-54" tabindex="-1"></a><span class="ex">gzcat</span> <span class="va">$RAW_DATA_DIR</span>/YPS606_WT_MOCK_REP1.fastq.gz <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n8</span></span></code></pre></div>
<pre><code>## /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw
## /Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 7257k  100 7257k    0     0  7961k      0 --:--:-- --:--:-- --:--:-- 7961k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6113k  100 6113k    0     0  8316k      0 --:--:-- --:--:-- --:--:-- 8316k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 7352k  100 7352k    0     0  6882k      0  0:00:01  0:00:01 --:--:-- 12.9M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6746k  100 6746k    0     0  9030k      0 --:--:-- --:--:-- --:--:-- 9030k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0 5774k    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 5774k  100 5774k    0     0  7207k      0 --:--:-- --:--:-- --:--:-- 19.7M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0 6438k    0 31831    0     0  45709      0  0:02:24 --:--:--  0:02:24 45709
 56 6438k   56 3609k    0     0  2184k      0  0:00:02  0:00:01  0:00:01 3742k
100 6438k  100 6438k    0     0  3502k      0  0:00:01  0:00:01 --:--:-- 5615k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 6908k  100 6908k    0     0  4237k      0  0:00:01  0:00:01 --:--:-- 8582k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6020k  100 6020k    0     0  8446k      0 --:--:-- --:--:-- --:--:-- 8446k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 5429k  100 5429k    0     0  6232k      0 --:--:-- --:--:-- --:--:-- 6232k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 5525k  100 5525k    0     0  7484k      0 --:--:-- --:--:-- --:--:-- 13.9M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
  7 6866k    7  494k    0     0   873k      0  0:00:07 --:--:--  0:00:07  873k
100 6866k  100 6866k    0     0  8877k      0 --:--:-- --:--:-- --:--:-- 30.0M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
 60 6797k   60 4081k    0     0  5477k      0  0:00:01 --:--:--  0:00:01 5477k
100 6797k  100 6797k    0     0  7781k      0 --:--:-- --:--:-- --:--:-- 20.7M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
 66 7521k   66 4996k    0     0  6212k      0  0:00:01 --:--:--  0:00:01 6212k
100 7521k  100 7521k    0     0  8430k      0 --:--:-- --:--:-- --:--:-- 28.3M
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6936k  100 6936k    0     0  8400k      0 --:--:-- --:--:-- --:--:-- 8400k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6426k  100 6426k    0     0  8192k      0 --:--:-- --:--:-- --:--:-- 8192k
##   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
##                                  Dload  Upload   Total   Spent    Left  Speed
## 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
## 
100 6601k  100 6601k    0     0  8491k      0 --:--:-- --:--:-- --:--:-- 8491k
## @K00242:669:HFYYJBBXY:2:1209:22455:7908 1:N:0:GTCCGCAC
## GCAATGGTTTACACCCCACCGTGAGATTAGTATGCAATTTAGATCCATTA
## +
## AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
## @K00242:669:HFYYJBBXY:2:1219:16254:46926 1:N:0:GTCCGCAC
## GTCTGATTTGTCTAGATTCTTCGCAAATTTCCAGCCTTCAGAGGCTTCGC
## +
## AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ</code></pre>
<p>We have the data downloaded onto our system now, so let’s first take a look at some of these files ourselves</p>
<p>The R package ShortRead allows us to look at and process raw fastq files. It has many more features than we will use today.</p>
</div>
<div id="examining-fastq" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Examining fastq<a href="working-with-sequences-raw-data-quality-control.html#examining-fastq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s take a look at a fastq file</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="working-with-sequences-raw-data-quality-control.html#cb51-1" tabindex="-1"></a><span class="co"># If you&#39;re using windows, put your username below and uncomment this code before continuing</span></span>
<span id="cb51-2"><a href="working-with-sequences-raw-data-quality-control.html#cb51-2" tabindex="-1"></a><span class="cf">if</span>(.Platform<span class="sc">$</span>OS.type <span class="sc">==</span> <span class="st">&quot;windows&quot;</span>) {</span>
<span id="cb51-3"><a href="working-with-sequences-raw-data-quality-control.html#cb51-3" tabindex="-1"></a>  <span class="fu">Sys.setenv</span>(<span class="at">R_USER =</span> <span class="st">&quot;C:/Users/$USERNAME&quot;</span>)</span>
<span id="cb51-4"><a href="working-with-sequences-raw-data-quality-control.html#cb51-4" tabindex="-1"></a>}</span>
<span id="cb51-5"><a href="working-with-sequences-raw-data-quality-control.html#cb51-5" tabindex="-1"></a></span>
<span id="cb51-6"><a href="working-with-sequences-raw-data-quality-control.html#cb51-6" tabindex="-1"></a><span class="co"># change this directory here to where you have the file saved</span></span>
<span id="cb51-7"><a href="working-with-sequences-raw-data-quality-control.html#cb51-7" tabindex="-1"></a>path_fastq_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">path.expand</span>(<span class="st">&quot;~/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_MOCK_REP1.fastq.gz&quot;</span>)</span>
<span id="cb51-8"><a href="working-with-sequences-raw-data-quality-control.html#cb51-8" tabindex="-1"></a></span>
<span id="cb51-9"><a href="working-with-sequences-raw-data-quality-control.html#cb51-9" tabindex="-1"></a></span>
<span id="cb51-10"><a href="working-with-sequences-raw-data-quality-control.html#cb51-10" tabindex="-1"></a></span>
<span id="cb51-11"><a href="working-with-sequences-raw-data-quality-control.html#cb51-11" tabindex="-1"></a>fastq_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">readFastq</span>(path_fastq_WT_MOCK_REP1)</span>
<span id="cb51-12"><a href="working-with-sequences-raw-data-quality-control.html#cb51-12" tabindex="-1"></a></span>
<span id="cb51-13"><a href="working-with-sequences-raw-data-quality-control.html#cb51-13" tabindex="-1"></a><span class="co"># file too big? swap readFastq() for:</span></span>
<span id="cb51-14"><a href="working-with-sequences-raw-data-quality-control.html#cb51-14" tabindex="-1"></a>subsampled_fastq_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">yield</span>(<span class="fu">FastqSampler</span>(path_fastq_WT_MOCK_REP1, <span class="at">n=</span><span class="dv">10000</span>)) <span class="co"># where n is the number of reads you want to sample</span></span>
<span id="cb51-15"><a href="working-with-sequences-raw-data-quality-control.html#cb51-15" tabindex="-1"></a><span class="co"># the fastq files we downloaded are smaller than a normal fastq file, </span></span>
<span id="cb51-16"><a href="working-with-sequences-raw-data-quality-control.html#cb51-16" tabindex="-1"></a><span class="co"># because they have been subsampled down from their full size for demonstration.</span></span></code></pre></div>
<p>A few quick ways to examine the fastq data object</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="working-with-sequences-raw-data-quality-control.html#cb52-1" tabindex="-1"></a><span class="co"># Typing the name of the object gives us a simple summary</span></span>
<span id="cb52-2"><a href="working-with-sequences-raw-data-quality-control.html#cb52-2" tabindex="-1"></a>fastq_WT_MOCK_REP1</span></code></pre></div>
<pre><code>## class: ShortReadQ
## length: 223565 reads; width: 50 cycles</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="working-with-sequences-raw-data-quality-control.html#cb54-1" tabindex="-1"></a><span class="co"># the length() function gives us the total number of reads</span></span>
<span id="cb54-2"><a href="working-with-sequences-raw-data-quality-control.html#cb54-2" tabindex="-1"></a><span class="fu">length</span>(fastq_WT_MOCK_REP1)</span></code></pre></div>
<pre><code>## [1] 223565</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="working-with-sequences-raw-data-quality-control.html#cb56-1" tabindex="-1"></a><span class="co"># We can use the width() function to find the size of each read/sequence in fastq</span></span>
<span id="cb56-2"><a href="working-with-sequences-raw-data-quality-control.html#cb56-2" tabindex="-1"></a><span class="fu">width</span>(fastq_WT_MOCK_REP1) <span class="sc">|&gt;</span> <span class="fu">head</span>() <span class="co"># add head() pipe to only print first 10</span></span></code></pre></div>
<pre><code>## [1] 50 50 50 50 50 50</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="working-with-sequences-raw-data-quality-control.html#cb58-1" tabindex="-1"></a><span class="co">#sread() - Retrieve sequence of reads.</span></span>
<span id="cb58-2"><a href="working-with-sequences-raw-data-quality-control.html#cb58-2" tabindex="-1"></a><span class="fu">sread</span>(fastq_WT_MOCK_REP1)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 223565:
##          width seq
##      [1]    50 GCAATGGTTTACACCCCACCGTGAGATTAGTATGCAATTTAGATCCATTA
##      [2]    50 GTCTGATTTGTCTAGATTCTTCGCAAATTTCCAGCCTTCAGAGGCTTCGC
##      [3]    50 CTTGAAGTAAGCTTCATCAGCTTCCAACATACCATCGAACCATGGCAACA
##      [4]    50 GCACCGGCAATCTTGTTACCCATAGCATCGAATTTATCTTCTTCGTCTTC
##      [5]    50 CTGTTACCAACTTGTTGTGACATCTTTCTAGTATAATTTTTAAAGTTCTA
##      ...   ... ...
## [223561]    50 GGTGATCAACTGGATTCATGGCAACACCACGGGTCTTTGGCCAAGAGTTT
## [223562]    50 CTTCCAAGTCGATGTATCTCTTGTGGATTCTCATTTCGTAGGTTTCCCAA
## [223563]    50 CGTTAGCATCAACTTCGAAGATAGCTTCCAAGACTGGTTCACCAGCTGGC
## [223564]    50 GCTTCTTTTCTTGGACCTTTTTCAATCTATAAAATTCTTCTCTGTCCAAC
## [223565]    50 ACCGAATGGAAGATTGGTCACCCTCGGAACCCATGATATCTTCGAATGGG</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="working-with-sequences-raw-data-quality-control.html#cb60-1" tabindex="-1"></a><span class="co">#quality() - Retrieve quality of reads as ASCII scores.</span></span>
<span id="cb60-2"><a href="working-with-sequences-raw-data-quality-control.html#cb60-2" tabindex="-1"></a><span class="fu">quality</span>(fastq_WT_MOCK_REP1)</span></code></pre></div>
<pre><code>## class: FastqQuality
## quality:
## BStringSet object of length 223565:
##          width seq
##      [1]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
##      [2]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
##      [3]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
##      [4]    50 AAFFFJJJJJJJJJJJJJJFJJFJJJJFJJJJJJJJJJJFJFJJJJJJJJ
##      [5]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJ
##      ...   ... ...
## [223561]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJ
## [223562]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
## [223563]    50 AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
## [223564]    50 A&lt;FFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
## [223565]    50 AAFFFJJJJJJJJJJJJJJJJJJJJ-7FJJJJJJJJJJJJJJJJJJJJJJ</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="working-with-sequences-raw-data-quality-control.html#cb62-1" tabindex="-1"></a><span class="co">#id() - Retrieve IDs of reads</span></span>
<span id="cb62-2"><a href="working-with-sequences-raw-data-quality-control.html#cb62-2" tabindex="-1"></a><span class="fu">id</span>(fastq_WT_MOCK_REP1)</span></code></pre></div>
<pre><code>## BStringSet object of length 223565:
##          width seq
##      [1]    53 K00242:669:HFYYJBBXY:2:1209:22455:7908 1:N:0:GTCCGCAC
##      [2]    54 K00242:669:HFYYJBBXY:2:1219:16254:46926 1:N:0:GTCCGCAC
##      [3]    53 K00242:669:HFYYJBBXY:2:2223:29985:7029 1:N:0:GTCCGCAC
##      [4]    52 K00242:669:HFYYJBBXY:2:2213:7770:7750 1:N:0:GTCCGCAC
##      [5]    54 K00242:669:HFYYJBBXY:2:1223:27813:29606 1:N:0:GTCCGCAC
##      ...   ... ...
## [223561]    54 K00242:669:HFYYJBBXY:2:1120:22820:46205 1:N:0:GTCCGCAC
## [223562]    54 K00242:669:HFYYJBBXY:2:2209:17239:49054 1:N:0:GTCCGCAC
## [223563]    53 K00242:669:HFYYJBBXY:2:2206:3244:20656 1:N:0:GTCCGCAC
## [223564]    54 K00242:669:HFYYJBBXY:2:2102:28229:16506 1:N:0:GTCCGCAC
## [223565]    53 K00242:669:HFYYJBBXY:2:2204:17381:6712 1:N:0:GTCCGCAC</code></pre>
<p>The output of <code>sread()</code> is a DNAStringSet object, so we can use all of the commands from the Biostrings library on the output object.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="working-with-sequences-raw-data-quality-control.html#cb64-1" tabindex="-1"></a><span class="co"># first, let&#39;s save the output of sread as an object</span></span>
<span id="cb64-2"><a href="working-with-sequences-raw-data-quality-control.html#cb64-2" tabindex="-1"></a>sequence_of_reads <span class="ot">&lt;-</span> <span class="fu">sread</span>(fastq_WT_MOCK_REP1)</span>
<span id="cb64-3"><a href="working-with-sequences-raw-data-quality-control.html#cb64-3" tabindex="-1"></a></span>
<span id="cb64-4"><a href="working-with-sequences-raw-data-quality-control.html#cb64-4" tabindex="-1"></a><span class="co"># Now, let&#39;s use the biostrings function alphabetFrequency to see</span></span>
<span id="cb64-5"><a href="working-with-sequences-raw-data-quality-control.html#cb64-5" tabindex="-1"></a><span class="co"># the occurrence of nucleotide bases in reads.</span></span>
<span id="cb64-6"><a href="working-with-sequences-raw-data-quality-control.html#cb64-6" tabindex="-1"></a>alph_freq <span class="ot">&lt;-</span> <span class="fu">alphabetFrequency</span>(sequence_of_reads)</span>
<span id="cb64-7"><a href="working-with-sequences-raw-data-quality-control.html#cb64-7" tabindex="-1"></a></span>
<span id="cb64-8"><a href="working-with-sequences-raw-data-quality-control.html#cb64-8" tabindex="-1"></a><span class="co"># subset just the first two reads</span></span>
<span id="cb64-9"><a href="working-with-sequences-raw-data-quality-control.html#cb64-9" tabindex="-1"></a>alph_freq[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,]</span></code></pre></div>
<pre><code>##       A  C  G  T M R W S Y K V H D B N - + .
## [1,] 15 11  9 15 0 0 0 0 0 0 0 0 0 0 0 0 0 0
## [2,]  9 13 10 18 0 0 0 0 0 0 0 0 0 0 0 0 0 0</code></pre>
<p>We see most of the nucleotides are assigned to A, C, G, or T, with one base in each read an N.</p>
<p>A fundamental difference between fasta and fastq files is the Quality scores containined in fastQ.</p>
<p>Quality scores are stored as ASCII characters representing -log10 probability of base being wrong (Larger scores would be associated to more confident base calls).</p>
<p>A comprehensive description of phred quality can be found on the wiki page for FastQ.</p>
<p>To see the fastq encodings, we can run:</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="working-with-sequences-raw-data-quality-control.html#cb66-1" tabindex="-1"></a><span class="fu">encoding</span>(<span class="fu">quality</span>(fastq_WT_MOCK_REP1))</span></code></pre></div>
<pre><code>##  !  &quot;  #  $  %  &amp;  &#39;  (  )  *  +  ,  -  .  /  0  1  2  3  4  5  6  7  8  9  : 
##  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 
##  ;  &lt;  =  &gt;  ?  @  A  B  C  D  E  F  G  H  I  J  K  L  M  N  O  P  Q  R  S  T 
## 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 
##  U  V  W  X  Y  Z  [ \\  ]  ^  _  `  a  b  c  d  e  f  g  h  i  j  k  l  m  n 
## 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 
##  o  p  q  r  s  t  u  v  w  x  y  z  {  |  }  ~ 
## 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93</code></pre>
<p>The ShortRead package has many functions available to allow us to collect useful metrics from our ShortRead object.</p>
<p>One very useful function is the <code>alphabetByCycle()</code> function which provides a quick method to summarise base occurrence of cycles.</p>
<p>Here we apply <code>alphabetByCycle()</code> function to the sequence information and show the occurrence of main 4 bases over first 15 cycles.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="working-with-sequences-raw-data-quality-control.html#cb68-1" tabindex="-1"></a>alph_by_cycle <span class="ot">&lt;-</span> <span class="fu">alphabetByCycle</span>(sequence_of_reads)</span>
<span id="cb68-2"><a href="working-with-sequences-raw-data-quality-control.html#cb68-2" tabindex="-1"></a>alph_by_cycle[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>]</span></code></pre></div>
<pre><code>##         cycle
## alphabet  [,1]  [,2]  [,3]  [,4]  [,5]  [,6]   [,7]  [,8]  [,9] [,10] [,11]
##        A 21976 31733 40716 54511 67035 76250  52744 50858 52179 82996 65197
##        C 80979 59940 67520 56801 42469 38578  34934 49605 44865 33367 47832
##        G 91512 46565 44726 59238 53934 41211  34113 40354 43781 41308 50801
##        T 28656 85309 70603 52948 60117 67525 101774 82748 82740 65894 59735
##         cycle
## alphabet [,12] [,13] [,14] [,15]
##        A 56240 62322 63066 62605
##        C 50963 43951 43562 46231
##        G 45250 44735 43818 43027
##        T 71112 72557 73119 71702</code></pre>
<p>We can use the table function to identify the number of times a sequence appears in our FastQ file’s sequence reads.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="working-with-sequences-raw-data-quality-control.html#cb70-1" tabindex="-1"></a>readOccurence <span class="ot">&lt;-</span> <span class="fu">table</span>(sequence_of_reads)</span>
<span id="cb70-2"><a href="working-with-sequences-raw-data-quality-control.html#cb70-2" tabindex="-1"></a></span>
<span id="cb70-3"><a href="working-with-sequences-raw-data-quality-control.html#cb70-3" tabindex="-1"></a><span class="co"># see the top 3 sequences that appear the highest number of times</span></span>
<span id="cb70-4"><a href="working-with-sequences-raw-data-quality-control.html#cb70-4" tabindex="-1"></a><span class="fu">sort</span>(readOccurence,<span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## sequence_of_reads
## CTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT 
##                                                600 
## CCCCCCCCCTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT 
##                                                496 
## CCCCCCCCTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT 
##                                                392</code></pre>
<p>We can identify duplicated reads (potentially arising from PCR over amplification) by using the <code>srduplicated()</code> function and the ShortReadQ object.</p>
<p>This returns a logical vector identifying which reads’ sequences are duplicates (occur more than once in file). Note that the first time a sequence appears in file is not a duplicate but the second, third, fourth times etc are.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="working-with-sequences-raw-data-quality-control.html#cb72-1" tabindex="-1"></a>duplicates <span class="ot">&lt;-</span> <span class="fu">srduplicated</span>(fastq_WT_MOCK_REP1)</span>
<span id="cb72-2"><a href="working-with-sequences-raw-data-quality-control.html#cb72-2" tabindex="-1"></a>duplicates[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## [1] FALSE FALSE FALSE</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="working-with-sequences-raw-data-quality-control.html#cb74-1" tabindex="-1"></a><span class="co"># we can use table() to get a quick summary of the seq duplication rate</span></span>
<span id="cb74-2"><a href="working-with-sequences-raw-data-quality-control.html#cb74-2" tabindex="-1"></a><span class="fu">table</span>(duplicates)</span></code></pre></div>
<pre><code>## duplicates
##  FALSE   TRUE 
## 140931  82634</code></pre>
<p>The ShortRead package also contains a function to generate a simple quality control report.</p>
<p>The <code>qa()</code> function accepts a FastQ file and returns a FastqQA object.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="working-with-sequences-raw-data-quality-control.html#cb76-1" tabindex="-1"></a>qa_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">qa</span>(path_fastq_WT_MOCK_REP1)</span>
<span id="cb76-2"><a href="working-with-sequences-raw-data-quality-control.html#cb76-2" tabindex="-1"></a>qa_WT_MOCK_REP1</span></code></pre></div>
<pre><code>## class: FastqQA(10)
## QA elements (access with qa[[&quot;elt&quot;]]):
##   readCounts: data.frame(1 3)
##   baseCalls: data.frame(1 5)
##   readQualityScore: data.frame(512 4)
##   baseQuality: data.frame(95 3)
##   alignQuality: data.frame(1 3)
##   frequentSequences: data.frame(50 4)
##   sequenceDistribution: data.frame(76 4)
##   perCycle: list(2)
##     baseCall: data.frame(231 4)
##     quality: data.frame(322 5)
##   perTile: list(2)
##     readCounts: data.frame(0 4)
##     medianReadQualityScore: data.frame(0 4)
##   adapterContamination: data.frame(1 1)</code></pre>
<p>We can then use the report() function to generate a simple report.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="working-with-sequences-raw-data-quality-control.html#cb78-1" tabindex="-1"></a>myReport_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">report</span>(qa_WT_MOCK_REP1)</span>
<span id="cb78-2"><a href="working-with-sequences-raw-data-quality-control.html#cb78-2" tabindex="-1"></a>myReport_WT_MOCK_REP1</span></code></pre></div>
<pre><code>## [1] &quot;/var/folders/y1/f7wyg4vj50dgrg8drv0bn4nc0000gn/T//RtmpbCeGU7/file16b718f864a/index.html&quot;</code></pre>
<p>Finally we can review the report in a browser or use the browseURL function to open it in a browser from R.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="working-with-sequences-raw-data-quality-control.html#cb80-1" tabindex="-1"></a><span class="fu">browseURL</span>(myReport_WT_MOCK_REP1)</span></code></pre></div>
</div>
<div id="trimming" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Trimming<a href="working-with-sequences-raw-data-quality-control.html#trimming" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When we observe low quality at the end of reads we may wish to remove the low quality bases for later alignment to the genome. The <code>trimTails()</code> function trims reads from the 3’, removing bases which fall below a desired quality. The <code>trimTails()</code> function accepts arguments specifying the ShortReadQ object, the minimum number of successive bases required to be below quality cut-off for trimming and the actual cut-off score.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="working-with-sequences-raw-data-quality-control.html#cb81-1" tabindex="-1"></a>trimmed_fastq_WT_MOCK_REP1 <span class="ot">&lt;-</span> <span class="fu">trimTails</span>(fastq_WT_MOCK_REP1, <span class="co"># ShortReadQ object to trim</span></span>
<span id="cb81-2"><a href="working-with-sequences-raw-data-quality-control.html#cb81-2" tabindex="-1"></a>                          <span class="at">k=</span><span class="dv">10</span>, <span class="co"># integer number of failing letters to trigger trim</span></span>
<span id="cb81-3"><a href="working-with-sequences-raw-data-quality-control.html#cb81-3" tabindex="-1"></a>                          <span class="at">a=</span><span class="st">&quot;5&quot;</span>) <span class="co"># character giving letter at or below to &quot;fail&quot;</span></span>
<span id="cb81-4"><a href="working-with-sequences-raw-data-quality-control.html#cb81-4" tabindex="-1"></a>trimmed_fastq_WT_MOCK_REP1</span></code></pre></div>
<pre><code>## class: ShortReadQ
## length: 223565 reads; width: 16..50 cycles</code></pre>
<p>Now we have trimmed our FastQ reads, we can export these reads for further analysis using the writeFastq() function</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="working-with-sequences-raw-data-quality-control.html#cb83-1" tabindex="-1"></a><span class="fu">writeFastq</span>(trimmed_fastq_WT_MOCK_REP1,</span>
<span id="cb83-2"><a href="working-with-sequences-raw-data-quality-control.html#cb83-2" tabindex="-1"></a>           <span class="st">&quot;~/Desktop/Genomic_Data_Analysis/WT_MOCK_REP1_shortread_trimmed.fastq.gz&quot;</span>) <span class="co">#path to save file</span></span></code></pre></div>
<div id="automate-for-list-of-files" class="section level3 hasAnchor" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Automate for list of files<a href="working-with-sequences-raw-data-quality-control.html#automate-for-list-of-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are several utility programs that will provide you with QC and trim your data for you, with less input from you. We like fastp as it does some basic QC and trims your fastq files, and it does it very quickly. To make this available in R, it has been made available in the Bioconductor package Rfastp.</p>
<p>By default, fastp will make a html report to summarize your result. But the Rfastp wrapper allows you to look at some of them in R.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="working-with-sequences-raw-data-quality-control.html#cb84-1" tabindex="-1"></a><span class="co"># create a directory for the output to go into if not already present</span></span>
<span id="cb84-2"><a href="working-with-sequences-raw-data-quality-control.html#cb84-2" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">dirname</span>(<span class="fu">dirname</span>(path_fastq_WT_MOCK_REP1)), <span class="st">&quot;/Trimmed_rfastp&quot;</span>) </span>
<span id="cb84-3"><a href="working-with-sequences-raw-data-quality-control.html#cb84-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(output_dir)) {<span class="fu">dir.create</span>(output_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>)}</span>
<span id="cb84-4"><a href="working-with-sequences-raw-data-quality-control.html#cb84-4" tabindex="-1"></a></span>
<span id="cb84-5"><a href="working-with-sequences-raw-data-quality-control.html#cb84-5" tabindex="-1"></a><span class="co"># if we wanted to just run a single file, we would do so like this:</span></span>
<span id="cb84-6"><a href="working-with-sequences-raw-data-quality-control.html#cb84-6" tabindex="-1"></a>rfastp_report <span class="ot">&lt;-</span> <span class="fu">rfastp</span>(<span class="at">read1 =</span> path_fastq_WT_MOCK_REP1,</span>
<span id="cb84-7"><a href="working-with-sequences-raw-data-quality-control.html#cb84-7" tabindex="-1"></a>                        <span class="at">outputFastq =</span> <span class="fu">paste0</span>(output_dir, <span class="st">&quot;/YPS606_WT_MOCK_REP1&quot;</span>))</span>
<span id="cb84-8"><a href="working-with-sequences-raw-data-quality-control.html#cb84-8" tabindex="-1"></a></span>
<span id="cb84-9"><a href="working-with-sequences-raw-data-quality-control.html#cb84-9" tabindex="-1"></a><span class="co"># print out the qc summary for this sample</span></span>
<span id="cb84-10"><a href="working-with-sequences-raw-data-quality-control.html#cb84-10" tabindex="-1"></a>df_summary <span class="ot">&lt;-</span> <span class="fu">qcSummary</span>(rfastp_report)</span>
<span id="cb84-11"><a href="working-with-sequences-raw-data-quality-control.html#cb84-11" tabindex="-1"></a>df_summary <span class="sc">|&gt;</span> <span class="fu">print.data.frame</span>()</span></code></pre></div>
<pre><code>##                      Before_QC     After_QC
## total_reads       2.235650e+05 2.235390e+05
## total_bases       1.117825e+07 1.117695e+07
## q20_bases         1.110373e+07 1.110280e+07
## q30_bases         1.096271e+07 1.096189e+07
## q20_rate          9.933340e-01 9.933660e-01
## q30_rate          9.807180e-01 9.807590e-01
## read1_mean_length 5.000000e+01 5.000000e+01
## gc_content        4.165340e-01 4.165420e-01</code></pre>
</div>
</div>
<div id="batch-file-processing" class="section level2 hasAnchor" number="4.6">
<h2><span class="header-section-number">4.6</span> Batch file processing<a href="working-with-sequences-raw-data-quality-control.html#batch-file-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>That’s nice, but we rarely just have a single fastq file, and we’d like to look at them all at once. Luckily, we can do that with rfastp.</p>
<p>First, we need to get the locations of all of the files we downloaded earlier</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="working-with-sequences-raw-data-quality-control.html#cb86-1" tabindex="-1"></a><span class="co"># adjust to the path where you assigned in RAW_DATA_DIR if using different than default</span></span>
<span id="cb86-2"><a href="working-with-sequences-raw-data-quality-control.html#cb86-2" tabindex="-1"></a>fq_file_dir <span class="ot">&lt;-</span> <span class="fu">dirname</span>(path_fastq_WT_MOCK_REP1) <span class="co"># this just gets the path file is in.</span></span>
<span id="cb86-3"><a href="working-with-sequences-raw-data-quality-control.html#cb86-3" tabindex="-1"></a><span class="co"># crate a list of all of the files</span></span>
<span id="cb86-4"><a href="working-with-sequences-raw-data-quality-control.html#cb86-4" tabindex="-1"></a>fastq.files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> fq_file_dir, <span class="co"># where to look</span></span>
<span id="cb86-5"><a href="working-with-sequences-raw-data-quality-control.html#cb86-5" tabindex="-1"></a>                          <span class="at">pattern =</span> <span class="st">&quot;REP[0-9].fastq.gz$&quot;</span>, <span class="co"># the pattern of file name to find</span></span>
<span id="cb86-6"><a href="working-with-sequences-raw-data-quality-control.html#cb86-6" tabindex="-1"></a>                                  <span class="co"># Note, if you have other fastq files in the folder, they will also be included.</span></span>
<span id="cb86-7"><a href="working-with-sequences-raw-data-quality-control.html#cb86-7" tabindex="-1"></a>                          <span class="at">full.names =</span> <span class="cn">TRUE</span>) <span class="co"># save the full path to the file</span></span>
<span id="cb86-8"><a href="working-with-sequences-raw-data-quality-control.html#cb86-8" tabindex="-1"></a></span>
<span id="cb86-9"><a href="working-with-sequences-raw-data-quality-control.html#cb86-9" tabindex="-1"></a><span class="fu">print</span>(fastq.files)</span></code></pre></div>
<pre><code>##  [1] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_MSN24_ETOH_REP1.fastq.gz&quot;
##  [2] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_MSN24_ETOH_REP2.fastq.gz&quot;
##  [3] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_MSN24_ETOH_REP3.fastq.gz&quot;
##  [4] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_MSN24_ETOH_REP4.fastq.gz&quot;
##  [5] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_MSN24_MOCK_REP1.fastq.gz&quot;
##  [6] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_MSN24_MOCK_REP2.fastq.gz&quot;
##  [7] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_MSN24_MOCK_REP3.fastq.gz&quot;
##  [8] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_MSN24_MOCK_REP4.fastq.gz&quot;
##  [9] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_ETOH_REP1.fastq.gz&quot;   
## [10] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_ETOH_REP2.fastq.gz&quot;   
## [11] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_ETOH_REP3.fastq.gz&quot;   
## [12] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_ETOH_REP4.fastq.gz&quot;   
## [13] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_MOCK_REP1.fastq.gz&quot;   
## [14] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_MOCK_REP2.fastq.gz&quot;   
## [15] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_MOCK_REP3.fastq.gz&quot;   
## [16] &quot;/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Raw/YPS606_WT_MOCK_REP4.fastq.gz&quot;</code></pre>
<p>Now we have all of the file paths</p>
<p>We can loop through all of the files to perform filtering and trimming. Note there are many arguments that can be modified. Use ?rfastp to learn more.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="working-with-sequences-raw-data-quality-control.html#cb88-1" tabindex="-1"></a><span class="co"># run rfastp on all fastq files</span></span>
<span id="cb88-2"><a href="working-with-sequences-raw-data-quality-control.html#cb88-2" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fastq.files)) {</span>
<span id="cb88-3"><a href="working-with-sequences-raw-data-quality-control.html#cb88-3" tabindex="-1"></a>  <span class="co"># file path to single end read</span></span>
<span id="cb88-4"><a href="working-with-sequences-raw-data-quality-control.html#cb88-4" tabindex="-1"></a>  read1 <span class="ot">&lt;-</span> fastq.files[i]</span>
<span id="cb88-5"><a href="working-with-sequences-raw-data-quality-control.html#cb88-5" tabindex="-1"></a>  <span class="co"># assign output file (putting it inside of Data/Trimmed folder)</span></span>
<span id="cb88-6"><a href="working-with-sequences-raw-data-quality-control.html#cb88-6" tabindex="-1"></a>  output_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(output_dir,</span>
<span id="cb88-7"><a href="working-with-sequences-raw-data-quality-control.html#cb88-7" tabindex="-1"></a>                        <span class="st">&quot;/&quot;</span>,</span>
<span id="cb88-8"><a href="working-with-sequences-raw-data-quality-control.html#cb88-8" tabindex="-1"></a>                        <span class="fu">basename</span>(fastq.files[i]))</span>
<span id="cb88-9"><a href="working-with-sequences-raw-data-quality-control.html#cb88-9" tabindex="-1"></a>  json_report <span class="ot">&lt;-</span> <span class="fu">rfastp</span>(</span>
<span id="cb88-10"><a href="working-with-sequences-raw-data-quality-control.html#cb88-10" tabindex="-1"></a>    <span class="at">read1 =</span> read1,</span>
<span id="cb88-11"><a href="working-with-sequences-raw-data-quality-control.html#cb88-11" tabindex="-1"></a>    <span class="at">outputFastq =</span> <span class="fu">str_split</span>(output_name, <span class="fu">fixed</span>(<span class="st">&quot;.&quot;</span>))[[<span class="dv">1</span>]][<span class="dv">1</span>],</span>
<span id="cb88-12"><a href="working-with-sequences-raw-data-quality-control.html#cb88-12" tabindex="-1"></a>    <span class="at">disableTrimPolyG =</span> <span class="cn">FALSE</span>,</span>
<span id="cb88-13"><a href="working-with-sequences-raw-data-quality-control.html#cb88-13" tabindex="-1"></a>    <span class="co"># cutLowQualFront = TRUE,</span></span>
<span id="cb88-14"><a href="working-with-sequences-raw-data-quality-control.html#cb88-14" tabindex="-1"></a>    <span class="co"># cutFrontWindowSize = 3,</span></span>
<span id="cb88-15"><a href="working-with-sequences-raw-data-quality-control.html#cb88-15" tabindex="-1"></a>    <span class="co"># cutFrontMeanQual = 10,</span></span>
<span id="cb88-16"><a href="working-with-sequences-raw-data-quality-control.html#cb88-16" tabindex="-1"></a>    <span class="co"># cutLowQualTail = TRUE,</span></span>
<span id="cb88-17"><a href="working-with-sequences-raw-data-quality-control.html#cb88-17" tabindex="-1"></a>    <span class="at">cutTailWindowSize =</span> <span class="dv">1</span>,</span>
<span id="cb88-18"><a href="working-with-sequences-raw-data-quality-control.html#cb88-18" tabindex="-1"></a>    <span class="co"># cutTailMeanQual = 5,</span></span>
<span id="cb88-19"><a href="working-with-sequences-raw-data-quality-control.html#cb88-19" tabindex="-1"></a>    <span class="at">minReadLength =</span> <span class="dv">15</span>,</span>
<span id="cb88-20"><a href="working-with-sequences-raw-data-quality-control.html#cb88-20" tabindex="-1"></a>    <span class="co"># trimFrontRead1 = 10,</span></span>
<span id="cb88-21"><a href="working-with-sequences-raw-data-quality-control.html#cb88-21" tabindex="-1"></a>    <span class="co"># adapterSequenceRead1 = &#39;GTGTCAGTCACTTCCAGCGG&#39;</span></span>
<span id="cb88-22"><a href="working-with-sequences-raw-data-quality-control.html#cb88-22" tabindex="-1"></a>  )</span>
<span id="cb88-23"><a href="working-with-sequences-raw-data-quality-control.html#cb88-23" tabindex="-1"></a>  </span>
<span id="cb88-24"><a href="working-with-sequences-raw-data-quality-control.html#cb88-24" tabindex="-1"></a>  <span class="co"># Print the output file link in the R Markdown document</span></span>
<span id="cb88-25"><a href="working-with-sequences-raw-data-quality-control.html#cb88-25" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(</span>
<span id="cb88-26"><a href="working-with-sequences-raw-data-quality-control.html#cb88-26" tabindex="-1"></a>    <span class="st">&quot;[Processing Complete - &quot;</span>,</span>
<span id="cb88-27"><a href="working-with-sequences-raw-data-quality-control.html#cb88-27" tabindex="-1"></a>    <span class="fu">basename</span>(output_name),</span>
<span id="cb88-28"><a href="working-with-sequences-raw-data-quality-control.html#cb88-28" tabindex="-1"></a>    <span class="st">&quot;](&quot;</span>,</span>
<span id="cb88-29"><a href="working-with-sequences-raw-data-quality-control.html#cb88-29" tabindex="-1"></a>    output_name,</span>
<span id="cb88-30"><a href="working-with-sequences-raw-data-quality-control.html#cb88-30" tabindex="-1"></a>    <span class="st">&quot;)</span><span class="sc">\n\n</span><span class="st">&quot;</span></span>
<span id="cb88-31"><a href="working-with-sequences-raw-data-quality-control.html#cb88-31" tabindex="-1"></a>  ))</span>
<span id="cb88-32"><a href="working-with-sequences-raw-data-quality-control.html#cb88-32" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## [Processing Complete - YPS606_MSN24_ETOH_REP1.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP1.fastq.gz)
## 
## [Processing Complete - YPS606_MSN24_ETOH_REP2.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP2.fastq.gz)
## 
## [Processing Complete - YPS606_MSN24_ETOH_REP3.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP3.fastq.gz)
## 
## [Processing Complete - YPS606_MSN24_ETOH_REP4.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_ETOH_REP4.fastq.gz)
## 
## [Processing Complete - YPS606_MSN24_MOCK_REP1.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP1.fastq.gz)
## 
## [Processing Complete - YPS606_MSN24_MOCK_REP2.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP2.fastq.gz)
## 
## [Processing Complete - YPS606_MSN24_MOCK_REP3.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP3.fastq.gz)
## 
## [Processing Complete - YPS606_MSN24_MOCK_REP4.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_MSN24_MOCK_REP4.fastq.gz)
## 
## [Processing Complete - YPS606_WT_ETOH_REP1.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP1.fastq.gz)
## 
## [Processing Complete - YPS606_WT_ETOH_REP2.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP2.fastq.gz)
## 
## [Processing Complete - YPS606_WT_ETOH_REP3.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP3.fastq.gz)
## 
## [Processing Complete - YPS606_WT_ETOH_REP4.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_ETOH_REP4.fastq.gz)
## 
## [Processing Complete - YPS606_WT_MOCK_REP1.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP1.fastq.gz)
## 
## [Processing Complete - YPS606_WT_MOCK_REP2.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP2.fastq.gz)
## 
## [Processing Complete - YPS606_WT_MOCK_REP3.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP3.fastq.gz)
## 
## [Processing Complete - YPS606_WT_MOCK_REP4.fastq.gz](/Users/clstacy/Desktop/Genomic_Data_Analysis/Data/Trimmed_rfastp/YPS606_WT_MOCK_REP4.fastq.gz)</code></pre>
<div id="running-rfastp-creates-several-files" class="section level3 hasAnchor" number="4.6.1">
<h3><span class="header-section-number">4.6.1</span> Running RfastP creates several files:<a href="working-with-sequences-raw-data-quality-control.html#running-rfastp-creates-several-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>XXX_R1.fastq.gz - FASTQ with poor quality reads filtered out</p></li>
<li><p>XXX.html - HTML file contains a QC report</p></li>
<li><p>XXX.json - JSON file with all the summary statistics</p></li>
</ol>
</div>
</div>
<div id="qc-and-adapters" class="section level2 hasAnchor" number="4.7">
<h2><span class="header-section-number">4.7</span> QC and adapters<a href="working-with-sequences-raw-data-quality-control.html#qc-and-adapters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Another common tool for quality control is called FastQC, useable via command line or GUI, available at <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a> or via pip or conda install in the command line.</p>
<p>To use this tool, let’s get conda running on your computer. NOTE: Anaconda is already installed on computers in the computer lab. If you are using your own computer, you’ll need to have conda installed (<a href="https://docs.conda.io/en/main/miniconda.html">link to learn more</a>)</p>
<p>First, we need to open a terminal window. We will copy code from the below code chunk into the terminal window.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb90-1"><a href="working-with-sequences-raw-data-quality-control.html#cb90-1" tabindex="-1"></a><span class="bu">.</span> /opt/anaconda3/bin/activate <span class="kw">&amp;&amp;</span> <span class="ex">conda</span> init</span>
<span id="cb90-2"><a href="working-with-sequences-raw-data-quality-control.html#cb90-2" tabindex="-1"></a><span class="co">#. /opt/anaconda3/bin/activate &amp;&amp; conda activate /opt/anaconda3;</span></span>
<span id="cb90-3"><a href="working-with-sequences-raw-data-quality-control.html#cb90-3" tabindex="-1"></a></span>
<span id="cb90-4"><a href="working-with-sequences-raw-data-quality-control.html#cb90-4" tabindex="-1"></a><span class="co"># run this command in terminal to make sure conda is activated</span></span>
<span id="cb90-5"><a href="working-with-sequences-raw-data-quality-control.html#cb90-5" tabindex="-1"></a><span class="fu">which</span> conda</span>
<span id="cb90-6"><a href="working-with-sequences-raw-data-quality-control.html#cb90-6" tabindex="-1"></a></span>
<span id="cb90-7"><a href="working-with-sequences-raw-data-quality-control.html#cb90-7" tabindex="-1"></a><span class="co"># copy these 4 lines into terminal and run them</span></span>
<span id="cb90-8"><a href="working-with-sequences-raw-data-quality-control.html#cb90-8" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels defaults</span>
<span id="cb90-9"><a href="working-with-sequences-raw-data-quality-control.html#cb90-9" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--append</span> channels bioconda</span>
<span id="cb90-10"><a href="working-with-sequences-raw-data-quality-control.html#cb90-10" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--append</span> channels conda-forge</span>
<span id="cb90-11"><a href="working-with-sequences-raw-data-quality-control.html#cb90-11" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--set</span> channel_priority strict</span>
<span id="cb90-12"><a href="working-with-sequences-raw-data-quality-control.html#cb90-12" tabindex="-1"></a></span>
<span id="cb90-13"><a href="working-with-sequences-raw-data-quality-control.html#cb90-13" tabindex="-1"></a><span class="co"># check channel order</span></span>
<span id="cb90-14"><a href="working-with-sequences-raw-data-quality-control.html#cb90-14" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--show</span> channels</span></code></pre></div>
<p>Now, we need to create a conda environment with our packages. You can do so with the code below. This may take a couple of minutes the first time we run it.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb91-1"><a href="working-with-sequences-raw-data-quality-control.html#cb91-1" tabindex="-1"></a><span class="co"># create an enviornment for our QC packages</span></span>
<span id="cb91-2"><a href="working-with-sequences-raw-data-quality-control.html#cb91-2" tabindex="-1"></a><span class="cf">if</span> <span class="ex">conda</span> info <span class="at">--envs</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-q</span> QC<span class="kw">;</span> <span class="cf">then</span> <span class="bu">echo</span> <span class="st">&quot;environment &#39;QC&#39; already exists&quot;</span><span class="kw">;</span> <span class="cf">else</span> <span class="ex">conda</span> create <span class="at">-y</span> <span class="at">-n</span> QC fastqc multiqc<span class="kw">;</span> <span class="cf">fi</span></span>
<span id="cb91-3"><a href="working-with-sequences-raw-data-quality-control.html#cb91-3" tabindex="-1"></a></span>
<span id="cb91-4"><a href="working-with-sequences-raw-data-quality-control.html#cb91-4" tabindex="-1"></a><span class="co"># see available conda environments</span></span>
<span id="cb91-5"><a href="working-with-sequences-raw-data-quality-control.html#cb91-5" tabindex="-1"></a><span class="ex">conda</span> env list</span>
<span id="cb91-6"><a href="working-with-sequences-raw-data-quality-control.html#cb91-6" tabindex="-1"></a></span>
<span id="cb91-7"><a href="working-with-sequences-raw-data-quality-control.html#cb91-7" tabindex="-1"></a><span class="co"># activate our QC environment</span></span>
<span id="cb91-8"><a href="working-with-sequences-raw-data-quality-control.html#cb91-8" tabindex="-1"></a><span class="ex">conda</span> activate QC</span>
<span id="cb91-9"><a href="working-with-sequences-raw-data-quality-control.html#cb91-9" tabindex="-1"></a></span>
<span id="cb91-10"><a href="working-with-sequences-raw-data-quality-control.html#cb91-10" tabindex="-1"></a><span class="co"># make sure desired packages are working</span></span>
<span id="cb91-11"><a href="working-with-sequences-raw-data-quality-control.html#cb91-11" tabindex="-1"></a><span class="fu">which</span> fastqc</span>
<span id="cb91-12"><a href="working-with-sequences-raw-data-quality-control.html#cb91-12" tabindex="-1"></a><span class="fu">which</span> multiqc</span>
<span id="cb91-13"><a href="working-with-sequences-raw-data-quality-control.html#cb91-13" tabindex="-1"></a></span>
<span id="cb91-14"><a href="working-with-sequences-raw-data-quality-control.html#cb91-14" tabindex="-1"></a><span class="co"># get the versions of each software</span></span>
<span id="cb91-15"><a href="working-with-sequences-raw-data-quality-control.html#cb91-15" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-v</span></span>
<span id="cb91-16"><a href="working-with-sequences-raw-data-quality-control.html#cb91-16" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">--version</span></span>
<span id="cb91-17"><a href="working-with-sequences-raw-data-quality-control.html#cb91-17" tabindex="-1"></a></span>
<span id="cb91-18"><a href="working-with-sequences-raw-data-quality-control.html#cb91-18" tabindex="-1"></a><span class="co"># it&#39;s always good coding practice to deactivate a conda environment at the end of a chunk</span></span>
<span id="cb91-19"><a href="working-with-sequences-raw-data-quality-control.html#cb91-19" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
<pre><code>## environment &#39;QC&#39; already exists
## # conda environments:
## #
##                          /Users/clstacy/Library/r-miniconda
##                          /Users/clstacy/Library/r-miniconda-arm64
##                          /Users/clstacy/Library/r-miniconda-arm64/envs/r-reticulate
##                          /Users/clstacy/Library/r-miniconda/envs/r-reticulate
## base                  *  /Users/clstacy/anaconda3
## QC                       /Users/clstacy/anaconda3/envs/QC
## mageck                   /Users/clstacy/anaconda3/envs/mageck
## salmon                   /Users/clstacy/anaconda3/envs/salmon
##                          /Users/clstacy/opt/anaconda3
##                          /Users/clstacy/opt/anaconda3/envs/colony_count_nn
##                          /Users/clstacy/opt/anaconda3/envs/tlcc
## 
## /Users/clstacy/anaconda3/envs/QC/bin/fastqc
## /Users/clstacy/anaconda3/envs/QC/bin/multiqc
## FastQC v0.12.1
## multiqc, version 1.15</code></pre>
</div>
<div id="running-fastqc" class="section level2 hasAnchor" number="4.8">
<h2><span class="header-section-number">4.8</span> Running fastqc<a href="working-with-sequences-raw-data-quality-control.html#running-fastqc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb93"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb93-1"><a href="working-with-sequences-raw-data-quality-control.html#cb93-1" tabindex="-1"></a><span class="co">#WARNING: variables in bash you&#39;ve saved in previous chunks won&#39;t be retained in later chunks</span></span>
<span id="cb93-2"><a href="working-with-sequences-raw-data-quality-control.html#cb93-2" tabindex="-1"></a><span class="co"># We need to set a variable for the folder above raw and trimmed files.</span></span>
<span id="cb93-3"><a href="working-with-sequences-raw-data-quality-control.html#cb93-3" tabindex="-1"></a><span class="va">DATA_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/Data&quot;</span></span>
<span id="cb93-4"><a href="working-with-sequences-raw-data-quality-control.html#cb93-4" tabindex="-1"></a><span class="va">QC_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/QC&quot;</span></span>
<span id="cb93-5"><a href="working-with-sequences-raw-data-quality-control.html#cb93-5" tabindex="-1"></a><span class="co"># Activate conda QC environment</span></span>
<span id="cb93-6"><a href="working-with-sequences-raw-data-quality-control.html#cb93-6" tabindex="-1"></a><span class="ex">conda</span> activate QC</span>
<span id="cb93-7"><a href="working-with-sequences-raw-data-quality-control.html#cb93-7" tabindex="-1"></a></span>
<span id="cb93-8"><a href="working-with-sequences-raw-data-quality-control.html#cb93-8" tabindex="-1"></a><span class="co"># show which version of fastqc is active</span></span>
<span id="cb93-9"><a href="working-with-sequences-raw-data-quality-control.html#cb93-9" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-v</span></span>
<span id="cb93-10"><a href="working-with-sequences-raw-data-quality-control.html#cb93-10" tabindex="-1"></a></span>
<span id="cb93-11"><a href="working-with-sequences-raw-data-quality-control.html#cb93-11" tabindex="-1"></a><span class="co"># Function to check if a command is installed, we use this next.</span></span>
<span id="cb93-12"><a href="working-with-sequences-raw-data-quality-control.html#cb93-12" tabindex="-1"></a><span class="fu">command_exists()</span> <span class="kw">{</span></span>
<span id="cb93-13"><a href="working-with-sequences-raw-data-quality-control.html#cb93-13" tabindex="-1"></a>  <span class="bu">command</span> <span class="at">-v</span> <span class="st">&quot;</span><span class="va">$1</span><span class="st">&quot;</span> <span class="op">&gt;</span>/dev/null <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span></span>
<span id="cb93-14"><a href="working-with-sequences-raw-data-quality-control.html#cb93-14" tabindex="-1"></a><span class="kw">}</span></span>
<span id="cb93-15"><a href="working-with-sequences-raw-data-quality-control.html#cb93-15" tabindex="-1"></a></span>
<span id="cb93-16"><a href="working-with-sequences-raw-data-quality-control.html#cb93-16" tabindex="-1"></a><span class="cf">if</span> <span class="ex">command_exists</span> fastqc<span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb93-17"><a href="working-with-sequences-raw-data-quality-control.html#cb93-17" tabindex="-1"></a>  <span class="co"># Continue if fastqc is installed</span></span>
<span id="cb93-18"><a href="working-with-sequences-raw-data-quality-control.html#cb93-18" tabindex="-1"></a>  <span class="co"># first, make sure we have the folders to store the fastqc outputs</span></span>
<span id="cb93-19"><a href="working-with-sequences-raw-data-quality-control.html#cb93-19" tabindex="-1"></a>  <span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$QC_DIR</span>/fastqc/Raw</span>
<span id="cb93-20"><a href="working-with-sequences-raw-data-quality-control.html#cb93-20" tabindex="-1"></a>  <span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$QC_DIR</span>/fastqc/Trimmed</span>
<span id="cb93-21"><a href="working-with-sequences-raw-data-quality-control.html#cb93-21" tabindex="-1"></a>  <span class="co"># run fastqc on the raw data files</span></span>
<span id="cb93-22"><a href="working-with-sequences-raw-data-quality-control.html#cb93-22" tabindex="-1"></a>  <span class="ex">fastqc</span> <span class="va">$DATA_DIR</span>/Raw/<span class="pp">*</span>.fastq.gz <span class="at">-o</span> <span class="va">$QC_DIR</span>/fastqc/Raw</span>
<span id="cb93-23"><a href="working-with-sequences-raw-data-quality-control.html#cb93-23" tabindex="-1"></a>  <span class="co"># run fastqc on the trimmed data files</span></span>
<span id="cb93-24"><a href="working-with-sequences-raw-data-quality-control.html#cb93-24" tabindex="-1"></a>  <span class="ex">fastqc</span> <span class="va">$DATA_DIR</span>/Trimmed_rfastp/<span class="pp">*</span>.fastq.gz <span class="at">-o</span> <span class="va">$QC_DIR</span>/fastqc/Trimmed</span>
<span id="cb93-25"><a href="working-with-sequences-raw-data-quality-control.html#cb93-25" tabindex="-1"></a>  <span class="cf">if</span> <span class="bu">[</span> <span class="va">$?</span> <span class="ot">-ne</span> 0 <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb93-26"><a href="working-with-sequences-raw-data-quality-control.html#cb93-26" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;FastQC execution failed. It didn&#39;t work.&quot;</span></span>
<span id="cb93-27"><a href="working-with-sequences-raw-data-quality-control.html#cb93-27" tabindex="-1"></a>  <span class="cf">fi</span></span>
<span id="cb93-28"><a href="working-with-sequences-raw-data-quality-control.html#cb93-28" tabindex="-1"></a><span class="cf">else</span></span>
<span id="cb93-29"><a href="working-with-sequences-raw-data-quality-control.html#cb93-29" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">&quot;FastQC is not installed.&quot;</span></span>
<span id="cb93-30"><a href="working-with-sequences-raw-data-quality-control.html#cb93-30" tabindex="-1"></a><span class="cf">fi</span></span>
<span id="cb93-31"><a href="working-with-sequences-raw-data-quality-control.html#cb93-31" tabindex="-1"></a></span>
<span id="cb93-32"><a href="working-with-sequences-raw-data-quality-control.html#cb93-32" tabindex="-1"></a><span class="co"># deactivate QC conda environment</span></span>
<span id="cb93-33"><a href="working-with-sequences-raw-data-quality-control.html#cb93-33" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
<pre><code>## FastQC v0.12.1
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## Started analysis of YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 5% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 10% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 15% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 20% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 25% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 30% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 35% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 40% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 45% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 50% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 55% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 60% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 65% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 70% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 75% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 80% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 85% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 90% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Approx 95% complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Analysis complete for YPS606_MSN24_ETOH_REP1.fastq.gz
## Warning: the fonts &quot;Times&quot; and &quot;Times&quot; are not available for the Java logical font &quot;Serif&quot;, which may have unexpected appearance or behavior. Re-enable the &quot;Times&quot; font to remove this warning.
## Started analysis of YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 5% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 10% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 15% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 20% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 25% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 30% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 35% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 40% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 45% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 50% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 55% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 60% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 65% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 70% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 75% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 80% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 85% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 90% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Approx 95% complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Analysis complete for YPS606_MSN24_ETOH_REP2.fastq.gz
## Started analysis of YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 5% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 10% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 15% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 20% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 25% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 30% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 35% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 40% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 45% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 50% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 55% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 60% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 65% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 70% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 75% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 80% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 85% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 90% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Approx 95% complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Analysis complete for YPS606_MSN24_ETOH_REP3.fastq.gz
## Started analysis of YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 5% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 10% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 15% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 20% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 25% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 30% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 35% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 40% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 45% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 50% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 55% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 60% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 65% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 70% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 75% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 80% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 85% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 90% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Approx 95% complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Analysis complete for YPS606_MSN24_ETOH_REP4.fastq.gz
## Started analysis of YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 5% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 10% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 15% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 20% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 25% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 30% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 35% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 40% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 45% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 50% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 55% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 60% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 65% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 70% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 75% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 80% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 85% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 90% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Approx 95% complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Analysis complete for YPS606_MSN24_MOCK_REP1.fastq.gz
## Started analysis of YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 5% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 10% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 15% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 20% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 25% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 30% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 35% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 40% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 45% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 50% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 55% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 60% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 65% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 70% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 75% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 80% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 85% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 90% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Approx 95% complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Analysis complete for YPS606_MSN24_MOCK_REP2.fastq.gz
## Started analysis of YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 5% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 10% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 15% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 20% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 25% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 30% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 35% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 40% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 45% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 50% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 55% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 60% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 65% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 70% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 75% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 80% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 85% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 90% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 95% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Approx 100% complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Analysis complete for YPS606_MSN24_MOCK_REP3.fastq.gz
## Started analysis of YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 5% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 10% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 15% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 20% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 25% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 30% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 35% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 40% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 45% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 50% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 55% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 60% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 65% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 70% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 75% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 80% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 85% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 90% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Approx 95% complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Analysis complete for YPS606_MSN24_MOCK_REP4.fastq.gz
## Started analysis of YPS606_WT_ETOH_REP1.fastq.gz
## Approx 5% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 10% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 15% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 20% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 25% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 30% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 35% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 40% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 45% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 50% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 55% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 60% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 65% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 70% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 75% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 80% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 85% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 90% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Approx 95% complete for YPS606_WT_ETOH_REP1.fastq.gz
## Analysis complete for YPS606_WT_ETOH_REP1.fastq.gz
## Started analysis of YPS606_WT_ETOH_REP2.fastq.gz
## Approx 5% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 10% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 15% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 20% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 25% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 30% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 35% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 40% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 45% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 50% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 55% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 60% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 65% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 70% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 75% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 80% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 85% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 90% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Approx 95% complete for YPS606_WT_ETOH_REP2.fastq.gz
## Analysis complete for YPS606_WT_ETOH_REP2.fastq.gz
## Started analysis of YPS606_WT_ETOH_REP3.fastq.gz
## Approx 5% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 10% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 15% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 20% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 25% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 30% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 35% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 40% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 45% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 50% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 55% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 60% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 65% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 70% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 75% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 80% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 85% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 90% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Approx 95% complete for YPS606_WT_ETOH_REP3.fastq.gz
## Analysis complete for YPS606_WT_ETOH_REP3.fastq.gz
## Started analysis of YPS606_WT_ETOH_REP4.fastq.gz
## Approx 5% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 10% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 15% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 20% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 25% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 30% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 35% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 40% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 45% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 50% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 55% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 60% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 65% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 70% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 75% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 80% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 85% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 90% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Approx 95% complete for YPS606_WT_ETOH_REP4.fastq.gz
## Analysis complete for YPS606_WT_ETOH_REP4.fastq.gz
## Started analysis of YPS606_WT_MOCK_REP1.fastq.gz
## Approx 5% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 10% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 15% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 20% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 25% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 30% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 35% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 40% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 45% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 50% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 55% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 60% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 65% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 70% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 75% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 80% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 85% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 90% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Approx 95% complete for YPS606_WT_MOCK_REP1.fastq.gz
## Analysis complete for YPS606_WT_MOCK_REP1.fastq.gz
## Started analysis of YPS606_WT_MOCK_REP2.fastq.gz
## Approx 5% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 10% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 15% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 20% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 25% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 30% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 35% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 40% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 45% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 50% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 55% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 60% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 65% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 70% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 75% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 80% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 85% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 90% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Approx 95% complete for YPS606_WT_MOCK_REP2.fastq.gz
## Analysis complete for YPS606_WT_MOCK_REP2.fastq.gz
## Started analysis of YPS606_WT_MOCK_REP3.fastq.gz
## Approx 5% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 10% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 15% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 20% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 25% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 30% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 35% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 40% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 45% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 50% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 55% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 60% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 65% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 70% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 75% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 80% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 85% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 90% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Approx 95% complete for YPS606_WT_MOCK_REP3.fastq.gz
## Analysis complete for YPS606_WT_MOCK_REP3.fastq.gz
## Started analysis of YPS606_WT_MOCK_REP4.fastq.gz
## Approx 5% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 10% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 15% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 20% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 25% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 30% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 35% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 40% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 45% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 50% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 55% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 60% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 65% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 70% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 75% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 80% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 85% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 90% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Approx 95% complete for YPS606_WT_MOCK_REP4.fastq.gz
## Analysis complete for YPS606_WT_MOCK_REP4.fastq.gz
## application/octet-stream
## application/octet-stream
## Started analysis of YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## application/octet-stream
## Approx 5% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 10% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 15% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 20% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 25% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 30% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 35% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 40% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 45% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 50% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 55% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 60% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 65% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 70% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 75% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 80% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 85% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 90% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Approx 95% complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Analysis complete for YPS606_MSN24_ETOH_REP1_R1.fastq.gz
## Warning: the fonts &quot;Times&quot; and &quot;Times&quot; are not available for the Java logical font &quot;Serif&quot;, which may have unexpected appearance or behavior. Re-enable the &quot;Times&quot; font to remove this warning.
## Started analysis of YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 5% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 10% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 15% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 20% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 25% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 30% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 35% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 40% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 45% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 50% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 55% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 60% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 65% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 70% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 75% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 80% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 85% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 90% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Approx 95% complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Analysis complete for YPS606_MSN24_ETOH_REP2_R1.fastq.gz
## Started analysis of YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 5% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 10% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 15% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 20% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 25% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 30% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 35% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 40% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 45% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 50% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 55% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 60% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 65% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 70% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 75% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 80% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 85% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 90% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Approx 95% complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Analysis complete for YPS606_MSN24_ETOH_REP3_R1.fastq.gz
## Started analysis of YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 5% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 10% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 15% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 20% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 25% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 30% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 35% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 40% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 45% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 50% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 55% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 60% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 65% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 70% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 75% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 80% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 85% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 90% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Approx 95% complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Analysis complete for YPS606_MSN24_ETOH_REP4_R1.fastq.gz
## Started analysis of YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 5% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 10% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 15% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 20% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 25% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 30% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 35% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 40% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 45% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 50% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 55% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 60% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 65% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 70% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 75% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 80% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 85% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 90% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Approx 95% complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Analysis complete for YPS606_MSN24_MOCK_REP1_R1.fastq.gz
## Started analysis of YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 5% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 10% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 15% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 20% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 25% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 30% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 35% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 40% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 45% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 50% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 55% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 60% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 65% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 70% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 75% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 80% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 85% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 90% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Approx 95% complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Analysis complete for YPS606_MSN24_MOCK_REP2_R1.fastq.gz
## Started analysis of YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 5% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 10% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 15% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 20% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 25% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 30% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 35% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 40% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 45% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 50% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 55% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 60% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 65% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 70% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 75% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 80% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 85% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 90% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 95% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Approx 100% complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Analysis complete for YPS606_MSN24_MOCK_REP3_R1.fastq.gz
## Started analysis of YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 5% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 10% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 15% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 20% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 25% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 30% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 35% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 40% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 45% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 50% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 55% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 60% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 65% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 70% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 75% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 80% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 85% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 90% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Approx 95% complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Analysis complete for YPS606_MSN24_MOCK_REP4_R1.fastq.gz
## Started analysis of YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 5% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 10% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 15% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 20% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 25% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 30% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 35% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 40% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 45% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 50% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 55% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 60% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 65% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 70% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 75% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 80% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 85% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 90% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Approx 95% complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Analysis complete for YPS606_WT_ETOH_REP1_R1.fastq.gz
## Started analysis of YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 5% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 10% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 15% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 20% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 25% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 30% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 35% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 40% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 45% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 50% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 55% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 60% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 65% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 70% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 75% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 80% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 85% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 90% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Approx 95% complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Analysis complete for YPS606_WT_ETOH_REP2_R1.fastq.gz
## Started analysis of YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 5% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 10% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 15% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 20% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 25% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 30% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 35% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 40% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 45% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 50% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 55% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 60% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 65% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 70% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 75% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 80% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 85% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 90% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Approx 95% complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Analysis complete for YPS606_WT_ETOH_REP3_R1.fastq.gz
## Started analysis of YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 5% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 10% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 15% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 20% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 25% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 30% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 35% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 40% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 45% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 50% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 55% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 60% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 65% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 70% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 75% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 80% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 85% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 90% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Approx 95% complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Analysis complete for YPS606_WT_ETOH_REP4_R1.fastq.gz
## Started analysis of YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 5% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 10% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 15% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 20% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 25% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 30% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 35% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 40% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 45% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 50% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 55% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 60% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 65% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 70% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 75% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 80% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 85% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 90% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Approx 95% complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Analysis complete for YPS606_WT_MOCK_REP1_R1.fastq.gz
## Started analysis of YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 5% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 10% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 15% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 20% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 25% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 30% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 35% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 40% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 45% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 50% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 55% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 60% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 65% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 70% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 75% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 80% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 85% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 90% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Approx 95% complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Analysis complete for YPS606_WT_MOCK_REP2_R1.fastq.gz
## Started analysis of YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 5% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 10% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 15% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 20% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 25% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 30% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 35% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 40% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 45% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 50% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 55% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 60% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 65% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 70% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 75% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 80% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 85% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 90% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Approx 95% complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Analysis complete for YPS606_WT_MOCK_REP3_R1.fastq.gz
## Started analysis of YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 5% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 10% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 15% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 20% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 25% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 30% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 35% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 40% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 45% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 50% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 55% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 60% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 65% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 70% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 75% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 80% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 85% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 90% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Approx 95% complete for YPS606_WT_MOCK_REP4_R1.fastq.gz
## Analysis complete for YPS606_WT_MOCK_REP4_R1.fastq.gz</code></pre>
<p>This link shows the fastqc output for the trimmed WT_MOCK_REP1.fastq.gz</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="working-with-sequences-raw-data-quality-control.html#cb95-1" tabindex="-1"></a><span class="fu">browseURL</span>(<span class="st">&quot;~/Desktop/Genomic_Data_Analysis/QC/fastqc/Trimmed/YPS606_WT_MOCK_REP1_R1_fastqc.html&quot;</span>)</span></code></pre></div>
<p>We could do this for each of the html fastq files to see how they all look but with a large sample size that takes a long time and can lead to missing important information.</p>
</div>
<div id="multiqc-for-qc-on-mutliple-samples" class="section level2 hasAnchor" number="4.9">
<h2><span class="header-section-number">4.9</span> Multiqc for QC on mutliple samples<a href="working-with-sequences-raw-data-quality-control.html#multiqc-for-qc-on-mutliple-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One of our favorite ways to analyze multiple samples simultaneously is <a href="https://multiqc.info/">MultiQC</a> a software that combines fastQC (and other) reports</p>
<p>Here is the code to run it:</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb96-1"><a href="working-with-sequences-raw-data-quality-control.html#cb96-1" tabindex="-1"></a><span class="co"># Be sure to change this file path to the path you want to run multiqc</span></span>
<span id="cb96-2"><a href="working-with-sequences-raw-data-quality-control.html#cb96-2" tabindex="-1"></a><span class="va">QC_DIR</span><span class="op">=</span><span class="st">&quot;/Users/</span><span class="va">$USER</span><span class="st">/Desktop/Genomic_Data_Analysis/QC&quot;</span></span>
<span id="cb96-3"><a href="working-with-sequences-raw-data-quality-control.html#cb96-3" tabindex="-1"></a></span>
<span id="cb96-4"><a href="working-with-sequences-raw-data-quality-control.html#cb96-4" tabindex="-1"></a><span class="co"># activate QC environment</span></span>
<span id="cb96-5"><a href="working-with-sequences-raw-data-quality-control.html#cb96-5" tabindex="-1"></a><span class="ex">conda</span> activate QC</span>
<span id="cb96-6"><a href="working-with-sequences-raw-data-quality-control.html#cb96-6" tabindex="-1"></a></span>
<span id="cb96-7"><a href="working-with-sequences-raw-data-quality-control.html#cb96-7" tabindex="-1"></a><span class="co"># run multiqc on all of the fastqc outputs</span></span>
<span id="cb96-8"><a href="working-with-sequences-raw-data-quality-control.html#cb96-8" tabindex="-1"></a><span class="ex">multiqc</span> <span class="va">$QC_DIR</span>/fastqc <span class="at">-o</span> <span class="va">$QC_DIR</span> <span class="at">-m</span> fastqc <span class="at">-f</span></span></code></pre></div>
<pre><code>## 
##   /// MultiQC 🔍 | v1.15
## 
## |           multiqc | MultiQC Version v1.16 now available!
## |           multiqc | Only using modules: fastqc
## |           multiqc | Search path : /Users/clstacy/Desktop/Genomic_Data_Analysis/QC/fastqc
## |         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 66/66  
## |            fastqc | Found 32 reports
## |           multiqc | Report      : ../../../Desktop/Genomic_Data_Analysis/QC/multiqc_report.html   (overwritten)
## |           multiqc | Data        : ../../../Desktop/Genomic_Data_Analysis/QC/multiqc_data   (overwritten)
## |           multiqc | MultiQC complete</code></pre>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="working-with-sequences-raw-data-quality-control.html#cb98-1" tabindex="-1"></a>path_multiqc <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Genomic_Data_Analysis/QC/multiqc_report.html&quot;</span></span>
<span id="cb98-2"><a href="working-with-sequences-raw-data-quality-control.html#cb98-2" tabindex="-1"></a><span class="fu">browseURL</span>(path_multiqc)</span></code></pre></div>
<p>Be sure to knit this file into a pdf or html file once you’re finished.</p>
<p>System information for reproducibility:</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="working-with-sequences-raw-data-quality-control.html#cb99-1" tabindex="-1"></a>pander<span class="sc">::</span><span class="fu">pander</span>(<span class="fu">sessionInfo</span>())</span></code></pre></div>
<p><strong>R version 4.3.1 (2023-06-16)</strong></p>
<p><strong>Platform:</strong> aarch64-apple-darwin20 (64-bit)</p>
<p><strong>locale:</strong>
en_US.UTF-8||en_US.UTF-8||en_US.UTF-8||C||en_US.UTF-8||en_US.UTF-8</p>
<p><strong>attached base packages:</strong>
<em>stats4</em>, <em>stats</em>, <em>graphics</em>, <em>grDevices</em>, <em>utils</em>, <em>datasets</em>, <em>methods</em> and <em>base</em></p>
<p><strong>other attached packages:</strong>
<em>ShortRead(v.1.58.0)</em>, <em>GenomicAlignments(v.1.36.0)</em>, <em>SummarizedExperiment(v.1.30.2)</em>, <em>MatrixGenerics(v.1.12.3)</em>, <em>matrixStats(v.1.0.0)</em>, <em>Rsamtools(v.2.16.0)</em>, <em>GenomicRanges(v.1.52.1)</em>, <em>Biostrings(v.2.68.1)</em>, <em>GenomeInfoDb(v.1.36.4)</em>, <em>XVector(v.0.40.0)</em>, <em>BiocParallel(v.1.34.2)</em>, <em>Rfastp(v.1.10.0)</em>, <em>org.Sc.sgd.db(v.3.17.0)</em>, <em>AnnotationDbi(v.1.62.2)</em>, <em>IRanges(v.2.34.1)</em>, <em>S4Vectors(v.0.38.2)</em>, <em>Biobase(v.2.60.0)</em>, <em>BiocGenerics(v.0.46.0)</em>, <em>clusterProfiler(v.4.8.2)</em>, <em>ggVennDiagram(v.1.2.3)</em>, <em>tidytree(v.0.4.5)</em>, <em>igraph(v.1.5.1)</em>, <em>janitor(v.2.2.0)</em>, <em>BiocManager(v.1.30.22)</em>, <em>pander(v.0.6.5)</em>, <em>knitr(v.1.44)</em>, <em>here(v.1.0.1)</em>, <em>lubridate(v.1.9.3)</em>, <em>forcats(v.1.0.0)</em>, <em>stringr(v.1.5.0)</em>, <em>dplyr(v.1.1.3)</em>, <em>purrr(v.1.0.2)</em>, <em>readr(v.2.1.4)</em>, <em>tidyr(v.1.3.0)</em>, <em>tibble(v.3.2.1)</em>, <em>ggplot2(v.3.4.4)</em>, <em>tidyverse(v.2.0.0)</em> and <em>pacman(v.0.5.1)</em></p>
<p><strong>loaded via a namespace (and not attached):</strong>
<em>RColorBrewer(v.1.1-3)</em>, <em>rstudioapi(v.0.15.0)</em>, <em>jsonlite(v.1.8.7)</em>, <em>magrittr(v.2.0.3)</em>, <em>farver(v.2.1.1)</em>, <em>rmarkdown(v.2.25)</em>, <em>ragg(v.1.2.6)</em>, <em>fs(v.1.6.3)</em>, <em>zlibbioc(v.1.46.0)</em>, <em>vctrs(v.0.6.4)</em>, <em>memoise(v.2.0.1)</em>, <em>RCurl(v.1.98-1.12)</em>, <em>ggtree(v.3.8.2)</em>, <em>S4Arrays(v.1.0.6)</em>, <em>htmltools(v.0.5.6.1)</em>, <em>curl(v.5.1.0)</em>, <em>gridGraphics(v.0.5-1)</em>, <em>sass(v.0.4.7)</em>, <em>KernSmooth(v.2.23-22)</em>, <em>bslib(v.0.5.1)</em>, <em>plyr(v.1.8.9)</em>, <em>cachem(v.1.0.8)</em>, <em>lifecycle(v.1.0.3)</em>, <em>pkgconfig(v.2.0.3)</em>, <em>Matrix(v.1.6-1.1)</em>, <em>R6(v.2.5.1)</em>, <em>fastmap(v.1.1.1)</em>, <em>gson(v.0.1.0)</em>, <em>GenomeInfoDbData(v.1.2.10)</em>, <em>snakecase(v.0.11.1)</em>, <em>digest(v.0.6.33)</em>, <em>aplot(v.0.2.2)</em>, <em>enrichplot(v.1.20.0)</em>, <em>colorspace(v.2.1-0)</em>, <em>patchwork(v.1.1.3)</em>, <em>rprojroot(v.2.0.3)</em>, <em>textshaping(v.0.3.7)</em>, <em>RSQLite(v.2.3.1)</em>, <em>hwriter(v.1.3.2.1)</em>, <em>labeling(v.0.4.3)</em>, <em>fansi(v.1.0.5)</em>, <em>timechange(v.0.2.0)</em>, <em>abind(v.1.4-5)</em>, <em>httr(v.1.4.7)</em>, <em>polyclip(v.1.10-6)</em>, <em>compiler(v.4.3.1)</em>, <em>proxy(v.0.4-27)</em>, <em>bit64(v.4.0.5)</em>, <em>withr(v.2.5.1)</em>, <em>downloader(v.0.4)</em>, <em>viridis(v.0.6.4)</em>, <em>DBI(v.1.1.3)</em>, <em>ggforce(v.0.4.1)</em>, <em>MASS(v.7.3-60)</em>, <em>DelayedArray(v.0.26.7)</em>, <em>rjson(v.0.2.21)</em>, <em>classInt(v.0.4-10)</em>, <em>HDO.db(v.0.99.1)</em>, <em>units(v.0.8-4)</em>, <em>tools(v.4.3.1)</em>, <em>ape(v.5.7-1)</em>, <em>scatterpie(v.0.2.1)</em>, <em>glue(v.1.6.2)</em>, <em>nlme(v.3.1-163)</em>, <em>GOSemSim(v.2.26.1)</em>, <em>sf(v.1.0-14)</em>, <em>grid(v.4.3.1)</em>, <em>shadowtext(v.0.1.2)</em>, <em>reshape2(v.1.4.4)</em>, <em>fgsea(v.1.26.0)</em>, <em>generics(v.0.1.3)</em>, <em>gtable(v.0.3.4)</em>, <em>tzdb(v.0.4.0)</em>, <em>class(v.7.3-22)</em>, <em>data.table(v.1.14.8)</em>, <em>hms(v.1.1.3)</em>, <em>tidygraph(v.1.2.3)</em>, <em>utf8(v.1.2.3)</em>, <em>ggrepel(v.0.9.4)</em>, <em>pillar(v.1.9.0)</em>, <em>yulab.utils(v.0.1.0)</em>, <em>vroom(v.1.6.4)</em>, <em>splines(v.4.3.1)</em>, <em>tweenr(v.2.0.2)</em>, <em>treeio(v.1.24.3)</em>, <em>lattice(v.0.21-9)</em>, <em>deldir(v.1.0-9)</em>, <em>bit(v.4.0.5)</em>, <em>tidyselect(v.1.2.0)</em>, <em>GO.db(v.3.17.0)</em>, <em>gridExtra(v.2.3)</em>, <em>bookdown(v.0.36)</em>, <em>xfun(v.0.40)</em>, <em>graphlayouts(v.1.0.1)</em>, <em>stringi(v.1.7.12)</em>, <em>lazyeval(v.0.2.2)</em>, <em>ggfun(v.0.1.3)</em>, <em>yaml(v.2.3.7)</em>, <em>evaluate(v.0.22)</em>, <em>codetools(v.0.2-19)</em>, <em>interp(v.1.1-4)</em>, <em>ggraph(v.2.1.0)</em>, <em>qvalue(v.2.32.0)</em>, <em>RVenn(v.1.1.0)</em>, <em>ggplotify(v.0.1.2)</em>, <em>cli(v.3.6.1)</em>, <em>systemfonts(v.1.0.5)</em>, <em>munsell(v.0.5.0)</em>, <em>jquerylib(v.0.1.4)</em>, <em>Rcpp(v.1.0.11)</em>, <em>png(v.0.1-8)</em>, <em>parallel(v.4.3.1)</em>, <em>blob(v.1.2.4)</em>, <em>jpeg(v.0.1-10)</em>, <em>latticeExtra(v.0.6-30)</em>, <em>DOSE(v.3.26.1)</em>, <em>bitops(v.1.0-7)</em>, <em>viridisLite(v.0.4.2)</em>, <em>e1071(v.1.7-13)</em>, <em>scales(v.1.2.1)</em>, <em>crayon(v.1.5.2)</em>, <em>rlang(v.1.1.1)</em>, <em>cowplot(v.1.1.1)</em>, <em>fastmatch(v.1.1-4)</em> and <em>KEGGREST(v.1.40.1)</em></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gene-ontology.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="read-mapping.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-Working_with_Sequences.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
